// @temalibrary/angular-core/dist/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const common_1 = require("@angular/common");
const core_component_1 = require("./src/core.component");
const core_directive_1 = require("./src/core.directive");
const core_pipe_1 = require("./src/core.pipe");
const core_service_1 = require("./src/core.service");
__export(require("./src/core.component"));
__export(require("./src/core.directive"));
__export(require("./src/core.pipe"));
__export(require("./src/core.service"));
let CoreModule = CoreModule_1 = class CoreModule {
    static forRoot() {
        return {
            ngModule: CoreModule_1,
            providers: [core_service_1.CoreService]
        };
    }
};
CoreModule = CoreModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule
        ],
        declarations: [
            core_component_1.CoreComponent,
            core_directive_1.CoreDirective,
            core_pipe_1.CorePipe
        ],
        exports: [
            core_component_1.CoreComponent,
            core_directive_1.CoreDirective,
            core_pipe_1.CorePipe
        ]
    })
], CoreModule);
exports.CoreModule = CoreModule;
var CoreModule_1;
