// igniteui-angular-core/ES5/AnchoredRadialSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { RadialBaseDescription } from "./RadialBaseDescription";
import { markType } from "./type";
var AnchoredRadialSeriesDescription = /** @class */ (function (_super) {
    tslib_1.__extends(AnchoredRadialSeriesDescription, _super);
    function AnchoredRadialSeriesDescription() {
        var _this = _super.call(this) || this;
        _this.ci = null;
        _this.ch = null;
        _this.cg = null;
        _this.cf = null;
        _this.b9 = 0;
        _this.cb = 0;
        _this.cc = 0;
        return _this;
    }
    AnchoredRadialSeriesDescription.prototype.get_type = function () {
        return "AnchoredRadialSeries";
    };
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "valueMemberPath", {
        get: function () {
            return this.ci;
        },
        set: function (a) {
            this.ci = a;
            this.e("ValueMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "trendLineType", {
        get: function () {
            return this.ch;
        },
        set: function (a) {
            this.ch = a;
            this.e("TrendLineType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "trendLineBrush", {
        get: function () {
            return this.cg;
        },
        set: function (a) {
            this.cg = a;
            this.e("TrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "actualTrendLineBrush", {
        get: function () {
            return this.cf;
        },
        set: function (a) {
            this.cf = a;
            this.e("ActualTrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "trendLineThickness", {
        get: function () {
            return this.b9;
        },
        set: function (a) {
            this.b9 = a;
            this.e("TrendLineThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "trendLinePeriod", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            this.cb = a;
            this.e("TrendLinePeriod");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AnchoredRadialSeriesDescription.prototype, "trendLineZIndex", {
        get: function () {
            return this.cc;
        },
        set: function (a) {
            this.cc = a;
            this.e("TrendLineZIndex");
        },
        enumerable: true,
        configurable: true
    });
    AnchoredRadialSeriesDescription.$t = markType(AnchoredRadialSeriesDescription, 'AnchoredRadialSeriesDescription', RadialBaseDescription.$);
    return AnchoredRadialSeriesDescription;
}(RadialBaseDescription));
export { AnchoredRadialSeriesDescription };
