// igniteui-angular-core/ES5/BaseDOMEventProxy_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { EventProxy } from "./EventProxy";
import { Point_$type, runOn, markType, PointUtil, String_$type, Delegate_$type } from "./type";
import { Rect } from "./Rect";
import { MouseEventArgs } from "./MouseEventArgs";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
var BaseDOMEventProxy = /** @class */ (function (_super) {
    tslib_1.__extends(BaseDOMEventProxy, _super);
    function BaseDOMEventProxy() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.bt = null;
        _this.bm = false;
        _this.bs = null;
        _this.bf = false;
        _this.bq = 0;
        _this.br = 0;
        _this.bo = 0;
        _this.bp = 0;
        _this.bu = 0;
        _this.cs = null;
        _this.cr = null;
        _this.be = false;
        _this.bv = -1;
        _this.bi = false;
        _this.bg = false;
        _this.bd = false;
        return _this;
    }
    BaseDOMEventProxy.prototype.d = function () {
        var a = new DOMEventProxy(this.bt, this.bs, this.bm);
        a.z = this.z;
        a.aj = this.aj;
        return a;
    };
    Object.defineProperty(BaseDOMEventProxy, "bn", {
        get: function () {
            return (BaseDOMEventProxy.bj || BaseDOMEventProxy.bk);
        },
        enumerable: true,
        configurable: true
    });
    BaseDOMEventProxy.prototype.bb = function (a) {
        return DOMEventProxy.ct(a);
    };
    BaseDOMEventProxy.prototype.cj = function (a) {
        if (!this.ab) {
            if (this.bh(a)) {
                a.preventDefault();
            }
        }
    };
    BaseDOMEventProxy.prototype.bh = function (e_) {
        return (e_.originalEvent.cancelable || e_.originalEvent.cancelable == undefined);
    };
    BaseDOMEventProxy.prototype.cf = function () {
        var source_ = this.bt.getNativeElement();
        BaseDOMEventProxy.ci(source_);
    };
    BaseDOMEventProxy.ci = function (source_) {
        if (BaseDOMEventProxy.bk) {
            source_.style.touchAction = 'none';
        }
        else {
            source_.style.msTouchAction = 'none';
        }
        source_.style.msUserSelect = 'none';
        source_.style.webkitTouchCallout = 'none';
        source_.style.webkitUserSelect = 'none';
        source_.style.khtmlUserSelect = 'none';
        source_.style.mozUserSelect = 'none';
        source_.style.userSelect = 'none';
    };
    BaseDOMEventProxy.cc = function (source_) {
        if (BaseDOMEventProxy.bk) {
            source_.style.touchAction = 'auto';
        }
        else {
            source_.style.msTouchAction = 'auto';
        }
        source_.style.msUserSelect = 'auto';
        source_.style.webkitTouchCallout = 'none';
        source_.style.webkitUserSelect = 'auto';
        source_.style.khtmlUserSelect = 'auto';
        source_.style.mozUserSelect = 'auto';
        source_.style.userSelect = 'auto';
    };
    BaseDOMEventProxy.prototype.cd = function () {
        var source_ = this.bt.getNativeElement();
        BaseDOMEventProxy.cc(source_);
    };
    BaseDOMEventProxy.prototype.ao = function () {
        _super.prototype.ao.call(this);
        if (!this.ab) {
            this.cf();
        }
        else {
            this.cd();
        }
    };
    BaseDOMEventProxy.prototype.b0 = function () {
        var ver_ = -1;
        var matchIE_ = /Trident\/([\d.]+)/;
        if ((matchIE_.exec(navigator.userAgent) != null)) {
            ver_ = (parseFloat(RegExp.$1));
        }
        return ver_;
    };
    BaseDOMEventProxy.prototype.bz = function () {
        var ver_ = -1;
        var matchIE_ = /Edge\/([\d.]+)/;
        if ((matchIE_.exec(navigator.userAgent) != null)) {
            ver_ = (parseFloat(RegExp.$1));
        }
        return ver_;
    };
    BaseDOMEventProxy.ch = function (e_) {
        var a = BaseDOMEventProxy.b1 >= 6;
        var oe_ = e_.originalEvent;
        if ((((typeof e_.pageX == 'undefined')) || a) && oe_.clientX != null) {
            var od_ = void 0;
            if ((e_.target !== undefined)) {
                od_ = e_.target.ownerDocument;
            }
            else {
                od_ = e_.originalEvent.target.ownerDocument;
            }
            var ed_ = od_ ? od_ : document;
            var doc_ = ed_.documentElement;
            var body_ = ed_.body;
            var clientX_ = oe_.clientX;
            var clientY_ = oe_.clientY;
            var scrollLeft_ = doc_ && doc_.scrollLeft || body_ && body_.scrollLeft || 0;
            var scrollTop_ = doc_ && doc_.scrollTop || body_ && body_.scrollTop || 0;
            var clientLeft_ = doc_ && doc_.clientLeft || body_ && body_.clientLeft || 0;
            var clientTop_ = doc_ && doc_.clientTop || body_ && body_.clientTop || 0;
            e_.pageX = clientX_ + (scrollLeft_ - clientLeft_);
            e_.pageY = clientY_ + (scrollTop_ - clientTop_);
        }
        if ((!e_.pageX)) {
            e_.pageX = e_.originalEvent.pageX;
        }
        if ((!e_.pageY)) {
            e_.pageY = e_.originalEvent.pageY;
        }
        if (((typeof e_.button == 'undefined')) && ((typeof oe_.button != 'undefined'))) {
            e_.button = oe_.button;
        }
    };
    BaseDOMEventProxy.ct = function (a) {
        var b = BaseDOMEventProxy.b1 >= 6;
        var source_ = a.getNativeElement();
        var d_ = source_.ownerDocument;
        var doc_ = d_ ? d_.documentElement : null;
        var z_ = doc_ ? doc_.msContentZoomFactor : null;
        var body_ = doc_.body;
        if ((z_ && z_ > 1) || b) {
            var rect_ = source_.getBoundingClientRect();
            var x_ = (rect_.left);
            var y_ = (rect_.top);
            var scrollLeft_ = (doc_ && doc_.scrollLeft || body_ && body_.scrollLeft || 0);
            var scrollTop_ = (doc_ && doc_.scrollTop || body_ && body_.scrollTop || 0);
            var clientLeft_ = (doc_ && doc_.clientLeft || body_ && body_.clientLeft || 0);
            var clientTop_ = (doc_ && doc_.clientTop || body_ && body_.clientTop || 0);
            var c = x_ + scrollLeft_ - clientLeft_;
            var d = y_ + scrollTop_ - clientTop_;
            return new Rect(0, c, d, 0, 0);
        }
        else {
            var e = a.getOffset();
            return new Rect(0, e.left, e.top, 0, 0);
        }
    };
    Object.defineProperty(BaseDOMEventProxy.prototype, "bx", {
        get: function () {
            return this.bu;
        },
        set: function (a) {
            this.bu = a;
        },
        enumerable: true,
        configurable: true
    });
    BaseDOMEventProxy.prototype.cm = function (a) {
        this.al = a;
        BaseDOMEventProxy.ch(a);
        var b = BaseDOMEventProxy.ct(this.bt);
        this.bq = a.pageX - b.left;
        this.br = a.pageY - b.top;
        this.bo = this.bq - this.bc.left;
        this.bp = this.br - this.bc.top;
        this.cs = { $type: Point_$type, x: this.bq, y: this.br };
        this.cr = { $type: Point_$type, x: this.bo, y: this.bp };
    };
    BaseDOMEventProxy.prototype.cn = function (e_) {
        this.al = e_;
        BaseDOMEventProxy.ch(e_);
        if (BaseDOMEventProxy.bn) {
            var a = e_.pageX;
            var b = e_.pageY;
            var c = BaseDOMEventProxy.ct(this.bt);
            this.bq = a - c.left;
            this.br = b - c.top;
        }
        else {
            this.bu = 0;
            if (!e_.originalEvent.targetTouches || e_.originalEvent.targetTouches.length < 1) {
                return;
            }
            ;
            this.bu = (e_.originalEvent.targetTouches.length);
            var d = (e_.originalEvent.targetTouches[0].pageX);
            var e = (e_.originalEvent.targetTouches[0].pageY);
            var f = BaseDOMEventProxy.ct(this.bt);
            this.bq = d - f.left;
            this.br = e - f.top;
        }
        this.bo = this.bq - this.bc.left;
        this.bp = this.br - this.bc.top;
        this.cs = { $type: Point_$type, x: this.bq, y: this.br };
        this.cr = { $type: Point_$type, x: this.bo, y: this.bp };
    };
    BaseDOMEventProxy.prototype.cb = function (e_) {
        this.cl(e_);
        if (this.aa(this.cs, false, true)) {
            var deltaY_ = 0;
            var ev_ = e_;
            deltaY_ = (ev_.originalEvent ? ev_.originalEvent.deltaY : ev_.deltaY);
            var deltaX_ = 0;
            deltaX_ = (ev_.originalEvent ? ev_.originalEvent.deltaX : ev_.deltaX);
            var a = (ev_.originalEvent ? (ev_.originalEvent.deltaMode == 1) : (ev_.deltaMode == 1));
            if (a) {
                deltaY_ = deltaY_ * 40;
                deltaX_ = deltaX_ * 40;
            }
            var b = this.af(this.cr, deltaY_);
            if (b) {
                this.cj(e_);
            }
        }
    };
    BaseDOMEventProxy.prototype.b9 = function (e_) {
        this.cl(e_);
        this.cm(e_);
        if (this.aa(this.cs, false, true)) {
            var ev_ = e_;
            var delta_ = 0;
            if (ev_.wheelDelta) {
                delta_ = ev_.wheelDelta / 120;
            }
            ;
            if (ev_.originalEvent && ev_.originalEvent.wheelDelta) {
                delta_ = ev_.originalEvent.wheelDelta / 120;
            }
            ;
            if (ev_.detail) {
                delta_ = -ev_.detail / 3;
            }
            ;
            if (ev_.originalEvent && ev_.originalEvent.detail) {
                delta_ = -ev_.originalEvent.detail / 3;
            }
            ;
            delta_ = delta_ / 10;
            var a = this.af(this.cr, delta_);
            if (a) {
                this.cj(e_);
            }
        }
    };
    BaseDOMEventProxy.prototype.cg = function () {
        if (this.be) {
            if (this.bv == -1) {
                this.bv = this.bs.setTimeout(runOn(this, this.ce), 500);
            }
            else {
                window.clearTimeout(this.bv);
                this.bv = -1;
                this.bv = this.bs.setTimeout(runOn(this, this.ce), 500);
            }
        }
    };
    BaseDOMEventProxy.prototype.b2 = function () {
        this.be = true;
        if (this.bv != -1) {
            window.clearTimeout(this.bv);
            this.bv = -1;
        }
    };
    BaseDOMEventProxy.prototype.ce = function () {
        this.be = false;
    };
    Object.defineProperty(BaseDOMEventProxy.prototype, "bl", {
        get: function () {
            return this.bg;
        },
        set: function (a) {
            this.bg = a;
        },
        enumerable: true,
        configurable: true
    });
    BaseDOMEventProxy.prototype.b7 = function (a) {
        if (this.bg || this.be) {
            return;
        }
        this.cm(a);
        if (this.bi) {
            this.bi = false;
            this.a0(this.cr);
        }
    };
    BaseDOMEventProxy.prototype.b8 = function (a) {
        if (this.bg || this.be) {
            return;
        }
        this.cm(a);
        this.cl(a);
        var b = new MouseEventArgs();
        b.position = this.cs;
        if (this.aa(this.cs, false, false)) {
            if (!this.bi) {
                this.bi = true;
                this.az(this.cs);
            }
            this.a1(this.cs, true, false);
            this.aq(this.cr, false);
        }
        else if (this.bi) {
            this.b7(a);
        }
    };
    BaseDOMEventProxy.prototype.ck = function (a) {
        this.bt.focus();
    };
    BaseDOMEventProxy.prototype.b6 = function (a) {
        if (this.bg || this.be) {
            return;
        }
        this.ck(a);
        this.cm(a);
        this.cl(a);
        if (this.aa(this.cs, false, false)) {
            this.bd = true;
            this.ay(this.cs);
            this.ar(this.cr, false);
            this.cj(a);
        }
    };
    BaseDOMEventProxy.prototype.co = function (a) {
        if (this.bg || this.be) {
            return;
        }
        if (this.bd) {
            this.ca(a);
        }
    };
    BaseDOMEventProxy.prototype.ca = function (a) {
        if (this.bg || this.be) {
            return;
        }
        this.cl(a);
        if (PointUtil.equals(this.cs, null) || PointUtil.equals(this.cr, null)) {
            this.cm(a);
        }
        var b = this.aa(this.cs, false, false) || this.bd || this.bi;
        this.bd = false;
        this.a2(this.cs);
        this.ap(this.cr, false);
        if (b) {
            this.cj(a);
        }
    };
    BaseDOMEventProxy.prototype.b3 = function (a) {
        this.b4(a, true);
    };
    BaseDOMEventProxy.prototype.b5 = function (a) {
        this.b4(a, false);
    };
    BaseDOMEventProxy.prototype.cq = function (a) {
        return BaseDOMEventProxy.cp(a);
    };
    BaseDOMEventProxy.cp = function (a) {
        var b;
        switch (a.which) {
            case 33:
                b = 10;
                break;
            case 34:
                b = 11;
                break;
            case 36:
                b = 13;
                break;
            case 37:
                b = 14;
                break;
            case 38:
                b = 15;
                break;
            case 39:
                b = 16;
                break;
            case 40:
                b = 17;
                break;
            case 9:
                b = 2;
                break;
            case 32:
                b = 9;
                break;
            case 13:
                b = 3;
                break;
            case 27:
                b = 8;
                break;
            case 16:
                b = 4;
                break;
            case 17:
                b = 5;
                break;
            case 18:
                b = 6;
                break;
            case 8:
                b = 1;
                break;
            case 35:
                b = 12;
                break;
            case 45:
                b = 18;
                break;
            case 46:
                b = 19;
                break;
            case 106:
                b = 78;
                break;
            case 107:
                b = 79;
                break;
            case 109:
                b = 80;
                break;
            case 110:
                b = 81;
                break;
            case 111:
                b = 82;
                break;
            case 220:
                b = 85;
                break;
            case 192:
                b = 86;
                break;
            case 191:
                b = 84;
                break;
            case 186:
                b = 83;
                break;
            case 189:
                b = 88;
                break;
            case 187:
                b = 87;
                break;
            default:
                if (112 <= a.which && a.which <= 123) {
                    b = (56 + a.which - 112);
                }
                else if (65 <= a.which && a.which <= 90) {
                    b = (30 + a.which - 65);
                }
                else if (48 <= a.which && a.which <= 57) {
                    b = (20 + a.which - 48);
                }
                else if (96 <= a.which && a.which <= 105) {
                    b = (68 + a.which - 96);
                }
                else {
                    b = 0;
                }
                break;
        }
        return b;
    };
    BaseDOMEventProxy.prototype.b4 = function (a, b) {
        var c = this.cq(a);
        var d = c != 0;
        this.cl(a);
        if (d) {
            var e = false;
            if (b) {
                e = this.ad(c);
            }
            else {
                e = this.ae(c);
            }
            if (e) {
                this.cj(a);
            }
        }
    };
    BaseDOMEventProxy.prototype.cl = function (a) {
        this.al = a;
        this.a9 = 0;
        if (a.shiftKey) {
            this.a9 = this.a9 | 4;
        }
        if (a.altKey) {
            this.a9 = this.a9 | 1;
        }
        if (a.ctrlKey) {
            this.a9 = this.a9 | 2;
        }
        this.ag = a.button == 2;
    };
    BaseDOMEventProxy.prototype.ba = function () {
        var a = BaseDOMEventProxy.ct(this.bt);
        var b = a.left;
        var c = a.top;
        return { $type: Point_$type, x: b, y: c };
    };
    BaseDOMEventProxy.$t = markType(BaseDOMEventProxy, 'BaseDOMEventProxy', EventProxy.$);
    BaseDOMEventProxy.bj = false;
    BaseDOMEventProxy.bk = false;
    BaseDOMEventProxy.b1 = 0;
    BaseDOMEventProxy.by = 0;
    return BaseDOMEventProxy;
}(EventProxy));
export { BaseDOMEventProxy };
var DOMEventProxy = /** @class */ (function (_super) {
    tslib_1.__extends(DOMEventProxy, _super);
    function DOMEventProxy(a, b, c) {
        var _this = _super.call(this) || this;
        _this.c5 = 0;
        _this.c6 = "";
        _this.cv = false;
        _this.c2 = null;
        _this.cy = new Dictionary$2(String_$type, List$1.$.specialize(Delegate_$type), 0);
        _this.dt = null;
        _this.c3 = -1;
        _this.du = null;
        _this.c0 = 0;
        _this.c1 = 0;
        _this.cz = 5;
        _this.cu = false;
        _this.ds = null;
        _this.bm = c;
        _this.bs = b;
        _this.aj = function (d) { return true; };
        _this.z = false;
        DOMEventProxy.c4++;
        _this.c5 = DOMEventProxy.c4;
        _this.c6 = ".DOMProxy" + _this.c5.toString();
        _this.a9 = 0;
        _this.bt = a;
        BaseDOMEventProxy.bj = false;
        try {
            BaseDOMEventProxy.bj = (window.navigator.msPointerEnabled && MSGesture !== undefined);
        }
        catch (d) {
        }
        BaseDOMEventProxy.bk = false;
        try {
            BaseDOMEventProxy.bk = (window.PointerEvent || window.navigator.pointerEnabled);
        }
        catch (e) {
        }
        BaseDOMEventProxy.b1 = _this.b0();
        BaseDOMEventProxy.by = _this.bz();
        _this.am(_this.bt, "");
        _this.ai = function (f) { return true; };
        return _this;
    }
    DOMEventProxy.prototype.c7 = function (a, b, c) {
    };
    DOMEventProxy.prototype.am = function (a, b) {
        var c = a;
        var d = this.c6 + b;
        if (!this.bs.supportsDOMEvents()) {
            return;
        }
        var e = new List$1(Delegate_$type, 0);
        this.cy.addItem(d, e);
        this.c7(a, b, e);
        if (!BaseDOMEventProxy.bn) {
            e.add(c.listen("mousemove", runOn(this, this.b8)));
            e.add(c.listen("mouseleave", runOn(this, this.b7)));
            e.add(c.listen("mousedown", runOn(this, this.b6)));
            e.add(c.listen("mouseup", runOn(this, this.ca)));
            e.add(this.bs.globalListen("window", "mouseup", runOn(this, this.co)));
        }
        e.add(c.listen("keydown", runOn(this, this.b3)));
        e.add(c.listen("keyup", runOn(this, this.b5)));
        var source_ = this.bt.getNativeElement();
        if (!this.ab) {
            BaseDOMEventProxy.ci(source_);
        }
        if (BaseDOMEventProxy.bn) {
            var f = this.bt.getNativeElement();
            if ((window.MSGesture !== undefined)) {
                var g = (new MSGesture());
                g.target = f;
                this.c2 = g;
                e.add(c.listen("MSGestureStart", runOn(this, this.da)));
                e.add(c.listen("MSGestureChange", runOn(this, this.c8)));
                e.add(c.listen("MSGestureEnd", runOn(this, this.c9)));
            }
            if (BaseDOMEventProxy.bk) {
                e.add(c.listen("pointerdown", runOn(this, this.dd)));
                e.add(c.listen("pointerup", runOn(this, this.dg)));
                e.add(c.listen("pointercancel", runOn(this, this.dc)));
                e.add(c.listen("pointermove", runOn(this, this.de)));
                e.add(c.listen("pointerout", runOn(this, this.df)));
                e.add(c.listen("lostpointercapture", runOn(this, this.db)));
            }
            else {
                e.add(c.listen("MSPointerDown", runOn(this, this.dd)));
                e.add(c.listen("MSPointerUp", runOn(this, this.dg)));
                e.add(c.listen("MSPointerCancel", runOn(this, this.dc)));
                e.add(c.listen("MSPointerMove", runOn(this, this.de)));
                e.add(c.listen("MSPointerOut", runOn(this, this.df)));
                e.add(c.listen("MSLostPointerCapture", runOn(this, this.db)));
            }
        }
        else {
            e.add(c.listen("gesturestart", runOn(this, this.da)));
            e.add(c.listen("gesturechange", runOn(this, this.c8)));
            e.add(c.listen("gestureend", runOn(this, this.c9)));
            e.add(c.listen("touchstart", runOn(this, this.dj)));
            e.add(c.listen("touchmove", runOn(this, this.di)));
            e.add(c.listen("touchend", runOn(this, this.dh)));
        }
        if (this.bm) {
            e.add(c.listen("wheel", runOn(this, this.cb)));
        }
        else {
            e.add(c.listen("DOMMouseScroll", runOn(this, this.b9)));
            e.add(c.listen("mousewheel", runOn(this, this.b9)));
        }
    };
    DOMEventProxy.prototype.dg = function (e_) {
        var a = this.cw(e_);
        if (a) {
            this.bx = this.bx - 1;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            var b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
        else {
            if (a) {
                this.dh(e_);
            }
            else {
                this.ca(e_);
            }
        }
    };
    DOMEventProxy.prototype.db = function (e_) {
        var a = this.cw(e_);
        if (a) {
            this.bx--;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            var b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
    };
    DOMEventProxy.prototype.dc = function (e_) {
        var a = this.cw(e_);
        if (a) {
            this.bx--;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            var b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
    };
    DOMEventProxy.prototype.dd = function (e_) {
        var a = this.cw(e_);
        if (this.c2 != null && a) {
            this.bx++;
            this.c2.addPointer((e_.originalEvent.pointerId));
        }
        var eventSource_ = this.bt.getNativeElement();
        if (BaseDOMEventProxy.bk) {
            eventSource_.setPointerCapture(e_.originalEvent.pointerId);
        }
        else {
            eventSource_.msSetPointerCapture(e_.originalEvent.pointerId);
        }
        if (this.bx > 1 && !this.cv && a) {
            this.cv = true;
            this.cn(e_);
            var b = (e_.originalEvent.scale);
            this.a4(this.cr, b);
        }
        else {
            if (a) {
                this.dj(e_);
            }
            else {
                this.b6(e_);
            }
        }
    };
    DOMEventProxy.prototype.cw = function (e_) {
        var pointerEvent_ = e_.originalEvent;
        var a = false;
        if (BaseDOMEventProxy.bk) {
            a = (pointerEvent_.pointerType == 'touch');
        }
        else {
            a = (pointerEvent_.pointerType == pointerEvent_.MSPOINTER_TYPE_TOUCH);
        }
        return a;
    };
    DOMEventProxy.prototype.de = function (e_) {
        var a = this.cw(e_);
        if (this.cv) {
            return;
        }
        if (a) {
            this.di(e_);
        }
        else {
            this.b8(e_);
        }
    };
    DOMEventProxy.prototype.df = function (e_) {
        var a = this.cw(e_);
        if (this.cv) {
            return;
        }
        if (a) {
        }
        else {
            this.b7(e_);
        }
    };
    DOMEventProxy.prototype.da = function (e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        this.dl();
        if (this.aa(this.cs, true, false)) {
            this.cj(e_);
            if (BaseDOMEventProxy.bn && this.bx < 2) {
                return;
            }
            var a = (e_.originalEvent.scale);
            this.a4(this.cr, a);
        }
    };
    DOMEventProxy.prototype.c8 = function (e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        if (this.aa(this.cs, true, false)) {
            this.cj(e_);
            if (BaseDOMEventProxy.bn && this.bx < 2) {
                return;
            }
            var a = (e_.originalEvent.scale);
            this.a3(this.cr, a);
        }
    };
    DOMEventProxy.prototype.c9 = function (e_) {
        this.cj(e_);
        var a = (e_.originalEvent.scale);
        this.bi = false;
        this.cl(e_);
        this.cn(e_);
        if (!BaseDOMEventProxy.bn || this.cv) {
            this.aw(this.cr, a);
        }
    };
    DOMEventProxy.prototype.dj = function (e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        this.dt = this.cr;
        if (this.aa(this.cs, true, false)) {
            if (!this.z) {
                this.cj(e_);
            }
            else {
                this.bl = true;
            }
            this.a1(this.cs, false, true);
            this.ay(this.cs);
            this.ar(this.cr, true);
            this.dp();
        }
    };
    DOMEventProxy.prototype.dp = function () {
        if (this.c3 == -1) {
            this.c0 = this.cs.x;
            this.c1 = this.cs.y;
            this.c3 = this.bs.setTimeout(runOn(this, this.dm), 1500);
        }
    };
    DOMEventProxy.prototype.dl = function () {
        if (Math.abs(this.c0 - this.cs.x) > this.cz || Math.abs(this.c1 - this.cs.y) > this.cz) {
            this.dq();
        }
    };
    DOMEventProxy.prototype.dq = function () {
        if (this.c3 != -1) {
            window.clearTimeout(this.c3);
            this.c3 = -1;
        }
    };
    DOMEventProxy.prototype.dm = function () {
        this.c3 = -1;
        if (this.bl) {
            this.bl = false;
            this.b2();
        }
        this.ax(this.cr);
    };
    DOMEventProxy.prototype.di = function (a) {
        this.bi = true;
        this.cl(a);
        this.cn(a);
        if (this.bl) {
            if (this.aa(this.cs, true, false) && this.cx()) {
                this.bl = false;
                this.b2();
            }
        }
        this.dl();
        this.dr();
        if (this.aa(this.cs, true, false) && this.bx == 1 && !this.bl) {
            this.cj(a);
            if (!this.cu) {
                this.cu = true;
                this.ds = this.cr;
                this.av(this.ds);
            }
            else {
                this.a1(this.cs, true, true);
                this.aq(this.cr, true);
                this.au(this.cr);
            }
        }
    };
    DOMEventProxy.prototype.cx = function () {
        var a = { $type: Point_$type, x: this.dt.x - this.cr.x, y: this.dt.y - this.cr.y };
        if (Math.abs(a.x) < 2 && Math.abs(a.y) < 2) {
            return false;
        }
        if (Math.abs(a.x) > Math.abs(a.y)) {
            a.y = 0;
        }
        if (Math.abs(a.y) > Math.abs(a.x)) {
            a.x = 0;
        }
        return this.aj(a);
    };
    DOMEventProxy.prototype.dh = function (a) {
        this.bi = false;
        this.cl(a);
        this.cn(a);
        this.dq();
        if (this.bl) {
            this.bl = false;
            this.b2();
        }
        this.cj(a);
        this.a1(this.cs, false, true);
        this.a2(this.cs);
        if (this.bx == 0) {
            this.dn(this.cs);
            this.cg();
        }
        if (this.cu && this.bx == 0) {
            this.cu = false;
            this.ds = null;
            this.at(this.cr);
            this.cg();
        }
        this.ap(this.cr, true);
    };
    DOMEventProxy.prototype.dn = function (a) {
        if (Rect.l_op_Equality(this.du, null)) {
            this.du = new Rect(0, a.x - 50, a.y - 50, 100, 100);
            this.bs.setTimeout(runOn(this, this.dr), 500);
        }
        else {
            if (a.x >= this.du.x && a.x <= this.du.right && a.y >= this.du.y && a.y <= this.du.bottom) {
                this.dr();
                this.as(a);
            }
        }
    };
    DOMEventProxy.prototype.dr = function () {
        this.du = null;
    };
    DOMEventProxy.prototype.dk = function (a) {
        if (this.cy.containsKey(a)) {
            var b = this.cy.item(a);
            this.cy.removeItem(a);
            for (var c = 0; c < b.count; c++) {
                b._inner[c]();
            }
            b.clear();
        }
    };
    DOMEventProxy.prototype.a8 = function (a, b) {
        var c = a;
        var d = this.c6 + b;
        this.dk(d);
    };
    DOMEventProxy.prototype.an = function () {
        if (this.bt == null) {
            return;
        }
        this.dk(this.c6);
        this.bt = null;
    };
    DOMEventProxy.$t = markType(DOMEventProxy, 'DOMEventProxy', BaseDOMEventProxy.$);
    DOMEventProxy.c4 = 0;
    return DOMEventProxy;
}(BaseDOMEventProxy));
export { DOMEventProxy };
