// igniteui-angular-core/ES5/CalloutLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
var CalloutLayerDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CalloutLayerDescription, _super);
    function CalloutLayerDescription() {
        var _this = _super.call(this) || this;
        _this.cv = null;
        _this.cy = null;
        _this.cz = null;
        _this.cu = null;
        _this.b0 = false;
        _this.ct = null;
        _this.cw = null;
        _this.cs = null;
        _this.bz = false;
        _this.b2 = false;
        _this.b1 = false;
        _this.cj = 0;
        _this.cr = null;
        _this.cn = null;
        _this.ck = 0;
        _this.cq = null;
        _this.cp = null;
        _this.b8 = 0;
        _this.ca = 0;
        _this.b9 = 0;
        _this.b7 = 0;
        _this.co = null;
        _this.cb = 0;
        _this.cc = 0;
        _this.cx = null;
        _this._CalloutStyleUpdatingRef = null;
        _this._CalloutContentUpdatingRef = null;
        _this._CalloutLabelUpdatingRef = null;
        _this._CalloutSeriesSelectingRef = null;
        return _this;
    }
    CalloutLayerDescription.prototype.get_type = function () {
        return "CalloutLayer";
    };
    Object.defineProperty(CalloutLayerDescription.prototype, "labelMemberPath", {
        get: function () {
            return this.cv;
        },
        set: function (a) {
            this.cv = a;
            this.e("LabelMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "xMemberPath", {
        get: function () {
            return this.cy;
        },
        set: function (a) {
            this.cy = a;
            this.e("XMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "yMemberPath", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            this.cz = a;
            this.e("YMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "keyMemberPath", {
        get: function () {
            return this.cu;
        },
        set: function (a) {
            this.cu = a;
            this.e("KeyMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "isCalloutOffsettingEnabled", {
        get: function () {
            return this.b0;
        },
        set: function (a) {
            this.b0 = a;
            this.e("IsCalloutOffsettingEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "contentMemberPath", {
        get: function () {
            return this.ct;
        },
        set: function (a) {
            this.ct = a;
            this.e("ContentMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "targetSeriesRef", {
        get: function () {
            return this.cw;
        },
        set: function (a) {
            this.cw = a;
            this.e("TargetSeriesRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "collisionChannel", {
        get: function () {
            return this.cs;
        },
        set: function (a) {
            this.cs = a;
            this.e("CollisionChannel");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "isAutoCalloutBehaviorEnabled", {
        get: function () {
            return this.bz;
        },
        set: function (a) {
            this.bz = a;
            this.e("IsAutoCalloutBehaviorEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "useValueForAutoCalloutLabels", {
        get: function () {
            return this.b2;
        },
        set: function (a) {
            this.b2 = a;
            this.e("UseValueForAutoCalloutLabels");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "isCustomCalloutStyleEnabled", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            this.b1 = a;
            this.e("IsCustomCalloutStyleEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "autoCalloutLabelPrecision", {
        get: function () {
            return this.cj;
        },
        set: function (a) {
            this.cj = a;
            this.e("AutoCalloutLabelPrecision");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutTextColor", {
        get: function () {
            return this.cr;
        },
        set: function (a) {
            this.cr = a;
            this.e("CalloutTextColor");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutBackground", {
        get: function () {
            return this.cn;
        },
        set: function (a) {
            this.cn = a;
            this.e("CalloutBackground");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutInterpolatedValuePrecision", {
        get: function () {
            return this.ck;
        },
        set: function (a) {
            this.ck = a;
            this.e("CalloutInterpolatedValuePrecision");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutOutline", {
        get: function () {
            return this.cq;
        },
        set: function (a) {
            this.cq = a;
            this.e("CalloutOutline");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutLeaderBrush", {
        get: function () {
            return this.cp;
        },
        set: function (a) {
            this.cp = a;
            this.e("CalloutLeaderBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutPaddingLeft", {
        get: function () {
            return this.b8;
        },
        set: function (a) {
            this.b8 = a;
            this.e("CalloutPaddingLeft");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutPaddingTop", {
        get: function () {
            return this.ca;
        },
        set: function (a) {
            this.ca = a;
            this.e("CalloutPaddingTop");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutPaddingRight", {
        get: function () {
            return this.b9;
        },
        set: function (a) {
            this.b9 = a;
            this.e("CalloutPaddingRight");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutPaddingBottom", {
        get: function () {
            return this.b7;
        },
        set: function (a) {
            this.b7 = a;
            this.e("CalloutPaddingBottom");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutCollisionMode", {
        get: function () {
            return this.co;
        },
        set: function (a) {
            this.co = a;
            this.e("CalloutCollisionMode");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutPositionPadding", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            this.cb = a;
            this.e("CalloutPositionPadding");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "calloutStrokeThickness", {
        get: function () {
            return this.cc;
        },
        set: function (a) {
            this.cc = a;
            this.e("CalloutStrokeThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "textStyle", {
        get: function () {
            return this.cx;
        },
        set: function (a) {
            this.cx = a;
            this.e("TextStyle");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "CalloutStyleUpdatingRef", {
        get: function () {
            return this._CalloutStyleUpdatingRef;
        },
        set: function (a) {
            this._CalloutStyleUpdatingRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "CalloutContentUpdatingRef", {
        get: function () {
            return this._CalloutContentUpdatingRef;
        },
        set: function (a) {
            this._CalloutContentUpdatingRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "CalloutLabelUpdatingRef", {
        get: function () {
            return this._CalloutLabelUpdatingRef;
        },
        set: function (a) {
            this._CalloutLabelUpdatingRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CalloutLayerDescription.prototype, "CalloutSeriesSelectingRef", {
        get: function () {
            return this._CalloutSeriesSelectingRef;
        },
        set: function (a) {
            this._CalloutSeriesSelectingRef = a;
        },
        enumerable: true,
        configurable: true
    });
    CalloutLayerDescription.$t = markType(CalloutLayerDescription, 'CalloutLayerDescription', AnnotationLayerDescription.$);
    return CalloutLayerDescription;
}(AnnotationLayerDescription));
export { CalloutLayerDescription };
