// igniteui-angular-core/ES5/CategoryAngleAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
var CategoryAngleAxisDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryAngleAxisDescription, _super);
    function CategoryAngleAxisDescription() {
        var _this = _super.call(this) || this;
        _this.c0 = 0;
        _this.cy = 0;
        _this.cw = 0;
        _this.cz = 0;
        _this.cx = 0;
        return _this;
    }
    CategoryAngleAxisDescription.prototype.get_type = function () {
        return "CategoryAngleAxis";
    };
    Object.defineProperty(CategoryAngleAxisDescription.prototype, "startAngleOffset", {
        get: function () {
            return this.c0;
        },
        set: function (a) {
            this.c0 = a;
            this.e("StartAngleOffset");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAngleAxisDescription.prototype, "interval", {
        get: function () {
            return this.cy;
        },
        set: function (a) {
            this.cy = a;
            this.e("Interval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAngleAxisDescription.prototype, "actualInterval", {
        get: function () {
            return this.cw;
        },
        set: function (a) {
            this.cw = a;
            this.e("ActualInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAngleAxisDescription.prototype, "minorInterval", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            this.cz = a;
            this.e("MinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAngleAxisDescription.prototype, "actualMinorInterval", {
        get: function () {
            return this.cx;
        },
        set: function (a) {
            this.cx = a;
            this.e("ActualMinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    CategoryAngleAxisDescription.$t = markType(CategoryAngleAxisDescription, 'CategoryAngleAxisDescription', CategoryAxisBaseDescription.$);
    return CategoryAngleAxisDescription;
}(CategoryAxisBaseDescription));
export { CategoryAngleAxisDescription };
