// igniteui-angular-core/ES5/CategoryItemHighlightLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
var CategoryItemHighlightLayerDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryItemHighlightLayerDescription, _super);
    function CategoryItemHighlightLayerDescription() {
        var _this = _super.call(this) || this;
        _this.b7 = null;
        _this.b0 = false;
        _this.b5 = null;
        _this.b6 = null;
        _this.b3 = 0;
        _this.bz = false;
        return _this;
    }
    CategoryItemHighlightLayerDescription.prototype.get_type = function () {
        return "CategoryItemHighlightLayer";
    };
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "targetSeriesRef", {
        get: function () {
            return this.b7;
        },
        set: function (a) {
            this.b7 = a;
            this.e("TargetSeriesRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "useInterpolation", {
        get: function () {
            return this.b0;
        },
        set: function (a) {
            this.b0 = a;
            this.e("UseInterpolation");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "highlightType", {
        get: function () {
            return this.b5;
        },
        set: function (a) {
            this.b5 = a;
            this.e("HighlightType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "markerTemplate", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("MarkerTemplate");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "bandHighlightWidth", {
        get: function () {
            return this.b3;
        },
        set: function (a) {
            this.b3 = a;
            this.e("BandHighlightWidth");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryItemHighlightLayerDescription.prototype, "skipUnknownValues", {
        get: function () {
            return this.bz;
        },
        set: function (a) {
            this.bz = a;
            this.e("SkipUnknownValues");
        },
        enumerable: true,
        configurable: true
    });
    CategoryItemHighlightLayerDescription.$t = markType(CategoryItemHighlightLayerDescription, 'CategoryItemHighlightLayerDescription', AnnotationLayerDescription.$);
    return CategoryItemHighlightLayerDescription;
}(AnnotationLayerDescription));
export { CategoryItemHighlightLayerDescription };
