// igniteui-angular-core/ES5/CategoryDateTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { TimeAxisBaseDescription } from "./TimeAxisBaseDescription";
import { markType } from "./type";
var CategoryDateTimeXAxisDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryDateTimeXAxisDescription, _super);
    function CategoryDateTimeXAxisDescription() {
        var _this = _super.call(this) || this;
        _this.dg = null;
        _this.da = 0;
        _this.c8 = 0;
        _this.db = 0;
        _this.c9 = 0;
        return _this;
    }
    CategoryDateTimeXAxisDescription.prototype.get_type = function () {
        return "CategoryDateTimeXAxis";
    };
    Object.defineProperty(CategoryDateTimeXAxisDescription.prototype, "displayType", {
        get: function () {
            return this.dg;
        },
        set: function (a) {
            this.dg = a;
            this.e("DisplayType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryDateTimeXAxisDescription.prototype, "interval", {
        get: function () {
            return this.da;
        },
        set: function (a) {
            this.da = a;
            this.e("Interval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryDateTimeXAxisDescription.prototype, "actualInterval", {
        get: function () {
            return this.c8;
        },
        set: function (a) {
            this.c8 = a;
            this.e("ActualInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryDateTimeXAxisDescription.prototype, "minorInterval", {
        get: function () {
            return this.db;
        },
        set: function (a) {
            this.db = a;
            this.e("MinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryDateTimeXAxisDescription.prototype, "actualMinorInterval", {
        get: function () {
            return this.c9;
        },
        set: function (a) {
            this.c9 = a;
            this.e("ActualMinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    CategoryDateTimeXAxisDescription.$t = markType(CategoryDateTimeXAxisDescription, 'CategoryDateTimeXAxisDescription', TimeAxisBaseDescription.$);
    return CategoryDateTimeXAxisDescription;
}(TimeAxisBaseDescription));
export { CategoryDateTimeXAxisDescription };
