// igniteui-angular-core/ES5/CategorySeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { MarkerSeriesDescription } from "./MarkerSeriesDescription";
import { markType } from "./type";
var CategorySeriesDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CategorySeriesDescription, _super);
    function CategorySeriesDescription() {
        var _this = _super.call(this) || this;
        _this.b4 = false;
        _this.b3 = false;
        _this.b6 = false;
        _this.cb = null;
        _this.b5 = false;
        _this._AssigningCategoryStyleRef = null;
        _this._AssigningCategoryMarkerStyleRef = null;
        return _this;
    }
    CategorySeriesDescription.prototype.get_type = function () {
        return "CategorySeries";
    };
    Object.defineProperty(CategorySeriesDescription.prototype, "isCustomCategoryStyleAllowed", {
        get: function () {
            return this.b4;
        },
        set: function (a) {
            this.b4 = a;
            this.e("IsCustomCategoryStyleAllowed");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "isCustomCategoryMarkerStyleAllowed", {
        get: function () {
            return this.b3;
        },
        set: function (a) {
            this.b3 = a;
            this.e("IsCustomCategoryMarkerStyleAllowed");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "useHighMarkerFidelity", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("UseHighMarkerFidelity");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "transitionInMode", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            this.cb = a;
            this.e("TransitionInMode");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "isTransitionInEnabled", {
        get: function () {
            return this.b5;
        },
        set: function (a) {
            this.b5 = a;
            this.e("IsTransitionInEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "AssigningCategoryStyleRef", {
        get: function () {
            return this._AssigningCategoryStyleRef;
        },
        set: function (a) {
            this._AssigningCategoryStyleRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategorySeriesDescription.prototype, "AssigningCategoryMarkerStyleRef", {
        get: function () {
            return this._AssigningCategoryMarkerStyleRef;
        },
        set: function (a) {
            this._AssigningCategoryMarkerStyleRef = a;
        },
        enumerable: true,
        configurable: true
    });
    CategorySeriesDescription.$t = markType(CategorySeriesDescription, 'CategorySeriesDescription', MarkerSeriesDescription.$);
    return CategorySeriesDescription;
}(MarkerSeriesDescription));
export { CategorySeriesDescription };
