// igniteui-angular-core/ES5/CategoryAxisBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { AxisDescription } from "./AxisDescription";
import { markType } from "./type";
var CategoryAxisBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryAxisBaseDescription, _super);
    function CategoryAxisBaseDescription() {
        var _this = _super.call(this) || this;
        _this.cs = 0;
        _this.co = 0;
        _this.cp = 0;
        _this.cm = false;
        _this.cu = null;
        return _this;
    }
    CategoryAxisBaseDescription.prototype.get_type = function () {
        return "CategoryAxisBase";
    };
    Object.defineProperty(CategoryAxisBaseDescription.prototype, "itemsCount", {
        get: function () {
            return this.cs;
        },
        set: function (a) {
            this.cs = a;
            this.e("ItemsCount");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAxisBaseDescription.prototype, "gap", {
        get: function () {
            return this.co;
        },
        set: function (a) {
            this.co = a;
            this.e("Gap");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAxisBaseDescription.prototype, "overlap", {
        get: function () {
            return this.cp;
        },
        set: function (a) {
            this.cp = a;
            this.e("Overlap");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAxisBaseDescription.prototype, "useClusteringMode", {
        get: function () {
            return this.cm;
        },
        set: function (a) {
            this.cm = a;
            this.e("UseClusteringMode");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CategoryAxisBaseDescription.prototype, "dataSourceRef", {
        get: function () {
            return this.cu;
        },
        set: function (a) {
            this.cu = a;
            this.e("DataSourceRef");
        },
        enumerable: true,
        configurable: true
    });
    CategoryAxisBaseDescription.$t = markType(CategoryAxisBaseDescription, 'CategoryAxisBaseDescription', AxisDescription.$);
    return CategoryAxisBaseDescription;
}(AxisDescription));
export { CategoryAxisBaseDescription };
