// igniteui-angular-core/ES5/ColumnDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
var ColumnDescription = /** @class */ (function (_super) {
    tslib_1.__extends(ColumnDescription, _super);
    function ColumnDescription() {
        var _this = _super.call(this) || this;
        _this.a9 = 0;
        _this.bb = 0;
        _this.ba = 0;
        _this.a8 = 0;
        _this.bj = null;
        _this.bk = null;
        _this.bi = null;
        _this.bg = null;
        _this.bl = null;
        _this.bh = null;
        _this.aw = null;
        _this.a6 = 0;
        _this.ay = false;
        _this.au = null;
        _this.az = false;
        _this.a0 = false;
        _this.a1 = false;
        _this.bm = null;
        return _this;
    }
    ColumnDescription.prototype.get_type = function () {
        return "Column";
    };
    Object.defineProperty(ColumnDescription.prototype, "paddingLeft", {
        get: function () {
            return this.a9;
        },
        set: function (a) {
            this.a9 = a;
            this.e("PaddingLeft");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingTop", {
        get: function () {
            return this.bb;
        },
        set: function (a) {
            this.bb = a;
            this.e("PaddingTop");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingRight", {
        get: function () {
            return this.ba;
        },
        set: function (a) {
            this.ba = a;
            this.e("PaddingRight");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingBottom", {
        get: function () {
            return this.a8;
        },
        set: function (a) {
            this.a8 = a;
            this.e("PaddingBottom");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "name", {
        get: function () {
            return this.bj;
        },
        set: function (a) {
            this.bj = a;
            this.e("Name");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "propertyPath", {
        get: function () {
            return this.bk;
        },
        set: function (a) {
            this.bk = a;
            this.e("PropertyPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "headerText", {
        get: function () {
            return this.bi;
        },
        set: function (a) {
            this.bi = a;
            this.e("HeaderText");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualHeaderText", {
        get: function () {
            return this.bg;
        },
        set: function (a) {
            this.bg = a;
            this.e("ActualHeaderText");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "selectedBackground", {
        get: function () {
            return this.bl;
        },
        set: function (a) {
            this.bl = a;
            this.e("SelectedBackground");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualSelectedBackground", {
        get: function () {
            return this.bh;
        },
        set: function (a) {
            this.bh = a;
            this.e("ActualSelectedBackground");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "animationSettings", {
        get: function () {
            return this.aw;
        },
        set: function (a) {
            this.aw = a;
            this.e("AnimationSettings");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "minWidth", {
        get: function () {
            return this.a6;
        },
        set: function (a) {
            this.a6 = a;
            this.e("MinWidth");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isAutoGenerated", {
        get: function () {
            return this.ay;
        },
        set: function (a) {
            this.ay = a;
            this.e("IsAutoGenerated");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "filter", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("Filter");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isFilteringEnabled", {
        get: function () {
            return this.az;
        },
        set: function (a) {
            this.az = a;
            this.e("IsFilteringEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isHidden", {
        get: function () {
            return this.a0;
        },
        set: function (a) {
            this.a0 = a;
            this.e("IsHidden");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "shouldRemoveWhenHidden", {
        get: function () {
            return this.a1;
        },
        set: function (a) {
            this.a1 = a;
            this.e("ShouldRemoveWhenHidden");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "sortDirection", {
        get: function () {
            return this.bm;
        },
        set: function (a) {
            this.bm = a;
            this.e("SortDirection");
        },
        enumerable: true,
        configurable: true
    });
    ColumnDescription.$t = markType(ColumnDescription, 'ColumnDescription', DefinitionBaseDescription.$);
    return ColumnDescription;
}(DefinitionBaseDescription));
export { ColumnDescription };
