// igniteui-angular-core/ES5/ComponentRenderer.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Base, Boolean_$type, typeCast, delegateCombine, markType } from "./type";
import { TypeDescriptionContext } from "./TypeDescriptionContext";
import { TypeDescriptionPropretyTransforms } from "./TypeDescriptionPropretyTransforms";
import { Dictionary$2 } from "./Dictionary$2";
import { Description } from "./Description";
import { DescriptionTreeNode } from "./DescriptionTreeNode";
import { ContainerState } from "./ContainerState";
import { Queue$1 } from "./Queue$1";
import { DescriptionTreeAction } from "./DescriptionTreeAction";
import { List$1 } from "./List$1";
import { DescriptionTreeBuilder } from "./DescriptionTreeBuilder";
import { DescriptionTreeReconciler } from "./DescriptionTreeReconciler";
import { NotImplementedException } from "./NotImplementedException";
import { DescriptionRef } from "./DescriptionRef";
import { ComponentRendererAdapter } from "./ComponentRendererAdapter";
var ComponentRenderer = /** @class */ (function (_super) {
    tslib_1.__extends(ComponentRenderer, _super);
    function ComponentRenderer() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this._context = null;
        _this.f = null;
        _this.z = new Dictionary$2(Base.$, Description.$, 0);
        _this.y = new Dictionary$2(Base.$, DescriptionTreeNode.$, 0);
        _this.ab = new Dictionary$2(Base.$, ContainerState.$, 0);
        _this.aa = new Dictionary$2(Base.$, Queue$1.$.specialize(Queue$1.$.specialize(DescriptionTreeAction.$)), 0);
        _this.x = new Dictionary$2(Base.$, Boolean_$type, 0);
        var a = _this.c();
        _this.b = a;
        _this.context = new TypeDescriptionContext(a, ComponentRenderer.platform);
        _this.f = new TypeDescriptionPropretyTransforms();
        return _this;
    }
    Object.defineProperty(ComponentRenderer.prototype, "context", {
        get: function () {
            return this._context;
        },
        set: function (a) {
            this._context = a;
        },
        enumerable: true,
        configurable: true
    });
    ComponentRenderer.prototype.c = function () {
        return (new ComponentRendererAdapter());
    };
    ComponentRenderer.prototype.render = function (a, b) {
        var _this = this;
        if (!this.ab.containsKey(b)) {
            this.ab.item(b, new ContainerState());
            this.ab.item(b).l = b;
        }
        this.z.item(b, a);
        this.bb(b, function () { return _this.bi(b); });
    };
    ComponentRenderer.prototype.bb = function (a, b) {
        this.b.onUIThread(a, b);
    };
    ComponentRenderer.prototype.bi = function (a) {
        if (this.z.containsKey(a)) {
            var b = this.z.item(a);
            this.z.removeItem(a);
            var c = DescriptionTreeBuilder.a(this.context, b);
            var d = null;
            if (this.y.containsKey(a)) {
                d = this.y.item(a);
            }
            var e = DescriptionTreeReconciler.b(d, c);
            this.y.item(a, c);
            this.bj(a, e);
        }
    };
    ComponentRenderer.prototype.provideRefValue = function (a, b, c) {
        var _this = this;
        this.bb(a, function () {
            if (!_this.ab.containsKey(a)) {
                _this.ab.addItem(a, ((function () {
                    var $ret = new ContainerState();
                    $ret.l = a;
                    return $ret;
                })()));
            }
            var d = _this.ab.item(a);
            d.q(b, c, true);
        });
    };
    ComponentRenderer.prototype.removeRefValue = function (a, b) {
        var _this = this;
        this.bb(a, function () {
            if (!_this.ab.containsKey(a)) {
                _this.ab.addItem(a, ((function () {
                    var $ret = new ContainerState();
                    $ret.l = a;
                    return $ret;
                })()));
            }
            var c = _this.ab.item(a);
            c.s(b, true);
        });
    };
    ComponentRenderer.prototype.clearRefValues = function (a) {
        var _this = this;
        this.bb(a, function () {
            if (!_this.ab.containsKey(a)) {
                _this.ab.addItem(a, ((function () {
                    var $ret = new ContainerState();
                    $ret.l = a;
                    return $ret;
                })()));
            }
            var b = _this.ab.item(a);
            b.p();
        });
    };
    ComponentRenderer.prototype.bj = function (a, b) {
        var c = this.ab.item(a);
        this.bc(a, c, b);
    };
    ComponentRenderer.prototype.bl = function (a, b) {
        this.a9(a, b, true);
    };
    ComponentRenderer.prototype.bo = function (a, b) {
        this.a9(a, b, false);
    };
    ComponentRenderer.prototype.a9 = function (a, b, c) {
        if (!c) {
            if (this.x.containsKey(a) && this.x.item(a)) {
                return;
            }
        }
        this.x.item(a, true);
        var d = true;
        while (d && this.aa.item(a).count > 0) {
            while (this.aa.item(a).count > 0 && this.aa.item(a).c().count < 1) {
                this.aa.item(a).b();
            }
            if (this.aa.item(a).count < 1) {
                break;
            }
            var e = this.aa.item(a).c();
            while (e.count > 0) {
                var f = e.b();
                d = this.l(a, b, f);
                if (!d) {
                    break;
                }
            }
        }
        if (d) {
            this.x.item(a, false);
        }
    };
    ComponentRenderer.prototype.bc = function (a, b, c) {
        var d = new Queue$1(DescriptionTreeAction.$);
        for (var e = 0; e < c.count; e++) {
            d.g(c._inner[e]);
        }
        if (!this.aa.containsKey(a)) {
            this.aa.addItem(a, new Queue$1(Queue$1.$.specialize(DescriptionTreeAction.$)));
        }
        this.aa.item(a).g(d);
        this.bo(a, b);
    };
    ComponentRenderer.prototype.bd = function (a, b, c) {
        for (var d = 0; d < c.count; d++) {
            this.l(a, b, c._inner[d]);
        }
    };
    ComponentRenderer.prototype.l = function (a, b, c) {
        switch (c.a) {
            case 5: return this.m(a, b, c);
            case 2: return this.n(a, b, c);
            case 4: return this.o(a, b, c);
            case 3: return this.p(a, b, c);
            case 1: return this.q(a, b, c);
            case 0: return this.r(a, b, c);
        }
        return true;
    };
    ComponentRenderer.prototype.r = function (a, b, c) {
        var d = this.ay(a, b, c);
        this.w(a, b, c, d);
        return true;
    };
    ComponentRenderer.prototype.w = function (a, b, c, d) {
        var e = this.av(c, a, b);
        e = this.f.b(ComponentRenderer.platform, e, c.c);
        if (c.c != null && c.c.a == 9) {
            this.v(a, b, c.h(ComponentRenderer.platform), c.c, e, d);
        }
        else {
            this.bm(a, b, c.h(ComponentRenderer.platform), e, c.g, d);
        }
        return true;
    };
    ComponentRenderer.prototype.v = function (a, b, c, d, e, f) {
        this.b.setOrUpdateCollectionOnTarget(a, c, d, this.context, f, e);
        return true;
    };
    ComponentRenderer.prototype.av = function (a, b, c) {
        if (a.c == null) {
            return a.f;
        }
        switch (a.c.a) {
            case 10: return this.ae(a, b, c);
            case 7: return this.af(a);
            case 4: return this.ag(a);
            case 6: return this.ah(a);
            case 9: return this.ae(a, b, c);
            case 5: return this.ai(a);
            case 16: return this.aj(a);
            case 20: return this.az(a, c);
            case 15: return null;
            case 3: return this.ak(a);
            case 19: return this.a0(a, c);
            case 8: return this.am(a, b, c);
            case 13: return null;
            case 18: return this.a1(a, c);
            case 1: return this.an(a);
            case 11: return this.ao(a);
            case 14: return this.ap(a);
            case 12: return this.aq(a);
            case 2: return this.ar(a);
            case 21: return this.as(a);
            case 17: return this.at(a);
            case 0: return null;
        }
        return null;
    };
    ComponentRenderer.prototype.at = function (a) {
        throw new NotImplementedException(0);
    };
    ComponentRenderer.prototype.as = function (a) {
        return a.f;
    };
    ComponentRenderer.prototype.ar = function (a) {
        return a.f != null ? a.f.toString() : null;
    };
    ComponentRenderer.prototype.aq = function (a) {
        var b = a.f;
        var width_ = b.width;
        var height_ = b.height;
        var c = { width: width_, height: height_ };
        return c;
    };
    ComponentRenderer.prototype.ap = function (a) {
        var b = a.f;
        var top_ = b.top;
        var left_ = b.left;
        var width_ = b.width;
        var height_ = b.height;
        var c = { left: left_, top: top_, width: width_, height: height_ };
        return c;
    };
    ComponentRenderer.prototype.ao = function (a) {
        var b = a.f;
        var x_ = b.x;
        var y_ = b.y;
        var c = { x: x_, y: y_ };
        return c;
    };
    ComponentRenderer.prototype.an = function (a) {
        return a.f;
    };
    ComponentRenderer.prototype.a1 = function (a, b) {
        return a.f == null ? null : b.a(a.f);
    };
    ComponentRenderer.prototype.am = function (a, b, c) {
        var d = this.au(a.c.l, a.f, b, c);
        return d;
    };
    ComponentRenderer.prototype.a0 = function (a, b) {
        return a.f == null ? null : b.a(a.f);
    };
    ComponentRenderer.prototype.ak = function (a) {
        return a.f;
    };
    ComponentRenderer.prototype.az = function (a, b) {
        return a.f == null ? null : b.a(a.f);
    };
    ComponentRenderer.prototype.a3 = function (a) {
        return a;
    };
    ComponentRenderer.prototype.aj = function (a) {
        var b = a.f;
        if (b == null) {
            return null;
        }
        var c = new Array(b.length);
        for (var d = 0; d < b.length; d++) {
            c[d] = this.a3(b[d]);
        }
        return c;
    };
    ComponentRenderer.prototype.ai = function (a) {
        return this.a3(a.f);
    };
    ComponentRenderer.prototype.ah = function (a) {
        var b = a.f;
        if (b == null) {
            return null;
        }
        var c = new Array(b.length);
        for (var d = 0; d < b.length; d++) {
            c[d] = this.a2(b[d]);
        }
        return c;
    };
    ComponentRenderer.prototype.a2 = function (a) {
        return a;
    };
    ComponentRenderer.prototype.ag = function (a) {
        return this.a2(a.f);
    };
    ComponentRenderer.prototype.af = function (a) {
        return a.f;
    };
    ComponentRenderer.prototype.ae = function (a, b, c) {
        if (a.f == null) {
            return null;
        }
        var d = a.f;
        var e = new List$1(Base.$, 0);
        var f = a.c.l;
        if (a.c.a == 9) {
            f = a.c.f;
        }
        for (var g = 0; g < d.length; g++) {
            var h = f;
            if (typeCast(DescriptionTreeNode.$, d[g]) !== null) {
                var i = d[g];
                if (i.b("Type")) {
                    h = i.a("Type").b;
                }
            }
            var j = this.au(h, d[g], b, c);
            e.add1(j);
        }
        return e.toArray();
    };
    ComponentRenderer.prototype.bm = function (a, b, c, d, e, f) {
        var _this = this;
        if (typeCast(DescriptionRef.$, d) !== null) {
            var g = false;
            if (e != null && typeof e === 'string') {
                if (this.j(b, e)) {
                    var h = this.a(b, e);
                    if (Base.equalsStatic(h, d)) {
                        g = true;
                    }
                    if (!g) {
                        this.a7(b, h);
                    }
                }
            }
            if (!g) {
                this.ba(b, d);
            }
            if (this.k(b, d)) {
                d = this.ax(b, d);
                this.bn(f, c, d);
            }
            else {
                var i = null;
                i = function (j, k) {
                    var l = _this.ax(b, j);
                    _this.bn(f, c, l);
                };
                var j = d;
                j.valueChanged = delegateCombine(j.valueChanged, i);
            }
            return;
        }
        this.bn(f, c, d);
    };
    ComponentRenderer.prototype.ba = function (a, b) {
        b.b++;
    };
    ComponentRenderer.prototype.a7 = function (a, b) {
        b.b--;
    };
    ComponentRenderer.prototype.k = function (a, b) {
        return a.d(b);
    };
    ComponentRenderer.prototype.j = function (a, b) {
        return a.c(b);
    };
    ComponentRenderer.prototype.a = function (a, b) {
        return a.b(b);
    };
    ComponentRenderer.prototype.ax = function (a, b) {
        return a.n(b);
    };
    ComponentRenderer.prototype.ay = function (a, b, c) {
        return b.m(c.b.f);
    };
    ComponentRenderer.prototype.q = function (a, b, c) {
        var d = this.ay(a, b, c);
        this.u(a, b, c, d);
        return true;
    };
    ComponentRenderer.prototype.u = function (a, b, c, d) {
        this.b.resetPropertyOnTarget(a, c.h(ComponentRenderer.platform), c.c, d);
        return true;
    };
    ComponentRenderer.prototype.p = function (a, b, c) {
        if (c.b == null) {
            return this.t(a, b, c);
        }
        else {
            var d = this.ay(a, b, c);
            this.bk(a, b, c, d);
        }
        return true;
    };
    ComponentRenderer.prototype.bk = function (a, b, c, d) {
        var e = c.h(ComponentRenderer.platform);
        var f = this.au(c.c.f, c.f, a, b);
        this.b.replaceItemInCollection(e, c.c, d, c.d, f);
    };
    ComponentRenderer.prototype.t = function (a, b, c) {
        var _this = this;
        var d = this.b.getRootObject(a);
        if (d != null) {
            this.a8(a, d, b);
        }
        if (!(typeCast(DescriptionTreeNode.$, c.f) !== null)) {
            throw new NotImplementedException(0);
        }
        var e = c.f;
        var f = false;
        this.b.replaceRootItem(a, e.h, this.context, function (g) {
            var h = _this.b.getRootObject(a);
            b.o(e.f, h, e);
            var i = _this.ac(e);
            _this.bd(a, b, i);
            f = true;
            if (g) {
                _this.bl(a, b);
            }
        });
        return f;
    };
    ComponentRenderer.prototype.a8 = function (a, b, c) {
        var d = this.ad(a, b, c);
        if (d != -1) {
            c.r(d);
        }
    };
    ComponentRenderer.prototype.ad = function (a, b, c) {
        return c.k(b);
    };
    ComponentRenderer.prototype.o = function (a, b, c) {
        if (c.b == null) {
            return this.s(a, b, c);
        }
        else {
            var d = this.ay(a, b, c);
            this.bf(a, b, c, d);
        }
        return true;
    };
    ComponentRenderer.prototype.bf = function (a, b, c, d) {
        var e = c.h(ComponentRenderer.platform);
        this.b.removeItemFromCollection(e, c.c, d, c.e);
    };
    ComponentRenderer.prototype.s = function (a, b, c) {
        var _this = this;
        var d = this.b.getRootObject(a);
        if (d != null) {
            this.a8(a, d, b);
        }
        var e = false;
        this.b.removeRootItem(a, this.context, function (f) {
            e = true;
            if (f) {
                _this.bl(a, b);
            }
        });
        return e;
    };
    ComponentRenderer.prototype.n = function (a, b, c) {
        if (c.b == null) {
            return this.g(a, b, c);
        }
        else {
            var d = this.ay(a, b, c);
            this.a4(a, b, c, d);
        }
        return true;
    };
    ComponentRenderer.prototype.g = function (a, b, c) {
        var _this = this;
        if (!(typeCast(DescriptionTreeNode.$, c.f) !== null)) {
            throw new NotImplementedException(0);
        }
        var d = c.f;
        var e = false;
        this.b.replaceRootItem(a, d.h, this.context, function (f) {
            var g = _this.b.getRootObject(a);
            b.o(d.f, g, d);
            var h = _this.ac(d);
            _this.bd(a, b, h);
            e = true;
            if (f) {
                _this.bl(a, b);
            }
        });
        return e;
    };
    ComponentRenderer.prototype.a4 = function (a, b, c, d) {
        var e = c.h(ComponentRenderer.platform);
        var f = c.c.f;
        if (typeCast(DescriptionTreeNode.$, c.f) !== null) {
            var g = c.f;
            if (g.b("Type")) {
                f = g.a("Type").b;
            }
        }
        var h = this.au(f, c.f, a, b);
        this.b.addItemToCollection(e, c.c, d, c.d, h);
    };
    ComponentRenderer.prototype.au = function (a, b, c, d) {
        var e = this.context.j(a, c);
        if (b == null) {
            return e;
        }
        if (typeCast(DescriptionTreeNode.$, b) !== null) {
            var f = b;
            d.o(f.f, e, f);
            var g = this.ac(f);
            this.bd(c, d, g);
        }
        if (typeof b === 'string' && a.toLowerCase() != "string") {
            return this.al(a, b);
        }
        return e;
    };
    ComponentRenderer.prototype.al = function (a, b) {
        return this.context.f(a, b);
    };
    ComponentRenderer.prototype.ac = function (a) {
        var b = a.e();
        var c = new List$1(DescriptionTreeAction.$, 0);
        for (var d = 0; d < b.count; d++) {
            var e = new DescriptionTreeAction();
            e.a = 0;
            e.i = b._inner[d].c;
            e.c = b._inner[d].a;
            e.b = a;
            e.f = b._inner[d].b;
            c.add(e);
        }
        return c;
    };
    ComponentRenderer.prototype.m = function (a, b, c) {
        if (c.b == null) {
            return this.i(a, b, c);
        }
        else {
            var d = this.ay(a, b, c);
            this.a5(a, b, c, d);
        }
        return true;
    };
    ComponentRenderer.prototype.a5 = function (a, b, c, d) {
        var e = c.h(ComponentRenderer.platform);
        this.b.clearCollection(d, e, c.c);
    };
    ComponentRenderer.prototype.bn = function (a, b, c) {
        this.b.setPropertyValue(a, b, c);
    };
    ComponentRenderer.prototype.aw = function (a, b) {
        return this.b.getPropertyValue(a, b);
    };
    ComponentRenderer.prototype.i = function (a, b, c) {
        return this.h(a, b);
    };
    ComponentRenderer.prototype.h = function (a, b) {
        var _this = this;
        var c = this.b.getRootObject(a);
        if (c != null) {
            this.a8(a, c, b);
        }
        var d = false;
        this.b.clearContainer(a, this.context, function (e) {
            d = true;
            if (e) {
                _this.bl(a, b);
            }
        });
        return d;
    };
    ComponentRenderer.$t = markType(ComponentRenderer, 'ComponentRenderer');
    ComponentRenderer.platform = 3;
    return ComponentRenderer;
}(Base));
export { ComponentRenderer };
