// igniteui-angular-core/ES5/DescriptionSerializer.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Base, NotSupportedException, typeCast, typeGetValue, markType } from "./type";
import { TypeDescriptionContext } from "./TypeDescriptionContext";
import { DescriptionTreeNode } from "./DescriptionTreeNode";
import { DescriptionSerializerBuilder } from "./DescriptionSerializerBuilder";
import { DescriptionTreeBuilder } from "./DescriptionTreeBuilder";
import { DescriptionResult } from "./DescriptionResult";
import { JsonDictionaryParser } from "./JsonDictionaryParser";
import { JsonDictionaryObject } from "./JsonDictionaryObject";
import { JsonDictionaryArray } from "./JsonDictionaryArray";
import { NotImplementedException } from "./NotImplementedException";
import { stringReplace } from "./string";
var DescriptionSerializer = /** @class */ (function (_super) {
    tslib_1.__extends(DescriptionSerializer, _super);
    function DescriptionSerializer() {
        return _super.call(this) || this;
    }
    DescriptionSerializer.prototype.f = function (a, b) {
        var c = DescriptionTreeBuilder.a(a, b);
        var d = new DescriptionSerializerBuilder();
        this.g(a, c, d);
        return d.toString();
    };
    DescriptionSerializer.prototype.c = function (a, b) {
        var c = new JsonDictionaryParser();
        var d = c.c(b);
        if (!(typeCast(JsonDictionaryObject.$, d) !== null)) {
            throw new NotSupportedException(1, "expected an object at the root of the json");
        }
        var e = d;
        return this.b(a, e);
    };
    DescriptionSerializer.prototype.b = function (a, b, c) {
        if (c === void 0) { c = 8; }
        if (!b.c("type")) {
            throw new NotSupportedException(1, "expected type to be defined for the json object");
        }
        var d = new DescriptionResult();
        var e = b.item("type").b;
        var f = a.i(e);
        d.result = f;
        if (f == null) {
            d.d("couldn't find registered description for type: " + e);
            return d;
        }
        var g = b.a();
        for (var h = 0; h < g.length; h++) {
            var i = g[h];
            var j = a.b(e, TypeDescriptionContext.l(i));
            if (j == null) {
                continue;
            }
            var k = b.item(i);
            if (typeCast(JsonDictionaryObject.$, k) !== null) {
                var l = k;
                var m = this.b(a, l, j.a);
                d.e(m);
                a.o(f, i, m.result);
            }
            else if (typeCast(JsonDictionaryArray.$, k) !== null) {
                var n = k;
                var o = "string";
                if (j.a == 10) {
                    o = j.k;
                }
                if (j.a == 9) {
                    o = j.f;
                }
                var p = this.a(d, a, n, o);
                a.o(f, i, p);
            }
            else {
                var q = j.k;
                if (q == null) {
                    q = "String";
                }
                var r = k;
                var s = this.d(d, a, r, q);
                a.o(f, i, s);
            }
        }
        return d;
    };
    DescriptionSerializer.prototype.a = function (a, b, c, d) {
        var e = new Array(c.a.length);
        for (var f = 0; f < c.a.length; f++) {
            var g = c.a[f];
            if (typeCast(JsonDictionaryObject.$, g) !== null) {
                var h = g;
                var i = this.b(b, h);
                a.e(i);
                e[f] = i.result;
            }
            else if (typeCast(JsonDictionaryArray.$, g) !== null) {
                var j = g;
                var k = this.a(a, b, j, d);
                e[f] = k;
            }
            else {
                var l = g;
                var m = this.d(a, b, l, d);
                e[f] = m;
            }
        }
        return e;
    };
    DescriptionSerializer.prototype.d = function (a, b, c, d) {
        switch (c.a) {
            case 1: return c.b;
            case 3: return c.b;
            case 0: switch (d.toLowerCase()) {
                case "int": return typeGetValue(c.b);
                case "double": return c.b;
                case "timespan": return c.b;
                case "short": return typeGetValue(c.b);
                case "long": return typeGetValue(c.b);
                case "float": return c.b;
                case "byte": return typeGetValue(c.b);
                default:
                    a.d("unexpected target type for number: " + d);
                    return c.b;
            }
            case 2:
                if (d.toLowerCase() == "datetime") {
                    var val_ = c.b;
                    var e = (new Date(val_));
                    return e;
                }
                return c.b;
            default:
                a.d("unexpected value type");
                return c.b;
        }
    };
    DescriptionSerializer.prototype.g = function (a, b, c) {
        c.f("{");
        c.i();
        c.f("\"type\": \"" + b.h + "\",");
        var d = b.e();
        for (var e = 0; e < d.count; e++) {
            var f = d._inner[e];
            this.h(a, f, c);
        }
        c.h();
        c.f("}");
    };
    DescriptionSerializer.prototype.e = function (a) {
        if (a == null) {
            return null;
        }
        return a.substr(0, 1).toLowerCase() + a.substr(1);
    };
    DescriptionSerializer.prototype.h = function (a, b, c) {
        if (b.c.toLowerCase() == "type") {
            return;
        }
        c.e("\"" + this.e(b.c) + "\": ");
        this.j(a, b, b.b, c);
        c.f(",");
    };
    DescriptionSerializer.prototype.j = function (a, b, c, d) {
        var e = 2;
        if (b.a != null) {
            e = b.a.a;
        }
        switch (e) {
            case 6:
            case 16:
            case 10:
                {
                    d.e("[");
                    var f = b.b;
                    var g = b.a.l;
                    if (b.a.a == 6 || b.a.a == 16) {
                        g = "String";
                    }
                    for (var h = 0; h < f.length; h++) {
                        if (h > 0) {
                            d.e(", ");
                        }
                        var i = g;
                        if (typeCast(DescriptionTreeNode.$, f[h]) !== null) {
                            this.g(a, f[h], d);
                        }
                        else {
                            this.i(a, g, f[h], d);
                        }
                    }
                    d.e("]");
                }
                break;
            case 7:
                this.i(a, "Boolean", c, d);
                break;
            case 9:
                {
                    d.e("[");
                    var j = b.b;
                    var k = b.a.f;
                    for (var l = 0; l < j.length; l++) {
                        if (l > 0) {
                            d.e(", ");
                        }
                        var m = k;
                        if (typeCast(DescriptionTreeNode.$, j[l]) !== null) {
                            this.g(a, j[l], d);
                        }
                        else {
                            this.i(a, k, j[l], d);
                        }
                    }
                    d.e("]");
                }
                break;
            case 4:
            case 18:
            case 19:
            case 20:
            case 5:
            case 2:
                this.i(a, "String", c, d);
                break;
            case 3:
                this.i(a, "DateTime", c, d);
                break;
            case 8:
                if (b.a.k.toLowerCase() == "string") {
                    this.i(a, "String", c, d);
                }
                else {
                    this.g(a, c, d);
                }
                break;
            case 1:
                this.i(a, "Double", c, d);
                break;
            case 14:
            case 12:
            case 11:
                this.g(a, c, d);
                break;
            case 21:
                this.i(a, "TimeSpan", c, d);
                break;
            case 17:
            case 0:
            case 15:
            case 13: throw new NotImplementedException(0);
        }
    };
    DescriptionSerializer.prototype.i = function (a, b, c, d) {
        switch (b.toLowerCase()) {
            case "string":
                d.e("\"" + stringReplace(c, "\"", "\\\"") + "\"");
                break;
            case "number":
            case "double":
            case "int":
            case "float":
            case "long":
            case "decimal":
                d.e(c.toString());
                break;
            case "timespan":
                d.e(c.toString());
                break;
            case "datetime":
                var v_ = c;
                d.e("\"" + (v_.toJSON()) + "\"");
                break;
        }
    };
    DescriptionSerializer.$t = markType(DescriptionSerializer, 'DescriptionSerializer');
    return DescriptionSerializer;
}(Base));
export { DescriptionSerializer };
