// igniteui-angular-core/ES5/FinancialPriceSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { FinancialSeriesDescription } from "./FinancialSeriesDescription";
import { markType } from "./type";
var FinancialPriceSeriesDescription = /** @class */ (function (_super) {
    tslib_1.__extends(FinancialPriceSeriesDescription, _super);
    function FinancialPriceSeriesDescription() {
        var _this = _super.call(this) || this;
        _this.cm = null;
        _this.cl = null;
        _this.ci = null;
        _this.ce = 0;
        _this.cg = 0;
        _this.cj = null;
        _this.ck = null;
        return _this;
    }
    FinancialPriceSeriesDescription.prototype.get_type = function () {
        return "FinancialPriceSeries";
    };
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "trendLineType", {
        get: function () {
            return this.cm;
        },
        set: function (a) {
            this.cm = a;
            this.e("TrendLineType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "trendLineBrush", {
        get: function () {
            return this.cl;
        },
        set: function (a) {
            this.cl = a;
            this.e("TrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "actualTrendLineBrush", {
        get: function () {
            return this.ci;
        },
        set: function (a) {
            this.ci = a;
            this.e("ActualTrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "trendLineThickness", {
        get: function () {
            return this.ce;
        },
        set: function (a) {
            this.ce = a;
            this.e("TrendLineThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "trendLinePeriod", {
        get: function () {
            return this.cg;
        },
        set: function (a) {
            this.cg = a;
            this.e("TrendLinePeriod");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "displayType", {
        get: function () {
            return this.cj;
        },
        set: function (a) {
            this.cj = a;
            this.e("DisplayType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialPriceSeriesDescription.prototype, "negativeOutline", {
        get: function () {
            return this.ck;
        },
        set: function (a) {
            this.ck = a;
            this.e("NegativeOutline");
        },
        enumerable: true,
        configurable: true
    });
    FinancialPriceSeriesDescription.$t = markType(FinancialPriceSeriesDescription, 'FinancialPriceSeriesDescription', FinancialSeriesDescription.$);
    return FinancialPriceSeriesDescription;
}(FinancialSeriesDescription));
export { FinancialPriceSeriesDescription };
