// igniteui-angular-core/ES5/FinancialSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
var FinancialSeriesDescription = /** @class */ (function (_super) {
    tslib_1.__extends(FinancialSeriesDescription, _super);
    function FinancialSeriesDescription() {
        var _this = _super.call(this) || this;
        _this.bw = null;
        _this.b0 = null;
        _this.b1 = null;
        _this.bx = null;
        _this.bu = null;
        _this.bv = null;
        _this.bt = null;
        _this.bz = null;
        _this.bp = false;
        _this.by = null;
        _this.bq = false;
        _this._AssigningCategoryStyleRef = null;
        _this._TypicalRef = null;
        _this._TypicalBasedOnRef = null;
        return _this;
    }
    FinancialSeriesDescription.prototype.get_type = function () {
        return "FinancialSeries";
    };
    Object.defineProperty(FinancialSeriesDescription.prototype, "negativeBrush", {
        get: function () {
            return this.bw;
        },
        set: function (a) {
            this.bw = a;
            this.e("NegativeBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "xAxisRef", {
        get: function () {
            return this.b0;
        },
        set: function (a) {
            this.b0 = a;
            this.e("XAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "yAxisRef", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            this.b1 = a;
            this.e("YAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "openMemberPath", {
        get: function () {
            return this.bx;
        },
        set: function (a) {
            this.bx = a;
            this.e("OpenMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "highMemberPath", {
        get: function () {
            return this.bu;
        },
        set: function (a) {
            this.bu = a;
            this.e("HighMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "lowMemberPath", {
        get: function () {
            return this.bv;
        },
        set: function (a) {
            this.bv = a;
            this.e("LowMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "closeMemberPath", {
        get: function () {
            return this.bt;
        },
        set: function (a) {
            this.bt = a;
            this.e("CloseMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "volumeMemberPath", {
        get: function () {
            return this.bz;
        },
        set: function (a) {
            this.bz = a;
            this.e("VolumeMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "isCustomCategoryStyleAllowed", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            this.bp = a;
            this.e("IsCustomCategoryStyleAllowed");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "transitionInMode", {
        get: function () {
            return this.by;
        },
        set: function (a) {
            this.by = a;
            this.e("TransitionInMode");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "isTransitionInEnabled", {
        get: function () {
            return this.bq;
        },
        set: function (a) {
            this.bq = a;
            this.e("IsTransitionInEnabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "AssigningCategoryStyleRef", {
        get: function () {
            return this._AssigningCategoryStyleRef;
        },
        set: function (a) {
            this._AssigningCategoryStyleRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "TypicalRef", {
        get: function () {
            return this._TypicalRef;
        },
        set: function (a) {
            this._TypicalRef = a;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FinancialSeriesDescription.prototype, "TypicalBasedOnRef", {
        get: function () {
            return this._TypicalBasedOnRef;
        },
        set: function (a) {
            this._TypicalBasedOnRef = a;
        },
        enumerable: true,
        configurable: true
    });
    FinancialSeriesDescription.$t = markType(FinancialSeriesDescription, 'FinancialSeriesDescription', SeriesDescription.$);
    return FinancialSeriesDescription;
}(SeriesDescription));
export { FinancialSeriesDescription };
