// igniteui-angular-core/ES5/HeaderDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
var HeaderDescription = /** @class */ (function (_super) {
    tslib_1.__extends(HeaderDescription, _super);
    function HeaderDescription() {
        var _this = _super.call(this) || this;
        _this.av = 0;
        _this.ax = 0;
        _this.aw = 0;
        _this.au = 0;
        return _this;
    }
    HeaderDescription.prototype.get_type = function () {
        return "Header";
    };
    Object.defineProperty(HeaderDescription.prototype, "paddingLeft", {
        get: function () {
            return this.av;
        },
        set: function (a) {
            this.av = a;
            this.e("PaddingLeft");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HeaderDescription.prototype, "paddingTop", {
        get: function () {
            return this.ax;
        },
        set: function (a) {
            this.ax = a;
            this.e("PaddingTop");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HeaderDescription.prototype, "paddingRight", {
        get: function () {
            return this.aw;
        },
        set: function (a) {
            this.aw = a;
            this.e("PaddingRight");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HeaderDescription.prototype, "paddingBottom", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("PaddingBottom");
        },
        enumerable: true,
        configurable: true
    });
    HeaderDescription.$t = markType(HeaderDescription, 'HeaderDescription', DefinitionBaseDescription.$);
    return HeaderDescription;
}(DefinitionBaseDescription));
export { HeaderDescription };
