// igniteui-angular-core/ES5/HighDensityScatterSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
var HighDensityScatterSeriesDescription = /** @class */ (function (_super) {
    tslib_1.__extends(HighDensityScatterSeriesDescription, _super);
    function HighDensityScatterSeriesDescription() {
        var _this = _super.call(this) || this;
        _this.b3 = null;
        _this.b5 = null;
        _this.b4 = null;
        _this.b6 = null;
        _this.bq = false;
        _this.bp = false;
        _this.bu = 0;
        _this.bt = 0;
        _this.b2 = null;
        _this.b1 = null;
        _this.bx = 0;
        _this.by = 0;
        return _this;
    }
    HighDensityScatterSeriesDescription.prototype.get_type = function () {
        return "HighDensityScatterSeries";
    };
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "xAxisRef", {
        get: function () {
            return this.b3;
        },
        set: function (a) {
            this.b3 = a;
            this.e("XAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "yAxisRef", {
        get: function () {
            return this.b5;
        },
        set: function (a) {
            this.b5 = a;
            this.e("YAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "xMemberPath", {
        get: function () {
            return this.b4;
        },
        set: function (a) {
            this.b4 = a;
            this.e("XMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "yMemberPath", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("YMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "useBruteForce", {
        get: function () {
            return this.bq;
        },
        set: function (a) {
            this.bq = a;
            this.e("UseBruteForce");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "progressiveLoad", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            this.bp = a;
            this.e("ProgressiveLoad");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "heatMinimum", {
        get: function () {
            return this.bu;
        },
        set: function (a) {
            this.bu = a;
            this.e("HeatMinimum");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "heatMaximum", {
        get: function () {
            return this.bt;
        },
        set: function (a) {
            this.bt = a;
            this.e("HeatMaximum");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "heatMinimumColor", {
        get: function () {
            return this.b2;
        },
        set: function (a) {
            this.b2 = a;
            this.e("HeatMinimumColor");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "heatMaximumColor", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            this.b1 = a;
            this.e("HeatMaximumColor");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "pointExtent", {
        get: function () {
            return this.bx;
        },
        set: function (a) {
            this.bx = a;
            this.e("PointExtent");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighDensityScatterSeriesDescription.prototype, "progressiveStatus", {
        get: function () {
            return this.by;
        },
        set: function (a) {
            this.by = a;
            this.e("ProgressiveStatus");
        },
        enumerable: true,
        configurable: true
    });
    HighDensityScatterSeriesDescription.$t = markType(HighDensityScatterSeriesDescription, 'HighDensityScatterSeriesDescription', SeriesDescription.$);
    return HighDensityScatterSeriesDescription;
}(SeriesDescription));
export { HighDensityScatterSeriesDescription };
