// igniteui-angular-core/ES5/JsonDictionaryObject.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { JsonDictionaryItem } from "./JsonDictionaryItem";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
import { String_$type, fromEnum, markType } from "./type";
var JsonDictionaryObject = /** @class */ (function (_super) {
    tslib_1.__extends(JsonDictionaryObject, _super);
    function JsonDictionaryObject() {
        var _this = _super.call(this) || this;
        _this.d = new Dictionary$2(String_$type, JsonDictionaryItem.$, 0);
        return _this;
    }
    JsonDictionaryObject.prototype.e = function (a, b) {
        this.d.item(a, b);
    };
    JsonDictionaryObject.prototype.g = function (a) {
        this.d.removeItem(a);
    };
    JsonDictionaryObject.prototype.f = function () {
        this.d.clear();
    };
    JsonDictionaryObject.prototype.item = function (a, b) {
        if (arguments.length === 2) {
            this.d.item(a, b);
            return b;
        }
        else {
            return this.d.item(a);
        }
    };
    JsonDictionaryObject.prototype.c = function (a) {
        return this.d.containsKey(a);
    };
    JsonDictionaryObject.prototype.a = function () {
        var a = new List$1(String_$type, 0);
        try {
            for (var _a = tslib_1.__values(fromEnum(this.d.keys)), _b = _a.next(); !_b.done; _b = _a.next()) {
                var b = _b.value;
                a.add(b);
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (_b && !_b.done && (_c = _a.return)) _c.call(_a);
            }
            finally { if (e_1) throw e_1.error; }
        }
        return a.toArray();
        var e_1, _c;
    };
    JsonDictionaryObject.$t = markType(JsonDictionaryObject, 'JsonDictionaryObject', JsonDictionaryItem.$);
    return JsonDictionaryObject;
}(JsonDictionaryItem));
export { JsonDictionaryObject };
