// igniteui-angular-core/ES5/NumericAxisBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { AxisDescription } from "./AxisDescription";
import { markType } from "./type";
var NumericAxisBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(NumericAxisBaseDescription, _super);
    function NumericAxisBaseDescription() {
        var _this = _super.call(this) || this;
        _this.c0 = 0;
        _this.cu = 0;
        _this.cx = 0;
        _this.cz = 0;
        _this.ct = 0;
        _this.cw = 0;
        _this.cy = 0;
        _this.cs = 0;
        _this.c1 = 0;
        _this.cv = 0;
        _this.c2 = 0;
        _this.co = false;
        _this.cn = false;
        _this.de = 0;
        _this.cm = false;
        return _this;
    }
    NumericAxisBaseDescription.prototype.get_type = function () {
        return "NumericAxisBase";
    };
    Object.defineProperty(NumericAxisBaseDescription.prototype, "minimumValue", {
        get: function () {
            return this.c0;
        },
        set: function (a) {
            this.c0 = a;
            this.e("MinimumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualMinimumValue", {
        get: function () {
            return this.cu;
        },
        set: function (a) {
            this.cu = a;
            this.e("ActualMinimumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualVisibleMinimumValue", {
        get: function () {
            return this.cx;
        },
        set: function (a) {
            this.cx = a;
            this.e("ActualVisibleMinimumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "maximumValue", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            this.cz = a;
            this.e("MaximumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualMaximumValue", {
        get: function () {
            return this.ct;
        },
        set: function (a) {
            this.ct = a;
            this.e("ActualMaximumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualVisibleMaximumValue", {
        get: function () {
            return this.cw;
        },
        set: function (a) {
            this.cw = a;
            this.e("ActualVisibleMaximumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "interval", {
        get: function () {
            return this.cy;
        },
        set: function (a) {
            this.cy = a;
            this.e("Interval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualInterval", {
        get: function () {
            return this.cs;
        },
        set: function (a) {
            this.cs = a;
            this.e("ActualInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "minorInterval", {
        get: function () {
            return this.c1;
        },
        set: function (a) {
            this.c1 = a;
            this.e("MinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualMinorInterval", {
        get: function () {
            return this.cv;
        },
        set: function (a) {
            this.cv = a;
            this.e("ActualMinorInterval");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "referenceValue", {
        get: function () {
            return this.c2;
        },
        set: function (a) {
            this.c2 = a;
            this.e("ReferenceValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "isLogarithmic", {
        get: function () {
            return this.co;
        },
        set: function (a) {
            this.co = a;
            this.e("IsLogarithmic");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "actualIsLogarithmic", {
        get: function () {
            return this.cn;
        },
        set: function (a) {
            this.cn = a;
            this.e("ActualIsLogarithmic");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "logarithmBase", {
        get: function () {
            return this.de;
        },
        set: function (a) {
            this.de = a;
            this.e("LogarithmBase");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericAxisBaseDescription.prototype, "abbreviateLargeNumbers", {
        get: function () {
            return this.cm;
        },
        set: function (a) {
            this.cm = a;
            this.e("AbbreviateLargeNumbers");
        },
        enumerable: true,
        configurable: true
    });
    NumericAxisBaseDescription.$t = markType(NumericAxisBaseDescription, 'NumericAxisBaseDescription', AxisDescription.$);
    return NumericAxisBaseDescription;
}(AxisDescription));
export { NumericAxisBaseDescription };
