// igniteui-angular-core/ES5/RingSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
var RingSeriesBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(RingSeriesBaseDescription, _super);
    function RingSeriesBaseDescription() {
        var _this = _super.call(this) || this;
        _this.aq = null;
        _this.af = null;
        _this.al = null;
        _this.ag = null;
        _this.ak = null;
        _this.ah = null;
        _this.ai = null;
        _this.o = 0;
        _this.n = 0;
        _this.aj = null;
        _this.m = 0;
        _this.p = 0;
        _this.an = null;
        _this.am = null;
        _this.ad = null;
        _this.ae = null;
        _this.l = 0;
        _this.t = 0;
        _this.ao = null;
        _this.ap = null;
        _this.s = 0;
        _this.r = 0;
        _this.f = null;
        _this.g = null;
        _this.j = false;
        _this.q = 0;
        _this._PropertyUpdatedRef = null;
        return _this;
    }
    RingSeriesBaseDescription.prototype.get_type = function () {
        return "RingSeriesBase";
    };
    Object.defineProperty(RingSeriesBaseDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "valueMemberPath", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("ValueMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "labelMemberPath", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("LabelMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "legendLabelMemberPath", {
        get: function () {
            return this.al;
        },
        set: function (a) {
            this.al = a;
            this.e("LegendLabelMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "labelsPosition", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("LabelsPosition");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineVisibility", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("LeaderLineVisibility");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineFill", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("LeaderLineFill");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineStroke", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("LeaderLineStroke");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineStrokeThickness", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("LeaderLineStrokeThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineOpacity", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("LeaderLineOpacity");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineType", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("LeaderLineType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "leaderLineMargin", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("LeaderLineMargin");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "othersCategoryThreshold", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("OthersCategoryThreshold");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "othersCategoryType", {
        get: function () {
            return this.an;
        },
        set: function (a) {
            this.an = a;
            this.e("OthersCategoryType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "othersCategoryText", {
        get: function () {
            return this.am;
        },
        set: function (a) {
            this.am = a;
            this.e("OthersCategoryText");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "formatLabelRef", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("FormatLabelRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "formatLegendLabelRef", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("FormatLegendLabelRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "labelExtent", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("LabelExtent");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "startAngle", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("StartAngle");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "selectedSliceFill", {
        get: function () {
            return this.ao;
        },
        set: function (a) {
            this.ao = a;
            this.e("SelectedSliceFill");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "selectedSliceStroke", {
        get: function () {
            return this.ap;
        },
        set: function (a) {
            this.ap = a;
            this.e("SelectedSliceStroke");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "selectedSliceStrokeThickness", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("SelectedSliceStrokeThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "selectedSliceOpacity", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("SelectedSliceOpacity");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "brushes", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Brushes");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "outlines", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Outlines");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "isSurfaceInteractionDisabled", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("IsSurfaceInteractionDisabled");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "radiusFactor", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("RadiusFactor");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RingSeriesBaseDescription.prototype, "PropertyUpdatedRef", {
        get: function () {
            return this._PropertyUpdatedRef;
        },
        set: function (a) {
            this._PropertyUpdatedRef = a;
        },
        enumerable: true,
        configurable: true
    });
    RingSeriesBaseDescription.$t = markType(RingSeriesBaseDescription, 'RingSeriesBaseDescription', Description.$);
    return RingSeriesBaseDescription;
}(Description));
export { RingSeriesBaseDescription };
