// igniteui-angular-core/ES5/ScatterBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { MarkerSeriesDescription } from "./MarkerSeriesDescription";
import { markType } from "./type";
var ScatterBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(ScatterBaseDescription, _super);
    function ScatterBaseDescription() {
        var _this = _super.call(this) || this;
        _this.cf = null;
        _this.ch = null;
        _this.cg = null;
        _this.ci = null;
        _this.ce = null;
        _this.cd = null;
        _this.cb = null;
        _this.b3 = 0;
        _this.b6 = 0;
        _this.cc = null;
        _this.b7 = 0;
        _this.b5 = 0;
        return _this;
    }
    ScatterBaseDescription.prototype.get_type = function () {
        return "ScatterBase";
    };
    Object.defineProperty(ScatterBaseDescription.prototype, "xAxisRef", {
        get: function () {
            return this.cf;
        },
        set: function (a) {
            this.cf = a;
            this.e("XAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "yAxisRef", {
        get: function () {
            return this.ch;
        },
        set: function (a) {
            this.ch = a;
            this.e("YAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "xMemberPath", {
        get: function () {
            return this.cg;
        },
        set: function (a) {
            this.cg = a;
            this.e("XMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "yMemberPath", {
        get: function () {
            return this.ci;
        },
        set: function (a) {
            this.ci = a;
            this.e("YMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "trendLineType", {
        get: function () {
            return this.ce;
        },
        set: function (a) {
            this.ce = a;
            this.e("TrendLineType");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "trendLineBrush", {
        get: function () {
            return this.cd;
        },
        set: function (a) {
            this.cd = a;
            this.e("TrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "actualTrendLineBrush", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            this.cb = a;
            this.e("ActualTrendLineBrush");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "trendLineThickness", {
        get: function () {
            return this.b3;
        },
        set: function (a) {
            this.b3 = a;
            this.e("TrendLineThickness");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "trendLinePeriod", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("TrendLinePeriod");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "markerCollisionAvoidance", {
        get: function () {
            return this.cc;
        },
        set: function (a) {
            this.cc = a;
            this.e("MarkerCollisionAvoidance");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "trendLineZIndex", {
        get: function () {
            return this.b7;
        },
        set: function (a) {
            this.b7 = a;
            this.e("TrendLineZIndex");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScatterBaseDescription.prototype, "maximumMarkers", {
        get: function () {
            return this.b5;
        },
        set: function (a) {
            this.b5 = a;
            this.e("MaximumMarkers");
        },
        enumerable: true,
        configurable: true
    });
    ScatterBaseDescription.$t = markType(ScatterBaseDescription, 'ScatterBaseDescription', MarkerSeriesDescription.$);
    return ScatterBaseDescription;
}(MarkerSeriesDescription));
export { ScatterBaseDescription };
