// igniteui-angular-core/ES5/ScrollerView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Base, runOn, delegateCombine, markType, INotifyPropertyChanged_$type, PropertyChangedEventArgs, Point_$type } from "./type";
import { CanvasGestureDOMEventProxy } from "./CanvasGestureDOMEventProxy";
import { Brush } from "./Brush";
import { Rect } from "./Rect";
import { truncate, isNaN_ } from "./number";
import { ScrollerScrollingEventArgs } from "./ScrollerScrollingEventArgs";
var ScrollerView = /** @class */ (function (_super) {
    tslib_1.__extends(ScrollerView, _super);
    function ScrollerView(a) {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.x = null;
        _this.ab = null;
        _this.w = null;
        _this.a = null;
        _this.aa = null;
        _this.y = null;
        _this.z = null;
        _this.ac = null;
        _this.d = false;
        _this.c = null;
        _this.h = false;
        _this.g = false;
        _this.s = 0;
        _this.p = 0;
        _this.t = -1;
        _this.q = -1;
        _this.f = false;
        _this.e = false;
        _this.r = 0;
        _this.o = 0;
        _this.b = a;
        return _this;
    }
    ScrollerView.prototype.ay = function () {
        return ((function () {
            var $ret = new Brush();
            $ret.fill = "#666666";
            return $ret;
        })());
    };
    ScrollerView.prototype.ar = function (a) {
        if (a == null) {
            if (this.x != null) {
                this.x = null;
            }
            if (this.a != null) {
                this.a.ai = null;
                this.a.an();
            }
            return;
        }
        this.x = a;
        var b = Math.round(this.x.rootWrapper.width());
        var c = Math.round(this.x.rootWrapper.height());
        this.aa = this.x.rootWrapper;
        this.aa.setStyleProperty("position", "relative");
        this.ab = this.x.createElement("div");
        this.ab.setStyleProperty("position", "relative");
        this.ab.setStyleProperty("width", "100%");
        this.ab.setStyleProperty("height", "100%");
        this.ab.setStyleProperty("overflow", "hidden");
        this.aa.append(this.ab);
        this.z = this.x.createElement("div");
        this.z.setStyleProperty("position", "absolute");
        this.z.setStyleProperty("height", this.b.scrollbarSpan + "px");
        this.z.setStyleProperty("width", this.b.minScrollbarSize + "px");
        this.z.setStyleProperty("background-color", this.b.scrollbarBrush._fill);
        this.z.setStyleProperty("display", "none");
        this.z.setStyleProperty("border-radius", "5px");
        this.z.setStyleProperty("z-index", "10000");
        this.z.setStyleProperty("bottom", "0px");
        this.aa.append(this.z);
        this.ac = this.x.createElement("div");
        this.ac.setStyleProperty("position", "absolute");
        this.ac.setStyleProperty("width", this.b.scrollbarSpan + "px");
        this.ac.setStyleProperty("height", this.b.minScrollbarSize + "px");
        this.ac.setStyleProperty("background-color", this.b.scrollbarBrush._fill);
        this.ac.setStyleProperty("display", "none");
        this.ac.setStyleProperty("border-radius", "5px");
        this.ac.setStyleProperty("z-index", "10000");
        this.ac.setStyleProperty("right", "0px");
        this.aa.append(this.ac);
        this.w = this.ab.getNativeElement();
        this.a = new CanvasGestureDOMEventProxy(this.x.rootWrapper, this.x, true);
        this.x.rootWrapper.setProperty("tabIndex", 1000);
        this.x.rootWrapper.setRawStyleProperty("outline", "none");
        this.a.cx = true;
        var d = this.a;
        d.onMouseWheel = delegateCombine(d.onMouseWheel, runOn(this, this.k));
        var e = this.a;
        e.onDragStarted = delegateCombine(e.onDragStarted, runOn(this, this.aj));
        var f = this.a;
        f.onDragDelta = delegateCombine(f.onDragDelta, runOn(this, this.ai));
        var g = this.a;
        g.onDragCompleted = delegateCombine(g.onDragCompleted, runOn(this, this.ah));
        var h = this.a;
        h.onFlingStarted = delegateCombine(h.onFlingStarted, runOn(this, this.i));
        var i = this.a;
        i.onContactStarted = delegateCombine(i.onContactStarted, runOn(this, this.ag));
        var j = this.a;
        j.onContactMoved = delegateCombine(j.onContactMoved, runOn(this, this.af));
        var k = this.a;
        k.onContactCompleted = delegateCombine(k.onContactCompleted, runOn(this, this.ae));
        this.a.z = true;
        this.a.ak = runOn(this, this.n);
        var l = this.a;
        l.onKeyDown = delegateCombine(l.onKeyDown, runOn(this, this.j));
        this.b.b0(b, c);
    };
    ScrollerView.prototype.j = function (a) {
        return this.b.n(a);
    };
    ScrollerView.prototype.n = function (a, b, c) {
        if (this.e || this.f) {
            return true;
        }
        if (c || b) {
            return true;
        }
        if (this.l(a, b)) {
            return true;
        }
        if (this.m(a, b)) {
            return true;
        }
        return false;
    };
    ScrollerView.prototype.ae = function (a, b) {
        this.e = false;
        this.f = false;
    };
    ScrollerView.prototype.af = function (a, b) {
        if (this.f) {
            var c = a.y - this.r;
            this.r = a.y;
            this.b.bz(c);
        }
        else if (this.e) {
            var d = a.x - this.o;
            this.o = a.x;
            this.b.bx(d);
        }
    };
    ScrollerView.prototype.ag = function (a, b) {
        this.d = false;
        this.ad();
        var c = this.m(a, b);
        var d = this.l(a, b);
        if (c) {
            this.f = true;
            this.r = a.y;
            this.ak();
        }
        else if (d) {
            this.e = true;
            this.o = a.x;
            this.ak();
        }
    };
    ScrollerView.prototype.l = function (a, b) {
        if (!this.g) {
            return false;
        }
        if (a.y >= this.b.viewportHeight - this.b.scrollbarSpan && a.y <= this.b.viewportHeight && a.x >= this.q - this.p / 2 && a.x <= this.q + this.p / 2) {
            return true;
        }
        return false;
    };
    ScrollerView.prototype.m = function (a, b) {
        if (!this.h) {
            return false;
        }
        if (a.x >= this.b.viewportWidth - this.b.scrollbarSpan && a.x <= this.b.viewportWidth && a.y >= this.t - this.s / 2 && a.y <= this.t + this.s / 2) {
            return true;
        }
        return false;
    };
    ScrollerView.prototype.i = function (a, b, c) {
        this.d = true;
        return true;
    };
    ScrollerView.prototype.ap = function (a) {
        this.a.bc = new Rect(0, 0, 0, this.b.viewportWidth, this.b.viewportHeight);
    };
    ScrollerView.prototype.aq = function (a) {
        this.a.bc = new Rect(0, 0, 0, this.b.viewportWidth, this.b.viewportHeight);
    };
    ScrollerView.prototype.an = function (a) {
        if (this.y != null) {
            this.y.setStyleProperty("height", a + "px");
        }
    };
    ScrollerView.prototype.ao = function (a) {
        if (this.y != null) {
            this.y.setStyleProperty("width", a + "px");
        }
    };
    ScrollerView.prototype.ad = function () {
    };
    ScrollerView.prototype.ah = function (a) {
        if (this.e || this.f) {
            return;
        }
        this.d = false;
        this.b.bu(a);
    };
    ScrollerView.prototype.ai = function (a) {
        if (this.e || this.f) {
            return;
        }
        this.b.bv(a);
    };
    ScrollerView.prototype.aw = function (a) {
        this.w.scrollTop = truncate(a);
    };
    ScrollerView.prototype.aj = function (a) {
        if (this.e || this.f) {
            return;
        }
        this.d = false;
        this.b.bw(a);
    };
    ScrollerView.prototype.u = function () {
        return this.w.scrollLeft;
    };
    ScrollerView.prototype.v = function () {
        return this.w.scrollTop;
    };
    ScrollerView.prototype.k = function (a, b) {
        this.d = false;
        this.ad();
        return this.b.p(a, b);
    };
    ScrollerView.prototype.av = function (a) {
        this.w.scrollLeft = truncate(a);
    };
    ScrollerView.prototype.as = function (a) {
        this.y = a;
        this.ab.append(this.y);
    };
    ScrollerView.prototype.at = function (a) {
        if (this.x == null) {
            return;
        }
        if (this.c == null) {
            this.c = this.x.getRequestAnimationFrame();
        }
        if (this.d) {
            a();
        }
        else {
            this.c(function () { return a(); });
        }
    };
    ScrollerView.prototype.am = function () {
        if (this.h) {
            this.h = false;
            this.ac.setStyleProperty("display", "none");
        }
    };
    ScrollerView.prototype.al = function () {
        if (this.g) {
            this.g = false;
            this.z.setStyleProperty("display", "none");
        }
    };
    ScrollerView.prototype.ax = function (a, b) {
        var c = false;
        if (!this.h) {
            c = true;
            this.h = true;
            this.ac.setStyleProperty("display", "block");
        }
        if (a != this.t || c) {
            this.t = a;
            if (this.h) {
                this.ac.setRawYPosition(a - b / 2);
            }
        }
        if (b != this.s || c) {
            this.s = b;
            if (this.h) {
                this.ac.setRawSize(this.b.scrollbarSpan, b);
            }
        }
    };
    ScrollerView.prototype.au = function (a, b) {
        var c = false;
        if (!this.g) {
            c = true;
            this.g = true;
            this.z.setStyleProperty("display", "block");
        }
        if (a != this.q || c) {
            this.q = a;
            if (this.g) {
                this.z.setRawXPosition(a - b / 2);
            }
        }
        if (b != this.p || c) {
            this.p = b;
            if (this.g) {
                this.z.setRawSize(b, this.b.scrollbarSpan);
            }
        }
    };
    ScrollerView.prototype.ak = function () {
        this.x.rootWrapper.focus();
    };
    ScrollerView.$t = markType(ScrollerView, 'ScrollerView');
    return ScrollerView;
}(Base));
export { ScrollerView };
var Scroller = /** @class */ (function (_super) {
    tslib_1.__extends(Scroller, _super);
    function Scroller() {
        var _this = _super.call(this) || this;
        _this.am = 49;
        _this.ae = NaN;
        _this.z = 49;
        _this.x = NaN;
        _this.al = 48;
        _this.ad = NaN;
        _this.y = 48;
        _this.w = NaN;
        _this.ac = 0;
        _this.af = 20;
        _this.ai = 8;
        _this.an = 0;
        _this.cb = null;
        _this.j = false;
        _this.ab = 0;
        _this.ap = 0;
        _this.ao = 0;
        _this.ak = 0;
        _this.aj = 0;
        _this.b = null;
        _this.h = true;
        _this.k = false;
        _this.aa = 0;
        _this.v = 0;
        _this.g = false;
        _this.e = false;
        _this.f = false;
        _this.cd = null;
        _this.d = false;
        _this.propertyChanged = null;
        _this.onScrolling = null;
        _this.ag = 0;
        _this.ah = 0;
        _this.i = false;
        _this.cb = _this.view.ay();
        return _this;
    }
    Object.defineProperty(Scroller.prototype, "smallVerticalChange", {
        get: function () {
            return this.am;
        },
        set: function (a) {
            var b = this.am;
            this.am = a;
            if (b != this.am) {
                this.by("SmallVerticalChange", b, this.am);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "largeVerticalChange", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            var b = this.ae;
            this.ae = a;
            if (b != this.ae) {
                this.by("LargeVerticalChange", b, this.ae);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "au", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            var b = this.z;
            this.z = a;
            if (b != this.z) {
                this.by("ActualSmallVerticalChange", b, this.z);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "as", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            var b = this.x;
            this.x = a;
            if (b != this.x) {
                this.by("ActualLargeVerticalChange", b, this.x);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "smallHorizontalChange", {
        get: function () {
            return this.al;
        },
        set: function (a) {
            var b = this.al;
            this.al = a;
            if (b != this.al) {
                this.by("SmallHorizontalChange", b, this.al);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "largeHorizontalChange", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            var b = this.ad;
            this.ad = a;
            if (b != this.ad) {
                this.by("LargeHorizontalChange", b, this.ad);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "at", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            var b = this.y;
            this.y = a;
            if (b != this.y) {
                this.by("ActualSmallHorizontalChange", b, this.y);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "ar", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            var b = this.w;
            this.w = a;
            if (b != this.w) {
                this.by("ActualLargeHorizontalChange", b, this.w);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "contentWidth", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            var b = this.ac;
            this.ac = a;
            if (b != this.ac) {
                this.by("ContentWidth", b, this.ac);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "minScrollbarSize", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            var b = this.af;
            this.af = a;
            if (b != this.af) {
                this.by("MinScrollbarSize", b, this.af);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "scrollbarSpan", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            var b = this.ai;
            this.ai = a;
            if (b != this.ai) {
                this.by("ScrollbarSpan", b, this.ai);
            }
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.n = function (a) {
        switch (a) {
            case 11: return this.q();
            case 10: return this.r();
            case 17: return this.m();
            case 15: return this.t();
            case 14: return this.o();
            case 16: return this.s();
        }
        return false;
    };
    Scroller.prototype.s = function () {
        if (this.scrollLeft + this.viewportWidth >= this.contentWidth) {
            return false;
        }
        this.b5(this.at, 0);
        return true;
    };
    Scroller.prototype.o = function () {
        if (this.scrollLeft <= 0) {
            return false;
        }
        this.b5(this.at * -1, 0);
        return true;
    };
    Scroller.prototype.t = function () {
        if (this.scrollTop <= 0) {
            return false;
        }
        this.b5(0, this.au * -1);
        return true;
    };
    Scroller.prototype.m = function () {
        if (this.scrollTop + this.viewportHeight >= this.contentHeight) {
            return false;
        }
        this.b5(0, this.au);
        return true;
    };
    Scroller.prototype.r = function () {
        if (this.scrollTop < 0) {
            return false;
        }
        this.b5(0, this.as * -1);
        return true;
    };
    Scroller.prototype.q = function () {
        if (this.scrollTop + this.viewportHeight >= this.contentHeight) {
            return false;
        }
        this.b5(0, this.as);
        return true;
    };
    Object.defineProperty(Scroller.prototype, "verticalTrackStartInset", {
        get: function () {
            return this.an;
        },
        set: function (a) {
            var b = this.an;
            this.an = a;
            if (b != this.an) {
                this.by("VerticalTrackStartInset", b, this.an);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "scrollbarBrush", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            var b = this.cb;
            this.cb = a;
            if (b != this.cb) {
                this.by("ScrollbarBrush", b, this.cb);
            }
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.by = function (a, b, c) {
        if (this.propertyChanged != null) {
            this.propertyChanged(this, new PropertyChangedEventArgs(a));
        }
        this.b1(a, b, c);
    };
    Object.defineProperty(Scroller.prototype, "u", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            if (!this.j) {
                this.view.aw(this.scrollTop);
                this.view.av(this.scrollLeft);
            }
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.b1 = function (a, b, c) {
        switch (a) {
            case "ContentWidth":
                this.view.ao(this.contentWidth);
                this.b8();
                break;
            case "ContentHeight":
                this.view.an(this.contentHeight);
                this.ca();
                break;
            case "ScrollTop":
                if (!this.u) {
                    this.view.aw(this.scrollTop);
                }
                break;
            case "ScrollLeft":
                if (!this.u) {
                    this.view.av(this.scrollLeft);
                }
                break;
            case "ViewportWidth":
                this.view.aq(this.viewportWidth);
                if (isNaN_(this.largeVerticalChange)) {
                    this.ar = this.viewportWidth;
                }
                break;
            case "ViewportHeight":
                this.view.ap(this.viewportHeight);
                if (isNaN_(this.largeVerticalChange)) {
                    this.as = this.viewportHeight;
                }
                break;
            case "ScrollbarBrush":
            case "ScrollbarSpan":
            case "MinScrollbarSize":
            case "VerticalTrackStartInset":
                this.ca();
                this.b8();
                break;
        }
    };
    Object.defineProperty(Scroller.prototype, "contentHeight", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            var b = this.ab;
            this.ab = a;
            if (b != this.ab) {
                this.by("ContentHeight", b, this.ab);
            }
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.bz = function (a) {
        var b = Math.max(this.minScrollbarSize, this.actualVerticalScrollbarHeight);
        var c = this.verticalTrackStartInset + (b / 2);
        var d = this.viewportHeight - (b / 2);
        var e = a / (d - c);
        var f = e * (this.contentHeight - this.viewportHeight);
        this.b5(0, f);
    };
    Scroller.prototype.bx = function (a) {
        var b = Math.max(this.minScrollbarSize, this.actualHorizontalScrollbarWidth);
        var c = 0 + (b / 2);
        var d = this.viewportWidth - (b / 2);
        var e = a / (d - c);
        var f = e * (this.contentWidth - this.viewportWidth);
        this.b5(f, 0);
    };
    Scroller.prototype.b0 = function (a, b) {
        this.ap = a;
        this.ao = b;
    };
    Object.defineProperty(Scroller.prototype, "viewportWidth", {
        get: function () {
            return this.ap;
        },
        set: function (a) {
            var b = this.ap;
            this.ap = a;
            if (b != this.ap) {
                this.by("ViewportWidth", b, this.ap);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "viewportHeight", {
        get: function () {
            return this.ao;
        },
        set: function (a) {
            var b = this.ao;
            this.ao = a;
            if (b != this.ao) {
                this.by("ViewportHeight", b, this.ao);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "scrollTop", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            var b = this.ak;
            this.ak = a;
            if (b != this.ak) {
                this.by("ScrollTop", b, this.ak);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "scrollLeft", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            var b = this.aj;
            this.aj = a;
            if (b != this.aj) {
                this.by("ScrollLeft", b, this.aj);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "view", {
        get: function () {
            if (this.b == null) {
                this.b = new ScrollerView(this);
            }
            return this.b;
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.provideContainer = function (a) {
        this.view.ar(a);
        this.ak = this.view.v();
        this.aj = this.view.u();
        this.b8();
        this.ca();
    };
    Scroller.prototype.provideContent = function (a) {
        this.view.as(a);
    };
    Scroller.prototype.p = function (a, b) {
        this.f = false;
        this.e = false;
        this.g = false;
        this.view.ad();
        this.b5(0, b);
        return true;
    };
    Object.defineProperty(Scroller.prototype, "lockScrollDirection", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = true;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "actualVerticalScrollbarHeight", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            var b = this.aa;
            this.aa = a;
            if (b != this.aa) {
                this.by("ActualVerticalScrollbarHeight", b, this.aa);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Scroller.prototype, "actualHorizontalScrollbarWidth", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            var b = this.v;
            this.v = a;
            if (b != this.v) {
                this.by("ActualHorizontalScrollbarWidth", b, this.v);
            }
        },
        enumerable: true,
        configurable: true
    });
    Scroller.prototype.focus = function () {
        this.view.ak();
    };
    Scroller.prototype.bw = function (a) {
        this.view.ak();
        this.g = false;
        this.e = false;
        this.f = false;
        this.d = true;
        this.cd = { $type: Point_$type, x: a.x, y: a.y };
    };
    Scroller.prototype.bv = function (a) {
        if (this.d) {
            var b = a.y - this.cd.y;
            var c = a.x - this.cd.x;
            if (this.lockScrollDirection && !this.f) {
                this.f = true;
                if (b > 0 || c > 0) {
                    if (b > c) {
                        this.g = true;
                    }
                    else {
                        this.g = true;
                    }
                }
            }
            this.cd = { $type: Point_$type, x: a.x, y: a.y };
            this.b5(-c, -b);
        }
    };
    Scroller.prototype.scrollTo = function (a, b) {
        this.ag = a - this.scrollLeft;
        this.ah = b - this.scrollTop;
        this.b4();
    };
    Scroller.prototype.b5 = function (a, b) {
        this.ag += a;
        this.ah += b;
        this.b4();
    };
    Scroller.prototype.b4 = function () {
        if (this.i) {
            return;
        }
        this.i = true;
        this.view.at(runOn(this, this.b7));
    };
    Scroller.prototype.b7 = function () {
        this.i = false;
        var a = this.ag;
        var b = this.ah;
        this.ag = 0;
        this.ah = 0;
        if (this.f) {
            if (this.g) {
                a = 0;
            }
            if (this.e) {
                b = 0;
            }
        }
        var c = Math.round(this.scrollTop + b);
        var d = Math.round(this.scrollLeft + a);
        if (c < 0) {
            c = 0;
            this.view.ad();
        }
        if (d < 0) {
            d = 0;
            this.view.ad();
        }
        if (c + this.viewportHeight > this.contentHeight) {
            c = this.contentHeight - this.viewportHeight;
            if (c < 0) {
                c = 0;
            }
            this.view.ad();
        }
        if (d + this.viewportWidth > this.contentWidth) {
            d = this.contentWidth - this.viewportWidth;
            if (d < 0) {
                d = 0;
            }
            this.view.ad();
        }
        b = c - this.scrollTop;
        a = d - this.scrollLeft;
        if (b != 0 || a != 0) {
            try {
                this.u = true;
                this.k = true;
                this.scrollTop = c;
                this.scrollLeft = d;
                this.ca();
                this.b8();
                this.k = false;
                if (this.onScrolling != null) {
                    this.onScrolling(this, ((function () {
                        var $ret = new ScrollerScrollingEventArgs();
                        $ret.deltaX = a;
                        $ret.deltaY = b;
                        return $ret;
                    })()));
                }
            }
            finally {
                this.u = false;
            }
        }
    };
    Scroller.prototype.ca = function () {
        this.b9(this.contentHeight, this.viewportHeight, this.scrollTop, true);
    };
    Scroller.prototype.b8 = function () {
        this.b9(this.contentWidth, this.viewportWidth, this.scrollLeft, false);
    };
    Scroller.prototype.b9 = function (a, b, c, d) {
        var e = Math.round((b / a) * b);
        if (e < this.minScrollbarSize) {
            e = this.minScrollbarSize;
        }
        if (e >= b) {
            if (d) {
                this.view.am();
            }
            else {
                this.view.al();
            }
            return;
        }
        var f = d ? this.verticalTrackStartInset : 0;
        var g = f + (e / 2);
        var h = b - (e / 2);
        var i = c / (a - b);
        var j = Math.round(g + (h - g) * i);
        if (d) {
            this.actualVerticalScrollbarHeight = e;
            this.view.ax(j, e);
        }
        else {
            this.actualHorizontalScrollbarWidth = e;
            this.view.au(j, e);
        }
    };
    Scroller.prototype.bu = function (a) {
        this.f = false;
        this.e = false;
        this.g = false;
        this.d = false;
    };
    Scroller.$t = markType(Scroller, 'Scroller', Base.$, [INotifyPropertyChanged_$type]);
    return Scroller;
}(Base));
export { Scroller };
