// igniteui-angular-core/ES5/SectionHeaderDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
var SectionHeaderDescription = /** @class */ (function (_super) {
    tslib_1.__extends(SectionHeaderDescription, _super);
    function SectionHeaderDescription() {
        var _this = _super.call(this) || this;
        _this.a3 = null;
        _this.a2 = null;
        _this.av = 0;
        _this.ax = 0;
        _this.aw = 0;
        _this.au = 0;
        return _this;
    }
    SectionHeaderDescription.prototype.get_type = function () {
        return "SectionHeader";
    };
    Object.defineProperty(SectionHeaderDescription.prototype, "selectedBackground", {
        get: function () {
            return this.a3;
        },
        set: function (a) {
            this.a3 = a;
            this.e("SelectedBackground");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SectionHeaderDescription.prototype, "actualSelectedBackground", {
        get: function () {
            return this.a2;
        },
        set: function (a) {
            this.a2 = a;
            this.e("ActualSelectedBackground");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SectionHeaderDescription.prototype, "paddingLeft", {
        get: function () {
            return this.av;
        },
        set: function (a) {
            this.av = a;
            this.e("PaddingLeft");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SectionHeaderDescription.prototype, "paddingTop", {
        get: function () {
            return this.ax;
        },
        set: function (a) {
            this.ax = a;
            this.e("PaddingTop");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SectionHeaderDescription.prototype, "paddingRight", {
        get: function () {
            return this.aw;
        },
        set: function (a) {
            this.aw = a;
            this.e("PaddingRight");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SectionHeaderDescription.prototype, "paddingBottom", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("PaddingBottom");
        },
        enumerable: true,
        configurable: true
    });
    SectionHeaderDescription.$t = markType(SectionHeaderDescription, 'SectionHeaderDescription', DefinitionBaseDescription.$);
    return SectionHeaderDescription;
}(DefinitionBaseDescription));
export { SectionHeaderDescription };
