// igniteui-angular-core/ES5/ShapeSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
var ShapeSeriesBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(ShapeSeriesBaseDescription, _super);
    function ShapeSeriesBaseDescription() {
        var _this = _super.call(this) || this;
        _this.br = null;
        _this.bs = null;
        _this.bt = null;
        _this.bp = 0;
        return _this;
    }
    ShapeSeriesBaseDescription.prototype.get_type = function () {
        return "ShapeSeriesBase";
    };
    Object.defineProperty(ShapeSeriesBaseDescription.prototype, "shapeMemberPath", {
        get: function () {
            return this.br;
        },
        set: function (a) {
            this.br = a;
            this.e("ShapeMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ShapeSeriesBaseDescription.prototype, "xAxisRef", {
        get: function () {
            return this.bs;
        },
        set: function (a) {
            this.bs = a;
            this.e("XAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ShapeSeriesBaseDescription.prototype, "yAxisRef", {
        get: function () {
            return this.bt;
        },
        set: function (a) {
            this.bt = a;
            this.e("YAxisRef");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ShapeSeriesBaseDescription.prototype, "shapeFilterResolution", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            this.bp = a;
            this.e("ShapeFilterResolution");
        },
        enumerable: true,
        configurable: true
    });
    ShapeSeriesBaseDescription.$t = markType(ShapeSeriesBaseDescription, 'ShapeSeriesBaseDescription', SeriesDescription.$);
    return ShapeSeriesBaseDescription;
}(SeriesDescription));
export { ShapeSeriesBaseDescription };
