// igniteui-angular-core/ES5/TimeAxisBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
var TimeAxisBaseDescription = /** @class */ (function (_super) {
    tslib_1.__extends(TimeAxisBaseDescription, _super);
    function TimeAxisBaseDescription() {
        var _this = _super.call(this) || this;
        _this.c6 = null;
        _this.cw = false;
        _this.cz = new Date();
        _this.cy = new Date();
        _this.c1 = new Date();
        _this.c0 = new Date();
        return _this;
    }
    TimeAxisBaseDescription.prototype.get_type = function () {
        return "TimeAxisBase";
    };
    Object.defineProperty(TimeAxisBaseDescription.prototype, "dateTimeMemberPath", {
        get: function () {
            return this.c6;
        },
        set: function (a) {
            this.c6 = a;
            this.e("DateTimeMemberPath");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TimeAxisBaseDescription.prototype, "isDataPreSorted", {
        get: function () {
            return this.cw;
        },
        set: function (a) {
            this.cw = a;
            this.e("IsDataPreSorted");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TimeAxisBaseDescription.prototype, "actualMinimumValue", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            this.cz = a;
            this.e("ActualMinimumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TimeAxisBaseDescription.prototype, "actualMaximumValue", {
        get: function () {
            return this.cy;
        },
        set: function (a) {
            this.cy = a;
            this.e("ActualMaximumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TimeAxisBaseDescription.prototype, "minimumValue", {
        get: function () {
            return this.c1;
        },
        set: function (a) {
            this.c1 = a;
            this.e("MinimumValue");
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TimeAxisBaseDescription.prototype, "maximumValue", {
        get: function () {
            return this.c0;
        },
        set: function (a) {
            this.c0 = a;
            this.e("MaximumValue");
        },
        enumerable: true,
        configurable: true
    });
    TimeAxisBaseDescription.$t = markType(TimeAxisBaseDescription, 'TimeAxisBaseDescription', CategoryAxisBaseDescription.$);
    return TimeAxisBaseDescription;
}(CategoryAxisBaseDescription));
export { TimeAxisBaseDescription };
