// igniteui-angular-core/ES5/XamMultiScaleImageView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Base, runOn, Delegate_$type, Type, markType, typeCast, INotifyPropertyChanged_$type, EventArgs, Point_$type } from "./type";
import { ISchedulableRender_$type } from "./ISchedulableRender";
import { List$1 } from "./List$1";
import { Image } from "./Image";
import { RenderingContext } from "./RenderingContext";
import { Dictionary$2 } from "./Dictionary$2";
import { Tile } from "./Tile";
import { Rect } from "./Rect";
import { CanvasRenderScheduler } from "./CanvasRenderScheduler";
import { StackPool$1 } from "./StackPool$1";
import { WriteableBitmap } from "./WriteableBitmap";
import { Uri } from "./Uri";
import { DownloadingMultiScaleImageEventArgs } from "./DownloadingMultiScaleImageEventArgs";
import { CanvasViewRenderer } from "./CanvasViewRenderer";
import { truncate, intDivide, logBase } from "./number";
import { DependencyObject } from "./DependencyObject";
import { Control } from "./Control";
import { DependencyProperty } from "./DependencyProperty";
import { IEasingFunction_$type } from "./IEasingFunction";
import { LRUCache$2 } from "./LRUCache$2";
import { TilePositionComparer } from "./TilePositionComparer";
import { Convert } from "./Convert";
import { MathUtil } from "./MathUtil";
import { PropertyMetadata } from "./PropertyMetadata";
import { dateNow } from "./date";
var XamMultiScaleImageView = /** @class */ (function (_super) {
    tslib_1.__extends(XamMultiScaleImageView, _super);
    function XamMultiScaleImageView(a) {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.h = null;
        _this.g = new Dictionary$2(Base.$, List$1.$.specialize(Delegate_$type), 0);
        _this.j = new List$1(Tile.$, 0);
        _this.i = new List$1(Tile.$, 0);
        _this.p = -1;
        _this.o = -1;
        _this.k = null;
        _this.m = null;
        _this.c = null;
        _this._isDirty = false;
        _this.au = Rect.empty;
        _this.l = null;
        _this.b = a;
        _this.b.ag.l(_this);
        _this.h = new List$1(Image.$, 0);
        _this.b.ao = ((function () {
            var $ret = new StackPool$1(Image.$);
            $ret.j = runOn(_this, _this.at);
            $ret.b = runOn(_this, _this.y);
            $ret.c = runOn(_this, _this.aa);
            $ret.d = runOn(_this, _this.z);
            return $ret;
        })());
        for (var b = 0; b < 4; ++b) {
            _this.i.add(null);
        }
        return _this;
    }
    XamMultiScaleImageView.prototype.at = function () {
        return new Image();
    };
    XamMultiScaleImageView.prototype.y = function (a) {
        this.h.add(a);
    };
    XamMultiScaleImageView.prototype.aa = function (a) {
        this.h.remove(a);
    };
    XamMultiScaleImageView.prototype.z = function (a) {
    };
    XamMultiScaleImageView.prototype.ao = function (a, b, c) {
        a.n = b;
        a.o = c;
    };
    XamMultiScaleImageView.prototype.r = function (a) {
        for (var b = 0; b < this.i.count; ++b) {
            if (a == this.i._inner[b]) {
                if (a.f != null && a.f.ac != null) {
                    var c = a.f.ac;
                    var d = c.c;
                    if (this.g.containsKey(d)) {
                        var e = this.g.item(d);
                        for (var f = 0; f < e.count; f++) {
                            e._inner[f]();
                        }
                        e.clear();
                        this.g.removeItem(d);
                    }
                }
                this.i._inner[b] = null;
                this.q();
                return;
            }
        }
        for (var g = 0; g < this.j.count; ++g) {
            if (this.j._inner[g] == a) {
                this.j.removeAt(g);
                break;
            }
        }
    };
    XamMultiScaleImageView.prototype.u = function (a) {
        this.j.add(a);
        this.j.aa(function (b, c) {
            var d = 0;
            var e = 0;
            if (b.e != null) {
                d = b.e.ac.b;
            }
            if (c.e != null) {
                e = c.e.ac.b;
            }
            if (d < e) {
                return -1;
            }
            else if (d > e) {
                return 1;
            }
            return 0;
        });
        this.q();
    };
    XamMultiScaleImageView.prototype.q = function () {
        var _this = this;
        var index_ = -1;
        if (this.j.count > 0) {
            for (var a = 0; a < this.i.count; ++a) {
                if (this.i._inner[a] == null) {
                    index_ = a;
                    break;
                }
            }
        }
        if (index_ >= 0) {
            this.i._inner[index_] = this.j._inner[0];
            var b = new WriteableBitmap(this.b.an.m, this.b.an.k);
            this.i._inner[index_].f.ac = b;
            var ele_ = this.l.createElement("img");
            var c_1 = ele_.getNativeElement();
            b.c = c_1;
            var self_ = this;
            var d = new List$1(Delegate_$type, 0);
            this.g.addItem(c_1, d);
            d.add(ele_.listen("load", function (e) { return _this.ab(e, index_); }));
            d.add(ele_.listen("readystatechange", function (e) { return _this.ab(e, index_); }));
            d.add(ele_.listen("error", function (e) { return _this.ac(e, index_); }));
            this.j.removeAt(0);
            var e = this.b.an.r(this.i._inner[index_].d + 8, this.i._inner[index_].b, this.i._inner[index_].c);
            var f_1 = e.value;
            var g = ((function () {
                var $ret = new DownloadingMultiScaleImageEventArgs();
                $ret.uri = f_1;
                $ret.image = c_1;
                return $ret;
            })());
            this.b.bk(g);
            f_1 = g.uri;
            if (f_1 != null) {
                c_1.src = f_1;
            }
        }
    };
    XamMultiScaleImageView.prototype.ab = function (a, b) {
        var e_ = a.originalEvent;
        var img_ = (e_.target);
        var c = (img_.complete || (img_.readyState == 'complete' && e_.type == 'readystatechange'));
        if (c) {
            this.downloadCompleted(a, b);
        }
    };
    XamMultiScaleImageView.prototype.ac = function (a, b) {
        this.downloadError(a, b);
    };
    XamMultiScaleImageView.prototype.downloadError = function (a, b) {
        var c = this.i._inner[b];
        this.i._inner[b] = null;
        if (c != null && c.f != null && c.f.ac != null) {
            var e_ = a.originalEvent;
            var d = (e_.target);
            if (this.g.containsKey(d)) {
                var e = this.g.item(d);
                for (var f = 0; f < e.count; f++) {
                    e._inner[f]();
                }
                e.clear();
                this.g.removeItem(d);
            }
        }
        this.q();
        this.ad();
    };
    XamMultiScaleImageView.prototype.downloadCompleted = function (a, b) {
        var c = this.i._inner[b];
        var d = true;
        if (c != null && c.f != null && c.f.ac != null) {
            d = false;
            var ele_ = c.f.ac.c;
            if (!ele_.complete) {
                d = true;
            }
            if ((ele_.width == 0 && ele_.height == 0)) {
                d = true;
            }
        }
        this.i._inner[b] = null;
        if (d) {
            this.q();
            this.ad();
            return;
        }
        var e_ = a.originalEvent;
        var e = (e_.target);
        if (this.g.containsKey(e)) {
            var f = this.g.item(e);
            for (var g = 0; g < f.count; g++) {
                f._inner[g]();
            }
            f.clear();
            this.g.removeItem(e);
        }
        this.b.be(c, c.f.ac);
        if (c.f != null) {
            this.b.bz(c);
        }
        this.q();
        this.ad();
    };
    XamMultiScaleImageView.prototype.a = function (a, b, c, d) {
        var e = new WriteableBitmap(a, a);
        e.c = b.c;
        if (!b.d.isEmpty) {
            c += truncate(Math.round(b.d.left));
            d += truncate(Math.round(b.d.top));
        }
        e.d = new Rect(0, c, d, a, a);
        return e;
    };
    XamMultiScaleImageView.prototype.am = function (a) {
        a.s = 0;
    };
    XamMultiScaleImageView.prototype.an = function (a) {
        a.s = 1;
    };
    XamMultiScaleImageView.prototype.f = function () {
        return true;
    };
    XamMultiScaleImageView.prototype.s = function (a) {
        if (this.b.ak != null) {
            this.b.ak.deferredRefresh();
        }
        else {
            if (this.k != null) {
                this.k.setTimeout(a, 0);
            }
            else {
                window.setTimeout(a, 0);
            }
        }
    };
    XamMultiScaleImageView.prototype.aq = function () {
        if (this.p == -1) {
            this.p = window.setInterval(runOn(this.b, this.b.by), 50);
        }
    };
    XamMultiScaleImageView.prototype.as = function () {
        if (this.p != -1) {
            window.clearInterval(this.p);
            this.p = -1;
        }
    };
    XamMultiScaleImageView.prototype.ap = function () {
        if (this.o == -1) {
            this.o = window.setInterval(runOn(this.b, this.b.bh), 50);
        }
    };
    XamMultiScaleImageView.prototype.ar = function () {
        if (this.o != -1) {
            window.clearInterval(this.o);
            this.o = -1;
        }
    };
    XamMultiScaleImageView.prototype.t = function () {
        if (this.p != -1) {
            this.ar();
            this.b.bn();
        }
    };
    XamMultiScaleImageView.prototype.ae = function (a) {
        this.k = a;
        this.k.rootWrapper.setStyleProperty("position", "relative");
        this.m = this.k.createElement("canvas");
        this.m.setStyleProperty("position", "absolute");
        this.m.setStyleProperty("top", "0px");
        this.m.setStyleProperty("left", "0px");
        this.k.append(this.m);
        this.c = new RenderingContext(new CanvasViewRenderer(), this.k.get2DCanvasContext(this.m));
        this.af(this.k.rootWrapper.width(), this.k.rootWrapper.height());
    };
    XamMultiScaleImageView.prototype.af = function (a, b) {
        this.m.setAttribute("width", truncate((a * this.b.a1)).toString());
        this.m.setAttribute("height", truncate((b * this.b.a1)).toString());
        this.m.setStyleProperty("width", a.toString() + "px");
        this.m.setStyleProperty("height", b.toString() + "px");
        this.b.b9 = new Rect(0, 0, 0, a, b);
        this.b.bu();
    };
    XamMultiScaleImageView.prototype.aj = function () {
        this.ad();
    };
    Object.defineProperty(XamMultiScaleImageView.prototype, "index", {
        get: function () {
            return 0;
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImageView.prototype.postRender = function () {
    };
    Object.defineProperty(XamMultiScaleImageView.prototype, "isDirty", {
        get: function () {
            return this._isDirty;
        },
        set: function (a) {
            this._isDirty = a;
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImageView.prototype.ad = function () {
        if (this.c == null) {
            return;
        }
        if (!this.isDirty) {
            this.isDirty = true;
            this.b.ag.n();
        }
    };
    XamMultiScaleImageView.prototype.undirty = function (a) {
        this.isDirty = false;
        this.ak();
    };
    XamMultiScaleImageView.prototype.ak = function () {
        if (this.c == null) {
            return;
        }
        if (this.c.d && this.b.a1 != 1) {
            this.c.z();
            this.c.aa(this.b.a1, this.b.a1);
        }
        if (!this.au.isEmpty) {
            this.c.k(this.au.left, this.au.top, this.au.width, this.au.height);
        }
        this.au = this.b.b9;
        for (var a = 0; a < this.h.count; a++) {
            var b = this.h._inner[a];
            if (b.s == 0) {
                this.al(b);
            }
        }
        for (var c = 0; c < this.h.count; c++) {
            var d = this.h._inner[c];
            if (d.s == 1) {
                this.al(d);
            }
        }
        if (this.c.d && this.b.a1 != 1) {
            this.c.y();
        }
        this.b.bj();
        this.b.bl();
    };
    XamMultiScaleImageView.prototype.d = function () {
        var a = true;
        for (var b = 0; b < this.i.count; b++) {
            if (this.i._inner[b] != null) {
                a = false;
            }
        }
        return a;
    };
    XamMultiScaleImageView.prototype.al = function (a) {
        if (this.c == null) {
            return;
        }
        var b = a.ac;
        var c = a._opacity;
        if (b == null || b.c == null) {
            return;
        }
        if (this.e(b.c)) {
            return;
        }
        if (!b.d.isEmpty) {
            if (b.d.width < 1 || b.d.height < 1) {
                return;
            }
            this.c.n(b.c, c, b.d.left, b.d.top, b.d.width, b.d.height, Math.round(a.n + this.b.b9.left), Math.round(a.o + this.b.b9.top), a.width, a.height);
        }
        else {
            this.c.m(b.c, c, Math.round(a.n + this.b.b9.left), Math.round(a.o + this.b.b9.top), a.width, a.height);
        }
    };
    XamMultiScaleImageView.prototype.e = function (a) {
        var img_ = a;
        if (!img_.complete) {
            return true;
        }
        if ((img_.width == 0 && img_.height == 0)) {
            return true;
        }
        return false;
    };
    XamMultiScaleImageView.prototype.x = function () {
        this.ad();
    };
    XamMultiScaleImageView.prototype.ag = function (a) {
        this.c = a;
        this.ad();
    };
    XamMultiScaleImageView.prototype.ah = function (a) {
        this.l = a;
    };
    XamMultiScaleImageView.prototype.ai = function (a) {
        this.b.b9 = a;
        this.b.bu();
    };
    XamMultiScaleImageView.prototype.preRender = function () {
    };
    XamMultiScaleImageView.prototype.isValid = function () {
        return true;
    };
    XamMultiScaleImageView.$t = markType(XamMultiScaleImageView, 'XamMultiScaleImageView', Base.$, [ISchedulableRender_$type]);
    return XamMultiScaleImageView;
}(Base));
export { XamMultiScaleImageView };
var XamMultiScaleTileSource = /** @class */ (function (_super) {
    tslib_1.__extends(XamMultiScaleTileSource, _super);
    function XamMultiScaleTileSource(a, b, c, d, e) {
        var _this = _super.call(this) || this;
        _this.o = 0;
        _this.n = 0;
        _this.m = 0;
        _this.k = 0;
        _this.l = 0;
        _this.j = null;
        _this.o = a;
        _this.n = b;
        _this.m = c;
        _this.k = d;
        _this.l = e;
        return _this;
    }
    Object.defineProperty(XamMultiScaleTileSource.prototype, "q", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.t(0, 0, 0, 0);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(XamMultiScaleTileSource.prototype, "p", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.t(0, 0, 0, 0);
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleTileSource.prototype.r = function (a, b, c) {
        var d = new List$1(Base.$, 0);
        this.s(a, b, c, d);
        var e = null;
        if (d.count > 0) {
            e = typeCast(Uri.$, d._inner[0]);
        }
        return e;
    };
    XamMultiScaleTileSource.prototype.t = function (a, b, c, d) {
        if (this.j != null) {
            this.j.bi(a, b, c, d);
        }
    };
    XamMultiScaleTileSource.$t = markType(XamMultiScaleTileSource, 'XamMultiScaleTileSource', DependencyObject.$);
    return XamMultiScaleTileSource;
}(DependencyObject));
export { XamMultiScaleTileSource };
var XamMultiScaleImage = /** @class */ (function (_super) {
    tslib_1.__extends(XamMultiScaleImage, _super);
    function XamMultiScaleImage() {
        var _this = _super.call(this) || this;
        _this.ao = null;
        _this.am = null;
        _this.aj = null;
        _this.b6 = null;
        _this.az = 0;
        _this.propertyChanged = null;
        _this.a7 = 0;
        _this.a8 = 0;
        _this.ax = new Date();
        _this.b7 = null;
        _this.a0 = 0;
        _this.av = new List$1(Tile.$, 0);
        _this.at = false;
        _this.ap = new LRUCache$2(Tile.$, WriteableBitmap.$, 1, 0x7FFFFFFF, new TilePositionComparer());
        _this.aw = new List$1(Tile.$, 0);
        _this.imageTilesReady = null;
        _this.ar = false;
        _this.b9 = null;
        _this.ag = null;
        _this.imagesChanged = null;
        _this.downloadingImage = null;
        _this.ay = 1;
        _this.b9 = Rect.empty;
        _this.ag = new CanvasRenderScheduler();
        _this.am = new XamMultiScaleImageView(_this);
        _this.ab = XamMultiScaleImage.$;
        _this.b6 = _this.b8;
        _this.az = _this.a2;
        return _this;
    }
    Object.defineProperty(XamMultiScaleImage.prototype, "ak", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            if (this.aj != null) {
                this.aj.unRegister(this);
            }
            this.aj = a;
            if (this.aj != null) {
                this.aj.register(this, runOn(this, this.bv));
            }
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImage.prototype.bm = function (a) {
        if (this.propertyChanged != null) {
            this.propertyChanged(this, a);
        }
        switch (a.propertyName) {
            case XamMultiScaleImage.$$p[0]:
                if (this.an != null) {
                    this.an.j = this;
                }
                this.bt();
                this.bw();
                this.bu();
                break;
            case XamMultiScaleImage.$$p[3]:
                this.bx();
                break;
            case XamMultiScaleImage.$$p[4]:
                this.bx();
                break;
            case XamMultiScaleImage.$$p[2]:
                if (!this.au) {
                    this.am.t();
                }
                break;
        }
    };
    XamMultiScaleImage.prototype.bw = function () {
        this.b0();
        if (this.an != null) {
            this.a7 = Convert.toInt32(logBase(this.an.m, 2));
            this.a8 = Convert.toInt32(logBase(this.an.q, 2));
        }
    };
    XamMultiScaleImage.prototype.bi = function (a, b, c, d) {
        this.bt();
        this.bw();
        this.bu();
    };
    XamMultiScaleImage.prototype.bx = function () {
        if (this.au) {
            this.ax = dateNow();
            this.b7 = this.b6;
            this.a0 = this.az;
            this.am.aq();
        }
        else {
            this.b6 = this.b8;
            this.az = this.a2;
            this.bu();
        }
    };
    XamMultiScaleImage.prototype.by = function () {
        var a = 2;
        var b = dateNow().getTime() - this.ax.getTime();
        var c = b / 1000;
        var d = MathUtil.b((c) / a, 0, 1);
        var e = this.ai != null ? this.ai.ease(d) : d;
        var f = 1 - e;
        this.az = this.a0 * f + this.a2 * e;
        this.b6 = { $type: Point_$type, x: this.b7.x * f + this.b8.x * e, y: this.b7.y * f + this.b8.y * e };
        if (d >= 1) {
            this.am.as();
        }
        else {
        }
        this.bu();
    };
    XamMultiScaleImage.prototype.a5 = function (a, b, c) {
        for (var d = 0; d < this.av.count; ++d) {
            if (this.av._inner[d].b == a && this.av._inner[d].c == b && this.av._inner[d].d == c) {
                return d;
            }
        }
        return -1;
    };
    XamMultiScaleImage.prototype.bv = function (a) {
        var _this = this;
        this.at = false;
        this.ar = false;
        if (this.an == null || !this.am.f() || this.b9.width == 0 || this.b9.height == 0) {
            return;
        }
        var b = truncate(Math.ceil(this.b9.width / this.an.m));
        var c = truncate(Math.max(1, Math.floor(-logBase(this.az / b, 2))));
        if (c >= this.a8 - 8) {
            c = (this.a8 - 8) - 1;
        }
        var d = truncate(Math.round(Math.pow(2, c)));
        var e = this.az;
        var f = this.b9.height * e / this.b9.width;
        var g = this.an.q / Math.pow(2, c);
        var h = this.an.p / Math.pow(2, c);
        var i = Math.max(truncate(Math.floor((this.b6.x * this.an.q) / g)), 0);
        var j = Math.min(truncate(Math.ceil(((this.b6.x + e) * this.an.q) / g)), d);
        var k = Math.max(truncate(Math.floor((this.b6.y * this.an.p) / h)), 0);
        var l = Math.min(truncate(Math.ceil(((this.b6.y + f) * this.an.q) / h)), d);
        var m = ((i * g) - (this.b6.x * this.an.q)) / g;
        var n = ((k * h) - (this.b6.y * this.an.p)) / h;
        var o = (e * this.an.q / g) * (this.an.m / this.b9.width);
        var p = new List$1(Tile.$, 0);
        var _loop_1 = function (q) {
            var _loop_2 = function (r) {
                var s = this_1.a5(q, r, c);
                if (s >= 0) {
                    p.add(this_1.av._inner[s]);
                    this_1.av.removeAt(s);
                }
                else {
                    p.add(((function () {
                        var $ret = new Tile();
                        $ret.b = q;
                        $ret.c = r;
                        $ret.d = c;
                        return $ret;
                    })()));
                }
            };
            for (var r = k; r < l; ++r) {
                _loop_2(r);
            }
        };
        var this_1 = this;
        for (var q = i; q < j; ++q) {
            _loop_1(q);
        }
        this.ao.f = true;
        this.b0();
        this.av = p;
        var _loop_3 = function (t) {
            if (this_2.av._inner[t].f == null) {
                this_2.av._inner[t].f = this_2.ao.a();
                this_2.av._inner[t].f._opacity = 1;
                this_2.am.am(this_2.av._inner[t].f);
                var u = this_2.al(this_2.av._inner[t]);
                if (u != null) {
                    this_2.av._inner[t].f.ac = u;
                }
                else {
                    var v = null;
                    var w = ((function () {
                        var $ret = new Tile();
                        $ret.b = _this.av._inner[t].b;
                        $ret.c = _this.av._inner[t].c;
                        $ret.d = _this.av._inner[t].d;
                        return $ret;
                    })());
                    while (w.d >= 0 && v == null) {
                        w.b = w.b >> 1;
                        w.c = w.c >> 1;
                        w.d = w.d - 1;
                        v = this_2.al(w);
                    }
                    if (v != null) {
                        var x = truncate(Math.pow(2, this_2.av._inner[t].d - w.d));
                        var y = intDivide(256, x);
                        var z = y * (this_2.av._inner[t].b % x);
                        var aa = y * (this_2.av._inner[t].c % x);
                        this_2.av._inner[t].e = this_2.ao.a();
                        this_2.av._inner[t].e._opacity = 1;
                        this_2.am.an(this_2.av._inner[t].e);
                        u = this_2.am.a(y, v, z, aa);
                        this_2.av._inner[t].e.ac = u;
                    }
                    this_2.am.u(this_2.av._inner[t]);
                }
            }
            var ab = this_2.an.m / o;
            var ac = this_2.an.k / o;
            var ad = (this_2.av._inner[t].b - i + m) * ab;
            var ae = (this_2.av._inner[t].c - k + n) * ac;
            this_2.av._inner[t].f.width = ab + 0.5;
            this_2.av._inner[t].f.height = ac + 0.5;
            this_2.am.ao(this_2.av._inner[t].f, ad, ae);
            if (this_2.av._inner[t].e != null) {
                this_2.av._inner[t].e.width = ab + 0.5;
                this_2.av._inner[t].e.height = ac + 0.5;
                this_2.am.ao(this_2.av._inner[t].e, ad, ae);
            }
        };
        var this_2 = this;
        for (var t = 0; t < this.av.count; ++t) {
            _loop_3(t);
        }
        this.ao.f = false;
        this.am.aj();
    };
    XamMultiScaleImage.prototype.bu = function () {
        if (this.an == null || !this.am.f() || this.b9.width == 0 || this.b9.height == 0) {
            return;
        }
        if (this.at) {
            return;
        }
        this.at = true;
        this.am.s(runOn(this, this.bv));
    };
    XamMultiScaleImage.prototype.b0 = function () {
        for (var a = 0; a < this.av.count; ++a) {
            this.am.r(this.av._inner[a]);
            this.bf(this.av._inner[a]);
            if (this.av._inner[a].f != null) {
                this.ao.n(this.av._inner[a].f);
                this.av._inner[a].f.ac = null;
                this.av._inner[a].f = null;
            }
        }
    };
    XamMultiScaleImage.prototype.bt = function () {
        this.aq.i();
    };
    XamMultiScaleImage.prototype.al = function (a) {
        return this.aq.item(a);
    };
    XamMultiScaleImage.prototype.be = function (a, b) {
        this.aq.item(a, b);
    };
    Object.defineProperty(XamMultiScaleImage.prototype, "aq", {
        get: function () {
            return this.ap;
        },
        set: function (a) {
            this.ap = a;
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImage.prototype.as = function () {
        return this.aw.count != 0;
    };
    XamMultiScaleImage.prototype.bj = function () {
        var a = true;
        if (this.as()) {
            a = false;
        }
        if (!this.am.d()) {
            a = false;
        }
        if (a && !this.ar) {
            if (this.imageTilesReady != null) {
                this.imageTilesReady(this, new EventArgs());
            }
        }
        this.ar = a;
    };
    XamMultiScaleImage.prototype.bz = function (a) {
        if (a.e != null) {
            a.a = dateNow();
            this.aw.add(a);
            this.am.ap();
        }
        else {
            this.bj();
        }
    };
    XamMultiScaleImage.prototype.bf = function (a) {
        if (a.e != null) {
            this.ao.n(a.e);
            a.e.ac = null;
            a.e = null;
            for (var b = 0; b < this.aw.count; ++b) {
                if (this.aw._inner[b] == a) {
                    this.aw.removeAt(b);
                    break;
                }
            }
            if (this.aw.count == 0) {
                this.am.ar();
            }
        }
    };
    XamMultiScaleImage.prototype.bh = function () {
        var a = dateNow();
        var b = 0.5;
        for (var c = 0; c < this.aw.count;) {
            var d = a.getTime() - this.aw._inner[c].a.getTime();
            var e = d / 1000;
            var f = (e) / b;
            f = MathUtil.b(f, 0, 1);
            this.aw._inner[c].e._opacity = 1 - f;
            if (f >= 1) {
                this.ao.n(this.aw._inner[c].e);
                this.aw._inner[c].e.ac = null;
                this.aw._inner[c].e = null;
                this.aw.removeAt(c);
            }
            else {
                ++c;
            }
            this.am.x();
        }
        if (this.aw.count == 0) {
            this.am.ar();
        }
    };
    XamMultiScaleImage.prototype.bn = function () {
        this.az = this.a2;
        this.b6 = this.b8;
        this.bu();
    };
    XamMultiScaleImage.prototype.bp = function (a) {
        var b = a;
        this.ag.j(b);
        this.am.ae(a);
    };
    XamMultiScaleImage.prototype.bq = function (a) {
        this.am.ag(a);
    };
    XamMultiScaleImage.prototype.br = function (a) {
        this.am.ah(a);
    };
    XamMultiScaleImage.prototype.bs = function (a) {
        this.am.ai(a);
    };
    XamMultiScaleImage.prototype.bl = function () {
        if (this.imagesChanged != null) {
            this.imagesChanged(this, new EventArgs());
        }
    };
    XamMultiScaleImage.prototype.bg = function () {
        this.at = false;
    };
    XamMultiScaleImage.prototype.bk = function (a) {
        if (this.downloadingImage != null) {
            this.downloadingImage(this, a);
        }
    };
    Object.defineProperty(XamMultiScaleImage.prototype, "a1", {
        get: function () {
            return this.ay;
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImage.prototype.bo = function (a) {
        this.ay = a;
    };
    Object.defineProperty(XamMultiScaleImage.prototype, "a6", {
        get: function () {
            return this.aq.h;
        },
        set: function (a) {
            var b = a != this.a6;
            if (b) {
                this.aq = new LRUCache$2(Tile.$, WriteableBitmap.$, 1, a, new TilePositionComparer());
            }
        },
        enumerable: true,
        configurable: true
    });
    XamMultiScaleImage.$t = markType(XamMultiScaleImage, 'XamMultiScaleImage', Control.$, [INotifyPropertyChanged_$type]);
    return XamMultiScaleImage;
}(Control));
export { XamMultiScaleImage };
Type.dep(DependencyProperty, PropertyMetadata, XamMultiScaleImage, 'bm', ['Source:an:b1', [XamMultiScaleTileSource.$, null], 'SpringsEasingFunction:ai:b2', [IEasingFunction_$type, null], 'UseSprings:au:b3', [0, false], 'ViewportOrigin:b8:b4', [Point_$type, { $type: Point_$type, x: 0, y: 0 }], 'ViewportWidth:a2:b5', [1, 1]]);
