// igniteui-angular-core/ES2015/AnchoredCategorySeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategorySeriesDescription } from "./CategorySeriesDescription";
import { markType } from "./type";
export class AnchoredCategorySeriesDescription extends CategorySeriesDescription {
    constructor() {
        super();
        this.co = null;
        this.cn = null;
        this.cm = null;
        this.cj = null;
        this.cf = 0;
        this.ch = 0;
        this.cl = null;
        this.ck = null;
    }
    get_type() {
        return "AnchoredCategorySeries";
    }
    get valueMemberPath() {
        return this.co;
    }
    set valueMemberPath(a) {
        this.co = a;
        this.e("ValueMemberPath");
    }
    get trendLineType() {
        return this.cn;
    }
    set trendLineType(a) {
        this.cn = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.cm;
    }
    set trendLineBrush(a) {
        this.cm = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.cj;
    }
    set actualTrendLineBrush(a) {
        this.cj = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.cf;
    }
    set trendLineThickness(a) {
        this.cf = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.ch;
    }
    set trendLinePeriod(a) {
        this.ch = a;
        this.e("TrendLinePeriod");
    }
    get markerCollisionAvoidance() {
        return this.cl;
    }
    set markerCollisionAvoidance(a) {
        this.cl = a;
        this.e("MarkerCollisionAvoidance");
    }
    get consolidatedItemHitTestBehavior() {
        return this.ck;
    }
    set consolidatedItemHitTestBehavior(a) {
        this.ck = a;
        this.e("ConsolidatedItemHitTestBehavior");
    }
}
AnchoredCategorySeriesDescription.$t = markType(AnchoredCategorySeriesDescription, 'AnchoredCategorySeriesDescription', CategorySeriesDescription.$);
