// igniteui-angular-core/ES2015/AnchoredRadialSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { RadialBaseDescription } from "./RadialBaseDescription";
import { markType } from "./type";
export class AnchoredRadialSeriesDescription extends RadialBaseDescription {
    constructor() {
        super();
        this.ci = null;
        this.ch = null;
        this.cg = null;
        this.cf = null;
        this.b9 = 0;
        this.cb = 0;
        this.cc = 0;
    }
    get_type() {
        return "AnchoredRadialSeries";
    }
    get valueMemberPath() {
        return this.ci;
    }
    set valueMemberPath(a) {
        this.ci = a;
        this.e("ValueMemberPath");
    }
    get trendLineType() {
        return this.ch;
    }
    set trendLineType(a) {
        this.ch = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.cg;
    }
    set trendLineBrush(a) {
        this.cg = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.cf;
    }
    set actualTrendLineBrush(a) {
        this.cf = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.b9;
    }
    set trendLineThickness(a) {
        this.b9 = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.cb;
    }
    set trendLinePeriod(a) {
        this.cb = a;
        this.e("TrendLinePeriod");
    }
    get trendLineZIndex() {
        return this.cc;
    }
    set trendLineZIndex(a) {
        this.cc = a;
        this.e("TrendLineZIndex");
    }
}
AnchoredRadialSeriesDescription.$t = markType(AnchoredRadialSeriesDescription, 'AnchoredRadialSeriesDescription', RadialBaseDescription.$);
