// igniteui-angular-core/ES2015/AnnotationLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class AnnotationLayerDescription extends SeriesDescription {
    constructor() {
        super();
        this.bt = false;
        this.bu = false;
        this.bp = null;
        this.br = false;
        this.bs = false;
    }
    get_type() {
        return "AnnotationLayer";
    }
    get useIndex() {
        return this.bt;
    }
    set useIndex(a) {
        this.bt = a;
        this.e("UseIndex");
    }
    get useLegend() {
        return this.bu;
    }
    set useLegend(a) {
        this.bu = a;
        this.e("UseLegend");
    }
    get cursorPosition() {
        return this.bp;
    }
    set cursorPosition(a) {
        this.bp = a;
        this.e("CursorPosition");
    }
    get isDefaultCrosshairDisabled() {
        return this.br;
    }
    set isDefaultCrosshairDisabled(a) {
        this.br = a;
        this.e("IsDefaultCrosshairDisabled");
    }
    get shouldRenderAsOverlay() {
        return this.bs;
    }
    set shouldRenderAsOverlay(a) {
        this.bs = a;
        this.e("ShouldRenderAsOverlay");
    }
}
AnnotationLayerDescription.$t = markType(AnnotationLayerDescription, 'AnnotationLayerDescription', SeriesDescription.$);
