// igniteui-angular-core/ES2015/AxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class AxisDescription extends Description {
    constructor() {
        super();
        this.a7 = null;
        this.bk = null;
        this.bi = null;
        this.a3 = null;
        this.z = 0;
        this.bh = null;
        this.bf = null;
        this.a1 = null;
        this.x = 0;
        this.bg = null;
        this.a2 = null;
        this.y = 0;
        this.bj = null;
        this.ab = 0;
        this.aa = 0;
        this.g = false;
        this.h = false;
        this.k = false;
        this.az = 0;
        this.bc = null;
        this.bb = null;
        this.ba = null;
        this.be = null;
        this.i = false;
        this.r = 0;
        this.t = 0;
        this.u = 0;
        this.v = 0;
        this.w = 0;
        this.s = 0;
        this.a9 = null;
        this.bd = null;
        this.a5 = null;
        this.a6 = null;
        this.bn = null;
        this.bp = null;
        this.bo = null;
        this.bm = null;
        this.br = null;
        this.j = false;
        this.ac = 0;
        this.ae = 0;
        this.af = 0;
        this.ag = 0;
        this.ah = 0;
        this.ad = 0;
        this.bl = null;
        this.bq = null;
        this.a4 = null;
        this.f = false;
        this._RangeChangedRef = null;
        this._RenderRequestedRef = null;
        this.a8 = null;
    }
    get_type() {
        return "Axis";
    }
    get type() {
        return this.get_type();
    }
    get formatLabelRef() {
        return this.a7;
    }
    set formatLabelRef(a) {
        this.a7 = a;
        this.e("FormatLabelRef");
    }
    get title() {
        return this.bk;
    }
    set title(a) {
        this.bk = a;
        this.e("Title");
    }
    get stroke() {
        return this.bi;
    }
    set stroke(a) {
        this.bi = a;
        this.e("Stroke");
    }
    get actualStroke() {
        return this.a3;
    }
    set actualStroke(a) {
        this.a3 = a;
        this.e("ActualStroke");
    }
    get strokeThickness() {
        return this.z;
    }
    set strokeThickness(a) {
        this.z = a;
        this.e("StrokeThickness");
    }
    get strip() {
        return this.bh;
    }
    set strip(a) {
        this.bh = a;
        this.e("Strip");
    }
    get majorStroke() {
        return this.bf;
    }
    set majorStroke(a) {
        this.bf = a;
        this.e("MajorStroke");
    }
    get actualMajorStroke() {
        return this.a1;
    }
    set actualMajorStroke(a) {
        this.a1 = a;
        this.e("ActualMajorStroke");
    }
    get majorStrokeThickness() {
        return this.x;
    }
    set majorStrokeThickness(a) {
        this.x = a;
        this.e("MajorStrokeThickness");
    }
    get minorStroke() {
        return this.bg;
    }
    set minorStroke(a) {
        this.bg = a;
        this.e("MinorStroke");
    }
    get actualMinorStroke() {
        return this.a2;
    }
    set actualMinorStroke(a) {
        this.a2 = a;
        this.e("ActualMinorStroke");
    }
    get minorStrokeThickness() {
        return this.y;
    }
    set minorStrokeThickness(a) {
        this.y = a;
        this.e("MinorStrokeThickness");
    }
    get tickStroke() {
        return this.bj;
    }
    set tickStroke(a) {
        this.bj = a;
        this.e("TickStroke");
    }
    get tickStrokeThickness() {
        return this.ab;
    }
    set tickStrokeThickness(a) {
        this.ab = a;
        this.e("TickStrokeThickness");
    }
    get tickLength() {
        return this.aa;
    }
    set tickLength(a) {
        this.aa = a;
        this.e("TickLength");
    }
    get isDisabled() {
        return this.g;
    }
    set isDisabled(a) {
        this.g = a;
        this.e("IsDisabled");
    }
    get isInverted() {
        return this.h;
    }
    set isInverted(a) {
        this.h = a;
        this.e("IsInverted");
    }
    get useEnhancedIntervalManagement() {
        return this.k;
    }
    set useEnhancedIntervalManagement(a) {
        this.k = a;
        this.e("UseEnhancedIntervalManagement");
    }
    get enhancedIntervalMinimumCharacters() {
        return this.az;
    }
    set enhancedIntervalMinimumCharacters(a) {
        this.az = a;
        this.e("EnhancedIntervalMinimumCharacters");
    }
    get labelTextStyle() {
        return this.bc;
    }
    set labelTextStyle(a) {
        this.bc = a;
        this.e("LabelTextStyle");
    }
    get labelTextColor() {
        return this.bb;
    }
    set labelTextColor(a) {
        this.bb = a;
        this.e("LabelTextColor");
    }
    get labelLocation() {
        return this.ba;
    }
    set labelLocation(a) {
        this.ba = a;
        this.e("LabelLocation");
    }
    get labelVisibility() {
        return this.be;
    }
    set labelVisibility(a) {
        this.be = a;
        this.e("LabelVisibility");
    }
    get labelShowFirstLabel() {
        return this.i;
    }
    set labelShowFirstLabel(a) {
        this.i = a;
        this.e("LabelShowFirstLabel");
    }
    get labelAngle() {
        return this.r;
    }
    set labelAngle(a) {
        this.r = a;
        this.e("LabelAngle");
    }
    get labelExtent() {
        return this.t;
    }
    set labelExtent(a) {
        this.t = a;
        this.e("LabelExtent");
    }
    get labelLeftMargin() {
        return this.u;
    }
    set labelLeftMargin(a) {
        this.u = a;
        this.e("LabelLeftMargin");
    }
    get labelRightMargin() {
        return this.v;
    }
    set labelRightMargin(a) {
        this.v = a;
        this.e("LabelRightMargin");
    }
    get labelTopMargin() {
        return this.w;
    }
    set labelTopMargin(a) {
        this.w = a;
        this.e("LabelTopMargin");
    }
    get labelBottomMargin() {
        return this.s;
    }
    set labelBottomMargin(a) {
        this.s = a;
        this.e("LabelBottomMargin");
    }
    get labelHorizontalAlignment() {
        return this.a9;
    }
    set labelHorizontalAlignment(a) {
        this.a9 = a;
        this.e("LabelHorizontalAlignment");
    }
    get labelVerticalAlignment() {
        return this.bd;
    }
    set labelVerticalAlignment(a) {
        this.bd = a;
        this.e("LabelVerticalAlignment");
    }
    get crossingAxisRef() {
        return this.a5;
    }
    set crossingAxisRef(a) {
        this.a5 = a;
        this.e("CrossingAxisRef");
    }
    get crossingValueRef() {
        return this.a6;
    }
    set crossingValueRef(a) {
        this.a6 = a;
        this.e("CrossingValueRef");
    }
    get titlePosition() {
        return this.bn;
    }
    set titlePosition(a) {
        this.bn = a;
        this.e("TitlePosition");
    }
    get titleTextStyle() {
        return this.bp;
    }
    set titleTextStyle(a) {
        this.bp = a;
        this.e("TitleTextStyle");
    }
    get titleTextColor() {
        return this.bo;
    }
    set titleTextColor(a) {
        this.bo = a;
        this.e("TitleTextColor");
    }
    get titleLocation() {
        return this.bm;
    }
    set titleLocation(a) {
        this.bm = a;
        this.e("TitleLocation");
    }
    get titleVisibility() {
        return this.br;
    }
    set titleVisibility(a) {
        this.br = a;
        this.e("TitleVisibility");
    }
    get titleShowFirstLabel() {
        return this.j;
    }
    set titleShowFirstLabel(a) {
        this.j = a;
        this.e("TitleShowFirstLabel");
    }
    get titleAngle() {
        return this.ac;
    }
    set titleAngle(a) {
        this.ac = a;
        this.e("TitleAngle");
    }
    get titleExtent() {
        return this.ae;
    }
    set titleExtent(a) {
        this.ae = a;
        this.e("TitleExtent");
    }
    get titleLeftMargin() {
        return this.af;
    }
    set titleLeftMargin(a) {
        this.af = a;
        this.e("TitleLeftMargin");
    }
    get titleRightMargin() {
        return this.ag;
    }
    set titleRightMargin(a) {
        this.ag = a;
        this.e("TitleRightMargin");
    }
    get titleTopMargin() {
        return this.ah;
    }
    set titleTopMargin(a) {
        this.ah = a;
        this.e("TitleTopMargin");
    }
    get titleBottomMargin() {
        return this.ad;
    }
    set titleBottomMargin(a) {
        this.ad = a;
        this.e("TitleBottomMargin");
    }
    get titleHorizontalAlignment() {
        return this.bl;
    }
    set titleHorizontalAlignment(a) {
        this.bl = a;
        this.e("TitleHorizontalAlignment");
    }
    get titleVerticalAlignment() {
        return this.bq;
    }
    set titleVerticalAlignment(a) {
        this.bq = a;
        this.e("TitleVerticalAlignment");
    }
    get coercionMethodsRef() {
        return this.a4;
    }
    set coercionMethodsRef(a) {
        this.a4 = a;
        this.e("CoercionMethodsRef");
    }
    get expectFunctions() {
        return this.f;
    }
    set expectFunctions(a) {
        this.f = a;
        this.e("ExpectFunctions");
    }
    get RangeChangedRef() {
        return this._RangeChangedRef;
    }
    set RangeChangedRef(a) {
        this._RangeChangedRef = a;
    }
    get RenderRequestedRef() {
        return this._RenderRequestedRef;
    }
    set RenderRequestedRef(a) {
        this._RenderRequestedRef = a;
    }
    get label() {
        return this.a8;
    }
    set label(a) {
        this.a8 = a;
        this.e("Label");
    }
}
AxisDescription.$t = markType(AxisDescription, 'AxisDescription', Description.$);
