// igniteui-angular-core/ES2015/BaseDOMEventProxy_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { EventProxy } from "./EventProxy";
import { Point_$type, runOn, markType, PointUtil, String_$type, Delegate_$type } from "./type";
import { Rect } from "./Rect";
import { MouseEventArgs } from "./MouseEventArgs";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
export class BaseDOMEventProxy extends EventProxy {
    constructor() {
        super(...arguments);
        this.bt = null;
        this.bm = false;
        this.bs = null;
        this.bf = false;
        this.bq = 0;
        this.br = 0;
        this.bo = 0;
        this.bp = 0;
        this.bu = 0;
        this.cs = null;
        this.cr = null;
        this.be = false;
        this.bv = -1;
        this.bi = false;
        this.bg = false;
        this.bd = false;
    }
    d() {
        let a = new DOMEventProxy(this.bt, this.bs, this.bm);
        a.z = this.z;
        a.aj = this.aj;
        return a;
    }
    static get bn() {
        return (BaseDOMEventProxy.bj || BaseDOMEventProxy.bk);
    }
    bb(a) {
        return DOMEventProxy.ct(a);
    }
    cj(a) {
        if (!this.ab) {
            if (this.bh(a)) {
                a.preventDefault();
            }
        }
    }
    bh(e_) {
        return (e_.originalEvent.cancelable || e_.originalEvent.cancelable == undefined);
    }
    cf() {
        let source_ = this.bt.getNativeElement();
        BaseDOMEventProxy.ci(source_);
    }
    static ci(source_) {
        if (BaseDOMEventProxy.bk) {
            source_.style.touchAction = 'none';
        }
        else {
            source_.style.msTouchAction = 'none';
        }
        source_.style.msUserSelect = 'none';
        source_.style.webkitTouchCallout = 'none';
        source_.style.webkitUserSelect = 'none';
        source_.style.khtmlUserSelect = 'none';
        source_.style.mozUserSelect = 'none';
        source_.style.userSelect = 'none';
    }
    static cc(source_) {
        if (BaseDOMEventProxy.bk) {
            source_.style.touchAction = 'auto';
        }
        else {
            source_.style.msTouchAction = 'auto';
        }
        source_.style.msUserSelect = 'auto';
        source_.style.webkitTouchCallout = 'none';
        source_.style.webkitUserSelect = 'auto';
        source_.style.khtmlUserSelect = 'auto';
        source_.style.mozUserSelect = 'auto';
        source_.style.userSelect = 'auto';
    }
    cd() {
        let source_ = this.bt.getNativeElement();
        BaseDOMEventProxy.cc(source_);
    }
    ao() {
        super.ao();
        if (!this.ab) {
            this.cf();
        }
        else {
            this.cd();
        }
    }
    b0() {
        let ver_ = -1;
        let matchIE_ = /Trident\/([\d.]+)/;
        if ((matchIE_.exec(navigator.userAgent) != null)) {
            ver_ = (parseFloat(RegExp.$1));
        }
        return ver_;
    }
    bz() {
        let ver_ = -1;
        let matchIE_ = /Edge\/([\d.]+)/;
        if ((matchIE_.exec(navigator.userAgent) != null)) {
            ver_ = (parseFloat(RegExp.$1));
        }
        return ver_;
    }
    static ch(e_) {
        let a = BaseDOMEventProxy.b1 >= 6;
        let oe_ = e_.originalEvent;
        if ((((typeof e_.pageX == 'undefined')) || a) && oe_.clientX != null) {
            let od_;
            if ((e_.target !== undefined)) {
                od_ = e_.target.ownerDocument;
            }
            else {
                od_ = e_.originalEvent.target.ownerDocument;
            }
            let ed_ = od_ ? od_ : document;
            let doc_ = ed_.documentElement;
            let body_ = ed_.body;
            let clientX_ = oe_.clientX;
            let clientY_ = oe_.clientY;
            let scrollLeft_ = doc_ && doc_.scrollLeft || body_ && body_.scrollLeft || 0;
            let scrollTop_ = doc_ && doc_.scrollTop || body_ && body_.scrollTop || 0;
            let clientLeft_ = doc_ && doc_.clientLeft || body_ && body_.clientLeft || 0;
            let clientTop_ = doc_ && doc_.clientTop || body_ && body_.clientTop || 0;
            e_.pageX = clientX_ + (scrollLeft_ - clientLeft_);
            e_.pageY = clientY_ + (scrollTop_ - clientTop_);
        }
        if ((!e_.pageX)) {
            e_.pageX = e_.originalEvent.pageX;
        }
        if ((!e_.pageY)) {
            e_.pageY = e_.originalEvent.pageY;
        }
        if (((typeof e_.button == 'undefined')) && ((typeof oe_.button != 'undefined'))) {
            e_.button = oe_.button;
        }
    }
    static ct(a) {
        let b = BaseDOMEventProxy.b1 >= 6;
        let source_ = a.getNativeElement();
        let d_ = source_.ownerDocument;
        let doc_ = d_ ? d_.documentElement : null;
        let z_ = doc_ ? doc_.msContentZoomFactor : null;
        let body_ = doc_.body;
        if ((z_ && z_ > 1) || b) {
            let rect_ = source_.getBoundingClientRect();
            let x_ = (rect_.left);
            let y_ = (rect_.top);
            let scrollLeft_ = (doc_ && doc_.scrollLeft || body_ && body_.scrollLeft || 0);
            let scrollTop_ = (doc_ && doc_.scrollTop || body_ && body_.scrollTop || 0);
            let clientLeft_ = (doc_ && doc_.clientLeft || body_ && body_.clientLeft || 0);
            let clientTop_ = (doc_ && doc_.clientTop || body_ && body_.clientTop || 0);
            let c = x_ + scrollLeft_ - clientLeft_;
            let d = y_ + scrollTop_ - clientTop_;
            return new Rect(0, c, d, 0, 0);
        }
        else {
            let e = a.getOffset();
            return new Rect(0, e.left, e.top, 0, 0);
        }
    }
    get bx() {
        return this.bu;
    }
    set bx(a) {
        this.bu = a;
    }
    cm(a) {
        this.al = a;
        BaseDOMEventProxy.ch(a);
        let b = BaseDOMEventProxy.ct(this.bt);
        this.bq = a.pageX - b.left;
        this.br = a.pageY - b.top;
        this.bo = this.bq - this.bc.left;
        this.bp = this.br - this.bc.top;
        this.cs = { $type: Point_$type, x: this.bq, y: this.br };
        this.cr = { $type: Point_$type, x: this.bo, y: this.bp };
    }
    cn(e_) {
        this.al = e_;
        BaseDOMEventProxy.ch(e_);
        if (BaseDOMEventProxy.bn) {
            let a = e_.pageX;
            let b = e_.pageY;
            let c = BaseDOMEventProxy.ct(this.bt);
            this.bq = a - c.left;
            this.br = b - c.top;
        }
        else {
            this.bu = 0;
            if (!e_.originalEvent.targetTouches || e_.originalEvent.targetTouches.length < 1) {
                return;
            }
            ;
            this.bu = (e_.originalEvent.targetTouches.length);
            let d = (e_.originalEvent.targetTouches[0].pageX);
            let e = (e_.originalEvent.targetTouches[0].pageY);
            let f = BaseDOMEventProxy.ct(this.bt);
            this.bq = d - f.left;
            this.br = e - f.top;
        }
        this.bo = this.bq - this.bc.left;
        this.bp = this.br - this.bc.top;
        this.cs = { $type: Point_$type, x: this.bq, y: this.br };
        this.cr = { $type: Point_$type, x: this.bo, y: this.bp };
    }
    cb(e_) {
        this.cl(e_);
        if (this.aa(this.cs, false, true)) {
            let deltaY_ = 0;
            let ev_ = e_;
            deltaY_ = (ev_.originalEvent ? ev_.originalEvent.deltaY : ev_.deltaY);
            let deltaX_ = 0;
            deltaX_ = (ev_.originalEvent ? ev_.originalEvent.deltaX : ev_.deltaX);
            let a = (ev_.originalEvent ? (ev_.originalEvent.deltaMode == 1) : (ev_.deltaMode == 1));
            if (a) {
                deltaY_ = deltaY_ * 40;
                deltaX_ = deltaX_ * 40;
            }
            let b = this.af(this.cr, deltaY_);
            if (b) {
                this.cj(e_);
            }
        }
    }
    b9(e_) {
        this.cl(e_);
        this.cm(e_);
        if (this.aa(this.cs, false, true)) {
            let ev_ = e_;
            let delta_ = 0;
            if (ev_.wheelDelta) {
                delta_ = ev_.wheelDelta / 120;
            }
            ;
            if (ev_.originalEvent && ev_.originalEvent.wheelDelta) {
                delta_ = ev_.originalEvent.wheelDelta / 120;
            }
            ;
            if (ev_.detail) {
                delta_ = -ev_.detail / 3;
            }
            ;
            if (ev_.originalEvent && ev_.originalEvent.detail) {
                delta_ = -ev_.originalEvent.detail / 3;
            }
            ;
            delta_ = delta_ / 10;
            let a = this.af(this.cr, delta_);
            if (a) {
                this.cj(e_);
            }
        }
    }
    cg() {
        if (this.be) {
            if (this.bv == -1) {
                this.bv = this.bs.setTimeout(runOn(this, this.ce), 500);
            }
            else {
                window.clearTimeout(this.bv);
                this.bv = -1;
                this.bv = this.bs.setTimeout(runOn(this, this.ce), 500);
            }
        }
    }
    b2() {
        this.be = true;
        if (this.bv != -1) {
            window.clearTimeout(this.bv);
            this.bv = -1;
        }
    }
    ce() {
        this.be = false;
    }
    get bl() {
        return this.bg;
    }
    set bl(a) {
        this.bg = a;
    }
    b7(a) {
        if (this.bg || this.be) {
            return;
        }
        this.cm(a);
        if (this.bi) {
            this.bi = false;
            this.a0(this.cr);
        }
    }
    b8(a) {
        if (this.bg || this.be) {
            return;
        }
        this.cm(a);
        this.cl(a);
        let b = new MouseEventArgs();
        b.position = this.cs;
        if (this.aa(this.cs, false, false)) {
            if (!this.bi) {
                this.bi = true;
                this.az(this.cs);
            }
            this.a1(this.cs, true, false);
            this.aq(this.cr, false);
        }
        else if (this.bi) {
            this.b7(a);
        }
    }
    ck(a) {
        this.bt.focus();
    }
    b6(a) {
        if (this.bg || this.be) {
            return;
        }
        this.ck(a);
        this.cm(a);
        this.cl(a);
        if (this.aa(this.cs, false, false)) {
            this.bd = true;
            this.ay(this.cs);
            this.ar(this.cr, false);
            this.cj(a);
        }
    }
    co(a) {
        if (this.bg || this.be) {
            return;
        }
        if (this.bd) {
            this.ca(a);
        }
    }
    ca(a) {
        if (this.bg || this.be) {
            return;
        }
        this.cl(a);
        if (PointUtil.equals(this.cs, null) || PointUtil.equals(this.cr, null)) {
            this.cm(a);
        }
        let b = this.aa(this.cs, false, false) || this.bd || this.bi;
        this.bd = false;
        this.a2(this.cs);
        this.ap(this.cr, false);
        if (b) {
            this.cj(a);
        }
    }
    b3(a) {
        this.b4(a, true);
    }
    b5(a) {
        this.b4(a, false);
    }
    cq(a) {
        return BaseDOMEventProxy.cp(a);
    }
    static cp(a) {
        let b;
        switch (a.which) {
            case 33:
                b = 10;
                break;
            case 34:
                b = 11;
                break;
            case 36:
                b = 13;
                break;
            case 37:
                b = 14;
                break;
            case 38:
                b = 15;
                break;
            case 39:
                b = 16;
                break;
            case 40:
                b = 17;
                break;
            case 9:
                b = 2;
                break;
            case 32:
                b = 9;
                break;
            case 13:
                b = 3;
                break;
            case 27:
                b = 8;
                break;
            case 16:
                b = 4;
                break;
            case 17:
                b = 5;
                break;
            case 18:
                b = 6;
                break;
            case 8:
                b = 1;
                break;
            case 35:
                b = 12;
                break;
            case 45:
                b = 18;
                break;
            case 46:
                b = 19;
                break;
            case 106:
                b = 78;
                break;
            case 107:
                b = 79;
                break;
            case 109:
                b = 80;
                break;
            case 110:
                b = 81;
                break;
            case 111:
                b = 82;
                break;
            case 220:
                b = 85;
                break;
            case 192:
                b = 86;
                break;
            case 191:
                b = 84;
                break;
            case 186:
                b = 83;
                break;
            case 189:
                b = 88;
                break;
            case 187:
                b = 87;
                break;
            default:
                if (112 <= a.which && a.which <= 123) {
                    b = (56 + a.which - 112);
                }
                else if (65 <= a.which && a.which <= 90) {
                    b = (30 + a.which - 65);
                }
                else if (48 <= a.which && a.which <= 57) {
                    b = (20 + a.which - 48);
                }
                else if (96 <= a.which && a.which <= 105) {
                    b = (68 + a.which - 96);
                }
                else {
                    b = 0;
                }
                break;
        }
        return b;
    }
    b4(a, b) {
        let c = this.cq(a);
        let d = c != 0;
        this.cl(a);
        if (d) {
            let e = false;
            if (b) {
                e = this.ad(c);
            }
            else {
                e = this.ae(c);
            }
            if (e) {
                this.cj(a);
            }
        }
    }
    cl(a) {
        this.al = a;
        this.a9 = 0;
        if (a.shiftKey) {
            this.a9 = this.a9 | 4;
        }
        if (a.altKey) {
            this.a9 = this.a9 | 1;
        }
        if (a.ctrlKey) {
            this.a9 = this.a9 | 2;
        }
        this.ag = a.button == 2;
    }
    ba() {
        let a = BaseDOMEventProxy.ct(this.bt);
        let b = a.left;
        let c = a.top;
        return { $type: Point_$type, x: b, y: c };
    }
}
BaseDOMEventProxy.$t = markType(BaseDOMEventProxy, 'BaseDOMEventProxy', EventProxy.$);
BaseDOMEventProxy.bj = false;
BaseDOMEventProxy.bk = false;
BaseDOMEventProxy.b1 = 0;
BaseDOMEventProxy.by = 0;
export class DOMEventProxy extends BaseDOMEventProxy {
    constructor(a, b, c) {
        super();
        this.c5 = 0;
        this.c6 = "";
        this.cv = false;
        this.c2 = null;
        this.cy = new Dictionary$2(String_$type, List$1.$.specialize(Delegate_$type), 0);
        this.dt = null;
        this.c3 = -1;
        this.du = null;
        this.c0 = 0;
        this.c1 = 0;
        this.cz = 5;
        this.cu = false;
        this.ds = null;
        this.bm = c;
        this.bs = b;
        this.aj = (d) => true;
        this.z = false;
        DOMEventProxy.c4++;
        this.c5 = DOMEventProxy.c4;
        this.c6 = ".DOMProxy" + this.c5.toString();
        this.a9 = 0;
        this.bt = a;
        BaseDOMEventProxy.bj = false;
        try {
            BaseDOMEventProxy.bj = (window.navigator.msPointerEnabled && MSGesture !== undefined);
        }
        catch (d) {
        }
        BaseDOMEventProxy.bk = false;
        try {
            BaseDOMEventProxy.bk = (window.PointerEvent || window.navigator.pointerEnabled);
        }
        catch (e) {
        }
        BaseDOMEventProxy.b1 = this.b0();
        BaseDOMEventProxy.by = this.bz();
        this.am(this.bt, "");
        this.ai = (f) => true;
    }
    c7(a, b, c) {
    }
    am(a, b) {
        let c = a;
        let d = this.c6 + b;
        if (!this.bs.supportsDOMEvents()) {
            return;
        }
        let e = new List$1(Delegate_$type, 0);
        this.cy.addItem(d, e);
        this.c7(a, b, e);
        if (!BaseDOMEventProxy.bn) {
            e.add(c.listen("mousemove", runOn(this, this.b8)));
            e.add(c.listen("mouseleave", runOn(this, this.b7)));
            e.add(c.listen("mousedown", runOn(this, this.b6)));
            e.add(c.listen("mouseup", runOn(this, this.ca)));
            e.add(this.bs.globalListen("window", "mouseup", runOn(this, this.co)));
        }
        e.add(c.listen("keydown", runOn(this, this.b3)));
        e.add(c.listen("keyup", runOn(this, this.b5)));
        let source_ = this.bt.getNativeElement();
        if (!this.ab) {
            BaseDOMEventProxy.ci(source_);
        }
        if (BaseDOMEventProxy.bn) {
            let f = this.bt.getNativeElement();
            if ((window.MSGesture !== undefined)) {
                let g = (new MSGesture());
                g.target = f;
                this.c2 = g;
                e.add(c.listen("MSGestureStart", runOn(this, this.da)));
                e.add(c.listen("MSGestureChange", runOn(this, this.c8)));
                e.add(c.listen("MSGestureEnd", runOn(this, this.c9)));
            }
            if (BaseDOMEventProxy.bk) {
                e.add(c.listen("pointerdown", runOn(this, this.dd)));
                e.add(c.listen("pointerup", runOn(this, this.dg)));
                e.add(c.listen("pointercancel", runOn(this, this.dc)));
                e.add(c.listen("pointermove", runOn(this, this.de)));
                e.add(c.listen("pointerout", runOn(this, this.df)));
                e.add(c.listen("lostpointercapture", runOn(this, this.db)));
            }
            else {
                e.add(c.listen("MSPointerDown", runOn(this, this.dd)));
                e.add(c.listen("MSPointerUp", runOn(this, this.dg)));
                e.add(c.listen("MSPointerCancel", runOn(this, this.dc)));
                e.add(c.listen("MSPointerMove", runOn(this, this.de)));
                e.add(c.listen("MSPointerOut", runOn(this, this.df)));
                e.add(c.listen("MSLostPointerCapture", runOn(this, this.db)));
            }
        }
        else {
            e.add(c.listen("gesturestart", runOn(this, this.da)));
            e.add(c.listen("gesturechange", runOn(this, this.c8)));
            e.add(c.listen("gestureend", runOn(this, this.c9)));
            e.add(c.listen("touchstart", runOn(this, this.dj)));
            e.add(c.listen("touchmove", runOn(this, this.di)));
            e.add(c.listen("touchend", runOn(this, this.dh)));
        }
        if (this.bm) {
            e.add(c.listen("wheel", runOn(this, this.cb)));
        }
        else {
            e.add(c.listen("DOMMouseScroll", runOn(this, this.b9)));
            e.add(c.listen("mousewheel", runOn(this, this.b9)));
        }
    }
    dg(e_) {
        let a = this.cw(e_);
        if (a) {
            this.bx = this.bx - 1;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            let b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
        else {
            if (a) {
                this.dh(e_);
            }
            else {
                this.ca(e_);
            }
        }
    }
    db(e_) {
        let a = this.cw(e_);
        if (a) {
            this.bx--;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            let b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
    }
    dc(e_) {
        let a = this.cw(e_);
        if (a) {
            this.bx--;
            if (this.bx < 0) {
                this.bx = 0;
            }
        }
        if (this.bx < 2 && this.cv && a) {
            this.cv = false;
            let b = (e_.originalEvent.scale);
            this.aw(this.cr, b);
        }
    }
    dd(e_) {
        let a = this.cw(e_);
        if (this.c2 != null && a) {
            this.bx++;
            this.c2.addPointer((e_.originalEvent.pointerId));
        }
        let eventSource_ = this.bt.getNativeElement();
        if (BaseDOMEventProxy.bk) {
            eventSource_.setPointerCapture(e_.originalEvent.pointerId);
        }
        else {
            eventSource_.msSetPointerCapture(e_.originalEvent.pointerId);
        }
        if (this.bx > 1 && !this.cv && a) {
            this.cv = true;
            this.cn(e_);
            let b = (e_.originalEvent.scale);
            this.a4(this.cr, b);
        }
        else {
            if (a) {
                this.dj(e_);
            }
            else {
                this.b6(e_);
            }
        }
    }
    cw(e_) {
        let pointerEvent_ = e_.originalEvent;
        let a = false;
        if (BaseDOMEventProxy.bk) {
            a = (pointerEvent_.pointerType == 'touch');
        }
        else {
            a = (pointerEvent_.pointerType == pointerEvent_.MSPOINTER_TYPE_TOUCH);
        }
        return a;
    }
    de(e_) {
        let a = this.cw(e_);
        if (this.cv) {
            return;
        }
        if (a) {
            this.di(e_);
        }
        else {
            this.b8(e_);
        }
    }
    df(e_) {
        let a = this.cw(e_);
        if (this.cv) {
            return;
        }
        if (a) {
        }
        else {
            this.b7(e_);
        }
    }
    da(e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        this.dl();
        if (this.aa(this.cs, true, false)) {
            this.cj(e_);
            if (BaseDOMEventProxy.bn && this.bx < 2) {
                return;
            }
            let a = (e_.originalEvent.scale);
            this.a4(this.cr, a);
        }
    }
    c8(e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        if (this.aa(this.cs, true, false)) {
            this.cj(e_);
            if (BaseDOMEventProxy.bn && this.bx < 2) {
                return;
            }
            let a = (e_.originalEvent.scale);
            this.a3(this.cr, a);
        }
    }
    c9(e_) {
        this.cj(e_);
        let a = (e_.originalEvent.scale);
        this.bi = false;
        this.cl(e_);
        this.cn(e_);
        if (!BaseDOMEventProxy.bn || this.cv) {
            this.aw(this.cr, a);
        }
    }
    dj(e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        this.dt = this.cr;
        if (this.aa(this.cs, true, false)) {
            if (!this.z) {
                this.cj(e_);
            }
            else {
                this.bl = true;
            }
            this.a1(this.cs, false, true);
            this.ay(this.cs);
            this.ar(this.cr, true);
            this.dp();
        }
    }
    dp() {
        if (this.c3 == -1) {
            this.c0 = this.cs.x;
            this.c1 = this.cs.y;
            this.c3 = this.bs.setTimeout(runOn(this, this.dm), 1500);
        }
    }
    dl() {
        if (Math.abs(this.c0 - this.cs.x) > this.cz || Math.abs(this.c1 - this.cs.y) > this.cz) {
            this.dq();
        }
    }
    dq() {
        if (this.c3 != -1) {
            window.clearTimeout(this.c3);
            this.c3 = -1;
        }
    }
    dm() {
        this.c3 = -1;
        if (this.bl) {
            this.bl = false;
            this.b2();
        }
        this.ax(this.cr);
    }
    di(a) {
        this.bi = true;
        this.cl(a);
        this.cn(a);
        if (this.bl) {
            if (this.aa(this.cs, true, false) && this.cx()) {
                this.bl = false;
                this.b2();
            }
        }
        this.dl();
        this.dr();
        if (this.aa(this.cs, true, false) && this.bx == 1 && !this.bl) {
            this.cj(a);
            if (!this.cu) {
                this.cu = true;
                this.ds = this.cr;
                this.av(this.ds);
            }
            else {
                this.a1(this.cs, true, true);
                this.aq(this.cr, true);
                this.au(this.cr);
            }
        }
    }
    cx() {
        let a = { $type: Point_$type, x: this.dt.x - this.cr.x, y: this.dt.y - this.cr.y };
        if (Math.abs(a.x) < 2 && Math.abs(a.y) < 2) {
            return false;
        }
        if (Math.abs(a.x) > Math.abs(a.y)) {
            a.y = 0;
        }
        if (Math.abs(a.y) > Math.abs(a.x)) {
            a.x = 0;
        }
        return this.aj(a);
    }
    dh(a) {
        this.bi = false;
        this.cl(a);
        this.cn(a);
        this.dq();
        if (this.bl) {
            this.bl = false;
            this.b2();
        }
        this.cj(a);
        this.a1(this.cs, false, true);
        this.a2(this.cs);
        if (this.bx == 0) {
            this.dn(this.cs);
            this.cg();
        }
        if (this.cu && this.bx == 0) {
            this.cu = false;
            this.ds = null;
            this.at(this.cr);
            this.cg();
        }
        this.ap(this.cr, true);
    }
    dn(a) {
        if (Rect.l_op_Equality(this.du, null)) {
            this.du = new Rect(0, a.x - 50, a.y - 50, 100, 100);
            this.bs.setTimeout(runOn(this, this.dr), 500);
        }
        else {
            if (a.x >= this.du.x && a.x <= this.du.right && a.y >= this.du.y && a.y <= this.du.bottom) {
                this.dr();
                this.as(a);
            }
        }
    }
    dr() {
        this.du = null;
    }
    dk(a) {
        if (this.cy.containsKey(a)) {
            let b = this.cy.item(a);
            this.cy.removeItem(a);
            for (let c = 0; c < b.count; c++) {
                b._inner[c]();
            }
            b.clear();
        }
    }
    a8(a, b) {
        let c = a;
        let d = this.c6 + b;
        this.dk(d);
    }
    an() {
        if (this.bt == null) {
            return;
        }
        this.dk(this.c6);
        this.bt = null;
    }
}
DOMEventProxy.$t = markType(DOMEventProxy, 'DOMEventProxy', BaseDOMEventProxy.$);
DOMEventProxy.c4 = 0;
