// igniteui-angular-core/ES2015/BubbleSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ScatterBaseDescription } from "./ScatterBaseDescription";
import { markType } from "./type";
export class BubbleSeriesDescription extends ScatterBaseDescription {
    constructor() {
        super();
        this.cx = null;
        this.ct = null;
        this.cw = null;
        this.cv = null;
        this.cr = null;
    }
    get_type() {
        return "BubbleSeries";
    }
    get radiusMemberPath() {
        return this.cx;
    }
    set radiusMemberPath(a) {
        this.cx = a;
        this.e("RadiusMemberPath");
    }
    get radiusScale() {
        return this.ct;
    }
    set radiusScale(a) {
        this.ct = a;
        this.e("RadiusScale");
    }
    get labelMemberPath() {
        return this.cw;
    }
    set labelMemberPath(a) {
        this.cw = a;
        this.e("LabelMemberPath");
    }
    get fillMemberPath() {
        return this.cv;
    }
    set fillMemberPath(a) {
        this.cv = a;
        this.e("FillMemberPath");
    }
    get fillScale() {
        return this.cr;
    }
    set fillScale(a) {
        this.cr = a;
        this.e("FillScale");
    }
}
BubbleSeriesDescription.$t = markType(BubbleSeriesDescription, 'BubbleSeriesDescription', ScatterBaseDescription.$);
