// igniteui-angular-core/ES2015/CalloutLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
export class CalloutLayerDescription extends AnnotationLayerDescription {
    constructor() {
        super();
        this.cv = null;
        this.cy = null;
        this.cz = null;
        this.cu = null;
        this.b0 = false;
        this.ct = null;
        this.cw = null;
        this.cs = null;
        this.bz = false;
        this.b2 = false;
        this.b1 = false;
        this.cj = 0;
        this.cr = null;
        this.cn = null;
        this.ck = 0;
        this.cq = null;
        this.cp = null;
        this.b8 = 0;
        this.ca = 0;
        this.b9 = 0;
        this.b7 = 0;
        this.co = null;
        this.cb = 0;
        this.cc = 0;
        this.cx = null;
        this._CalloutStyleUpdatingRef = null;
        this._CalloutContentUpdatingRef = null;
        this._CalloutLabelUpdatingRef = null;
        this._CalloutSeriesSelectingRef = null;
    }
    get_type() {
        return "CalloutLayer";
    }
    get labelMemberPath() {
        return this.cv;
    }
    set labelMemberPath(a) {
        this.cv = a;
        this.e("LabelMemberPath");
    }
    get xMemberPath() {
        return this.cy;
    }
    set xMemberPath(a) {
        this.cy = a;
        this.e("XMemberPath");
    }
    get yMemberPath() {
        return this.cz;
    }
    set yMemberPath(a) {
        this.cz = a;
        this.e("YMemberPath");
    }
    get keyMemberPath() {
        return this.cu;
    }
    set keyMemberPath(a) {
        this.cu = a;
        this.e("KeyMemberPath");
    }
    get isCalloutOffsettingEnabled() {
        return this.b0;
    }
    set isCalloutOffsettingEnabled(a) {
        this.b0 = a;
        this.e("IsCalloutOffsettingEnabled");
    }
    get contentMemberPath() {
        return this.ct;
    }
    set contentMemberPath(a) {
        this.ct = a;
        this.e("ContentMemberPath");
    }
    get targetSeriesRef() {
        return this.cw;
    }
    set targetSeriesRef(a) {
        this.cw = a;
        this.e("TargetSeriesRef");
    }
    get collisionChannel() {
        return this.cs;
    }
    set collisionChannel(a) {
        this.cs = a;
        this.e("CollisionChannel");
    }
    get isAutoCalloutBehaviorEnabled() {
        return this.bz;
    }
    set isAutoCalloutBehaviorEnabled(a) {
        this.bz = a;
        this.e("IsAutoCalloutBehaviorEnabled");
    }
    get useValueForAutoCalloutLabels() {
        return this.b2;
    }
    set useValueForAutoCalloutLabels(a) {
        this.b2 = a;
        this.e("UseValueForAutoCalloutLabels");
    }
    get isCustomCalloutStyleEnabled() {
        return this.b1;
    }
    set isCustomCalloutStyleEnabled(a) {
        this.b1 = a;
        this.e("IsCustomCalloutStyleEnabled");
    }
    get autoCalloutLabelPrecision() {
        return this.cj;
    }
    set autoCalloutLabelPrecision(a) {
        this.cj = a;
        this.e("AutoCalloutLabelPrecision");
    }
    get calloutTextColor() {
        return this.cr;
    }
    set calloutTextColor(a) {
        this.cr = a;
        this.e("CalloutTextColor");
    }
    get calloutBackground() {
        return this.cn;
    }
    set calloutBackground(a) {
        this.cn = a;
        this.e("CalloutBackground");
    }
    get calloutInterpolatedValuePrecision() {
        return this.ck;
    }
    set calloutInterpolatedValuePrecision(a) {
        this.ck = a;
        this.e("CalloutInterpolatedValuePrecision");
    }
    get calloutOutline() {
        return this.cq;
    }
    set calloutOutline(a) {
        this.cq = a;
        this.e("CalloutOutline");
    }
    get calloutLeaderBrush() {
        return this.cp;
    }
    set calloutLeaderBrush(a) {
        this.cp = a;
        this.e("CalloutLeaderBrush");
    }
    get calloutPaddingLeft() {
        return this.b8;
    }
    set calloutPaddingLeft(a) {
        this.b8 = a;
        this.e("CalloutPaddingLeft");
    }
    get calloutPaddingTop() {
        return this.ca;
    }
    set calloutPaddingTop(a) {
        this.ca = a;
        this.e("CalloutPaddingTop");
    }
    get calloutPaddingRight() {
        return this.b9;
    }
    set calloutPaddingRight(a) {
        this.b9 = a;
        this.e("CalloutPaddingRight");
    }
    get calloutPaddingBottom() {
        return this.b7;
    }
    set calloutPaddingBottom(a) {
        this.b7 = a;
        this.e("CalloutPaddingBottom");
    }
    get calloutCollisionMode() {
        return this.co;
    }
    set calloutCollisionMode(a) {
        this.co = a;
        this.e("CalloutCollisionMode");
    }
    get calloutPositionPadding() {
        return this.cb;
    }
    set calloutPositionPadding(a) {
        this.cb = a;
        this.e("CalloutPositionPadding");
    }
    get calloutStrokeThickness() {
        return this.cc;
    }
    set calloutStrokeThickness(a) {
        this.cc = a;
        this.e("CalloutStrokeThickness");
    }
    get textStyle() {
        return this.cx;
    }
    set textStyle(a) {
        this.cx = a;
        this.e("TextStyle");
    }
    get CalloutStyleUpdatingRef() {
        return this._CalloutStyleUpdatingRef;
    }
    set CalloutStyleUpdatingRef(a) {
        this._CalloutStyleUpdatingRef = a;
    }
    get CalloutContentUpdatingRef() {
        return this._CalloutContentUpdatingRef;
    }
    set CalloutContentUpdatingRef(a) {
        this._CalloutContentUpdatingRef = a;
    }
    get CalloutLabelUpdatingRef() {
        return this._CalloutLabelUpdatingRef;
    }
    set CalloutLabelUpdatingRef(a) {
        this._CalloutLabelUpdatingRef = a;
    }
    get CalloutSeriesSelectingRef() {
        return this._CalloutSeriesSelectingRef;
    }
    set CalloutSeriesSelectingRef(a) {
        this._CalloutSeriesSelectingRef = a;
    }
}
CalloutLayerDescription.$t = markType(CalloutLayerDescription, 'CalloutLayerDescription', AnnotationLayerDescription.$);
