// igniteui-angular-core/ES2015/CanvasGestureDOMEventProxy.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BaseDOMEventProxy } from "./BaseDOMEventProxy";
import { TypeRegistrar, runOn, delegateCombine, Point_$type, String_$type, Delegate_$type, Number_$type, markType } from "./type";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
import { TouchPointInfo } from "./TouchPointInfo";
import { DOMExecutionContext } from "./DOMExecutionContext";
import { TouchGestureRecognizer } from "./TouchGestureRecognizer";
export class CanvasGestureDOMEventProxy extends BaseDOMEventProxy {
    constructor(a, b, c) {
        super();
        this.c8 = 0;
        this.c9 = "";
        this.cu = null;
        this.c6 = null;
        this.c1 = new Dictionary$2(String_$type, List$1.$.specialize(Delegate_$type), 0);
        this.dx = null;
        this.cv = false;
        this.c2 = new List$1(Number_$type, 0);
        this.c3 = new List$1(Point_$type, 0);
        this.cx = false;
        this.c4 = new List$1(TouchPointInfo.$, 0);
        this.bm = c;
        this.bs = b;
        this.cu = new TouchGestureRecognizer(new DOMExecutionContext(b));
        let d = this.cu;
        d.holdEnding = delegateCombine(d.holdEnding, runOn(this, this.db));
        let e = this.cu;
        e.holdStarting = delegateCombine(e.holdStarting, runOn(this, this.dc));
        let f = this.cu;
        f.zoomStarting = delegateCombine(f.zoomStarting, runOn(this, this.dj));
        let g = this.cu;
        g.zoomEnding = delegateCombine(g.zoomEnding, runOn(this, this.di));
        let h = this.cu;
        h.panStarting = delegateCombine(h.panStarting, runOn(this, this.dh));
        let i = this.cu;
        i.panEnding = delegateCombine(i.panEnding, runOn(this, this.dg));
        let j = this.cu;
        j.manipulationStarting = delegateCombine(j.manipulationStarting, runOn(this, this.df));
        let k = this.cu;
        k.manipulationChanging = delegateCombine(k.manipulationChanging, runOn(this, this.dd));
        let l = this.cu;
        l.manipulationEnding = delegateCombine(l.manipulationEnding, runOn(this, this.de));
        let m = this.cu;
        m.doubleTapped = delegateCombine(m.doubleTapped, runOn(this, this.da));
        let n = this.cu;
        n.flingStarting = delegateCombine(n.flingStarting, runOn(this, this.cw));
        this.aj = (o) => true;
        this.z = true;
        CanvasGestureDOMEventProxy.c7++;
        this.c8 = CanvasGestureDOMEventProxy.c7;
        this.c9 = ".DOMProxy" + this.c8.toString();
        this.a9 = 0;
        this.bt = a;
        BaseDOMEventProxy.bj = false;
        try {
            BaseDOMEventProxy.bj = (window.navigator.msPointerEnabled && MSGesture !== undefined);
        }
        catch (o) {
        }
        BaseDOMEventProxy.bk = false;
        try {
            BaseDOMEventProxy.bk = (window.PointerEvent || window.navigator.pointerEnabled);
        }
        catch (p) {
        }
        if (BaseDOMEventProxy.bn) {
            this.z = false;
        }
        BaseDOMEventProxy.b1 = this.b0();
        BaseDOMEventProxy.by = this.bz();
        this.am(this.bt, "");
        this.ai = (q) => true;
    }
    d() {
        let a = new CanvasGestureDOMEventProxy(this.bt, this.bs, this.bm);
        a.z = this.z;
        a.aj = this.aj;
        return a;
    }
    cw(a, b) {
        return this.ac({ $type: Point_$type, x: b.a, y: b.b }, b.h, b.i);
    }
    da(a, b) {
        this.as({ $type: Point_$type, x: b.a, y: b.b });
    }
    get_ah() {
        return true;
    }
    de(a, b) {
        if (this.cu.n == 0) {
            this.aw({ $type: Point_$type, x: b.a, y: b.b }, b.d);
            this.cg();
        }
    }
    dd(a, b) {
        if (this.cu.n == 2) {
            this.a3({ $type: Point_$type, x: b.a, y: b.b }, b.d);
            this.a6({ $type: Point_$type, x: b.a, y: b.b }, b.d, b.e, b.c);
        }
        if (this.cu.n == 1 || this.cu.n == 5) {
            this.aq({ $type: Point_$type, x: b.a, y: b.b }, true);
            this.au({ $type: Point_$type, x: b.a, y: b.b });
        }
    }
    df(a, b) {
    }
    dg(a, b) {
        this.cv = false;
        if (this.cu.n == 0) {
            this.ap({ $type: Point_$type, x: b.a, y: b.b }, true);
            this.at({ $type: Point_$type, x: b.a, y: b.b });
        }
    }
    dh(a, b) {
        this.av({ $type: Point_$type, x: b.a, y: b.b });
    }
    di(a, b) {
        this.a5({ $type: Point_$type, x: b.a, y: b.b }, b.d, b.e, b.c);
    }
    dj(a, b) {
        this.a4({ $type: Point_$type, x: b.a, y: b.b }, 1);
        this.a7({ $type: Point_$type, x: b.a, y: b.b }, 1, 1, 1);
    }
    dc(a, b) {
        this.cv = true;
        this.ax({ $type: Point_$type, x: b.a, y: b.b });
    }
    db(a, b) {
        if (this.cu.n == 0) {
            this.ap({ $type: Point_$type, x: b.a, y: b.b }, true);
        }
    }
    static cz() {
        BaseDOMEventProxy.bj = false;
        try {
            BaseDOMEventProxy.bj = (window.navigator.msPointerEnabled && MSGesture !== undefined);
        }
        catch (a) {
        }
        BaseDOMEventProxy.bk = false;
        try {
            BaseDOMEventProxy.bk = (window.PointerEvent || window.navigator.pointerEnabled);
        }
        catch (b) {
        }
        return BaseDOMEventProxy.bn;
    }
    dk(a, b, c) {
    }
    am(a, b) {
        let c = a;
        if (!this.bs.supportsDOMEvents()) {
            return;
        }
        let d = this.c9 + b;
        let e = new List$1(Delegate_$type, 0);
        this.c1.addItem(d, e);
        this.dk(a, b, e);
        if (!BaseDOMEventProxy.bn) {
            e.add(c.listen("mousemove", runOn(this, this.b8)));
            e.add(c.listen("mouseleave", runOn(this, this.b7)));
            e.add(c.listen("mousedown", runOn(this, this.b6)));
            e.add(c.listen("mouseup", runOn(this, this.ca)));
            e.add(this.bs.globalListen("window", "mouseup", runOn(this, this.co)));
        }
        e.add(c.listen("keydown", runOn(this, this.b3)));
        e.add(c.listen("keyup", runOn(this, this.b5)));
        if (BaseDOMEventProxy.bn) {
            let source_ = this.bt.getNativeElement();
            BaseDOMEventProxy.ci(source_);
            let f = this.bt.getNativeElement();
            if (BaseDOMEventProxy.bk) {
                e.add(c.listen("pointerdown", runOn(this, this.dn)));
                e.add(c.listen("pointerup", runOn(this, this.dr)));
                e.add(c.listen("pointercancel", runOn(this, this.dm)));
                e.add(c.listen("pointermove", runOn(this, this.dp)));
                e.add(c.listen("pointerout", runOn(this, this.dq)));
                e.add(c.listen("lostpointercapture", runOn(this, this.dl)));
            }
            else {
                e.add(c.listen("MSPointerDown", runOn(this, this.dn)));
                e.add(c.listen("MSPointerUp", runOn(this, this.dr)));
                e.add(c.listen("MSPointerCancel", runOn(this, this.dm)));
                e.add(c.listen("MSPointerMove", runOn(this, this.dp)));
                e.add(c.listen("MSPointerOut", runOn(this, this.dq)));
                e.add(c.listen("MSLostPointerCapture", runOn(this, this.dl)));
            }
        }
        else {
            e.add(c.listen("touchstart", runOn(this, this.du)));
            e.add(c.listen("touchmove", runOn(this, this.dt)));
            e.add(c.listen("touchend", runOn(this, this.ds)));
        }
        if (this.bm) {
            e.add(c.listen("wheel", runOn(this, this.cb)));
        }
        else {
            e.add(c.listen("DOMMouseScroll", runOn(this, this.b9)));
            e.add(c.listen("mousewheel", runOn(this, this.b9)));
        }
    }
    dr(e_) {
        let a = this.cy(e_);
        let b = this.c2.indexOf((e_.originalEvent.pointerId));
        if (b >= 0) {
            this.c2.removeAt(b);
            this.c3.removeAt(b);
            this.bx = this.c2.count;
        }
        if (a) {
            this.ds(e_);
        }
        else {
            this.ca(e_);
        }
    }
    dl(e_) {
        let a = this.cy(e_);
    }
    dm(e_) {
        let a = this.cy(e_);
        let b = this.c2.indexOf((e_.originalEvent.pointerId));
        if (b >= 0) {
            this.c2.removeAt(b);
            this.c3.removeAt(b);
            this.bx = this.c2.count;
        }
    }
    dn(e_) {
        let a = this.cy(e_);
        BaseDOMEventProxy.ch(e_);
        this.c2.add((e_.originalEvent.pointerId));
        this.c3.add({ $type: Point_$type, x: e_.pageX, y: e_.pageY });
        this.bx = this.c2.count;
        if (a) {
            this.cn(e_);
        }
        else {
            this.cm(e_);
        }
        this.cl(e_);
        if (!this.cx && this.aa(this.cs, a, false)) {
            let eventSource_ = this.bt.getNativeElement();
            if (BaseDOMEventProxy.bk) {
                eventSource_.setPointerCapture(e_.originalEvent.pointerId);
            }
            else {
                eventSource_.msSetPointerCapture(e_.originalEvent.pointerId);
            }
        }
        if (a) {
            this.du(e_);
        }
        else {
            this.b6(e_);
        }
    }
    cy(e_) {
        let pointerEvent_ = e_.originalEvent;
        let a = false;
        if (BaseDOMEventProxy.bk) {
            a = (pointerEvent_.pointerType == 'touch');
        }
        else {
            a = (pointerEvent_.pointerType == pointerEvent_.MSPOINTER_TYPE_TOUCH);
        }
        return a;
    }
    dp(e_) {
        let a = this.cy(e_);
        BaseDOMEventProxy.ch(e_);
        this.cl(e_);
        this.cn(e_);
        let b = this.c2.indexOf((e_.originalEvent.pointerId));
        if (b >= 0) {
            this.c3._inner[b] = { $type: Point_$type, x: e_.pageX, y: e_.pageY };
        }
        if (this.cx && this.aa(this.cs, a, false)) {
            let eventSource_ = this.bt.getNativeElement();
            if (BaseDOMEventProxy.bk) {
                eventSource_.setPointerCapture(e_.originalEvent.pointerId);
            }
            else {
                eventSource_.msSetPointerCapture(e_.originalEvent.pointerId);
            }
        }
        if (a) {
            this.dt(e_);
        }
        else {
            this.b8(e_);
        }
    }
    dq(e_) {
        let a = this.cy(e_);
        if (a) {
        }
        else {
            this.b7(e_);
        }
    }
    du(e_) {
        this.bi = true;
        this.cl(e_);
        this.cn(e_);
        this.dx = this.cr;
        if (this.aa(this.cs, true, false)) {
            if (!this.z || this.bx > 1) {
                if (this.bl) {
                    this.bl = false;
                    this.b2();
                }
                this.cj(e_);
            }
            else {
                this.bl = true;
                this.cv = false;
            }
            this.a1(this.cs, false, true);
            this.ay(this.cs);
            this.ar(this.cr, true);
            this.cu.a6(this.c5(e_));
        }
    }
    c5(e_) {
        let a = BaseDOMEventProxy.ct(this.bt);
        let b = this.bc;
        this.c4.clear();
        if (BaseDOMEventProxy.bn) {
            let c = this.c2.count;
            for (let j_ = 0; j_ < c; j_++) {
                let d = this.c2._inner[j_];
                let e = this.c3._inner[j_];
                let f = e.x;
                let g = e.y;
                let h = new TouchPointInfo(1);
                h.a = (f - a.left) - b.left;
                h.b = (g - a.top) - b.top;
                this.c4.add(h);
            }
        }
        else {
            let i = (e_.originalEvent.targetTouches.length);
            for (let j_ = 0; j_ < i; j_++) {
                let j = (e_.originalEvent.targetTouches[j_].pageX);
                let k = (e_.originalEvent.targetTouches[j_].pageY);
                let l = new TouchPointInfo(1);
                l.a = (j - a.left) - b.left;
                l.b = (k - a.top) - b.top;
                this.c4.add(l);
            }
        }
        return this.c4;
    }
    dt(a) {
        this.bi = true;
        this.cl(a);
        this.cn(a);
        if (this.bl) {
            if ((this.aa(this.cs, true, false) && this.c0()) || this.cv == true || this.cu.n == 2) {
                this.bl = false;
                this.b2();
            }
        }
        if ((this.cv || this.cu.n == 2 || this.aa(this.cs, true, false)) && !this.bl) {
            this.cv = true;
            this.cj(a);
            this.cu.a6(this.c5(a));
        }
    }
    c0() {
        let a = { $type: Point_$type, x: this.dx.x - this.cr.x, y: this.dx.y - this.cr.y };
        if (Math.abs(a.x) > Math.abs(a.y)) {
            a.y = 0;
        }
        if (Math.abs(a.y) > Math.abs(a.x)) {
            a.x = 0;
        }
        return this.aj(a);
    }
    ds(a) {
        this.bi = false;
        this.cl(a);
        this.cn(a);
        if (this.bl) {
            this.bl = false;
            this.b2();
        }
        let b = this.cu.n;
        let c = this.aa(this.cs, true, false) || b != 0;
        if (this.cu.n != 2 && this.cu.n != 5) {
            this.a1(this.cs, false, true);
            this.a2(this.cs);
        }
        let d = this.cu.n == 2;
        if (d) {
            c = true;
            this.ap({ $type: Point_$type, x: this.cu.as, y: this.cu.at }, true);
        }
        this.cu.a6(this.c5(a));
        if (this.cu.n == 0) {
            this.cg();
        }
        if (!d && this.cu.n == 0) {
            this.ap({ $type: Point_$type, x: this.cu.as, y: this.cu.at }, true);
        }
        if (this.cu.n != 0) {
            c = true;
        }
        if (c) {
            this.cj(a);
        }
    }
    dv(a) {
        if (this.c1.containsKey(a)) {
            let b = this.c1.item(a);
            this.c1.removeItem(a);
            for (let c = 0; c < b.count; c++) {
                b._inner[c]();
            }
            b.clear();
        }
    }
    a8(a, b) {
        let c = a;
        let d = this.c9 + b;
        this.dv(d);
    }
    an() {
        if (this.bt == null) {
            return;
        }
        this.dv(this.c9);
        this.bt = null;
    }
    static register() {
        TypeRegistrar.register("CanvasGestureDOMEventProxy", CanvasGestureDOMEventProxy.$);
    }
}
CanvasGestureDOMEventProxy.$t = markType(CanvasGestureDOMEventProxy, 'CanvasGestureDOMEventProxy', BaseDOMEventProxy.$);
CanvasGestureDOMEventProxy.c7 = 0;
