// igniteui-angular-core/ES2015/CategoryAngleAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
export class CategoryAngleAxisDescription extends CategoryAxisBaseDescription {
    constructor() {
        super();
        this.c0 = 0;
        this.cy = 0;
        this.cw = 0;
        this.cz = 0;
        this.cx = 0;
    }
    get_type() {
        return "CategoryAngleAxis";
    }
    get startAngleOffset() {
        return this.c0;
    }
    set startAngleOffset(a) {
        this.c0 = a;
        this.e("StartAngleOffset");
    }
    get interval() {
        return this.cy;
    }
    set interval(a) {
        this.cy = a;
        this.e("Interval");
    }
    get actualInterval() {
        return this.cw;
    }
    set actualInterval(a) {
        this.cw = a;
        this.e("ActualInterval");
    }
    get minorInterval() {
        return this.cz;
    }
    set minorInterval(a) {
        this.cz = a;
        this.e("MinorInterval");
    }
    get actualMinorInterval() {
        return this.cx;
    }
    set actualMinorInterval(a) {
        this.cx = a;
        this.e("ActualMinorInterval");
    }
}
CategoryAngleAxisDescription.$t = markType(CategoryAngleAxisDescription, 'CategoryAngleAxisDescription', CategoryAxisBaseDescription.$);
