// igniteui-angular-core/ES2015/CategoryChartDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XYChartDescription } from "./XYChartDescription";
import { markType } from "./type";
export class CategoryChartDescription extends XYChartDescription {
    constructor() {
        super();
        this.hp = 0;
        this.hv = null;
        this.ht = null;
        this.hu = null;
        this.g1 = false;
        this.hw = null;
        this.hx = null;
        this.ha = 0;
        this.hb = 0;
        this.g9 = 0;
        this.hc = 0;
        this.hd = 0;
        this.g3 = false;
        this.hq = 0;
        this.hf = 0;
        this.he = 0;
        this.hg = 0;
        this.gv = null;
        this.gw = null;
        this.g2 = false;
        this.gz = false;
        this.g0 = false;
    }
    get_type() {
        return "CategoryChart";
    }
    get transitionInDuration() {
        return this.hp;
    }
    set transitionInDuration(a) {
        this.hp = a;
        this.e("TransitionInDuration");
    }
    get transitionInEasingFunctionRef() {
        return this.hv;
    }
    set transitionInEasingFunctionRef(a) {
        this.hv = a;
        this.e("TransitionInEasingFunctionRef");
    }
    get chartType() {
        return this.ht;
    }
    set chartType(a) {
        this.ht = a;
        this.e("ChartType");
    }
    get markerCollisionAvoidance() {
        return this.hu;
    }
    set markerCollisionAvoidance(a) {
        this.hu = a;
        this.e("MarkerCollisionAvoidance");
    }
    get isTransitionInEnabled() {
        return this.g1;
    }
    set isTransitionInEnabled(a) {
        this.g1 = a;
        this.e("IsTransitionInEnabled");
    }
    get transitionInMode() {
        return this.hw;
    }
    set transitionInMode(a) {
        this.hw = a;
        this.e("TransitionInMode");
    }
    get transitionInSpeedType() {
        return this.hx;
    }
    set transitionInSpeedType(a) {
        this.hx = a;
        this.e("TransitionInSpeedType");
    }
    get xAxisInterval() {
        return this.ha;
    }
    set xAxisInterval(a) {
        this.ha = a;
        this.e("XAxisInterval");
    }
    get xAxisMinorInterval() {
        return this.hb;
    }
    set xAxisMinorInterval(a) {
        this.hb = a;
        this.e("XAxisMinorInterval");
    }
    get xAxisGap() {
        return this.g9;
    }
    set xAxisGap(a) {
        this.g9 = a;
        this.e("XAxisGap");
    }
    get xAxisOverlap() {
        return this.hc;
    }
    set xAxisOverlap(a) {
        this.hc = a;
        this.e("XAxisOverlap");
    }
    get yAxisInterval() {
        return this.hd;
    }
    set yAxisInterval(a) {
        this.hd = a;
        this.e("YAxisInterval");
    }
    get yAxisIsLogarithmic() {
        return this.g3;
    }
    set yAxisIsLogarithmic(a) {
        this.g3 = a;
        this.e("YAxisIsLogarithmic");
    }
    get yAxisLogarithmBase() {
        return this.hq;
    }
    set yAxisLogarithmBase(a) {
        this.hq = a;
        this.e("YAxisLogarithmBase");
    }
    get yAxisMinimumValue() {
        return this.hf;
    }
    set yAxisMinimumValue(a) {
        this.hf = a;
        this.e("YAxisMinimumValue");
    }
    get yAxisMaximumValue() {
        return this.he;
    }
    set yAxisMaximumValue(a) {
        this.he = a;
        this.e("YAxisMaximumValue");
    }
    get yAxisMinorInterval() {
        return this.hg;
    }
    set yAxisMinorInterval(a) {
        this.hg = a;
        this.e("YAxisMinorInterval");
    }
    get negativeBrushes() {
        return this.gv;
    }
    set negativeBrushes(a) {
        this.gv = a;
        this.e("NegativeBrushes");
    }
    get negativeOutlines() {
        return this.gw;
    }
    set negativeOutlines(a) {
        this.gw = a;
        this.e("NegativeOutlines");
    }
    get yAxisAbbreviateLargeNumbers() {
        return this.g2;
    }
    set yAxisAbbreviateLargeNumbers(a) {
        this.g2 = a;
        this.e("YAxisAbbreviateLargeNumbers");
    }
    get isCategoryHighlightingEnabled() {
        return this.gz;
    }
    set isCategoryHighlightingEnabled(a) {
        this.gz = a;
        this.e("IsCategoryHighlightingEnabled");
    }
    get isItemHighlightingEnabled() {
        return this.g0;
    }
    set isItemHighlightingEnabled(a) {
        this.g0 = a;
        this.e("IsItemHighlightingEnabled");
    }
}
CategoryChartDescription.$t = markType(CategoryChartDescription, 'CategoryChartDescription', XYChartDescription.$);
