// igniteui-angular-core/ES2015/CategorySeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { MarkerSeriesDescription } from "./MarkerSeriesDescription";
import { markType } from "./type";
export class CategorySeriesDescription extends MarkerSeriesDescription {
    constructor() {
        super();
        this.b4 = false;
        this.b3 = false;
        this.b6 = false;
        this.cb = null;
        this.b5 = false;
        this._AssigningCategoryStyleRef = null;
        this._AssigningCategoryMarkerStyleRef = null;
    }
    get_type() {
        return "CategorySeries";
    }
    get isCustomCategoryStyleAllowed() {
        return this.b4;
    }
    set isCustomCategoryStyleAllowed(a) {
        this.b4 = a;
        this.e("IsCustomCategoryStyleAllowed");
    }
    get isCustomCategoryMarkerStyleAllowed() {
        return this.b3;
    }
    set isCustomCategoryMarkerStyleAllowed(a) {
        this.b3 = a;
        this.e("IsCustomCategoryMarkerStyleAllowed");
    }
    get useHighMarkerFidelity() {
        return this.b6;
    }
    set useHighMarkerFidelity(a) {
        this.b6 = a;
        this.e("UseHighMarkerFidelity");
    }
    get transitionInMode() {
        return this.cb;
    }
    set transitionInMode(a) {
        this.cb = a;
        this.e("TransitionInMode");
    }
    get isTransitionInEnabled() {
        return this.b5;
    }
    set isTransitionInEnabled(a) {
        this.b5 = a;
        this.e("IsTransitionInEnabled");
    }
    get AssigningCategoryStyleRef() {
        return this._AssigningCategoryStyleRef;
    }
    set AssigningCategoryStyleRef(a) {
        this._AssigningCategoryStyleRef = a;
    }
    get AssigningCategoryMarkerStyleRef() {
        return this._AssigningCategoryMarkerStyleRef;
    }
    set AssigningCategoryMarkerStyleRef(a) {
        this._AssigningCategoryMarkerStyleRef = a;
    }
}
CategorySeriesDescription.$t = markType(CategorySeriesDescription, 'CategorySeriesDescription', MarkerSeriesDescription.$);
