// igniteui-angular-core/ES2015/CategoryItemHighlightLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
export class CategoryItemHighlightLayerDescription extends AnnotationLayerDescription {
    constructor() {
        super();
        this.b7 = null;
        this.b0 = false;
        this.b5 = null;
        this.b6 = null;
        this.b3 = 0;
        this.bz = false;
    }
    get_type() {
        return "CategoryItemHighlightLayer";
    }
    get targetSeriesRef() {
        return this.b7;
    }
    set targetSeriesRef(a) {
        this.b7 = a;
        this.e("TargetSeriesRef");
    }
    get useInterpolation() {
        return this.b0;
    }
    set useInterpolation(a) {
        this.b0 = a;
        this.e("UseInterpolation");
    }
    get highlightType() {
        return this.b5;
    }
    set highlightType(a) {
        this.b5 = a;
        this.e("HighlightType");
    }
    get markerTemplate() {
        return this.b6;
    }
    set markerTemplate(a) {
        this.b6 = a;
        this.e("MarkerTemplate");
    }
    get bandHighlightWidth() {
        return this.b3;
    }
    set bandHighlightWidth(a) {
        this.b3 = a;
        this.e("BandHighlightWidth");
    }
    get skipUnknownValues() {
        return this.bz;
    }
    set skipUnknownValues(a) {
        this.bz = a;
        this.e("SkipUnknownValues");
    }
}
CategoryItemHighlightLayerDescription.$t = markType(CategoryItemHighlightLayerDescription, 'CategoryItemHighlightLayerDescription', AnnotationLayerDescription.$);
