// igniteui-angular-core/ES2015/CategoryDateTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { TimeAxisBaseDescription } from "./TimeAxisBaseDescription";
import { markType } from "./type";
export class CategoryDateTimeXAxisDescription extends TimeAxisBaseDescription {
    constructor() {
        super();
        this.dg = null;
        this.da = 0;
        this.c8 = 0;
        this.db = 0;
        this.c9 = 0;
    }
    get_type() {
        return "CategoryDateTimeXAxis";
    }
    get displayType() {
        return this.dg;
    }
    set displayType(a) {
        this.dg = a;
        this.e("DisplayType");
    }
    get interval() {
        return this.da;
    }
    set interval(a) {
        this.da = a;
        this.e("Interval");
    }
    get actualInterval() {
        return this.c8;
    }
    set actualInterval(a) {
        this.c8 = a;
        this.e("ActualInterval");
    }
    get minorInterval() {
        return this.db;
    }
    set minorInterval(a) {
        this.db = a;
        this.e("MinorInterval");
    }
    get actualMinorInterval() {
        return this.c9;
    }
    set actualMinorInterval(a) {
        this.c9 = a;
        this.e("ActualMinorInterval");
    }
}
CategoryDateTimeXAxisDescription.$t = markType(CategoryDateTimeXAxisDescription, 'CategoryDateTimeXAxisDescription', TimeAxisBaseDescription.$);
