// igniteui-angular-core/ES2015/CellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class CellInfoDescription extends Description {
    constructor() {
        super();
        this.bo = null;
        this.au = 0;
        this.m = false;
        this.q = false;
        this.h = false;
        this.k = false;
        this.r = false;
        this.i = false;
        this.l = false;
        this.f = null;
        this.n = false;
        this.o = false;
        this.j = false;
        this.bk = null;
        this.aj = 0;
        this.ak = 0;
        this.a0 = 0;
        this.a1 = 0;
        this.ae = 0;
        this.af = 0;
        this.ad = 0;
        this.av = 0;
        this.a2 = 0;
        this.p = false;
        this.ah = 0;
        this.be = null;
        this.bq = null;
        this.bd = null;
        this.bc = null;
        this.bl = null;
        this.bn = null;
        this.bi = null;
        this.bf = null;
        this.ag = 0;
        this.bh = null;
        this.bj = null;
        this.bp = null;
        this._fontFamily = null;
        this._fontSize = 0;
        this._fontStyle = null;
        this.bg = null;
        this.ai = 0;
        this.ax = 0;
        this.az = 0;
        this.ay = 0;
        this.aw = 0;
        this.bm = null;
    }
    get_type() {
        return "CellInfo";
    }
    get type() {
        return this.get_type();
    }
    get styleKey() {
        return this.bo;
    }
    set styleKey(a) {
        this.bo = a;
        this.e("StyleKey");
    }
    get dataRow() {
        return this.au;
    }
    set dataRow(a) {
        this.au = a;
        this.e("DataRow");
    }
    get isPositionDirty() {
        return this.m;
    }
    set isPositionDirty(a) {
        this.m = a;
        this.e("IsPositionDirty");
    }
    get isSizeDirty() {
        return this.q;
    }
    set isSizeDirty(a) {
        this.q = a;
        this.e("IsSizeDirty");
    }
    get isContentDirty() {
        return this.h;
    }
    set isContentDirty(a) {
        this.h = a;
        this.e("IsContentDirty");
    }
    get isLayerDirty() {
        return this.k;
    }
    set isLayerDirty(a) {
        this.k = a;
        this.e("IsLayerDirty");
    }
    get isStateDirty() {
        return this.r;
    }
    set isStateDirty(a) {
        this.r = a;
        this.e("IsStateDirty");
    }
    get isDataDirty() {
        return this.i;
    }
    set isDataDirty(a) {
        this.i = a;
        this.e("IsDataDirty");
    }
    get isPlaceholdContentNeeded() {
        return this.l;
    }
    set isPlaceholdContentNeeded(a) {
        this.l = a;
        this.e("IsPlaceholdContentNeeded");
    }
    get actionManager() {
        return this.f;
    }
    set actionManager(a) {
        this.f = a;
        this.e("ActionManager");
    }
    get isRowPinned() {
        return this.n;
    }
    set isRowPinned(a) {
        this.n = a;
        this.e("IsRowPinned");
    }
    get isRowSticky() {
        return this.o;
    }
    set isRowSticky(a) {
        this.o = a;
        this.e("IsRowSticky");
    }
    get isLastStickyRow() {
        return this.j;
    }
    set isLastStickyRow(a) {
        this.j = a;
        this.e("IsLastStickyRow");
    }
    get rowItemRef() {
        return this.bk;
    }
    set rowItemRef(a) {
        this.bk = a;
        this.e("RowItemRef");
    }
    get x() {
        return this.aj;
    }
    set x(a) {
        this.aj = a;
        this.e("X");
    }
    get y() {
        return this.ak;
    }
    set y(a) {
        this.ak = a;
        this.e("Y");
    }
    get snappedX() {
        return this.a0;
    }
    set snappedX(a) {
        this.a0 = a;
        this.e("SnappedX");
    }
    get snappedY() {
        return this.a1;
    }
    set snappedY(a) {
        this.a1 = a;
        this.e("SnappedY");
    }
    get contentOpacity() {
        return this.ae;
    }
    set contentOpacity(a) {
        this.ae = a;
        this.e("ContentOpacity");
    }
    get opacity() {
        return this.af;
    }
    set opacity(a) {
        this.af = a;
        this.e("Opacity");
    }
    get actualOpacity() {
        return this.ad;
    }
    set actualOpacity(a) {
        this.ad = a;
        this.e("ActualOpacity");
    }
    get height() {
        return this.av;
    }
    set height(a) {
        this.av = a;
        this.e("Height");
    }
    get width() {
        return this.a2;
    }
    set width(a) {
        this.a2 = a;
        this.e("Width");
    }
    get isSelected() {
        return this.p;
    }
    set isSelected(a) {
        this.p = a;
        this.e("IsSelected");
    }
    get selectedStatus() {
        return this.ah;
    }
    set selectedStatus(a) {
        this.ah = a;
        this.e("SelectedStatus");
    }
    get horizontalAlignment() {
        return this.be;
    }
    set horizontalAlignment(a) {
        this.be = a;
        this.e("HorizontalAlignment");
    }
    get verticalAlignment() {
        return this.bq;
    }
    set verticalAlignment(a) {
        this.bq = a;
        this.e("VerticalAlignment");
    }
    get background() {
        return this.bd;
    }
    set background(a) {
        this.bd = a;
        this.e("Background");
    }
    get actualBackground() {
        return this.bc;
    }
    set actualBackground(a) {
        this.bc = a;
        this.e("ActualBackground");
    }
    get selectedBackground() {
        return this.bl;
    }
    set selectedBackground(a) {
        this.bl = a;
        this.e("SelectedBackground");
    }
    get stickyRowBackground() {
        return this.bn;
    }
    set stickyRowBackground(a) {
        this.bn = a;
        this.e("StickyRowBackground");
    }
    get pinnedRowBackground() {
        return this.bi;
    }
    set pinnedRowBackground(a) {
        this.bi = a;
        this.e("PinnedRowBackground");
    }
    get lastStickyRowBackground() {
        return this.bf;
    }
    set lastStickyRowBackground(a) {
        this.bf = a;
        this.e("LastStickyRowBackground");
    }
    get pinnedRowOpacity() {
        return this.ag;
    }
    set pinnedRowOpacity(a) {
        this.ag = a;
        this.e("PinnedRowOpacity");
    }
    get originalValueRef() {
        return this.bh;
    }
    set originalValueRef(a) {
        this.bh = a;
        this.e("OriginalValueRef");
    }
    get renderValue() {
        return this.bj;
    }
    set renderValue(a) {
        this.bj = a;
        this.e("RenderValue");
    }
    get textColor() {
        return this.bp;
    }
    set textColor(a) {
        this.bp = a;
        this.e("TextColor");
    }
    get fontFamily() {
        return this._fontFamily;
    }
    set fontFamily(a) {
        this._fontFamily = a;
    }
    get fontSize() {
        return this._fontSize;
    }
    set fontSize(a) {
        this._fontSize = a;
    }
    get fontStyle() {
        return this._fontStyle;
    }
    set fontStyle(a) {
        this._fontStyle = a;
    }
    get lineBreakMode() {
        return this.bg;
    }
    set lineBreakMode(a) {
        this.bg = a;
        this.e("LineBreakMode");
    }
    get virtualizationPercentage() {
        return this.ai;
    }
    set virtualizationPercentage(a) {
        this.ai = a;
        this.e("VirtualizationPercentage");
    }
    get paddingLeft() {
        return this.ax;
    }
    set paddingLeft(a) {
        this.ax = a;
        this.e("PaddingLeft");
    }
    get paddingTop() {
        return this.az;
    }
    set paddingTop(a) {
        this.az = a;
        this.e("PaddingTop");
    }
    get paddingRight() {
        return this.ay;
    }
    set paddingRight(a) {
        this.ay = a;
        this.e("PaddingRight");
    }
    get paddingBottom() {
        return this.aw;
    }
    set paddingBottom(a) {
        this.aw = a;
        this.e("PaddingBottom");
    }
    get sortDirection() {
        return this.bm;
    }
    set sortDirection(a) {
        this.bm = a;
        this.e("SortDirection");
    }
}
CellInfoDescription.$t = markType(CellInfoDescription, 'CellInfoDescription', Description.$);
