// igniteui-angular-core/ES2015/ColumnSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { HorizontalAnchoredCategorySeriesDescription } from "./HorizontalAnchoredCategorySeriesDescription";
import { markType } from "./type";
export class ColumnSeriesDescription extends HorizontalAnchoredCategorySeriesDescription {
    constructor() {
        super();
        this.cz = 0;
        this.c0 = 0;
        this.c3 = null;
    }
    get_type() {
        return "ColumnSeries";
    }
    get radiusX() {
        return this.cz;
    }
    set radiusX(a) {
        this.cz = a;
        this.e("RadiusX");
    }
    get radiusY() {
        return this.c0;
    }
    set radiusY(a) {
        this.c0 = a;
        this.e("RadiusY");
    }
    get consolidatedColumnVerticalPosition() {
        return this.c3;
    }
    set consolidatedColumnVerticalPosition(a) {
        this.c3 = a;
        this.e("ConsolidatedColumnVerticalPosition");
    }
}
ColumnSeriesDescription.$t = markType(ColumnSeriesDescription, 'ColumnSeriesDescription', HorizontalAnchoredCategorySeriesDescription.$);
