// igniteui-angular-core/ES2015/CrosshairLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
export class CrosshairLayerDescription extends AnnotationLayerDescription {
    constructor() {
        super();
        this.ct = null;
        this.cw = null;
        this.cv = null;
        this.b1 = false;
        this.bz = false;
        this.c0 = null;
        this.cy = null;
        this.cp = 0;
        this.cz = null;
        this.b6 = 0;
        this.b8 = 0;
        this.b7 = 0;
        this.b5 = 0;
        this.cb = 0;
        this.cd = 0;
        this.cc = 0;
        this.ca = 0;
        this.b9 = 0;
        this.c3 = null;
        this.c1 = null;
        this.cq = 0;
        this.c2 = null;
        this.ce = 0;
        this.cx = null;
        this.cu = null;
        this.b0 = false;
    }
    get_type() {
        return "CrosshairLayer";
    }
    get horizontalLineStroke() {
        return this.ct;
    }
    set horizontalLineStroke(a) {
        this.ct = a;
        this.e("HorizontalLineStroke");
    }
    get verticalLineStroke() {
        return this.cw;
    }
    set verticalLineStroke(a) {
        this.cw = a;
        this.e("VerticalLineStroke");
    }
    get targetSeriesRef() {
        return this.cv;
    }
    set targetSeriesRef(a) {
        this.cv = a;
        this.e("TargetSeriesRef");
    }
    get useInterpolation() {
        return this.b1;
    }
    set useInterpolation(a) {
        this.b1 = a;
        this.e("UseInterpolation");
    }
    get isAxisAnnotationEnabled() {
        return this.bz;
    }
    set isAxisAnnotationEnabled(a) {
        this.bz = a;
        this.e("IsAxisAnnotationEnabled");
    }
    get xAxisAnnotationTextColor() {
        return this.c0;
    }
    set xAxisAnnotationTextColor(a) {
        this.c0 = a;
        this.e("XAxisAnnotationTextColor");
    }
    get xAxisAnnotationBackground() {
        return this.cy;
    }
    set xAxisAnnotationBackground(a) {
        this.cy = a;
        this.e("XAxisAnnotationBackground");
    }
    get xAxisAnnotationInterpolatedValuePrecision() {
        return this.cp;
    }
    set xAxisAnnotationInterpolatedValuePrecision(a) {
        this.cp = a;
        this.e("XAxisAnnotationInterpolatedValuePrecision");
    }
    get xAxisAnnotationOutline() {
        return this.cz;
    }
    set xAxisAnnotationOutline(a) {
        this.cz = a;
        this.e("XAxisAnnotationOutline");
    }
    get xAxisAnnotationPaddingLeft() {
        return this.b6;
    }
    set xAxisAnnotationPaddingLeft(a) {
        this.b6 = a;
        this.e("XAxisAnnotationPaddingLeft");
    }
    get xAxisAnnotationPaddingTop() {
        return this.b8;
    }
    set xAxisAnnotationPaddingTop(a) {
        this.b8 = a;
        this.e("XAxisAnnotationPaddingTop");
    }
    get xAxisAnnotationPaddingRight() {
        return this.b7;
    }
    set xAxisAnnotationPaddingRight(a) {
        this.b7 = a;
        this.e("XAxisAnnotationPaddingRight");
    }
    get xAxisAnnotationPaddingBottom() {
        return this.b5;
    }
    set xAxisAnnotationPaddingBottom(a) {
        this.b5 = a;
        this.e("XAxisAnnotationPaddingBottom");
    }
    get yAxisAnnotationPaddingLeft() {
        return this.cb;
    }
    set yAxisAnnotationPaddingLeft(a) {
        this.cb = a;
        this.e("YAxisAnnotationPaddingLeft");
    }
    get yAxisAnnotationPaddingTop() {
        return this.cd;
    }
    set yAxisAnnotationPaddingTop(a) {
        this.cd = a;
        this.e("YAxisAnnotationPaddingTop");
    }
    get yAxisAnnotationPaddingRight() {
        return this.cc;
    }
    set yAxisAnnotationPaddingRight(a) {
        this.cc = a;
        this.e("YAxisAnnotationPaddingRight");
    }
    get yAxisAnnotationPaddingBottom() {
        return this.ca;
    }
    set yAxisAnnotationPaddingBottom(a) {
        this.ca = a;
        this.e("YAxisAnnotationPaddingBottom");
    }
    get xAxisAnnotationStrokeThickness() {
        return this.b9;
    }
    set xAxisAnnotationStrokeThickness(a) {
        this.b9 = a;
        this.e("XAxisAnnotationStrokeThickness");
    }
    get yAxisAnnotationTextColor() {
        return this.c3;
    }
    set yAxisAnnotationTextColor(a) {
        this.c3 = a;
        this.e("YAxisAnnotationTextColor");
    }
    get yAxisAnnotationBackground() {
        return this.c1;
    }
    set yAxisAnnotationBackground(a) {
        this.c1 = a;
        this.e("YAxisAnnotationBackground");
    }
    get yAxisAnnotationInterpolatedValuePrecision() {
        return this.cq;
    }
    set yAxisAnnotationInterpolatedValuePrecision(a) {
        this.cq = a;
        this.e("YAxisAnnotationInterpolatedValuePrecision");
    }
    get yAxisAnnotationOutline() {
        return this.c2;
    }
    set yAxisAnnotationOutline(a) {
        this.c2 = a;
        this.e("YAxisAnnotationOutline");
    }
    get yAxisAnnotationStrokeThickness() {
        return this.ce;
    }
    set yAxisAnnotationStrokeThickness(a) {
        this.ce = a;
        this.e("YAxisAnnotationStrokeThickness");
    }
    get verticalLineVisibility() {
        return this.cx;
    }
    set verticalLineVisibility(a) {
        this.cx = a;
        this.e("VerticalLineVisibility");
    }
    get horizontalLineVisibility() {
        return this.cu;
    }
    set horizontalLineVisibility(a) {
        this.cu = a;
        this.e("HorizontalLineVisibility");
    }
    get skipUnknownValues() {
        return this.b0;
    }
    set skipUnknownValues(a) {
        this.b0 = a;
        this.e("SkipUnknownValues");
    }
}
CrosshairLayerDescription.$t = markType(CrosshairLayerDescription, 'CrosshairLayerDescription', AnnotationLayerDescription.$);
