// igniteui-angular-core/ES2015/CustomPaletteColorScaleDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ColorScaleDescription } from "./ColorScaleDescription";
import { markType } from "./type";
export class CustomPaletteColorScaleDescription extends ColorScaleDescription {
    constructor() {
        super();
        this.k = 0;
        this.j = 0;
        this.h = null;
        this.n = null;
    }
    get_type() {
        return "CustomPaletteColorScale";
    }
    get minimumValue() {
        return this.k;
    }
    set minimumValue(a) {
        this.k = a;
        this.e("MinimumValue");
    }
    get maximumValue() {
        return this.j;
    }
    set maximumValue(a) {
        this.j = a;
        this.e("MaximumValue");
    }
    get palette() {
        return this.h;
    }
    set palette(a) {
        this.h = a;
        this.e("Palette");
    }
    get interpolationMode() {
        return this.n;
    }
    set interpolationMode(a) {
        this.n = a;
        this.e("InterpolationMode");
    }
}
CustomPaletteColorScaleDescription.$t = markType(CustomPaletteColorScaleDescription, 'CustomPaletteColorScaleDescription', ColorScaleDescription.$);
