// igniteui-angular-core/ES2015/DateTimeCellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CellInfoDescription } from "./CellInfoDescription";
import { markType } from "./type";
export class DateTimeCellInfoDescription extends CellInfoDescription {
    constructor() {
        super();
        this.cb = new Date();
        this.ce = null;
        this.cd = null;
        this.b9 = false;
    }
    get_type() {
        return "DateTimeCellInfo";
    }
    get dateTimeValue() {
        return this.cb;
    }
    set dateTimeValue(a) {
        this.cb = a;
        this.e("DateTimeValue");
    }
    get formatOverrideRef() {
        return this.ce;
    }
    set formatOverrideRef(a) {
        this.ce = a;
        this.e("FormatOverrideRef");
    }
    get dateTimeFormat() {
        return this.cd;
    }
    set dateTimeFormat(a) {
        this.cd = a;
        this.e("DateTimeFormat");
    }
    get isOffsetValue() {
        return this.b9;
    }
    set isOffsetValue(a) {
        this.b9 = a;
        this.e("IsOffsetValue");
    }
}
DateTimeCellInfoDescription.$t = markType(DateTimeCellInfoDescription, 'DateTimeCellInfoDescription', CellInfoDescription.$);
