// igniteui-angular-core/ES2015/DefinitionBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class DefinitionBaseDescription extends Description {
    constructor() {
        super();
        this.s = null;
        this.m = null;
        this.x = null;
        this.q = null;
        this.w = null;
        this.p = null;
        this.u = null;
        this.n = null;
        this.g = 0;
        this.t = null;
        this.z = null;
        this.v = null;
        this.o = null;
        this.y = null;
        this.r = null;
        this.h = 0;
        this.f = 0;
        this._fontFamily = null;
        this._fontSize = 0;
        this._fontStyle = null;
        this._DataBindingRef = null;
        this._DataBoundRef = null;
        this._CellStyleKeyRequestedRef = null;
    }
    get_type() {
        return "DefinitionBase";
    }
    get type() {
        return this.get_type();
    }
    get background() {
        return this.s;
    }
    set background(a) {
        this.s = a;
        this.e("Background");
    }
    get actualBackground() {
        return this.m;
    }
    set actualBackground(a) {
        this.m = a;
        this.e("ActualBackground");
    }
    get stickyRowBackground() {
        return this.x;
    }
    set stickyRowBackground(a) {
        this.x = a;
        this.e("StickyRowBackground");
    }
    get actualStickyRowBackground() {
        return this.q;
    }
    set actualStickyRowBackground(a) {
        this.q = a;
        this.e("ActualStickyRowBackground");
    }
    get pinnedRowBackground() {
        return this.w;
    }
    set pinnedRowBackground(a) {
        this.w = a;
        this.e("PinnedRowBackground");
    }
    get actualPinnedRowBackground() {
        return this.p;
    }
    set actualPinnedRowBackground(a) {
        this.p = a;
        this.e("ActualPinnedRowBackground");
    }
    get lastStickyRowBackground() {
        return this.u;
    }
    set lastStickyRowBackground(a) {
        this.u = a;
        this.e("LastStickyRowBackground");
    }
    get actualLastStickyRowBackground() {
        return this.n;
    }
    set actualLastStickyRowBackground(a) {
        this.n = a;
        this.e("ActualLastStickyRowBackground");
    }
    get contentOpacity() {
        return this.g;
    }
    set contentOpacity(a) {
        this.g = a;
        this.e("ContentOpacity");
    }
    get horizontalAlignment() {
        return this.t;
    }
    set horizontalAlignment(a) {
        this.t = a;
        this.e("HorizontalAlignment");
    }
    get verticalAlignment() {
        return this.z;
    }
    set verticalAlignment(a) {
        this.z = a;
        this.e("VerticalAlignment");
    }
    get lineBreakMode() {
        return this.v;
    }
    set lineBreakMode(a) {
        this.v = a;
        this.e("LineBreakMode");
    }
    get actualLineBreakMode() {
        return this.o;
    }
    set actualLineBreakMode(a) {
        this.o = a;
        this.e("ActualLineBreakMode");
    }
    get textColor() {
        return this.y;
    }
    set textColor(a) {
        this.y = a;
        this.e("TextColor");
    }
    get actualTextColor() {
        return this.r;
    }
    set actualTextColor(a) {
        this.r = a;
        this.e("ActualTextColor");
    }
    get pinnedRowOpacity() {
        return this.h;
    }
    set pinnedRowOpacity(a) {
        this.h = a;
        this.e("PinnedRowOpacity");
    }
    get actualPinnedRowOpacity() {
        return this.f;
    }
    set actualPinnedRowOpacity(a) {
        this.f = a;
        this.e("ActualPinnedRowOpacity");
    }
    get fontFamily() {
        return this._fontFamily;
    }
    set fontFamily(a) {
        this._fontFamily = a;
    }
    get fontSize() {
        return this._fontSize;
    }
    set fontSize(a) {
        this._fontSize = a;
    }
    get fontStyle() {
        return this._fontStyle;
    }
    set fontStyle(a) {
        this._fontStyle = a;
    }
    get DataBindingRef() {
        return this._DataBindingRef;
    }
    set DataBindingRef(a) {
        this._DataBindingRef = a;
    }
    get DataBoundRef() {
        return this._DataBoundRef;
    }
    set DataBoundRef(a) {
        this._DataBoundRef = a;
    }
    get CellStyleKeyRequestedRef() {
        return this._CellStyleKeyRequestedRef;
    }
    set CellStyleKeyRequestedRef(a) {
        this._CellStyleKeyRequestedRef = a;
    }
}
DefinitionBaseDescription.$t = markType(DefinitionBaseDescription, 'DefinitionBaseDescription', Description.$);
