// igniteui-angular-core/ES2015/DescriptionRef.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "./type";
import { DescriptionRefValueChangedEventArgs } from "./DescriptionRefValueChangedEventArgs";
export class DescriptionRef extends Base {
    constructor(a) {
        super();
        this.c = null;
        this.b = 0;
        this.valueChanged = null;
        this.b = 0;
        this.c = a;
    }
    get d() {
        return this.c;
    }
    e(a, b) {
        if (this.valueChanged != null) {
            let c = new DescriptionRefValueChangedEventArgs();
            c.b = a;
            c.a = b;
            this.valueChanged(this, c);
        }
    }
}
DescriptionRef.$t = markType(DescriptionRef, 'DescriptionRef');
