// igniteui-angular-core/ES2015/FinancialIndicatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FinancialSeriesDescription } from "./FinancialSeriesDescription";
import { markType } from "./type";
export class FinancialIndicatorDescription extends FinancialSeriesDescription {
    constructor() {
        super();
        this.cl = null;
        this.cg = 0;
        this.cn = null;
        this.cm = null;
        this.ck = null;
        this.ce = 0;
        this.ch = 0;
    }
    get_type() {
        return "FinancialIndicator";
    }
    get displayType() {
        return this.cl;
    }
    set displayType(a) {
        this.cl = a;
        this.e("DisplayType");
    }
    get ignoreFirst() {
        return this.cg;
    }
    set ignoreFirst(a) {
        this.cg = a;
        this.e("IgnoreFirst");
    }
    get trendLineType() {
        return this.cn;
    }
    set trendLineType(a) {
        this.cn = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.cm;
    }
    set trendLineBrush(a) {
        this.cm = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.ck;
    }
    set actualTrendLineBrush(a) {
        this.ck = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.ce;
    }
    set trendLineThickness(a) {
        this.ce = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.ch;
    }
    set trendLinePeriod(a) {
        this.ch = a;
        this.e("TrendLinePeriod");
    }
}
FinancialIndicatorDescription.$t = markType(FinancialIndicatorDescription, 'FinancialIndicatorDescription', FinancialSeriesDescription.$);
