// igniteui-angular-core/ES2015/FinalValueLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
export class FinalValueLayerDescription extends AnnotationLayerDescription {
    constructor() {
        super();
        this.cf = null;
        this.ce = null;
        this.cd = null;
        this.cb = null;
        this.b9 = 0;
        this.cc = null;
        this.b0 = 0;
        this.b2 = 0;
        this.b1 = 0;
        this.bz = 0;
        this.b3 = 0;
    }
    get_type() {
        return "FinalValueLayer";
    }
    get targetSeriesRef() {
        return this.cf;
    }
    set targetSeriesRef(a) {
        this.cf = a;
        this.e("TargetSeriesRef");
    }
    get finalValueSelectionMode() {
        return this.ce;
    }
    set finalValueSelectionMode(a) {
        this.ce = a;
        this.e("FinalValueSelectionMode");
    }
    get axisAnnotationTextColor() {
        return this.cd;
    }
    set axisAnnotationTextColor(a) {
        this.cd = a;
        this.e("AxisAnnotationTextColor");
    }
    get axisAnnotationBackground() {
        return this.cb;
    }
    set axisAnnotationBackground(a) {
        this.cb = a;
        this.e("AxisAnnotationBackground");
    }
    get axisAnnotationInterpolatedValuePrecision() {
        return this.b9;
    }
    set axisAnnotationInterpolatedValuePrecision(a) {
        this.b9 = a;
        this.e("AxisAnnotationInterpolatedValuePrecision");
    }
    get axisAnnotationOutline() {
        return this.cc;
    }
    set axisAnnotationOutline(a) {
        this.cc = a;
        this.e("AxisAnnotationOutline");
    }
    get axisAnnotationPaddingLeft() {
        return this.b0;
    }
    set axisAnnotationPaddingLeft(a) {
        this.b0 = a;
        this.e("AxisAnnotationPaddingLeft");
    }
    get axisAnnotationPaddingTop() {
        return this.b2;
    }
    set axisAnnotationPaddingTop(a) {
        this.b2 = a;
        this.e("AxisAnnotationPaddingTop");
    }
    get axisAnnotationPaddingRight() {
        return this.b1;
    }
    set axisAnnotationPaddingRight(a) {
        this.b1 = a;
        this.e("AxisAnnotationPaddingRight");
    }
    get axisAnnotationPaddingBottom() {
        return this.bz;
    }
    set axisAnnotationPaddingBottom(a) {
        this.bz = a;
        this.e("AxisAnnotationPaddingBottom");
    }
    get axisAnnotationStrokeThickness() {
        return this.b3;
    }
    set axisAnnotationStrokeThickness(a) {
        this.b3 = a;
        this.e("AxisAnnotationStrokeThickness");
    }
}
FinalValueLayerDescription.$t = markType(FinalValueLayerDescription, 'FinalValueLayerDescription', AnnotationLayerDescription.$);
