// igniteui-angular-core/ES2015/FinancialPriceSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FinancialSeriesDescription } from "./FinancialSeriesDescription";
import { markType } from "./type";
export class FinancialPriceSeriesDescription extends FinancialSeriesDescription {
    constructor() {
        super();
        this.cm = null;
        this.cl = null;
        this.ci = null;
        this.ce = 0;
        this.cg = 0;
        this.cj = null;
        this.ck = null;
    }
    get_type() {
        return "FinancialPriceSeries";
    }
    get trendLineType() {
        return this.cm;
    }
    set trendLineType(a) {
        this.cm = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.cl;
    }
    set trendLineBrush(a) {
        this.cl = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.ci;
    }
    set actualTrendLineBrush(a) {
        this.ci = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.ce;
    }
    set trendLineThickness(a) {
        this.ce = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.cg;
    }
    set trendLinePeriod(a) {
        this.cg = a;
        this.e("TrendLinePeriod");
    }
    get displayType() {
        return this.cj;
    }
    set displayType(a) {
        this.cj = a;
        this.e("DisplayType");
    }
    get negativeOutline() {
        return this.ck;
    }
    set negativeOutline(a) {
        this.ck = a;
        this.e("NegativeOutline");
    }
}
FinancialPriceSeriesDescription.$t = markType(FinancialPriceSeriesDescription, 'FinancialPriceSeriesDescription', FinancialSeriesDescription.$);
