// igniteui-angular-core/ES2015/FinancialSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class FinancialSeriesDescription extends SeriesDescription {
    constructor() {
        super();
        this.bw = null;
        this.b0 = null;
        this.b1 = null;
        this.bx = null;
        this.bu = null;
        this.bv = null;
        this.bt = null;
        this.bz = null;
        this.bp = false;
        this.by = null;
        this.bq = false;
        this._AssigningCategoryStyleRef = null;
        this._TypicalRef = null;
        this._TypicalBasedOnRef = null;
    }
    get_type() {
        return "FinancialSeries";
    }
    get negativeBrush() {
        return this.bw;
    }
    set negativeBrush(a) {
        this.bw = a;
        this.e("NegativeBrush");
    }
    get xAxisRef() {
        return this.b0;
    }
    set xAxisRef(a) {
        this.b0 = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.b1;
    }
    set yAxisRef(a) {
        this.b1 = a;
        this.e("YAxisRef");
    }
    get openMemberPath() {
        return this.bx;
    }
    set openMemberPath(a) {
        this.bx = a;
        this.e("OpenMemberPath");
    }
    get highMemberPath() {
        return this.bu;
    }
    set highMemberPath(a) {
        this.bu = a;
        this.e("HighMemberPath");
    }
    get lowMemberPath() {
        return this.bv;
    }
    set lowMemberPath(a) {
        this.bv = a;
        this.e("LowMemberPath");
    }
    get closeMemberPath() {
        return this.bt;
    }
    set closeMemberPath(a) {
        this.bt = a;
        this.e("CloseMemberPath");
    }
    get volumeMemberPath() {
        return this.bz;
    }
    set volumeMemberPath(a) {
        this.bz = a;
        this.e("VolumeMemberPath");
    }
    get isCustomCategoryStyleAllowed() {
        return this.bp;
    }
    set isCustomCategoryStyleAllowed(a) {
        this.bp = a;
        this.e("IsCustomCategoryStyleAllowed");
    }
    get transitionInMode() {
        return this.by;
    }
    set transitionInMode(a) {
        this.by = a;
        this.e("TransitionInMode");
    }
    get isTransitionInEnabled() {
        return this.bq;
    }
    set isTransitionInEnabled(a) {
        this.bq = a;
        this.e("IsTransitionInEnabled");
    }
    get AssigningCategoryStyleRef() {
        return this._AssigningCategoryStyleRef;
    }
    set AssigningCategoryStyleRef(a) {
        this._AssigningCategoryStyleRef = a;
    }
    get TypicalRef() {
        return this._TypicalRef;
    }
    set TypicalRef(a) {
        this._TypicalRef = a;
    }
    get TypicalBasedOnRef() {
        return this._TypicalBasedOnRef;
    }
    set TypicalBasedOnRef(a) {
        this._TypicalBasedOnRef = a;
    }
}
FinancialSeriesDescription.$t = markType(FinancialSeriesDescription, 'FinancialSeriesDescription', SeriesDescription.$);
