// igniteui-angular-core/ES2015/FinancialChartDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XYChartDescription } from "./XYChartDescription";
import { markType } from "./type";
export class FinancialChartDescription extends XYChartDescription {
    constructor() {
        super();
        this.h5 = 0;
        this.hl = false;
        this.h8 = 0;
        this.hr = false;
        this.iy = 0;
        this.ia = 0;
        this.ic = 0;
        this.ib = 0;
        this.id = 0;
        this.i6 = null;
        this.i7 = null;
        this.ja = null;
        this.hn = false;
        this.i5 = null;
        this.i8 = null;
        this.hp = false;
        this.gz = null;
        this.g4 = null;
        this.hq = false;
        this.h7 = 0;
        this.i9 = null;
        this.jb = null;
        this.g0 = null;
        this.g1 = null;
        this.g2 = null;
        this.g3 = null;
        this.g7 = null;
        this.g6 = null;
        this.gy = null;
        this.gw = null;
        this.h9 = 0;
        this.h6 = 0;
        this.h4 = 0;
        this.gx = null;
        this.iu = 0;
        this.h3 = 0;
        this.ix = 0;
        this.iv = 0;
        this.it = 0;
        this.iw = 0;
        this.g5 = null;
        this.gv = null;
        this.ho = false;
        this.jc = null;
        this.ie = 0;
        this.hm = false;
        this.h0 = new Date();
        this.hz = new Date();
        this._ApplyCustomIndicatorsRef = null;
    }
    get_type() {
        return "FinancialChart";
    }
    get leftMargin() {
        return this.h5;
    }
    set leftMargin(a) {
        this.h5 = a;
        this.e("LeftMargin");
    }
    get isHorizontalZoomEnabled() {
        return this.hl;
    }
    set isHorizontalZoomEnabled(a) {
        this.hl = a;
        this.e("IsHorizontalZoomEnabled");
    }
    get toolbarHeight() {
        return this.h8;
    }
    set toolbarHeight(a) {
        this.h8 = a;
        this.e("ToolbarHeight");
    }
    get yAxisIsLogarithmic() {
        return this.hr;
    }
    set yAxisIsLogarithmic(a) {
        this.hr = a;
        this.e("YAxisIsLogarithmic");
    }
    get yAxisLogarithmBase() {
        return this.iy;
    }
    set yAxisLogarithmBase(a) {
        this.iy = a;
        this.e("YAxisLogarithmBase");
    }
    get yAxisInterval() {
        return this.ia;
    }
    set yAxisInterval(a) {
        this.ia = a;
        this.e("YAxisInterval");
    }
    get yAxisMinimumValue() {
        return this.ic;
    }
    set yAxisMinimumValue(a) {
        this.ic = a;
        this.e("YAxisMinimumValue");
    }
    get yAxisMaximumValue() {
        return this.ib;
    }
    set yAxisMaximumValue(a) {
        this.ib = a;
        this.e("YAxisMaximumValue");
    }
    get yAxisMinorInterval() {
        return this.id;
    }
    set yAxisMinorInterval(a) {
        this.id = a;
        this.e("YAxisMinorInterval");
    }
    get volumeType() {
        return this.i6;
    }
    set volumeType(a) {
        this.i6 = a;
        this.e("VolumeType");
    }
    get xAxisMode() {
        return this.i7;
    }
    set xAxisMode(a) {
        this.i7 = a;
        this.e("XAxisMode");
    }
    get yAxisMode() {
        return this.ja;
    }
    set yAxisMode(a) {
        this.ja = a;
        this.e("YAxisMode");
    }
    get isToolbarVisible() {
        return this.hn;
    }
    set isToolbarVisible(a) {
        this.hn = a;
        this.e("IsToolbarVisible");
    }
    get chartType() {
        return this.i5;
    }
    set chartType(a) {
        this.i5 = a;
        this.e("ChartType");
    }
    get yAxisLabelHorizontalAlignment() {
        return this.i8;
    }
    set yAxisLabelHorizontalAlignment(a) {
        this.i8 = a;
        this.e("YAxisLabelHorizontalAlignment");
    }
    get isWindowSyncedToVisibleRange() {
        return this.hp;
    }
    set isWindowSyncedToVisibleRange(a) {
        this.hp = a;
        this.e("IsWindowSyncedToVisibleRange");
    }
    get indicatorTypes() {
        return this.gz;
    }
    set indicatorTypes(a) {
        this.gz = a;
        this.e("IndicatorTypes");
    }
    get overlayTypes() {
        return this.g4;
    }
    set overlayTypes(a) {
        this.g4 = a;
        this.e("OverlayTypes");
    }
    get yAxisAbbreviateLargeNumbers() {
        return this.hq;
    }
    set yAxisAbbreviateLargeNumbers(a) {
        this.hq = a;
        this.e("YAxisAbbreviateLargeNumbers");
    }
    get resolution() {
        return this.h7;
    }
    set resolution(a) {
        this.h7 = a;
        this.e("Resolution");
    }
    get yAxisLabelLocation() {
        return this.i9;
    }
    set yAxisLabelLocation(a) {
        this.i9 = a;
        this.e("YAxisLabelLocation");
    }
    get zoomSliderType() {
        return this.jb;
    }
    set zoomSliderType(a) {
        this.jb = a;
        this.e("ZoomSliderType");
    }
    get negativeBrushes() {
        return this.g0;
    }
    set negativeBrushes(a) {
        this.g0 = a;
        this.e("NegativeBrushes");
    }
    get negativeOutlines() {
        return this.g1;
    }
    set negativeOutlines(a) {
        this.g1 = a;
        this.e("NegativeOutlines");
    }
    get overlayBrushes() {
        return this.g2;
    }
    set overlayBrushes(a) {
        this.g2 = a;
        this.e("OverlayBrushes");
    }
    get overlayOutlines() {
        return this.g3;
    }
    set overlayOutlines(a) {
        this.g3 = a;
        this.e("OverlayOutlines");
    }
    get volumeOutlines() {
        return this.g7;
    }
    set volumeOutlines(a) {
        this.g7 = a;
        this.e("VolumeOutlines");
    }
    get volumeBrushes() {
        return this.g6;
    }
    set volumeBrushes(a) {
        this.g6 = a;
        this.e("VolumeBrushes");
    }
    get indicatorNegativeBrushes() {
        return this.gy;
    }
    set indicatorNegativeBrushes(a) {
        this.gy = a;
        this.e("IndicatorNegativeBrushes");
    }
    get indicatorBrushes() {
        return this.gw;
    }
    set indicatorBrushes(a) {
        this.gw = a;
        this.e("IndicatorBrushes");
    }
    get volumeThickness() {
        return this.h9;
    }
    set volumeThickness(a) {
        this.h9 = a;
        this.e("VolumeThickness");
    }
    get overlayThickness() {
        return this.h6;
    }
    set overlayThickness(a) {
        this.h6 = a;
        this.e("OverlayThickness");
    }
    get indicatorThickness() {
        return this.h4;
    }
    set indicatorThickness(a) {
        this.h4 = a;
        this.e("IndicatorThickness");
    }
    get indicatorDisplayTypes() {
        return this.gx;
    }
    set indicatorDisplayTypes(a) {
        this.gx = a;
        this.e("IndicatorDisplayTypes");
    }
    get indicatorPeriod() {
        return this.iu;
    }
    set indicatorPeriod(a) {
        this.iu = a;
        this.e("IndicatorPeriod");
    }
    get indicatorMultiplier() {
        return this.h3;
    }
    set indicatorMultiplier(a) {
        this.h3 = a;
        this.e("IndicatorMultiplier");
    }
    get indicatorSmoothingPeriod() {
        return this.ix;
    }
    set indicatorSmoothingPeriod(a) {
        this.ix = a;
        this.e("IndicatorSmoothingPeriod");
    }
    get indicatorShortPeriod() {
        return this.iv;
    }
    set indicatorShortPeriod(a) {
        this.iv = a;
        this.e("IndicatorShortPeriod");
    }
    get indicatorLongPeriod() {
        return this.it;
    }
    set indicatorLongPeriod(a) {
        this.it = a;
        this.e("IndicatorLongPeriod");
    }
    get indicatorSignalPeriod() {
        return this.iw;
    }
    set indicatorSignalPeriod(a) {
        this.iw = a;
        this.e("IndicatorSignalPeriod");
    }
    get rangeSelectorOptions() {
        return this.g5;
    }
    set rangeSelectorOptions(a) {
        this.g5 = a;
        this.e("RangeSelectorOptions");
    }
    get customIndicatorNames() {
        return this.gv;
    }
    set customIndicatorNames(a) {
        this.gv = a;
        this.e("CustomIndicatorNames");
    }
    get isVerticalZoomEnabled() {
        return this.ho;
    }
    set isVerticalZoomEnabled(a) {
        this.ho = a;
        this.e("IsVerticalZoomEnabled");
    }
    get zoomSliderXAxisMajorStroke() {
        return this.jc;
    }
    set zoomSliderXAxisMajorStroke(a) {
        this.jc = a;
        this.e("ZoomSliderXAxisMajorStroke");
    }
    get zoomSliderXAxisMajorStrokeThickness() {
        return this.ie;
    }
    set zoomSliderXAxisMajorStrokeThickness(a) {
        this.ie = a;
        this.e("ZoomSliderXAxisMajorStrokeThickness");
    }
    get isLegendVisible() {
        return this.hm;
    }
    set isLegendVisible(a) {
        this.hm = a;
        this.e("IsLegendVisible");
    }
    get xAxisMinimumValue() {
        return this.h0;
    }
    set xAxisMinimumValue(a) {
        this.h0 = a;
        this.e("XAxisMinimumValue");
    }
    get xAxisMaximumValue() {
        return this.hz;
    }
    set xAxisMaximumValue(a) {
        this.hz = a;
        this.e("XAxisMaximumValue");
    }
    get ApplyCustomIndicatorsRef() {
        return this._ApplyCustomIndicatorsRef;
    }
    set ApplyCustomIndicatorsRef(a) {
        this._ApplyCustomIndicatorsRef = a;
    }
}
FinancialChartDescription.$t = markType(FinancialChartDescription, 'FinancialChartDescription', XYChartDescription.$);
