// igniteui-angular-core/ES2015/FullStochasticOscillatorIndicatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicatorDescription } from "./StrategyBasedIndicatorDescription";
import { markType } from "./type";
export class FullStochasticOscillatorIndicatorDescription extends StrategyBasedIndicatorDescription {
    constructor() {
        super();
        this.cs = 0;
        this.ct = 0;
        this.cu = 0;
    }
    get_type() {
        return "FullStochasticOscillatorIndicator";
    }
    get period() {
        return this.cs;
    }
    set period(a) {
        this.cs = a;
        this.e("Period");
    }
    get smoothingPeriod() {
        return this.ct;
    }
    set smoothingPeriod(a) {
        this.ct = a;
        this.e("SmoothingPeriod");
    }
    get triggerPeriod() {
        return this.cu;
    }
    set triggerPeriod(a) {
        this.cu = a;
        this.e("TriggerPeriod");
    }
}
FullStochasticOscillatorIndicatorDescription.$t = markType(FullStochasticOscillatorIndicatorDescription, 'FullStochasticOscillatorIndicatorDescription', StrategyBasedIndicatorDescription.$);
