// igniteui-angular-core/ES2015/HighDensityScatterSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class HighDensityScatterSeriesDescription extends SeriesDescription {
    constructor() {
        super();
        this.b3 = null;
        this.b5 = null;
        this.b4 = null;
        this.b6 = null;
        this.bq = false;
        this.bp = false;
        this.bu = 0;
        this.bt = 0;
        this.b2 = null;
        this.b1 = null;
        this.bx = 0;
        this.by = 0;
    }
    get_type() {
        return "HighDensityScatterSeries";
    }
    get xAxisRef() {
        return this.b3;
    }
    set xAxisRef(a) {
        this.b3 = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.b5;
    }
    set yAxisRef(a) {
        this.b5 = a;
        this.e("YAxisRef");
    }
    get xMemberPath() {
        return this.b4;
    }
    set xMemberPath(a) {
        this.b4 = a;
        this.e("XMemberPath");
    }
    get yMemberPath() {
        return this.b6;
    }
    set yMemberPath(a) {
        this.b6 = a;
        this.e("YMemberPath");
    }
    get useBruteForce() {
        return this.bq;
    }
    set useBruteForce(a) {
        this.bq = a;
        this.e("UseBruteForce");
    }
    get progressiveLoad() {
        return this.bp;
    }
    set progressiveLoad(a) {
        this.bp = a;
        this.e("ProgressiveLoad");
    }
    get heatMinimum() {
        return this.bu;
    }
    set heatMinimum(a) {
        this.bu = a;
        this.e("HeatMinimum");
    }
    get heatMaximum() {
        return this.bt;
    }
    set heatMaximum(a) {
        this.bt = a;
        this.e("HeatMaximum");
    }
    get heatMinimumColor() {
        return this.b2;
    }
    set heatMinimumColor(a) {
        this.b2 = a;
        this.e("HeatMinimumColor");
    }
    get heatMaximumColor() {
        return this.b1;
    }
    set heatMaximumColor(a) {
        this.b1 = a;
        this.e("HeatMaximumColor");
    }
    get pointExtent() {
        return this.bx;
    }
    set pointExtent(a) {
        this.bx = a;
        this.e("PointExtent");
    }
    get progressiveStatus() {
        return this.by;
    }
    set progressiveStatus(a) {
        this.by = a;
        this.e("ProgressiveStatus");
    }
}
HighDensityScatterSeriesDescription.$t = markType(HighDensityScatterSeriesDescription, 'HighDensityScatterSeriesDescription', SeriesDescription.$);
