// igniteui-angular-core/ES2015/HorizontalRangeCategorySeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { RangeCategorySeriesDescription } from "./RangeCategorySeriesDescription";
import { markType } from "./type";
export class HorizontalRangeCategorySeriesDescription extends RangeCategorySeriesDescription {
    constructor() {
        super();
        this.cj = null;
        this.ck = null;
    }
    get_type() {
        return "HorizontalRangeCategorySeries";
    }
    get xAxisRef() {
        return this.cj;
    }
    set xAxisRef(a) {
        this.cj = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.ck;
    }
    set yAxisRef(a) {
        this.ck = a;
        this.e("YAxisRef");
    }
}
HorizontalRangeCategorySeriesDescription.$t = markType(HorizontalRangeCategorySeriesDescription, 'HorizontalRangeCategorySeriesDescription', RangeCategorySeriesDescription.$);
