// igniteui-angular-core/ES2015/HorizontalAnchoredCategorySeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnchoredCategorySeriesDescription } from "./AnchoredCategorySeriesDescription";
import { markType } from "./type";
export class HorizontalAnchoredCategorySeriesDescription extends AnchoredCategorySeriesDescription {
    constructor() {
        super();
        this.cv = null;
        this.cw = null;
    }
    get_type() {
        return "HorizontalAnchoredCategorySeries";
    }
    get xAxisRef() {
        return this.cv;
    }
    set xAxisRef(a) {
        this.cv = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.cw;
    }
    set yAxisRef(a) {
        this.cw = a;
        this.e("YAxisRef");
    }
}
HorizontalAnchoredCategorySeriesDescription.$t = markType(HorizontalAnchoredCategorySeriesDescription, 'HorizontalAnchoredCategorySeriesDescription', AnchoredCategorySeriesDescription.$);
