// igniteui-angular-core/ES2015/ImageCellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CellInfoDescription } from "./CellInfoDescription";
import { markType } from "./type";
export class ImageCellInfoDescription extends CellInfoDescription {
    constructor() {
        super();
        this.b9 = null;
        this.cb = null;
        this.ca = null;
    }
    get_type() {
        return "ImageCellInfo";
    }
    get imagePath() {
        return this.b9;
    }
    set imagePath(a) {
        this.b9 = a;
        this.e("ImagePath");
    }
    get imageStretchOption() {
        return this.cb;
    }
    set imageStretchOption(a) {
        this.cb = a;
        this.e("ImageStretchOption");
    }
    get imageResourceType() {
        return this.ca;
    }
    set imageResourceType(a) {
        this.ca = a;
        this.e("ImageResourceType");
    }
}
ImageCellInfoDescription.$t = markType(ImageCellInfoDescription, 'ImageCellInfoDescription', CellInfoDescription.$);
