// igniteui-angular-core/ES2015/ItemToolTipLayerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AnnotationLayerDescription } from "./AnnotationLayerDescription";
import { markType } from "./type";
export class ItemToolTipLayerDescription extends AnnotationLayerDescription {
    constructor() {
        super();
        this.b0 = false;
        this.bz = false;
    }
    get_type() {
        return "ItemToolTipLayer";
    }
    get useInterpolation() {
        return this.b0;
    }
    set useInterpolation(a) {
        this.b0 = a;
        this.e("UseInterpolation");
    }
    get skipUnknownValues() {
        return this.bz;
    }
    set skipUnknownValues(a) {
        this.bz = a;
        this.e("SkipUnknownValues");
    }
}
ItemToolTipLayerDescription.$t = markType(ItemToolTipLayerDescription, 'ItemToolTipLayerDescription', AnnotationLayerDescription.$);
