// igniteui-angular-core/ES2015/LegendBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class LegendBaseDescription extends Description {
    constructor() {
        super();
        this._LegendItemMouseLeftButtonDownRef = null;
        this._LegendItemMouseLeftButtonUpRef = null;
        this._LegendItemMouseEnterRef = null;
        this._LegendItemMouseLeaveRef = null;
        this._LegendItemMouseMoveRef = null;
    }
    get_type() {
        return "LegendBase";
    }
    get type() {
        return this.get_type();
    }
    get LegendItemMouseLeftButtonDownRef() {
        return this._LegendItemMouseLeftButtonDownRef;
    }
    set LegendItemMouseLeftButtonDownRef(a) {
        this._LegendItemMouseLeftButtonDownRef = a;
    }
    get LegendItemMouseLeftButtonUpRef() {
        return this._LegendItemMouseLeftButtonUpRef;
    }
    set LegendItemMouseLeftButtonUpRef(a) {
        this._LegendItemMouseLeftButtonUpRef = a;
    }
    get LegendItemMouseEnterRef() {
        return this._LegendItemMouseEnterRef;
    }
    set LegendItemMouseEnterRef(a) {
        this._LegendItemMouseEnterRef = a;
    }
    get LegendItemMouseLeaveRef() {
        return this._LegendItemMouseLeaveRef;
    }
    set LegendItemMouseLeaveRef(a) {
        this._LegendItemMouseLeaveRef = a;
    }
    get LegendItemMouseMoveRef() {
        return this._LegendItemMouseMoveRef;
    }
    set LegendItemMouseMoveRef(a) {
        this._LegendItemMouseMoveRef = a;
    }
}
LegendBaseDescription.$t = markType(LegendBaseDescription, 'LegendBaseDescription', Description.$);
