// igniteui-angular-core/ES2015/LegendDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LegendBaseDescription } from "./LegendBaseDescription";
import { markType } from "./type";
export class LegendDescription extends LegendBaseDescription {
    constructor() {
        super();
        this.l = null;
        this.m = null;
        this.n = null;
    }
    get_type() {
        return "Legend";
    }
    get orientation() {
        return this.l;
    }
    set orientation(a) {
        this.l = a;
        this.e("Orientation");
    }
    get textColor() {
        return this.m;
    }
    set textColor(a) {
        this.m = a;
        this.e("TextColor");
    }
    get textStyle() {
        return this.n;
    }
    set textStyle(a) {
        this.n = a;
        this.e("TextStyle");
    }
}
LegendDescription.$t = markType(LegendDescription, 'LegendDescription', LegendBaseDescription.$);
