// igniteui-angular-core/ES2015/LiveGridFilterDialogDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class LiveGridFilterDialogDescription extends Description {
    constructor() {
        super();
        this.f = false;
        this.i = 0;
        this.h = 0;
        this._DialogOpeningRef = null;
        this._FilterChangingRef = null;
        this._FilterChangedRef = null;
    }
    get_type() {
        return "LiveGridFilterDialog";
    }
    get type() {
        return this.get_type();
    }
    get isAnimationEnabled() {
        return this.f;
    }
    set isAnimationEnabled(a) {
        this.f = a;
        this.e("IsAnimationEnabled");
    }
    get pixelScalingRatio() {
        return this.i;
    }
    set pixelScalingRatio(a) {
        this.i = a;
        this.e("PixelScalingRatio");
    }
    get actualPixelScalingRatio() {
        return this.h;
    }
    set actualPixelScalingRatio(a) {
        this.h = a;
        this.e("ActualPixelScalingRatio");
    }
    get DialogOpeningRef() {
        return this._DialogOpeningRef;
    }
    set DialogOpeningRef(a) {
        this._DialogOpeningRef = a;
    }
    get FilterChangingRef() {
        return this._FilterChangingRef;
    }
    set FilterChangingRef(a) {
        this._FilterChangingRef = a;
    }
    get FilterChangedRef() {
        return this._FilterChangedRef;
    }
    set FilterChangedRef(a) {
        this._FilterChangedRef = a;
    }
}
LiveGridFilterDialogDescription.$t = markType(LiveGridFilterDialogDescription, 'LiveGridFilterDialogDescription', Description.$);
