// igniteui-angular-core/ES2015/LiveGridDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class LiveGridDescription extends Description {
    constructor() {
        super();
        this.n = null;
        this.g = null;
        this.af = 0;
        this.x = 0;
        this.ae = 0;
        this.av = null;
        this.p = false;
        this.ag = 0;
        this.s = false;
        this.ao = null;
        this.as = null;
        this.al = null;
        this.az = null;
        this.l = null;
        this.at = null;
        this.ar = null;
        this.aq = null;
        this.ap = null;
        this.a7 = null;
        this.q = false;
        this.r = false;
        this.am = null;
        this.a5 = null;
        this.y = 0;
        this.a8 = null;
        this.ay = null;
        this.ax = null;
        this.ak = null;
        this.an = null;
        this._cellFontFamily = null;
        this._cellFontSize = 0;
        this._cellFontStyle = null;
        this.a4 = null;
        this.a6 = null;
        this._sectionHeaderFontFamily = null;
        this._sectionHeaderFontSize = 0;
        this._sectionHeaderFontStyle = null;
        this.au = null;
        this.aw = null;
        this._headerFontFamily = null;
        this._headerFontSize = 0;
        this._headerFontStyle = null;
        this.a0 = null;
        this.a3 = null;
        this.a1 = null;
        this.a2 = null;
        this.h = null;
        this.f = null;
        this._FilterExpressionsChangingRef = null;
        this._FilterExpressionsChangedRef = null;
        this._CellPreviewPointerDownRef = null;
        this._CellPreviewPointerUpRef = null;
        this._CellPointerDownRef = null;
        this._CellPointerUpRef = null;
        this._CellClickedRef = null;
        this._ColumnsAutoGeneratedRef = null;
        this._SelectedItemsChangedRef = null;
        this._SelectedKeysChangedRef = null;
    }
    get_type() {
        return "LiveGrid";
    }
    get type() {
        return this.get_type();
    }
    get sectionHeader() {
        return this.n;
    }
    set sectionHeader(a) {
        this.n = a;
        this.e("SectionHeader");
    }
    get dataSourceDesiredProperties() {
        return this.g;
    }
    set dataSourceDesiredProperties(a) {
        this.g = a;
        this.e("DataSourceDesiredProperties");
    }
    get rowHeight() {
        return this.af;
    }
    set rowHeight(a) {
        this.af = a;
        this.e("RowHeight");
    }
    get defaultColumnMinWidth() {
        return this.x;
    }
    set defaultColumnMinWidth(a) {
        this.x = a;
        this.e("DefaultColumnMinWidth");
    }
    get headerHeight() {
        return this.ae;
    }
    set headerHeight(a) {
        this.ae = a;
        this.e("HeaderHeight");
    }
    get headerClickAction() {
        return this.av;
    }
    set headerClickAction(a) {
        this.av = a;
        this.e("HeaderClickAction");
    }
    get autoGenerateColumns() {
        return this.p;
    }
    set autoGenerateColumns(a) {
        this.p = a;
        this.e("AutoGenerateColumns");
    }
    get rowSeparatorHeight() {
        return this.ag;
    }
    set rowSeparatorHeight(a) {
        this.ag = a;
        this.e("RowSeparatorHeight");
    }
    get isPlaceholderRenderingEnabled() {
        return this.s;
    }
    set isPlaceholderRenderingEnabled(a) {
        this.s = a;
        this.e("IsPlaceholderRenderingEnabled");
    }
    get columnAddingAnimationMode() {
        return this.ao;
    }
    set columnAddingAnimationMode(a) {
        this.ao = a;
        this.e("ColumnAddingAnimationMode");
    }
    get columnPropertyUpdatingAnimationMode() {
        return this.as;
    }
    set columnPropertyUpdatingAnimationMode(a) {
        this.as = a;
        this.e("ColumnPropertyUpdatingAnimationMode");
    }
    get cellDataLoadedAnimationMode() {
        return this.al;
    }
    set cellDataLoadedAnimationMode(a) {
        this.al = a;
        this.e("CellDataLoadedAnimationMode");
    }
    get rowSelectionAnimationMode() {
        return this.az;
    }
    set rowSelectionAnimationMode(a) {
        this.az = a;
        this.e("RowSelectionAnimationMode");
    }
    get animationSettings() {
        return this.l;
    }
    set animationSettings(a) {
        this.l = a;
        this.e("AnimationSettings");
    }
    get columnShowingAnimationMode() {
        return this.at;
    }
    set columnShowingAnimationMode(a) {
        this.at = a;
        this.e("ColumnShowingAnimationMode");
    }
    get columnMovingAnimationMode() {
        return this.ar;
    }
    set columnMovingAnimationMode(a) {
        this.ar = a;
        this.e("ColumnMovingAnimationMode");
    }
    get columnHidingAnimationMode() {
        return this.aq;
    }
    set columnHidingAnimationMode(a) {
        this.aq = a;
        this.e("ColumnHidingAnimationMode");
    }
    get columnExchangingAnimationMode() {
        return this.ap;
    }
    set columnExchangingAnimationMode(a) {
        this.ap = a;
        this.e("ColumnExchangingAnimationMode");
    }
    get selectionMode() {
        return this.a7;
    }
    set selectionMode(a) {
        this.a7 = a;
        this.e("SelectionMode");
    }
    get isFilterUIEnabled() {
        return this.q;
    }
    set isFilterUIEnabled(a) {
        this.q = a;
        this.e("IsFilterUIEnabled");
    }
    get isGroupRowSticky() {
        return this.r;
    }
    set isGroupRowSticky(a) {
        this.r = a;
        this.e("IsGroupRowSticky");
    }
    get cellSelectedBackground() {
        return this.am;
    }
    set cellSelectedBackground(a) {
        this.am = a;
        this.e("CellSelectedBackground");
    }
    get sectionHeaderSelectedBackground() {
        return this.a5;
    }
    set sectionHeaderSelectedBackground(a) {
        this.a5 = a;
        this.e("SectionHeaderSelectedBackground");
    }
    get pinnedRowOpacity() {
        return this.y;
    }
    set pinnedRowOpacity(a) {
        this.y = a;
        this.e("PinnedRowOpacity");
    }
    get stickyRowBackground() {
        return this.a8;
    }
    set stickyRowBackground(a) {
        this.a8 = a;
        this.e("StickyRowBackground");
    }
    get pinnedRowBackground() {
        return this.ay;
    }
    set pinnedRowBackground(a) {
        this.ay = a;
        this.e("PinnedRowBackground");
    }
    get lastStickyRowBackground() {
        return this.ax;
    }
    set lastStickyRowBackground(a) {
        this.ax = a;
        this.e("LastStickyRowBackground");
    }
    get cellBackground() {
        return this.ak;
    }
    set cellBackground(a) {
        this.ak = a;
        this.e("CellBackground");
    }
    get cellTextColor() {
        return this.an;
    }
    set cellTextColor(a) {
        this.an = a;
        this.e("CellTextColor");
    }
    get cellFontFamily() {
        return this._cellFontFamily;
    }
    set cellFontFamily(a) {
        this._cellFontFamily = a;
    }
    get cellFontSize() {
        return this._cellFontSize;
    }
    set cellFontSize(a) {
        this._cellFontSize = a;
    }
    get cellFontStyle() {
        return this._cellFontStyle;
    }
    set cellFontStyle(a) {
        this._cellFontStyle = a;
    }
    get sectionHeaderBackground() {
        return this.a4;
    }
    set sectionHeaderBackground(a) {
        this.a4 = a;
        this.e("SectionHeaderBackground");
    }
    get sectionHeaderTextColor() {
        return this.a6;
    }
    set sectionHeaderTextColor(a) {
        this.a6 = a;
        this.e("SectionHeaderTextColor");
    }
    get sectionHeaderFontFamily() {
        return this._sectionHeaderFontFamily;
    }
    set sectionHeaderFontFamily(a) {
        this._sectionHeaderFontFamily = a;
    }
    get sectionHeaderFontSize() {
        return this._sectionHeaderFontSize;
    }
    set sectionHeaderFontSize(a) {
        this._sectionHeaderFontSize = a;
    }
    get sectionHeaderFontStyle() {
        return this._sectionHeaderFontStyle;
    }
    set sectionHeaderFontStyle(a) {
        this._sectionHeaderFontStyle = a;
    }
    get headerBackground() {
        return this.au;
    }
    set headerBackground(a) {
        this.au = a;
        this.e("HeaderBackground");
    }
    get headerTextColor() {
        return this.aw;
    }
    set headerTextColor(a) {
        this.aw = a;
        this.e("HeaderTextColor");
    }
    get headerFontFamily() {
        return this._headerFontFamily;
    }
    set headerFontFamily(a) {
        this._headerFontFamily = a;
    }
    get headerFontSize() {
        return this._headerFontSize;
    }
    set headerFontSize(a) {
        this._headerFontSize = a;
    }
    get headerFontStyle() {
        return this._headerFontStyle;
    }
    set headerFontStyle(a) {
        this._headerFontStyle = a;
    }
    get rowSeparatorBackground() {
        return this.a0;
    }
    set rowSeparatorBackground(a) {
        this.a0 = a;
        this.e("RowSeparatorBackground");
    }
    get rowSeparatorStickyRowBackground() {
        return this.a3;
    }
    set rowSeparatorStickyRowBackground(a) {
        this.a3 = a;
        this.e("RowSeparatorStickyRowBackground");
    }
    get rowSeparatorLastStickyRowBackground() {
        return this.a1;
    }
    set rowSeparatorLastStickyRowBackground(a) {
        this.a1 = a;
        this.e("RowSeparatorLastStickyRowBackground");
    }
    get rowSeparatorPinnedRowBackground() {
        return this.a2;
    }
    set rowSeparatorPinnedRowBackground(a) {
        this.a2 = a;
        this.e("RowSeparatorPinnedRowBackground");
    }
    get primaryKey() {
        return this.h;
    }
    set primaryKey(a) {
        this.h = a;
        this.e("PrimaryKey");
    }
    get actualPrimaryKey() {
        return this.f;
    }
    set actualPrimaryKey(a) {
        this.f = a;
        this.e("ActualPrimaryKey");
    }
    get FilterExpressionsChangingRef() {
        return this._FilterExpressionsChangingRef;
    }
    set FilterExpressionsChangingRef(a) {
        this._FilterExpressionsChangingRef = a;
    }
    get FilterExpressionsChangedRef() {
        return this._FilterExpressionsChangedRef;
    }
    set FilterExpressionsChangedRef(a) {
        this._FilterExpressionsChangedRef = a;
    }
    get CellPreviewPointerDownRef() {
        return this._CellPreviewPointerDownRef;
    }
    set CellPreviewPointerDownRef(a) {
        this._CellPreviewPointerDownRef = a;
    }
    get CellPreviewPointerUpRef() {
        return this._CellPreviewPointerUpRef;
    }
    set CellPreviewPointerUpRef(a) {
        this._CellPreviewPointerUpRef = a;
    }
    get CellPointerDownRef() {
        return this._CellPointerDownRef;
    }
    set CellPointerDownRef(a) {
        this._CellPointerDownRef = a;
    }
    get CellPointerUpRef() {
        return this._CellPointerUpRef;
    }
    set CellPointerUpRef(a) {
        this._CellPointerUpRef = a;
    }
    get CellClickedRef() {
        return this._CellClickedRef;
    }
    set CellClickedRef(a) {
        this._CellClickedRef = a;
    }
    get ColumnsAutoGeneratedRef() {
        return this._ColumnsAutoGeneratedRef;
    }
    set ColumnsAutoGeneratedRef(a) {
        this._ColumnsAutoGeneratedRef = a;
    }
    get SelectedItemsChangedRef() {
        return this._SelectedItemsChangedRef;
    }
    set SelectedItemsChangedRef(a) {
        this._SelectedItemsChangedRef = a;
    }
    get SelectedKeysChangedRef() {
        return this._SelectedKeysChangedRef;
    }
    set SelectedKeysChangedRef(a) {
        this._SelectedKeysChangedRef = a;
    }
}
LiveGridDescription.$t = markType(LiveGridDescription, 'LiveGridDescription', Description.$);
