// igniteui-angular-core/ES2015/MarkerSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class MarkerSeriesDescription extends SeriesDescription {
    constructor() {
        super();
        this.bv = null;
        this.bu = null;
        this.br = null;
        this.bs = null;
        this.bp = null;
        this.bt = null;
        this.bq = null;
    }
    get_type() {
        return "MarkerSeries";
    }
    get markerType() {
        return this.bv;
    }
    set markerType(a) {
        this.bv = a;
        this.e("MarkerType");
    }
    get markerTemplate() {
        return this.bu;
    }
    set markerTemplate(a) {
        this.bu = a;
        this.e("MarkerTemplate");
    }
    get actualMarkerTemplate() {
        return this.br;
    }
    set actualMarkerTemplate(a) {
        this.br = a;
        this.e("ActualMarkerTemplate");
    }
    get markerBrush() {
        return this.bs;
    }
    set markerBrush(a) {
        this.bs = a;
        this.e("MarkerBrush");
    }
    get actualMarkerBrush() {
        return this.bp;
    }
    set actualMarkerBrush(a) {
        this.bp = a;
        this.e("ActualMarkerBrush");
    }
    get markerOutline() {
        return this.bt;
    }
    set markerOutline(a) {
        this.bt = a;
        this.e("MarkerOutline");
    }
    get actualMarkerOutline() {
        return this.bq;
    }
    set actualMarkerOutline(a) {
        this.bq = a;
        this.e("ActualMarkerOutline");
    }
}
MarkerSeriesDescription.$t = markType(MarkerSeriesDescription, 'MarkerSeriesDescription', SeriesDescription.$);
