// igniteui-angular-core/ES2015/NullStream.js
//
// System.IO.StreamWriter.cs
//
// Authors:
//   Dietmar Maurer (dietmar@ximian.com)
//   Paolo Molaro (lupus@ximian.com)
//
// (C) Ximian, Inc.  http://www.ximian.com
//
import { Stream, markType } from "./type";
export class NullStream extends Stream {
    get_canRead() {
        return true;
    }
    get canRead() {
        return this.get_canRead();
    }
    get_canSeek() {
        return true;
    }
    get canSeek() {
        return this.get_canSeek();
    }
    get_canWrite() {
        return true;
    }
    get canWrite() {
        return this.get_canWrite();
    }
    get_length() {
        return 0;
    }
    get length() {
        return this.get_length();
    }
    get_position() {
        return 0;
    }
    set_position(a) {
    }
    get position() {
        return this.get_position();
    }
    set position(a) {
        this.set_position(a);
    }
    flush() {
    }
    read(a, b, c) {
        return 0;
    }
    readByte() {
        return -1;
    }
    seek(a, b) {
        return 0;
    }
    setLength(a) {
    }
    write(a, b, c) {
    }
    writeByte(a) {
    }
}
NullStream.$t = markType(NullStream, 'NullStream', Stream.$);
