// igniteui-angular-core/ES2015/NumericRadiusAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { NumericAxisBaseDescription } from "./NumericAxisBaseDescription";
import { markType } from "./type";
export class NumericRadiusAxisDescription extends NumericAxisBaseDescription {
    constructor() {
        super();
        this.dh = 0;
        this.dg = 0;
    }
    get_type() {
        return "NumericRadiusAxis";
    }
    get radiusExtentScale() {
        return this.dh;
    }
    set radiusExtentScale(a) {
        this.dh = a;
        this.e("RadiusExtentScale");
    }
    get innerRadiusExtentScale() {
        return this.dg;
    }
    set innerRadiusExtentScale(a) {
        this.dg = a;
        this.e("InnerRadiusExtentScale");
    }
}
NumericRadiusAxisDescription.$t = markType(NumericRadiusAxisDescription, 'NumericRadiusAxisDescription', NumericAxisBaseDescription.$);
