// igniteui-angular-core/ES2015/NumericCellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CellInfoDescription } from "./CellInfoDescription";
import { markType } from "./type";
export class NumericCellInfoDescription extends CellInfoDescription {
    constructor() {
        super();
        this.cd = 0;
        this.b9 = false;
        this.cl = null;
        this.cm = null;
        this.co = null;
        this.cn = null;
        this.cp = null;
        this.cf = 0;
        this.cg = 0;
        this.ch = 0;
        this.ca = false;
    }
    get_type() {
        return "NumericCellInfo";
    }
    get numericValue() {
        return this.cd;
    }
    set numericValue(a) {
        this.cd = a;
        this.e("NumericValue");
    }
    get hasDecimalValue() {
        return this.b9;
    }
    set hasDecimalValue(a) {
        this.b9 = a;
        this.e("HasDecimalValue");
    }
    get formatOverrideRef() {
        return this.cl;
    }
    set formatOverrideRef(a) {
        this.cl = a;
        this.e("FormatOverrideRef");
    }
    get negativePrefix() {
        return this.cm;
    }
    set negativePrefix(a) {
        this.cm = a;
        this.e("NegativePrefix");
    }
    get positivePrefix() {
        return this.co;
    }
    set positivePrefix(a) {
        this.co = a;
        this.e("PositivePrefix");
    }
    get negativeSuffix() {
        return this.cn;
    }
    set negativeSuffix(a) {
        this.cn = a;
        this.e("NegativeSuffix");
    }
    get positiveSuffix() {
        return this.cp;
    }
    set positiveSuffix(a) {
        this.cp = a;
        this.e("PositiveSuffix");
    }
    get maxFractionDigits() {
        return this.cf;
    }
    set maxFractionDigits(a) {
        this.cf = a;
        this.e("MaxFractionDigits");
    }
    get minFractionDigits() {
        return this.cg;
    }
    set minFractionDigits(a) {
        this.cg = a;
        this.e("MinFractionDigits");
    }
    get minIntegerDigits() {
        return this.ch;
    }
    set minIntegerDigits(a) {
        this.ch = a;
        this.e("MinIntegerDigits");
    }
    get showGroupingSeparator() {
        return this.ca;
    }
    set showGroupingSeparator(a) {
        this.ca = a;
        this.e("ShowGroupingSeparator");
    }
}
NumericCellInfoDescription.$t = markType(NumericCellInfoDescription, 'NumericCellInfoDescription', CellInfoDescription.$);
