// igniteui-angular-core/ES2015/NumericAxisBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AxisDescription } from "./AxisDescription";
import { markType } from "./type";
export class NumericAxisBaseDescription extends AxisDescription {
    constructor() {
        super();
        this.c0 = 0;
        this.cu = 0;
        this.cx = 0;
        this.cz = 0;
        this.ct = 0;
        this.cw = 0;
        this.cy = 0;
        this.cs = 0;
        this.c1 = 0;
        this.cv = 0;
        this.c2 = 0;
        this.co = false;
        this.cn = false;
        this.de = 0;
        this.cm = false;
    }
    get_type() {
        return "NumericAxisBase";
    }
    get minimumValue() {
        return this.c0;
    }
    set minimumValue(a) {
        this.c0 = a;
        this.e("MinimumValue");
    }
    get actualMinimumValue() {
        return this.cu;
    }
    set actualMinimumValue(a) {
        this.cu = a;
        this.e("ActualMinimumValue");
    }
    get actualVisibleMinimumValue() {
        return this.cx;
    }
    set actualVisibleMinimumValue(a) {
        this.cx = a;
        this.e("ActualVisibleMinimumValue");
    }
    get maximumValue() {
        return this.cz;
    }
    set maximumValue(a) {
        this.cz = a;
        this.e("MaximumValue");
    }
    get actualMaximumValue() {
        return this.ct;
    }
    set actualMaximumValue(a) {
        this.ct = a;
        this.e("ActualMaximumValue");
    }
    get actualVisibleMaximumValue() {
        return this.cw;
    }
    set actualVisibleMaximumValue(a) {
        this.cw = a;
        this.e("ActualVisibleMaximumValue");
    }
    get interval() {
        return this.cy;
    }
    set interval(a) {
        this.cy = a;
        this.e("Interval");
    }
    get actualInterval() {
        return this.cs;
    }
    set actualInterval(a) {
        this.cs = a;
        this.e("ActualInterval");
    }
    get minorInterval() {
        return this.c1;
    }
    set minorInterval(a) {
        this.c1 = a;
        this.e("MinorInterval");
    }
    get actualMinorInterval() {
        return this.cv;
    }
    set actualMinorInterval(a) {
        this.cv = a;
        this.e("ActualMinorInterval");
    }
    get referenceValue() {
        return this.c2;
    }
    set referenceValue(a) {
        this.c2 = a;
        this.e("ReferenceValue");
    }
    get isLogarithmic() {
        return this.co;
    }
    set isLogarithmic(a) {
        this.co = a;
        this.e("IsLogarithmic");
    }
    get actualIsLogarithmic() {
        return this.cn;
    }
    set actualIsLogarithmic(a) {
        this.cn = a;
        this.e("ActualIsLogarithmic");
    }
    get logarithmBase() {
        return this.de;
    }
    set logarithmBase(a) {
        this.de = a;
        this.e("LogarithmBase");
    }
    get abbreviateLargeNumbers() {
        return this.cm;
    }
    set abbreviateLargeNumbers(a) {
        this.cm = a;
        this.e("AbbreviateLargeNumbers");
    }
}
NumericAxisBaseDescription.$t = markType(NumericAxisBaseDescription, 'NumericAxisBaseDescription', AxisDescription.$);
