// igniteui-angular-core/ES2015/OrdinalTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryXAxisDescription } from "./CategoryXAxisDescription";
import { markType } from "./type";
export class OrdinalTimeXAxisDescription extends CategoryXAxisDescription {
    constructor() {
        super();
        this.c8 = null;
        this.c5 = new Date();
        this.c4 = new Date();
    }
    get_type() {
        return "OrdinalTimeXAxis";
    }
    get dateTimeMemberPath() {
        return this.c8;
    }
    set dateTimeMemberPath(a) {
        this.c8 = a;
        this.e("DateTimeMemberPath");
    }
    get minimumValue() {
        return this.c5;
    }
    set minimumValue(a) {
        this.c5 = a;
        this.e("MinimumValue");
    }
    get maximumValue() {
        return this.c4;
    }
    set maximumValue(a) {
        this.c4 = a;
        this.e("MaximumValue");
    }
}
OrdinalTimeXAxisDescription.$t = markType(OrdinalTimeXAxisDescription, 'OrdinalTimeXAxisDescription', CategoryXAxisDescription.$);
