// igniteui-angular-core/ES2015/PercentageVolumeOscillatorIndicatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicatorDescription } from "./StrategyBasedIndicatorDescription";
import { markType } from "./type";
export class PercentageVolumeOscillatorIndicatorDescription extends StrategyBasedIndicatorDescription {
    constructor() {
        super();
        this.ct = 0;
        this.cs = 0;
    }
    get_type() {
        return "PercentageVolumeOscillatorIndicator";
    }
    get shortPeriod() {
        return this.ct;
    }
    set shortPeriod(a) {
        this.ct = a;
        this.e("ShortPeriod");
    }
    get longPeriod() {
        return this.cs;
    }
    set longPeriod(a) {
        this.cs = a;
        this.e("LongPeriod");
    }
}
PercentageVolumeOscillatorIndicatorDescription.$t = markType(PercentageVolumeOscillatorIndicatorDescription, 'PercentageVolumeOscillatorIndicatorDescription', StrategyBasedIndicatorDescription.$);
