// igniteui-angular-core/ES2015/PieChartDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PieChartBaseDescription } from "./PieChartBaseDescription";
import { markType } from "./type";
export class PieChartDescription extends PieChartBaseDescription {
    constructor() {
        super();
        this.cc = null;
        this.cb = null;
        this.ca = null;
    }
    get width() {
        return this.cc;
    }
    set width(a) {
        this.cc = a;
        this.e("Width");
    }
    get height() {
        return this.cb;
    }
    set height(a) {
        this.cb = a;
        this.e("Height");
    }
    get dataSourceRef() {
        return this.ca;
    }
    set dataSourceRef(a) {
        this.ca = a;
        this.e("DataSourceRef");
    }
    get_type() {
        return "PieChart";
    }
}
PieChartDescription.$t = markType(PieChartDescription, 'PieChartDescription', PieChartBaseDescription.$);
