// igniteui-angular-core/ES2015/PieChartBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class PieChartBaseDescription extends Description {
    constructor() {
        super();
        this.v = 0;
        this.bf = null;
        this.aw = null;
        this.a3 = null;
        this.ay = null;
        this.ax = null;
        this.av = null;
        this.a0 = null;
        this.az = null;
        this.x = 0;
        this.be = null;
        this.aa = 0;
        this.a7 = null;
        this.a6 = null;
        this.u = 0;
        this.ac = 0;
        this.o = false;
        this.n = false;
        this.g = null;
        this.w = 0;
        this.af = 0;
        this.bc = null;
        this.a4 = null;
        this.a5 = null;
        this.z = 0;
        this.y = 0;
        this.a9 = null;
        this.ba = null;
        this.ae = 0;
        this.ad = 0;
        this.f = null;
        this.h = null;
        this.a2 = null;
        this.a1 = null;
        this.p = false;
        this.at = null;
        this.au = null;
        this.ab = 0;
        this.t = 0;
        this.bb = null;
        this.a8 = null;
        this.i = null;
        this.bd = null;
        this._LabelClickRef = null;
        this._SelectedItemChangingRef = null;
        this._SelectedItemsChangingRef = null;
        this._SelectedItemChangedRef = null;
        this._SelectedItemsChangedRef = null;
        this._SliceClickRef = null;
    }
    get_type() {
        return "PieChartBase";
    }
    get type() {
        return this.get_type();
    }
    get innerExtent() {
        return this.v;
    }
    set innerExtent(a) {
        this.v = a;
        this.e("InnerExtent");
    }
    get valueMemberPath() {
        return this.bf;
    }
    set valueMemberPath(a) {
        this.bf = a;
        this.e("ValueMemberPath");
    }
    get labelMemberPath() {
        return this.aw;
    }
    set labelMemberPath(a) {
        this.aw = a;
        this.e("LabelMemberPath");
    }
    get legendLabelMemberPath() {
        return this.a3;
    }
    set legendLabelMemberPath(a) {
        this.a3 = a;
        this.e("LegendLabelMemberPath");
    }
    get labelsPosition() {
        return this.ay;
    }
    set labelsPosition(a) {
        this.ay = a;
        this.e("LabelsPosition");
    }
    get labelOuterColor() {
        return this.ax;
    }
    set labelOuterColor(a) {
        this.ax = a;
        this.e("LabelOuterColor");
    }
    get labelInnerColor() {
        return this.av;
    }
    set labelInnerColor(a) {
        this.av = a;
        this.e("LabelInnerColor");
    }
    get leaderLineVisibility() {
        return this.a0;
    }
    set leaderLineVisibility(a) {
        this.a0 = a;
        this.e("LeaderLineVisibility");
    }
    get leaderLineType() {
        return this.az;
    }
    set leaderLineType(a) {
        this.az = a;
        this.e("LeaderLineType");
    }
    get leaderLineMargin() {
        return this.x;
    }
    set leaderLineMargin(a) {
        this.x = a;
        this.e("LeaderLineMargin");
    }
    get toolTipRef() {
        return this.be;
    }
    set toolTipRef(a) {
        this.be = a;
        this.e("ToolTipRef");
    }
    get othersCategoryThreshold() {
        return this.aa;
    }
    set othersCategoryThreshold(a) {
        this.aa = a;
        this.e("OthersCategoryThreshold");
    }
    get othersCategoryType() {
        return this.a7;
    }
    set othersCategoryType(a) {
        this.a7 = a;
        this.e("OthersCategoryType");
    }
    get othersCategoryText() {
        return this.a6;
    }
    set othersCategoryText(a) {
        this.a6 = a;
        this.e("OthersCategoryText");
    }
    get explodedRadius() {
        return this.u;
    }
    set explodedRadius(a) {
        this.u = a;
        this.e("ExplodedRadius");
    }
    get radiusFactor() {
        return this.ac;
    }
    set radiusFactor(a) {
        this.ac = a;
        this.e("RadiusFactor");
    }
    get allowSliceSelection() {
        return this.o;
    }
    set allowSliceSelection(a) {
        this.o = a;
        this.e("AllowSliceSelection");
    }
    get allowSliceExplosion() {
        return this.n;
    }
    set allowSliceExplosion(a) {
        this.n = a;
        this.e("AllowSliceExplosion");
    }
    get explodedSlices() {
        return this.g;
    }
    set explodedSlices(a) {
        this.g = a;
        this.e("ExplodedSlices");
    }
    get labelExtent() {
        return this.w;
    }
    set labelExtent(a) {
        this.w = a;
        this.e("LabelExtent");
    }
    get startAngle() {
        return this.af;
    }
    set startAngle(a) {
        this.af = a;
        this.e("StartAngle");
    }
    get sweepDirection() {
        return this.bc;
    }
    set sweepDirection(a) {
        this.bc = a;
        this.e("SweepDirection");
    }
    get othersCategoryFill() {
        return this.a4;
    }
    set othersCategoryFill(a) {
        this.a4 = a;
        this.e("OthersCategoryFill");
    }
    get othersCategoryStroke() {
        return this.a5;
    }
    set othersCategoryStroke(a) {
        this.a5 = a;
        this.e("OthersCategoryStroke");
    }
    get othersCategoryStrokeThickness() {
        return this.z;
    }
    set othersCategoryStrokeThickness(a) {
        this.z = a;
        this.e("OthersCategoryStrokeThickness");
    }
    get othersCategoryOpacity() {
        return this.y;
    }
    set othersCategoryOpacity(a) {
        this.y = a;
        this.e("OthersCategoryOpacity");
    }
    get selectedSliceFill() {
        return this.a9;
    }
    set selectedSliceFill(a) {
        this.a9 = a;
        this.e("SelectedSliceFill");
    }
    get selectedSliceStroke() {
        return this.ba;
    }
    set selectedSliceStroke(a) {
        this.ba = a;
        this.e("SelectedSliceStroke");
    }
    get selectedSliceStrokeThickness() {
        return this.ae;
    }
    set selectedSliceStrokeThickness(a) {
        this.ae = a;
        this.e("SelectedSliceStrokeThickness");
    }
    get selectedSliceOpacity() {
        return this.ad;
    }
    set selectedSliceOpacity(a) {
        this.ad = a;
        this.e("SelectedSliceOpacity");
    }
    get brushes() {
        return this.f;
    }
    set brushes(a) {
        this.f = a;
        this.e("Brushes");
    }
    get outlines() {
        return this.h;
    }
    set outlines(a) {
        this.h = a;
        this.e("Outlines");
    }
    get legendItemTemplate() {
        return this.a2;
    }
    set legendItemTemplate(a) {
        this.a2 = a;
        this.e("LegendItemTemplate");
    }
    get legendItemBadgeTemplate() {
        return this.a1;
    }
    set legendItemBadgeTemplate(a) {
        this.a1 = a;
        this.e("LegendItemBadgeTemplate");
    }
    get isSurfaceInteractionDisabled() {
        return this.p;
    }
    set isSurfaceInteractionDisabled(a) {
        this.p = a;
        this.e("IsSurfaceInteractionDisabled");
    }
    get formatLabelRef() {
        return this.at;
    }
    set formatLabelRef(a) {
        this.at = a;
        this.e("FormatLabelRef");
    }
    get formatLegendLabelRef() {
        return this.au;
    }
    set formatLegendLabelRef(a) {
        this.au = a;
        this.e("FormatLegendLabelRef");
    }
    get pixelScalingRatio() {
        return this.ab;
    }
    set pixelScalingRatio(a) {
        this.ab = a;
        this.e("PixelScalingRatio");
    }
    get actualPixelScalingRatio() {
        return this.t;
    }
    set actualPixelScalingRatio(a) {
        this.t = a;
        this.e("ActualPixelScalingRatio");
    }
    get selectionMode() {
        return this.bb;
    }
    set selectionMode(a) {
        this.bb = a;
        this.e("SelectionMode");
    }
    get selectedItemRef() {
        return this.a8;
    }
    set selectedItemRef(a) {
        this.a8 = a;
        this.e("SelectedItemRef");
    }
    get selectedItems() {
        return this.i;
    }
    set selectedItems(a) {
        this.i = a;
        this.e("SelectedItems");
    }
    get textStyle() {
        return this.bd;
    }
    set textStyle(a) {
        this.bd = a;
        this.e("TextStyle");
    }
    get LabelClickRef() {
        return this._LabelClickRef;
    }
    set LabelClickRef(a) {
        this._LabelClickRef = a;
    }
    get SelectedItemChangingRef() {
        return this._SelectedItemChangingRef;
    }
    set SelectedItemChangingRef(a) {
        this._SelectedItemChangingRef = a;
    }
    get SelectedItemsChangingRef() {
        return this._SelectedItemsChangingRef;
    }
    set SelectedItemsChangingRef(a) {
        this._SelectedItemsChangingRef = a;
    }
    get SelectedItemChangedRef() {
        return this._SelectedItemChangedRef;
    }
    set SelectedItemChangedRef(a) {
        this._SelectedItemChangedRef = a;
    }
    get SelectedItemsChangedRef() {
        return this._SelectedItemsChangedRef;
    }
    set SelectedItemsChangedRef(a) {
        this._SelectedItemsChangedRef = a;
    }
    get SliceClickRef() {
        return this._SliceClickRef;
    }
    set SliceClickRef(a) {
        this._SliceClickRef = a;
    }
}
PieChartBaseDescription.$t = markType(PieChartBaseDescription, 'PieChartBaseDescription', Description.$);
