// igniteui-angular-core/ES2015/PolarBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { MarkerSeriesDescription } from "./MarkerSeriesDescription";
import { markType } from "./type";
export class PolarBaseDescription extends MarkerSeriesDescription {
    constructor() {
        super();
        this.ch = null;
        this.cj = null;
        this.cg = null;
        this.ci = null;
        this.b4 = false;
        this.b9 = 0;
        this.cl = null;
        this.ck = null;
        this.cf = null;
        this.b7 = 0;
        this.ca = 0;
        this.cb = 0;
        this.b3 = false;
    }
    get_type() {
        return "PolarBase";
    }
    get angleMemberPath() {
        return this.ch;
    }
    set angleMemberPath(a) {
        this.ch = a;
        this.e("AngleMemberPath");
    }
    get radiusMemberPath() {
        return this.cj;
    }
    set radiusMemberPath(a) {
        this.cj = a;
        this.e("RadiusMemberPath");
    }
    get angleAxisRef() {
        return this.cg;
    }
    set angleAxisRef(a) {
        this.cg = a;
        this.e("AngleAxisRef");
    }
    get radiusAxisRef() {
        return this.ci;
    }
    set radiusAxisRef(a) {
        this.ci = a;
        this.e("RadiusAxisRef");
    }
    get useCartesianInterpolation() {
        return this.b4;
    }
    set useCartesianInterpolation(a) {
        this.b4 = a;
        this.e("UseCartesianInterpolation");
    }
    get maximumMarkers() {
        return this.b9;
    }
    set maximumMarkers(a) {
        this.b9 = a;
        this.e("MaximumMarkers");
    }
    get trendLineType() {
        return this.cl;
    }
    set trendLineType(a) {
        this.cl = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.ck;
    }
    set trendLineBrush(a) {
        this.ck = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.cf;
    }
    set actualTrendLineBrush(a) {
        this.cf = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.b7;
    }
    set trendLineThickness(a) {
        this.b7 = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.ca;
    }
    set trendLinePeriod(a) {
        this.ca = a;
        this.e("TrendLinePeriod");
    }
    get trendLineZIndex() {
        return this.cb;
    }
    set trendLineZIndex(a) {
        this.cb = a;
        this.e("TrendLineZIndex");
    }
    get clipSeriesToBounds() {
        return this.b3;
    }
    set clipSeriesToBounds(a) {
        this.b3 = a;
        this.e("ClipSeriesToBounds");
    }
}
PolarBaseDescription.$t = markType(PolarBaseDescription, 'PolarBaseDescription', MarkerSeriesDescription.$);
