// igniteui-angular-core/ES2015/RingSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class RingSeriesBaseDescription extends Description {
    constructor() {
        super();
        this.aq = null;
        this.af = null;
        this.al = null;
        this.ag = null;
        this.ak = null;
        this.ah = null;
        this.ai = null;
        this.o = 0;
        this.n = 0;
        this.aj = null;
        this.m = 0;
        this.p = 0;
        this.an = null;
        this.am = null;
        this.ad = null;
        this.ae = null;
        this.l = 0;
        this.t = 0;
        this.ao = null;
        this.ap = null;
        this.s = 0;
        this.r = 0;
        this.f = null;
        this.g = null;
        this.j = false;
        this.q = 0;
        this._PropertyUpdatedRef = null;
    }
    get_type() {
        return "RingSeriesBase";
    }
    get type() {
        return this.get_type();
    }
    get valueMemberPath() {
        return this.aq;
    }
    set valueMemberPath(a) {
        this.aq = a;
        this.e("ValueMemberPath");
    }
    get labelMemberPath() {
        return this.af;
    }
    set labelMemberPath(a) {
        this.af = a;
        this.e("LabelMemberPath");
    }
    get legendLabelMemberPath() {
        return this.al;
    }
    set legendLabelMemberPath(a) {
        this.al = a;
        this.e("LegendLabelMemberPath");
    }
    get labelsPosition() {
        return this.ag;
    }
    set labelsPosition(a) {
        this.ag = a;
        this.e("LabelsPosition");
    }
    get leaderLineVisibility() {
        return this.ak;
    }
    set leaderLineVisibility(a) {
        this.ak = a;
        this.e("LeaderLineVisibility");
    }
    get leaderLineFill() {
        return this.ah;
    }
    set leaderLineFill(a) {
        this.ah = a;
        this.e("LeaderLineFill");
    }
    get leaderLineStroke() {
        return this.ai;
    }
    set leaderLineStroke(a) {
        this.ai = a;
        this.e("LeaderLineStroke");
    }
    get leaderLineStrokeThickness() {
        return this.o;
    }
    set leaderLineStrokeThickness(a) {
        this.o = a;
        this.e("LeaderLineStrokeThickness");
    }
    get leaderLineOpacity() {
        return this.n;
    }
    set leaderLineOpacity(a) {
        this.n = a;
        this.e("LeaderLineOpacity");
    }
    get leaderLineType() {
        return this.aj;
    }
    set leaderLineType(a) {
        this.aj = a;
        this.e("LeaderLineType");
    }
    get leaderLineMargin() {
        return this.m;
    }
    set leaderLineMargin(a) {
        this.m = a;
        this.e("LeaderLineMargin");
    }
    get othersCategoryThreshold() {
        return this.p;
    }
    set othersCategoryThreshold(a) {
        this.p = a;
        this.e("OthersCategoryThreshold");
    }
    get othersCategoryType() {
        return this.an;
    }
    set othersCategoryType(a) {
        this.an = a;
        this.e("OthersCategoryType");
    }
    get othersCategoryText() {
        return this.am;
    }
    set othersCategoryText(a) {
        this.am = a;
        this.e("OthersCategoryText");
    }
    get formatLabelRef() {
        return this.ad;
    }
    set formatLabelRef(a) {
        this.ad = a;
        this.e("FormatLabelRef");
    }
    get formatLegendLabelRef() {
        return this.ae;
    }
    set formatLegendLabelRef(a) {
        this.ae = a;
        this.e("FormatLegendLabelRef");
    }
    get labelExtent() {
        return this.l;
    }
    set labelExtent(a) {
        this.l = a;
        this.e("LabelExtent");
    }
    get startAngle() {
        return this.t;
    }
    set startAngle(a) {
        this.t = a;
        this.e("StartAngle");
    }
    get selectedSliceFill() {
        return this.ao;
    }
    set selectedSliceFill(a) {
        this.ao = a;
        this.e("SelectedSliceFill");
    }
    get selectedSliceStroke() {
        return this.ap;
    }
    set selectedSliceStroke(a) {
        this.ap = a;
        this.e("SelectedSliceStroke");
    }
    get selectedSliceStrokeThickness() {
        return this.s;
    }
    set selectedSliceStrokeThickness(a) {
        this.s = a;
        this.e("SelectedSliceStrokeThickness");
    }
    get selectedSliceOpacity() {
        return this.r;
    }
    set selectedSliceOpacity(a) {
        this.r = a;
        this.e("SelectedSliceOpacity");
    }
    get brushes() {
        return this.f;
    }
    set brushes(a) {
        this.f = a;
        this.e("Brushes");
    }
    get outlines() {
        return this.g;
    }
    set outlines(a) {
        this.g = a;
        this.e("Outlines");
    }
    get isSurfaceInteractionDisabled() {
        return this.j;
    }
    set isSurfaceInteractionDisabled(a) {
        this.j = a;
        this.e("IsSurfaceInteractionDisabled");
    }
    get radiusFactor() {
        return this.q;
    }
    set radiusFactor(a) {
        this.q = a;
        this.e("RadiusFactor");
    }
    get PropertyUpdatedRef() {
        return this._PropertyUpdatedRef;
    }
    set PropertyUpdatedRef(a) {
        this._PropertyUpdatedRef = a;
    }
}
RingSeriesBaseDescription.$t = markType(RingSeriesBaseDescription, 'RingSeriesBaseDescription', Description.$);
