// igniteui-angular-core/ES2015/ScatterBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { MarkerSeriesDescription } from "./MarkerSeriesDescription";
import { markType } from "./type";
export class ScatterBaseDescription extends MarkerSeriesDescription {
    constructor() {
        super();
        this.cf = null;
        this.ch = null;
        this.cg = null;
        this.ci = null;
        this.ce = null;
        this.cd = null;
        this.cb = null;
        this.b3 = 0;
        this.b6 = 0;
        this.cc = null;
        this.b7 = 0;
        this.b5 = 0;
    }
    get_type() {
        return "ScatterBase";
    }
    get xAxisRef() {
        return this.cf;
    }
    set xAxisRef(a) {
        this.cf = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.ch;
    }
    set yAxisRef(a) {
        this.ch = a;
        this.e("YAxisRef");
    }
    get xMemberPath() {
        return this.cg;
    }
    set xMemberPath(a) {
        this.cg = a;
        this.e("XMemberPath");
    }
    get yMemberPath() {
        return this.ci;
    }
    set yMemberPath(a) {
        this.ci = a;
        this.e("YMemberPath");
    }
    get trendLineType() {
        return this.ce;
    }
    set trendLineType(a) {
        this.ce = a;
        this.e("TrendLineType");
    }
    get trendLineBrush() {
        return this.cd;
    }
    set trendLineBrush(a) {
        this.cd = a;
        this.e("TrendLineBrush");
    }
    get actualTrendLineBrush() {
        return this.cb;
    }
    set actualTrendLineBrush(a) {
        this.cb = a;
        this.e("ActualTrendLineBrush");
    }
    get trendLineThickness() {
        return this.b3;
    }
    set trendLineThickness(a) {
        this.b3 = a;
        this.e("TrendLineThickness");
    }
    get trendLinePeriod() {
        return this.b6;
    }
    set trendLinePeriod(a) {
        this.b6 = a;
        this.e("TrendLinePeriod");
    }
    get markerCollisionAvoidance() {
        return this.cc;
    }
    set markerCollisionAvoidance(a) {
        this.cc = a;
        this.e("MarkerCollisionAvoidance");
    }
    get trendLineZIndex() {
        return this.b7;
    }
    set trendLineZIndex(a) {
        this.b7 = a;
        this.e("TrendLineZIndex");
    }
    get maximumMarkers() {
        return this.b5;
    }
    set maximumMarkers(a) {
        this.b5 = a;
        this.e("MaximumMarkers");
    }
}
ScatterBaseDescription.$t = markType(ScatterBaseDescription, 'ScatterBaseDescription', MarkerSeriesDescription.$);
