// igniteui-angular-core/ES2015/ScatterPolygonSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ShapeSeriesBaseDescription } from "./ShapeSeriesBaseDescription";
import { markType } from "./type";
export class ScatterPolygonSeriesDescription extends ShapeSeriesBaseDescription {
    constructor() {
        super();
        this.bz = null;
        this.bx = null;
        this.b8 = null;
        this.b7 = null;
        this.b3 = null;
        this.b4 = null;
        this.b1 = null;
        this.b6 = null;
        this.b2 = null;
        this.b5 = null;
    }
    get_type() {
        return "ScatterPolygonSeries";
    }
    get shapeStyleSelector() {
        return this.bz;
    }
    set shapeStyleSelector(a) {
        this.bz = a;
        this.e("ShapeStyleSelector");
    }
    get shapeStyle() {
        return this.bx;
    }
    set shapeStyle(a) {
        this.bx = a;
        this.e("ShapeStyle");
    }
    get markerType() {
        return this.b8;
    }
    set markerType(a) {
        this.b8 = a;
        this.e("MarkerType");
    }
    get markerTemplate() {
        return this.b7;
    }
    set markerTemplate(a) {
        this.b7 = a;
        this.e("MarkerTemplate");
    }
    get actualMarkerTemplate() {
        return this.b3;
    }
    set actualMarkerTemplate(a) {
        this.b3 = a;
        this.e("ActualMarkerTemplate");
    }
    get markerBrush() {
        return this.b4;
    }
    set markerBrush(a) {
        this.b4 = a;
        this.e("MarkerBrush");
    }
    get actualMarkerBrush() {
        return this.b1;
    }
    set actualMarkerBrush(a) {
        this.b1 = a;
        this.e("ActualMarkerBrush");
    }
    get markerOutline() {
        return this.b6;
    }
    set markerOutline(a) {
        this.b6 = a;
        this.e("MarkerOutline");
    }
    get actualMarkerOutline() {
        return this.b2;
    }
    set actualMarkerOutline(a) {
        this.b2 = a;
        this.e("ActualMarkerOutline");
    }
    get markerCollisionAvoidance() {
        return this.b5;
    }
    set markerCollisionAvoidance(a) {
        this.b5 = a;
        this.e("MarkerCollisionAvoidance");
    }
}
ScatterPolygonSeriesDescription.$t = markType(ScatterPolygonSeriesDescription, 'ScatterPolygonSeriesDescription', ShapeSeriesBaseDescription.$);
