// igniteui-angular-core/ES2015/ScatterPolylineSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ShapeSeriesBaseDescription } from "./ShapeSeriesBaseDescription";
import { markType } from "./type";
export class ScatterPolylineSeriesDescription extends ShapeSeriesBaseDescription {
    constructor() {
        super();
        this.bz = null;
        this.bx = null;
    }
    get_type() {
        return "ScatterPolylineSeries";
    }
    get shapeStyleSelector() {
        return this.bz;
    }
    set shapeStyleSelector(a) {
        this.bz = a;
        this.e("ShapeStyleSelector");
    }
    get shapeStyle() {
        return this.bx;
    }
    set shapeStyle(a) {
        this.bx = a;
        this.e("ShapeStyle");
    }
}
ScatterPolylineSeriesDescription.$t = markType(ScatterPolylineSeriesDescription, 'ScatterPolylineSeriesDescription', ShapeSeriesBaseDescription.$);
