// igniteui-angular-core/ES2015/ScatterTriangulationSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class ScatterTriangulationSeriesDescription extends SeriesDescription {
    constructor() {
        super();
        this.bt = null;
        this.bv = null;
        this.bs = null;
        this.bu = null;
        this.bp = null;
        this.bq = null;
        this.br = null;
    }
    get_type() {
        return "ScatterTriangulationSeries";
    }
    get xMemberPath() {
        return this.bt;
    }
    set xMemberPath(a) {
        this.bt = a;
        this.e("XMemberPath");
    }
    get yMemberPath() {
        return this.bv;
    }
    set yMemberPath(a) {
        this.bv = a;
        this.e("YMemberPath");
    }
    get xAxisRef() {
        return this.bs;
    }
    set xAxisRef(a) {
        this.bs = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.bu;
    }
    set yAxisRef(a) {
        this.bu = a;
        this.e("YAxisRef");
    }
    get triangleVertexMemberPath1() {
        return this.bp;
    }
    set triangleVertexMemberPath1(a) {
        this.bp = a;
        this.e("TriangleVertexMemberPath1");
    }
    get triangleVertexMemberPath2() {
        return this.bq;
    }
    set triangleVertexMemberPath2(a) {
        this.bq = a;
        this.e("TriangleVertexMemberPath2");
    }
    get triangleVertexMemberPath3() {
        return this.br;
    }
    set triangleVertexMemberPath3(a) {
        this.br = a;
        this.e("TriangleVertexMemberPath3");
    }
}
ScatterTriangulationSeriesDescription.$t = markType(ScatterTriangulationSeriesDescription, 'ScatterTriangulationSeriesDescription', SeriesDescription.$);
