// igniteui-angular-core/ES2015/ScatterContourSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ScatterTriangulationSeriesDescription } from "./ScatterTriangulationSeriesDescription";
import { markType } from "./type";
export class ScatterContourSeriesDescription extends ScatterTriangulationSeriesDescription {
    constructor() {
        super();
        this.b7 = null;
        this.b5 = null;
        this.b3 = null;
    }
    get_type() {
        return "ScatterContourSeries";
    }
    get valueMemberPath() {
        return this.b7;
    }
    set valueMemberPath(a) {
        this.b7 = a;
        this.e("ValueMemberPath");
    }
    get fillScale() {
        return this.b5;
    }
    set fillScale(a) {
        this.b5 = a;
        this.e("FillScale");
    }
    get valueResolver() {
        return this.b3;
    }
    set valueResolver(a) {
        this.b3 = a;
        this.e("ValueResolver");
    }
}
ScatterContourSeriesDescription.$t = markType(ScatterContourSeriesDescription, 'ScatterContourSeriesDescription', ScatterTriangulationSeriesDescription.$);
