// igniteui-angular-core/ES2015/ScrollerView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, runOn, delegateCombine, markType, INotifyPropertyChanged_$type, PropertyChangedEventArgs, Point_$type } from "./type";
import { CanvasGestureDOMEventProxy } from "./CanvasGestureDOMEventProxy";
import { Brush } from "./Brush";
import { Rect } from "./Rect";
import { truncate, isNaN_ } from "./number";
import { ScrollerScrollingEventArgs } from "./ScrollerScrollingEventArgs";
export class ScrollerView extends Base {
    constructor(a) {
        super();
        this.b = null;
        this.x = null;
        this.ab = null;
        this.w = null;
        this.a = null;
        this.aa = null;
        this.y = null;
        this.z = null;
        this.ac = null;
        this.d = false;
        this.c = null;
        this.h = false;
        this.g = false;
        this.s = 0;
        this.p = 0;
        this.t = -1;
        this.q = -1;
        this.f = false;
        this.e = false;
        this.r = 0;
        this.o = 0;
        this.b = a;
    }
    ay() {
        return ((() => {
            let $ret = new Brush();
            $ret.fill = "#666666";
            return $ret;
        })());
    }
    ar(a) {
        if (a == null) {
            if (this.x != null) {
                this.x = null;
            }
            if (this.a != null) {
                this.a.ai = null;
                this.a.an();
            }
            return;
        }
        this.x = a;
        let b = Math.round(this.x.rootWrapper.width());
        let c = Math.round(this.x.rootWrapper.height());
        this.aa = this.x.rootWrapper;
        this.aa.setStyleProperty("position", "relative");
        this.ab = this.x.createElement("div");
        this.ab.setStyleProperty("position", "relative");
        this.ab.setStyleProperty("width", "100%");
        this.ab.setStyleProperty("height", "100%");
        this.ab.setStyleProperty("overflow", "hidden");
        this.aa.append(this.ab);
        this.z = this.x.createElement("div");
        this.z.setStyleProperty("position", "absolute");
        this.z.setStyleProperty("height", this.b.scrollbarSpan + "px");
        this.z.setStyleProperty("width", this.b.minScrollbarSize + "px");
        this.z.setStyleProperty("background-color", this.b.scrollbarBrush._fill);
        this.z.setStyleProperty("display", "none");
        this.z.setStyleProperty("border-radius", "5px");
        this.z.setStyleProperty("z-index", "10000");
        this.z.setStyleProperty("bottom", "0px");
        this.aa.append(this.z);
        this.ac = this.x.createElement("div");
        this.ac.setStyleProperty("position", "absolute");
        this.ac.setStyleProperty("width", this.b.scrollbarSpan + "px");
        this.ac.setStyleProperty("height", this.b.minScrollbarSize + "px");
        this.ac.setStyleProperty("background-color", this.b.scrollbarBrush._fill);
        this.ac.setStyleProperty("display", "none");
        this.ac.setStyleProperty("border-radius", "5px");
        this.ac.setStyleProperty("z-index", "10000");
        this.ac.setStyleProperty("right", "0px");
        this.aa.append(this.ac);
        this.w = this.ab.getNativeElement();
        this.a = new CanvasGestureDOMEventProxy(this.x.rootWrapper, this.x, true);
        this.x.rootWrapper.setProperty("tabIndex", 1000);
        this.x.rootWrapper.setRawStyleProperty("outline", "none");
        this.a.cx = true;
        let d = this.a;
        d.onMouseWheel = delegateCombine(d.onMouseWheel, runOn(this, this.k));
        let e = this.a;
        e.onDragStarted = delegateCombine(e.onDragStarted, runOn(this, this.aj));
        let f = this.a;
        f.onDragDelta = delegateCombine(f.onDragDelta, runOn(this, this.ai));
        let g = this.a;
        g.onDragCompleted = delegateCombine(g.onDragCompleted, runOn(this, this.ah));
        let h = this.a;
        h.onFlingStarted = delegateCombine(h.onFlingStarted, runOn(this, this.i));
        let i = this.a;
        i.onContactStarted = delegateCombine(i.onContactStarted, runOn(this, this.ag));
        let j = this.a;
        j.onContactMoved = delegateCombine(j.onContactMoved, runOn(this, this.af));
        let k = this.a;
        k.onContactCompleted = delegateCombine(k.onContactCompleted, runOn(this, this.ae));
        this.a.z = true;
        this.a.ak = runOn(this, this.n);
        let l = this.a;
        l.onKeyDown = delegateCombine(l.onKeyDown, runOn(this, this.j));
        this.b.b0(b, c);
    }
    j(a) {
        return this.b.n(a);
    }
    n(a, b, c) {
        if (this.e || this.f) {
            return true;
        }
        if (c || b) {
            return true;
        }
        if (this.l(a, b)) {
            return true;
        }
        if (this.m(a, b)) {
            return true;
        }
        return false;
    }
    ae(a, b) {
        this.e = false;
        this.f = false;
    }
    af(a, b) {
        if (this.f) {
            let c = a.y - this.r;
            this.r = a.y;
            this.b.bz(c);
        }
        else if (this.e) {
            let d = a.x - this.o;
            this.o = a.x;
            this.b.bx(d);
        }
    }
    ag(a, b) {
        this.d = false;
        this.ad();
        let c = this.m(a, b);
        let d = this.l(a, b);
        if (c) {
            this.f = true;
            this.r = a.y;
            this.ak();
        }
        else if (d) {
            this.e = true;
            this.o = a.x;
            this.ak();
        }
    }
    l(a, b) {
        if (!this.g) {
            return false;
        }
        if (a.y >= this.b.viewportHeight - this.b.scrollbarSpan && a.y <= this.b.viewportHeight && a.x >= this.q - this.p / 2 && a.x <= this.q + this.p / 2) {
            return true;
        }
        return false;
    }
    m(a, b) {
        if (!this.h) {
            return false;
        }
        if (a.x >= this.b.viewportWidth - this.b.scrollbarSpan && a.x <= this.b.viewportWidth && a.y >= this.t - this.s / 2 && a.y <= this.t + this.s / 2) {
            return true;
        }
        return false;
    }
    i(a, b, c) {
        this.d = true;
        return true;
    }
    ap(a) {
        this.a.bc = new Rect(0, 0, 0, this.b.viewportWidth, this.b.viewportHeight);
    }
    aq(a) {
        this.a.bc = new Rect(0, 0, 0, this.b.viewportWidth, this.b.viewportHeight);
    }
    an(a) {
        if (this.y != null) {
            this.y.setStyleProperty("height", a + "px");
        }
    }
    ao(a) {
        if (this.y != null) {
            this.y.setStyleProperty("width", a + "px");
        }
    }
    ad() {
    }
    ah(a) {
        if (this.e || this.f) {
            return;
        }
        this.d = false;
        this.b.bu(a);
    }
    ai(a) {
        if (this.e || this.f) {
            return;
        }
        this.b.bv(a);
    }
    aw(a) {
        this.w.scrollTop = truncate(a);
    }
    aj(a) {
        if (this.e || this.f) {
            return;
        }
        this.d = false;
        this.b.bw(a);
    }
    u() {
        return this.w.scrollLeft;
    }
    v() {
        return this.w.scrollTop;
    }
    k(a, b) {
        this.d = false;
        this.ad();
        return this.b.p(a, b);
    }
    av(a) {
        this.w.scrollLeft = truncate(a);
    }
    as(a) {
        this.y = a;
        this.ab.append(this.y);
    }
    at(a) {
        if (this.x == null) {
            return;
        }
        if (this.c == null) {
            this.c = this.x.getRequestAnimationFrame();
        }
        if (this.d) {
            a();
        }
        else {
            this.c(() => a());
        }
    }
    am() {
        if (this.h) {
            this.h = false;
            this.ac.setStyleProperty("display", "none");
        }
    }
    al() {
        if (this.g) {
            this.g = false;
            this.z.setStyleProperty("display", "none");
        }
    }
    ax(a, b) {
        let c = false;
        if (!this.h) {
            c = true;
            this.h = true;
            this.ac.setStyleProperty("display", "block");
        }
        if (a != this.t || c) {
            this.t = a;
            if (this.h) {
                this.ac.setRawYPosition(a - b / 2);
            }
        }
        if (b != this.s || c) {
            this.s = b;
            if (this.h) {
                this.ac.setRawSize(this.b.scrollbarSpan, b);
            }
        }
    }
    au(a, b) {
        let c = false;
        if (!this.g) {
            c = true;
            this.g = true;
            this.z.setStyleProperty("display", "block");
        }
        if (a != this.q || c) {
            this.q = a;
            if (this.g) {
                this.z.setRawXPosition(a - b / 2);
            }
        }
        if (b != this.p || c) {
            this.p = b;
            if (this.g) {
                this.z.setRawSize(b, this.b.scrollbarSpan);
            }
        }
    }
    ak() {
        this.x.rootWrapper.focus();
    }
}
ScrollerView.$t = markType(ScrollerView, 'ScrollerView');
export class Scroller extends Base {
    constructor() {
        super();
        this.am = 49;
        this.ae = NaN;
        this.z = 49;
        this.x = NaN;
        this.al = 48;
        this.ad = NaN;
        this.y = 48;
        this.w = NaN;
        this.ac = 0;
        this.af = 20;
        this.ai = 8;
        this.an = 0;
        this.cb = null;
        this.j = false;
        this.ab = 0;
        this.ap = 0;
        this.ao = 0;
        this.ak = 0;
        this.aj = 0;
        this.b = null;
        this.h = true;
        this.k = false;
        this.aa = 0;
        this.v = 0;
        this.g = false;
        this.e = false;
        this.f = false;
        this.cd = null;
        this.d = false;
        this.propertyChanged = null;
        this.onScrolling = null;
        this.ag = 0;
        this.ah = 0;
        this.i = false;
        this.cb = this.view.ay();
    }
    get smallVerticalChange() {
        return this.am;
    }
    set smallVerticalChange(a) {
        let b = this.am;
        this.am = a;
        if (b != this.am) {
            this.by("SmallVerticalChange", b, this.am);
        }
    }
    get largeVerticalChange() {
        return this.ae;
    }
    set largeVerticalChange(a) {
        let b = this.ae;
        this.ae = a;
        if (b != this.ae) {
            this.by("LargeVerticalChange", b, this.ae);
        }
    }
    get au() {
        return this.z;
    }
    set au(a) {
        let b = this.z;
        this.z = a;
        if (b != this.z) {
            this.by("ActualSmallVerticalChange", b, this.z);
        }
    }
    get as() {
        return this.x;
    }
    set as(a) {
        let b = this.x;
        this.x = a;
        if (b != this.x) {
            this.by("ActualLargeVerticalChange", b, this.x);
        }
    }
    get smallHorizontalChange() {
        return this.al;
    }
    set smallHorizontalChange(a) {
        let b = this.al;
        this.al = a;
        if (b != this.al) {
            this.by("SmallHorizontalChange", b, this.al);
        }
    }
    get largeHorizontalChange() {
        return this.ad;
    }
    set largeHorizontalChange(a) {
        let b = this.ad;
        this.ad = a;
        if (b != this.ad) {
            this.by("LargeHorizontalChange", b, this.ad);
        }
    }
    get at() {
        return this.y;
    }
    set at(a) {
        let b = this.y;
        this.y = a;
        if (b != this.y) {
            this.by("ActualSmallHorizontalChange", b, this.y);
        }
    }
    get ar() {
        return this.w;
    }
    set ar(a) {
        let b = this.w;
        this.w = a;
        if (b != this.w) {
            this.by("ActualLargeHorizontalChange", b, this.w);
        }
    }
    get contentWidth() {
        return this.ac;
    }
    set contentWidth(a) {
        let b = this.ac;
        this.ac = a;
        if (b != this.ac) {
            this.by("ContentWidth", b, this.ac);
        }
    }
    get minScrollbarSize() {
        return this.af;
    }
    set minScrollbarSize(a) {
        let b = this.af;
        this.af = a;
        if (b != this.af) {
            this.by("MinScrollbarSize", b, this.af);
        }
    }
    get scrollbarSpan() {
        return this.ai;
    }
    set scrollbarSpan(a) {
        let b = this.ai;
        this.ai = a;
        if (b != this.ai) {
            this.by("ScrollbarSpan", b, this.ai);
        }
    }
    n(a) {
        switch (a) {
            case 11: return this.q();
            case 10: return this.r();
            case 17: return this.m();
            case 15: return this.t();
            case 14: return this.o();
            case 16: return this.s();
        }
        return false;
    }
    s() {
        if (this.scrollLeft + this.viewportWidth >= this.contentWidth) {
            return false;
        }
        this.b5(this.at, 0);
        return true;
    }
    o() {
        if (this.scrollLeft <= 0) {
            return false;
        }
        this.b5(this.at * -1, 0);
        return true;
    }
    t() {
        if (this.scrollTop <= 0) {
            return false;
        }
        this.b5(0, this.au * -1);
        return true;
    }
    m() {
        if (this.scrollTop + this.viewportHeight >= this.contentHeight) {
            return false;
        }
        this.b5(0, this.au);
        return true;
    }
    r() {
        if (this.scrollTop < 0) {
            return false;
        }
        this.b5(0, this.as * -1);
        return true;
    }
    q() {
        if (this.scrollTop + this.viewportHeight >= this.contentHeight) {
            return false;
        }
        this.b5(0, this.as);
        return true;
    }
    get verticalTrackStartInset() {
        return this.an;
    }
    set verticalTrackStartInset(a) {
        let b = this.an;
        this.an = a;
        if (b != this.an) {
            this.by("VerticalTrackStartInset", b, this.an);
        }
    }
    get scrollbarBrush() {
        return this.cb;
    }
    set scrollbarBrush(a) {
        let b = this.cb;
        this.cb = a;
        if (b != this.cb) {
            this.by("ScrollbarBrush", b, this.cb);
        }
    }
    by(a, b, c) {
        if (this.propertyChanged != null) {
            this.propertyChanged(this, new PropertyChangedEventArgs(a));
        }
        this.b1(a, b, c);
    }
    get u() {
        return this.j;
    }
    set u(a) {
        this.j = a;
        if (!this.j) {
            this.view.aw(this.scrollTop);
            this.view.av(this.scrollLeft);
        }
    }
    b1(a, b, c) {
        switch (a) {
            case "ContentWidth":
                this.view.ao(this.contentWidth);
                this.b8();
                break;
            case "ContentHeight":
                this.view.an(this.contentHeight);
                this.ca();
                break;
            case "ScrollTop":
                if (!this.u) {
                    this.view.aw(this.scrollTop);
                }
                break;
            case "ScrollLeft":
                if (!this.u) {
                    this.view.av(this.scrollLeft);
                }
                break;
            case "ViewportWidth":
                this.view.aq(this.viewportWidth);
                if (isNaN_(this.largeVerticalChange)) {
                    this.ar = this.viewportWidth;
                }
                break;
            case "ViewportHeight":
                this.view.ap(this.viewportHeight);
                if (isNaN_(this.largeVerticalChange)) {
                    this.as = this.viewportHeight;
                }
                break;
            case "ScrollbarBrush":
            case "ScrollbarSpan":
            case "MinScrollbarSize":
            case "VerticalTrackStartInset":
                this.ca();
                this.b8();
                break;
        }
    }
    get contentHeight() {
        return this.ab;
    }
    set contentHeight(a) {
        let b = this.ab;
        this.ab = a;
        if (b != this.ab) {
            this.by("ContentHeight", b, this.ab);
        }
    }
    bz(a) {
        let b = Math.max(this.minScrollbarSize, this.actualVerticalScrollbarHeight);
        let c = this.verticalTrackStartInset + (b / 2);
        let d = this.viewportHeight - (b / 2);
        let e = a / (d - c);
        let f = e * (this.contentHeight - this.viewportHeight);
        this.b5(0, f);
    }
    bx(a) {
        let b = Math.max(this.minScrollbarSize, this.actualHorizontalScrollbarWidth);
        let c = 0 + (b / 2);
        let d = this.viewportWidth - (b / 2);
        let e = a / (d - c);
        let f = e * (this.contentWidth - this.viewportWidth);
        this.b5(f, 0);
    }
    b0(a, b) {
        this.ap = a;
        this.ao = b;
    }
    get viewportWidth() {
        return this.ap;
    }
    set viewportWidth(a) {
        let b = this.ap;
        this.ap = a;
        if (b != this.ap) {
            this.by("ViewportWidth", b, this.ap);
        }
    }
    get viewportHeight() {
        return this.ao;
    }
    set viewportHeight(a) {
        let b = this.ao;
        this.ao = a;
        if (b != this.ao) {
            this.by("ViewportHeight", b, this.ao);
        }
    }
    get scrollTop() {
        return this.ak;
    }
    set scrollTop(a) {
        let b = this.ak;
        this.ak = a;
        if (b != this.ak) {
            this.by("ScrollTop", b, this.ak);
        }
    }
    get scrollLeft() {
        return this.aj;
    }
    set scrollLeft(a) {
        let b = this.aj;
        this.aj = a;
        if (b != this.aj) {
            this.by("ScrollLeft", b, this.aj);
        }
    }
    get view() {
        if (this.b == null) {
            this.b = new ScrollerView(this);
        }
        return this.b;
    }
    provideContainer(a) {
        this.view.ar(a);
        this.ak = this.view.v();
        this.aj = this.view.u();
        this.b8();
        this.ca();
    }
    provideContent(a) {
        this.view.as(a);
    }
    p(a, b) {
        this.f = false;
        this.e = false;
        this.g = false;
        this.view.ad();
        this.b5(0, b);
        return true;
    }
    get lockScrollDirection() {
        return this.h;
    }
    set lockScrollDirection(a) {
        this.h = true;
    }
    get actualVerticalScrollbarHeight() {
        return this.aa;
    }
    set actualVerticalScrollbarHeight(a) {
        let b = this.aa;
        this.aa = a;
        if (b != this.aa) {
            this.by("ActualVerticalScrollbarHeight", b, this.aa);
        }
    }
    get actualHorizontalScrollbarWidth() {
        return this.v;
    }
    set actualHorizontalScrollbarWidth(a) {
        let b = this.v;
        this.v = a;
        if (b != this.v) {
            this.by("ActualHorizontalScrollbarWidth", b, this.v);
        }
    }
    focus() {
        this.view.ak();
    }
    bw(a) {
        this.view.ak();
        this.g = false;
        this.e = false;
        this.f = false;
        this.d = true;
        this.cd = { $type: Point_$type, x: a.x, y: a.y };
    }
    bv(a) {
        if (this.d) {
            let b = a.y - this.cd.y;
            let c = a.x - this.cd.x;
            if (this.lockScrollDirection && !this.f) {
                this.f = true;
                if (b > 0 || c > 0) {
                    if (b > c) {
                        this.g = true;
                    }
                    else {
                        this.g = true;
                    }
                }
            }
            this.cd = { $type: Point_$type, x: a.x, y: a.y };
            this.b5(-c, -b);
        }
    }
    scrollTo(a, b) {
        this.ag = a - this.scrollLeft;
        this.ah = b - this.scrollTop;
        this.b4();
    }
    b5(a, b) {
        this.ag += a;
        this.ah += b;
        this.b4();
    }
    b4() {
        if (this.i) {
            return;
        }
        this.i = true;
        this.view.at(runOn(this, this.b7));
    }
    b7() {
        this.i = false;
        let a = this.ag;
        let b = this.ah;
        this.ag = 0;
        this.ah = 0;
        if (this.f) {
            if (this.g) {
                a = 0;
            }
            if (this.e) {
                b = 0;
            }
        }
        let c = Math.round(this.scrollTop + b);
        let d = Math.round(this.scrollLeft + a);
        if (c < 0) {
            c = 0;
            this.view.ad();
        }
        if (d < 0) {
            d = 0;
            this.view.ad();
        }
        if (c + this.viewportHeight > this.contentHeight) {
            c = this.contentHeight - this.viewportHeight;
            if (c < 0) {
                c = 0;
            }
            this.view.ad();
        }
        if (d + this.viewportWidth > this.contentWidth) {
            d = this.contentWidth - this.viewportWidth;
            if (d < 0) {
                d = 0;
            }
            this.view.ad();
        }
        b = c - this.scrollTop;
        a = d - this.scrollLeft;
        if (b != 0 || a != 0) {
            try {
                this.u = true;
                this.k = true;
                this.scrollTop = c;
                this.scrollLeft = d;
                this.ca();
                this.b8();
                this.k = false;
                if (this.onScrolling != null) {
                    this.onScrolling(this, ((() => {
                        let $ret = new ScrollerScrollingEventArgs();
                        $ret.deltaX = a;
                        $ret.deltaY = b;
                        return $ret;
                    })()));
                }
            }
            finally {
                this.u = false;
            }
        }
    }
    ca() {
        this.b9(this.contentHeight, this.viewportHeight, this.scrollTop, true);
    }
    b8() {
        this.b9(this.contentWidth, this.viewportWidth, this.scrollLeft, false);
    }
    b9(a, b, c, d) {
        let e = Math.round((b / a) * b);
        if (e < this.minScrollbarSize) {
            e = this.minScrollbarSize;
        }
        if (e >= b) {
            if (d) {
                this.view.am();
            }
            else {
                this.view.al();
            }
            return;
        }
        let f = d ? this.verticalTrackStartInset : 0;
        let g = f + (e / 2);
        let h = b - (e / 2);
        let i = c / (a - b);
        let j = Math.round(g + (h - g) * i);
        if (d) {
            this.actualVerticalScrollbarHeight = e;
            this.view.ax(j, e);
        }
        else {
            this.actualHorizontalScrollbarWidth = e;
            this.view.au(j, e);
        }
    }
    bu(a) {
        this.f = false;
        this.e = false;
        this.g = false;
        this.d = false;
    }
}
Scroller.$t = markType(Scroller, 'Scroller', Base.$, [INotifyPropertyChanged_$type]);
