// igniteui-angular-core/ES2015/SectionHeaderDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
export class SectionHeaderDescription extends DefinitionBaseDescription {
    constructor() {
        super();
        this.a3 = null;
        this.a2 = null;
        this.av = 0;
        this.ax = 0;
        this.aw = 0;
        this.au = 0;
    }
    get_type() {
        return "SectionHeader";
    }
    get selectedBackground() {
        return this.a3;
    }
    set selectedBackground(a) {
        this.a3 = a;
        this.e("SelectedBackground");
    }
    get actualSelectedBackground() {
        return this.a2;
    }
    set actualSelectedBackground(a) {
        this.a2 = a;
        this.e("ActualSelectedBackground");
    }
    get paddingLeft() {
        return this.av;
    }
    set paddingLeft(a) {
        this.av = a;
        this.e("PaddingLeft");
    }
    get paddingTop() {
        return this.ax;
    }
    set paddingTop(a) {
        this.ax = a;
        this.e("PaddingTop");
    }
    get paddingRight() {
        return this.aw;
    }
    set paddingRight(a) {
        this.aw = a;
        this.e("PaddingRight");
    }
    get paddingBottom() {
        return this.au;
    }
    set paddingBottom(a) {
        this.au = a;
        this.e("PaddingBottom");
    }
}
SectionHeaderDescription.$t = markType(SectionHeaderDescription, 'SectionHeaderDescription', DefinitionBaseDescription.$);
