// igniteui-angular-core/ES2015/SeriesViewerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
export class SeriesViewerDescription extends Description {
    constructor() {
        super();
        this.h = null;
        this.r = false;
        this.k = null;
        this.m = null;
        this.y = false;
        this.v = false;
        this.bs = 0;
        this.br = 0;
        this.t = false;
        this.bx = null;
        this.bv = null;
        this.u = false;
        this.cd = null;
        this.a3 = 0;
        this.b2 = null;
        this.by = null;
        this.b4 = null;
        this.bz = null;
        this.b0 = null;
        this.b3 = null;
        this.l = null;
        this.a1 = 0;
        this.a2 = 0;
        this.bw = null;
        this.ca = null;
        this.b6 = null;
        this.cc = null;
        this.b8 = null;
        this.cb = null;
        this.b7 = null;
        this.az = 0;
        this.ax = 0;
        this.ay = 0;
        this.aw = 0;
        this.av = 0;
        this.at = 0;
        this.au = 0;
        this.as = 0;
        this.b5 = null;
        this.b9 = null;
        this.x = false;
        this.w = false;
        this.a0 = 0;
        this.ap = 0;
        this.ar = 0;
        this.an = 0;
        this.am = 0;
        this.al = 0;
        this.s = false;
        this.aq = 0;
        this.ao = 0;
        this.b1 = null;
        this.ai = 0;
        this.ah = 0;
        this.j = null;
        this.aj = 0;
        this.ak = 0;
        this._SeriesCursorMouseMoveRef = null;
        this._SeriesMouseLeftButtonDownRef = null;
        this._SeriesMouseLeftButtonUpRef = null;
        this._SeriesMouseMoveRef = null;
        this._SeriesMouseEnterRef = null;
        this._SeriesMouseLeaveRef = null;
        this._WindowRectChangedRef = null;
        this._SizeChangedRef = null;
        this._GridAreaRectChangedRef = null;
        this.f = null;
    }
    get_type() {
        return "SeriesViewer";
    }
    get type() {
        return this.get_type();
    }
    get crosshairPoint() {
        return this.h;
    }
    set crosshairPoint(a) {
        this.h = a;
        this.e("CrosshairPoint");
    }
    get animateSeriesWhenAxisRangeChanges() {
        return this.r;
    }
    set animateSeriesWhenAxisRangeChanges(a) {
        this.r = a;
        this.e("AnimateSeriesWhenAxisRangeChanges");
    }
    get effectiveViewport() {
        return this.k;
    }
    set effectiveViewport(a) {
        this.k = a;
        this.e("EffectiveViewport");
    }
    get windowRect() {
        return this.m;
    }
    set windowRect(a) {
        this.m = a;
        this.e("WindowRect");
    }
    get useTiledZooming() {
        return this.y;
    }
    set useTiledZooming(a) {
        this.y = a;
        this.e("UseTiledZooming");
    }
    get preferHigherResolutionTiles() {
        return this.v;
    }
    set preferHigherResolutionTiles(a) {
        this.v = a;
        this.e("PreferHigherResolutionTiles");
    }
    get zoomTileCacheSize() {
        return this.bs;
    }
    set zoomTileCacheSize(a) {
        this.bs = a;
        this.e("ZoomTileCacheSize");
    }
    get highlightingTransitionDuration() {
        return this.br;
    }
    set highlightingTransitionDuration(a) {
        this.br = a;
        this.e("HighlightingTransitionDuration");
    }
    get isPagePanningAllowed() {
        return this.t;
    }
    set isPagePanningAllowed(a) {
        this.t = a;
        this.e("IsPagePanningAllowed");
    }
    get contentHitTestMode() {
        return this.bx;
    }
    set contentHitTestMode(a) {
        this.bx = a;
        this.e("ContentHitTestMode");
    }
    get actualContentHitTestMode() {
        return this.bv;
    }
    set actualContentHitTestMode(a) {
        this.bv = a;
        this.e("ActualContentHitTestMode");
    }
    get isSurfaceInteractionDisabled() {
        return this.u;
    }
    set isSurfaceInteractionDisabled(a) {
        this.u = a;
        this.e("IsSurfaceInteractionDisabled");
    }
    get windowResponse() {
        return this.cd;
    }
    set windowResponse(a) {
        this.cd = a;
        this.e("WindowResponse");
    }
    get windowRectMinWidth() {
        return this.a3;
    }
    set windowRectMinWidth(a) {
        this.a3 = a;
        this.e("WindowRectMinWidth");
    }
    get overviewPlusDetailPaneVisibility() {
        return this.b2;
    }
    set overviewPlusDetailPaneVisibility(a) {
        this.b2 = a;
        this.e("OverviewPlusDetailPaneVisibility");
    }
    get crosshairVisibility() {
        return this.by;
    }
    set crosshairVisibility(a) {
        this.by = a;
        this.e("CrosshairVisibility");
    }
    get plotAreaBackground() {
        return this.b4;
    }
    set plotAreaBackground(a) {
        this.b4 = a;
        this.e("PlotAreaBackground");
    }
    get defaultInteraction() {
        return this.bz;
    }
    set defaultInteraction(a) {
        this.bz = a;
        this.e("DefaultInteraction");
    }
    get dragModifier() {
        return this.b0;
    }
    set dragModifier(a) {
        this.b0 = a;
        this.e("DragModifier");
    }
    get panModifier() {
        return this.b3;
    }
    set panModifier(a) {
        this.b3 = a;
        this.e("PanModifier");
    }
    get previewRect() {
        return this.l;
    }
    set previewRect(a) {
        this.l = a;
        this.e("PreviewRect");
    }
    get windowPositionHorizontal() {
        return this.a1;
    }
    set windowPositionHorizontal(a) {
        this.a1 = a;
        this.e("WindowPositionHorizontal");
    }
    get windowPositionVertical() {
        return this.a2;
    }
    set windowPositionVertical(a) {
        this.a2 = a;
        this.e("WindowPositionVertical");
    }
    get chartTitle() {
        return this.bw;
    }
    set chartTitle(a) {
        this.bw = a;
        this.e("ChartTitle");
    }
    get titleHorizontalAlignment() {
        return this.ca;
    }
    set titleHorizontalAlignment(a) {
        this.ca = a;
        this.e("TitleHorizontalAlignment");
    }
    get subtitleHorizontalAlignment() {
        return this.b6;
    }
    set subtitleHorizontalAlignment(a) {
        this.b6 = a;
        this.e("SubtitleHorizontalAlignment");
    }
    get titleTextStyle() {
        return this.cc;
    }
    set titleTextStyle(a) {
        this.cc = a;
        this.e("TitleTextStyle");
    }
    get subtitleTextStyle() {
        return this.b8;
    }
    set subtitleTextStyle(a) {
        this.b8 = a;
        this.e("SubtitleTextStyle");
    }
    get titleTextColor() {
        return this.cb;
    }
    set titleTextColor(a) {
        this.cb = a;
        this.e("TitleTextColor");
    }
    get subtitleTextColor() {
        return this.b7;
    }
    set subtitleTextColor(a) {
        this.b7 = a;
        this.e("SubtitleTextColor");
    }
    get titleTopMargin() {
        return this.az;
    }
    set titleTopMargin(a) {
        this.az = a;
        this.e("TitleTopMargin");
    }
    get titleLeftMargin() {
        return this.ax;
    }
    set titleLeftMargin(a) {
        this.ax = a;
        this.e("TitleLeftMargin");
    }
    get titleRightMargin() {
        return this.ay;
    }
    set titleRightMargin(a) {
        this.ay = a;
        this.e("TitleRightMargin");
    }
    get titleBottomMargin() {
        return this.aw;
    }
    set titleBottomMargin(a) {
        this.aw = a;
        this.e("TitleBottomMargin");
    }
    get subtitleTopMargin() {
        return this.av;
    }
    set subtitleTopMargin(a) {
        this.av = a;
        this.e("SubtitleTopMargin");
    }
    get subtitleLeftMargin() {
        return this.at;
    }
    set subtitleLeftMargin(a) {
        this.at = a;
        this.e("SubtitleLeftMargin");
    }
    get subtitleRightMargin() {
        return this.au;
    }
    set subtitleRightMargin(a) {
        this.au = a;
        this.e("SubtitleRightMargin");
    }
    get subtitleBottomMargin() {
        return this.as;
    }
    set subtitleBottomMargin(a) {
        this.as = a;
        this.e("SubtitleBottomMargin");
    }
    get subtitle() {
        return this.b5;
    }
    set subtitle(a) {
        this.b5 = a;
        this.e("Subtitle");
    }
    get syncChannel() {
        return this.b9;
    }
    set syncChannel(a) {
        this.b9 = a;
        this.e("SyncChannel");
    }
    get synchronizeVertically() {
        return this.x;
    }
    set synchronizeVertically(a) {
        this.x = a;
        this.e("SynchronizeVertically");
    }
    get synchronizeHorizontally() {
        return this.w;
    }
    set synchronizeHorizontally(a) {
        this.w = a;
        this.e("SynchronizeHorizontally");
    }
    get topMargin() {
        return this.a0;
    }
    set topMargin(a) {
        this.a0 = a;
        this.e("TopMargin");
    }
    get leftMargin() {
        return this.ap;
    }
    set leftMargin(a) {
        this.ap = a;
        this.e("LeftMargin");
    }
    get rightMargin() {
        return this.ar;
    }
    set rightMargin(a) {
        this.ar = a;
        this.e("RightMargin");
    }
    get bottomMargin() {
        return this.an;
    }
    set bottomMargin(a) {
        this.an = a;
        this.e("BottomMargin");
    }
    get autoMarginWidth() {
        return this.am;
    }
    set autoMarginWidth(a) {
        this.am = a;
        this.e("AutoMarginWidth");
    }
    get autoMarginHeight() {
        return this.al;
    }
    set autoMarginHeight(a) {
        this.al = a;
        this.e("AutoMarginHeight");
    }
    get isAntiAliasingEnabledDuringInteraction() {
        return this.s;
    }
    set isAntiAliasingEnabledDuringInteraction(a) {
        this.s = a;
        this.e("IsAntiAliasingEnabledDuringInteraction");
    }
    get pixelScalingRatio() {
        return this.aq;
    }
    set pixelScalingRatio(a) {
        this.aq = a;
        this.e("PixelScalingRatio");
    }
    get interactionPixelScalingRatio() {
        return this.ao;
    }
    set interactionPixelScalingRatio(a) {
        this.ao = a;
        this.e("InteractionPixelScalingRatio");
    }
    get overviewPlusDetailPaneBackgroundImageUri() {
        return this.b1;
    }
    set overviewPlusDetailPaneBackgroundImageUri(a) {
        this.b1 = a;
        this.e("OverviewPlusDetailPaneBackgroundImageUri");
    }
    get actualPixelScalingRatio() {
        return this.ai;
    }
    set actualPixelScalingRatio(a) {
        this.ai = a;
        this.e("ActualPixelScalingRatio");
    }
    get actualInteractionPixelScalingRatio() {
        return this.ah;
    }
    set actualInteractionPixelScalingRatio(a) {
        this.ah = a;
        this.e("ActualInteractionPixelScalingRatio");
    }
    get actualWindowRect() {
        return this.j;
    }
    set actualWindowRect(a) {
        this.j = a;
        this.e("ActualWindowRect");
    }
    get actualWindowPositionHorizontal() {
        return this.aj;
    }
    set actualWindowPositionHorizontal(a) {
        this.aj = a;
        this.e("ActualWindowPositionHorizontal");
    }
    get actualWindowPositionVertical() {
        return this.ak;
    }
    set actualWindowPositionVertical(a) {
        this.ak = a;
        this.e("ActualWindowPositionVertical");
    }
    get SeriesCursorMouseMoveRef() {
        return this._SeriesCursorMouseMoveRef;
    }
    set SeriesCursorMouseMoveRef(a) {
        this._SeriesCursorMouseMoveRef = a;
    }
    get SeriesMouseLeftButtonDownRef() {
        return this._SeriesMouseLeftButtonDownRef;
    }
    set SeriesMouseLeftButtonDownRef(a) {
        this._SeriesMouseLeftButtonDownRef = a;
    }
    get SeriesMouseLeftButtonUpRef() {
        return this._SeriesMouseLeftButtonUpRef;
    }
    set SeriesMouseLeftButtonUpRef(a) {
        this._SeriesMouseLeftButtonUpRef = a;
    }
    get SeriesMouseMoveRef() {
        return this._SeriesMouseMoveRef;
    }
    set SeriesMouseMoveRef(a) {
        this._SeriesMouseMoveRef = a;
    }
    get SeriesMouseEnterRef() {
        return this._SeriesMouseEnterRef;
    }
    set SeriesMouseEnterRef(a) {
        this._SeriesMouseEnterRef = a;
    }
    get SeriesMouseLeaveRef() {
        return this._SeriesMouseLeaveRef;
    }
    set SeriesMouseLeaveRef(a) {
        this._SeriesMouseLeaveRef = a;
    }
    get WindowRectChangedRef() {
        return this._WindowRectChangedRef;
    }
    set WindowRectChangedRef(a) {
        this._WindowRectChangedRef = a;
    }
    get SizeChangedRef() {
        return this._SizeChangedRef;
    }
    set SizeChangedRef(a) {
        this._SizeChangedRef = a;
    }
    get GridAreaRectChangedRef() {
        return this._GridAreaRectChangedRef;
    }
    set GridAreaRectChangedRef(a) {
        this._GridAreaRectChangedRef = a;
    }
    get series() {
        return this.f;
    }
    set series(a) {
        this.f = a;
        this.e("Series");
    }
}
SeriesViewerDescription.$t = markType(SeriesViewerDescription, 'SeriesViewerDescription', Description.$);
