// igniteui-angular-core/ES2015/ShapefileConverter.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { DependencyObjectNotifier } from "./DependencyObjectNotifier";
import { IList$1_$type, BaseError, delegateCombine, typeCast, markType } from "./type";
import { ShapefileRecord } from "./ShapefileRecord";
import { INotifyCollectionChanged_$type } from "./INotifyCollectionChanged";
import { Rect } from "./Rect";
import { Uri } from "./Uri";
import { DependencyProperty } from "./DependencyProperty";
import { ObservableCollection$1 } from "./ObservableCollection$1";
import { AsyncCompletedEventArgs } from "./AsyncCompletedEventArgs";
import { BinaryReader } from "./BinaryReader";
import { BinaryFileDownloader } from "./BinaryFileDownloader";
import { ShapeFileUtil } from "./ShapeFileUtil";
import { PropertyMetadata } from "./PropertyMetadata";
export class ShapefileConverter extends DependencyObjectNotifier {
    constructor() {
        super();
        this.aa = null;
        this._shapeHeader = null;
        this.importCompleted = null;
        this.m = null;
        this.collectionChanged = null;
        this.records = new ObservableCollection$1(ShapefileRecord.$, 0);
        let a = this.records;
        a.collectionChanged = delegateCombine(a.collectionChanged, (a, b) => {
            if (this.collectionChanged != null) {
                this.collectionChanged(this, b);
            }
        });
    }
    get worldRect() {
        return this.aa;
    }
    set worldRect(a) {
        if (Rect.l_op_Inequality(this.aa, a)) {
            let b = this.aa;
            this.aa = a;
            this.x("WorldRect", b, this.aa);
        }
    }
    get shapeType() {
        return this.shapeHeader.a;
    }
    get shapeHeader() {
        return this._shapeHeader;
    }
    set shapeHeader(a) {
        this._shapeHeader = a;
    }
    get shapefileSource() {
        return typeCast(Uri.$, this.c(ShapefileConverter.shapefileSourceProperty));
    }
    set shapefileSource(a) {
        this.h(ShapefileConverter.shapefileSourceProperty, a);
    }
    get databaseSource() {
        return typeCast(Uri.$, this.c(ShapefileConverter.databaseSourceProperty));
    }
    set databaseSource(a) {
        this.h(ShapefileConverter.databaseSourceProperty, a);
    }
    x(a, b, c) {
        switch (a) {
            case "ShapefileSource":
            case "DatabaseSource":
                if (this.shapefileSource != null && this.databaseSource != null) {
                    this.u();
                }
                break;
        }
        this.j(a);
    }
    v(a) {
        if (this.importCompleted != null) {
            this.importCompleted(this, a);
        }
    }
    u() {
        let a = null;
        let b = null;
        let c = () => {
            if (a != null && b != null) {
                this.w(a, b);
                this.v(new AsyncCompletedEventArgs(null, false, null));
            }
        };
        BinaryFileDownloader.a(this.shapefileSource.value, (d) => {
            a = new BinaryReader(0, d, false);
            c();
        }, (d) => {
            throw new BaseError(1, "shape file download error: " + d);
        });
        BinaryFileDownloader.a(this.databaseSource.value, (d) => {
            b = new BinaryReader(0, d, false);
            c();
        }, (d) => {
            throw new BaseError(1, "dbf file download error: " + d);
        });
    }
    w(a, b) {
        this.shapeHeader = ShapeFileUtil.a(a, b);
        this.worldRect = this.shapeHeader.e;
        let c = a.currentPosition;
        let d = a.length;
        while (c < d) {
            let e = ShapeFileUtil.e(this.shapeHeader, a, b);
            let record_ = e;
            this.records.add(e);
            c = a.currentPosition;
            d = a.length;
        }
    }
    get records() {
        return this.m;
    }
    set records(a) {
        this.m = a;
    }
    indexOf(a) {
        return this.m.indexOf(a);
    }
    insert(a, b) {
        this.m.insert(a, b);
    }
    removeAt(a) {
        this.m.removeAt(a);
    }
    item(a, b) {
        if (arguments.length === 2) {
            this.m.item(a, b);
            return b;
        }
        else {
            return this.m._inner[a];
        }
    }
    add(a) {
        this.m.add(a);
    }
    clear() {
        this.m.clear();
    }
    contains(a) {
        return this.m.contains(a);
    }
    copyTo(a, b) {
        this.m.copyTo(a, b);
    }
    get count() {
        return this.m.count;
    }
    get isReadOnly() {
        return this.m.isReadOnly;
    }
    remove(a) {
        return this.m.remove(a);
    }
    getEnumerator() {
        return this.m.getEnumerator();
    }
    getEnumeratorObject() {
        return this.m.getEnumerator();
    }
}
ShapefileConverter.$t = markType(ShapefileConverter, 'ShapefileConverter', DependencyObjectNotifier.$, [IList$1_$type.specialize(ShapefileRecord.$), INotifyCollectionChanged_$type]);
ShapefileConverter.shapefileSourceProperty = DependencyProperty.i("ShapefileSource", Uri.$, ShapefileConverter.$, new PropertyMetadata(2, null, (a, b) => (typeCast(ShapefileConverter.$, a)).x("ShapefileSource", b.oldValue, b.newValue)));
ShapefileConverter.databaseSourceProperty = DependencyProperty.i("DatabaseSource", Uri.$, ShapefileConverter.$, new PropertyMetadata(2, null, (a, b) => (typeCast(ShapefileConverter.$, a)).x("DatabaseSource", b.oldValue, b.newValue)));
