// igniteui-angular-core/ES2015/ShapeSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { SeriesDescription } from "./SeriesDescription";
import { markType } from "./type";
export class ShapeSeriesBaseDescription extends SeriesDescription {
    constructor() {
        super();
        this.br = null;
        this.bs = null;
        this.bt = null;
        this.bp = 0;
    }
    get_type() {
        return "ShapeSeriesBase";
    }
    get shapeMemberPath() {
        return this.br;
    }
    set shapeMemberPath(a) {
        this.br = a;
        this.e("ShapeMemberPath");
    }
    get xAxisRef() {
        return this.bs;
    }
    set xAxisRef(a) {
        this.bs = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.bt;
    }
    set yAxisRef(a) {
        this.bt = a;
        this.e("YAxisRef");
    }
    get shapeFilterResolution() {
        return this.bp;
    }
    set shapeFilterResolution(a) {
        this.bp = a;
        this.e("ShapeFilterResolution");
    }
}
ShapeSeriesBaseDescription.$t = markType(ShapeSeriesBaseDescription, 'ShapeSeriesBaseDescription', SeriesDescription.$);
