// igniteui-angular-core/ES2015/StackedSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategorySeriesDescription } from "./CategorySeriesDescription";
import { markType } from "./type";
export class StackedSeriesBaseDescription extends CategorySeriesDescription {
    constructor() {
        super();
        this.ch = false;
        this.ci = false;
        this._SeriesCreatedRef = null;
        this.cf = null;
    }
    get_type() {
        return "StackedSeriesBase";
    }
    get autoGenerateSeries() {
        return this.ch;
    }
    set autoGenerateSeries(a) {
        this.ch = a;
        this.e("AutoGenerateSeries");
    }
    get reverseLegendOrder() {
        return this.ci;
    }
    set reverseLegendOrder(a) {
        this.ci = a;
        this.e("ReverseLegendOrder");
    }
    get SeriesCreatedRef() {
        return this._SeriesCreatedRef;
    }
    set SeriesCreatedRef(a) {
        this._SeriesCreatedRef = a;
    }
    get series() {
        return this.cf;
    }
    set series(a) {
        this.cf = a;
        this.e("Series");
    }
}
StackedSeriesBaseDescription.$t = markType(StackedSeriesBaseDescription, 'StackedSeriesBaseDescription', CategorySeriesDescription.$);
