// igniteui-angular-core/ES2015/TimeAxisBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
export class TimeAxisBaseDescription extends CategoryAxisBaseDescription {
    constructor() {
        super();
        this.c6 = null;
        this.cw = false;
        this.cz = new Date();
        this.cy = new Date();
        this.c1 = new Date();
        this.c0 = new Date();
    }
    get_type() {
        return "TimeAxisBase";
    }
    get dateTimeMemberPath() {
        return this.c6;
    }
    set dateTimeMemberPath(a) {
        this.c6 = a;
        this.e("DateTimeMemberPath");
    }
    get isDataPreSorted() {
        return this.cw;
    }
    set isDataPreSorted(a) {
        this.cw = a;
        this.e("IsDataPreSorted");
    }
    get actualMinimumValue() {
        return this.cz;
    }
    set actualMinimumValue(a) {
        this.cz = a;
        this.e("ActualMinimumValue");
    }
    get actualMaximumValue() {
        return this.cy;
    }
    set actualMaximumValue(a) {
        this.cy = a;
        this.e("ActualMaximumValue");
    }
    get minimumValue() {
        return this.c1;
    }
    set minimumValue(a) {
        this.c1 = a;
        this.e("MinimumValue");
    }
    get maximumValue() {
        return this.c0;
    }
    set maximumValue(a) {
        this.c0 = a;
        this.e("MaximumValue");
    }
}
TimeAxisBaseDescription.$t = markType(TimeAxisBaseDescription, 'TimeAxisBaseDescription', CategoryAxisBaseDescription.$);
