// igniteui-angular-core/ES2015/ValueBrushScaleDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BrushScaleDescription } from "./BrushScaleDescription";
import { markType } from "./type";
export class ValueBrushScaleDescription extends BrushScaleDescription {
    constructor() {
        super();
        this.m = 0;
        this.l = 0;
        this.j = false;
        this.p = 0;
    }
    get_type() {
        return "ValueBrushScale";
    }
    get minimumValue() {
        return this.m;
    }
    set minimumValue(a) {
        this.m = a;
        this.e("MinimumValue");
    }
    get maximumValue() {
        return this.l;
    }
    set maximumValue(a) {
        this.l = a;
        this.e("MaximumValue");
    }
    get isLogarithmic() {
        return this.j;
    }
    set isLogarithmic(a) {
        this.j = a;
        this.e("IsLogarithmic");
    }
    get logarithmBase() {
        return this.p;
    }
    set logarithmBase(a) {
        this.p = a;
        this.e("LogarithmBase");
    }
}
ValueBrushScaleDescription.$t = markType(ValueBrushScaleDescription, 'ValueBrushScaleDescription', BrushScaleDescription.$);
