// igniteui-angular-core/ES2015/VerticalStackedSeriesBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StackedSeriesBaseDescription } from "./StackedSeriesBaseDescription";
import { markType } from "./type";
export class VerticalStackedSeriesBaseDescription extends StackedSeriesBaseDescription {
    constructor() {
        super();
        this.cm = null;
        this.cn = null;
    }
    get_type() {
        return "VerticalStackedSeriesBase";
    }
    get xAxisRef() {
        return this.cm;
    }
    set xAxisRef(a) {
        this.cm = a;
        this.e("XAxisRef");
    }
    get yAxisRef() {
        return this.cn;
    }
    set yAxisRef(a) {
        this.cn = a;
        this.e("YAxisRef");
    }
}
VerticalStackedSeriesBaseDescription.$t = markType(VerticalStackedSeriesBaseDescription, 'VerticalStackedSeriesBaseDescription', StackedSeriesBaseDescription.$);
