// igniteui-angular-core/ES2015/WaterfallSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { HorizontalAnchoredCategorySeriesDescription } from "./HorizontalAnchoredCategorySeriesDescription";
import { markType } from "./type";
export class WaterfallSeriesDescription extends HorizontalAnchoredCategorySeriesDescription {
    constructor() {
        super();
        this.c3 = null;
        this.cz = 0;
        this.c0 = 0;
        this.c4 = null;
    }
    get_type() {
        return "WaterfallSeries";
    }
    get negativeBrush() {
        return this.c3;
    }
    set negativeBrush(a) {
        this.c3 = a;
        this.e("NegativeBrush");
    }
    get radiusX() {
        return this.cz;
    }
    set radiusX(a) {
        this.cz = a;
        this.e("RadiusX");
    }
    get radiusY() {
        return this.c0;
    }
    set radiusY(a) {
        this.c0 = a;
        this.e("RadiusY");
    }
    get negativeOutline() {
        return this.c4;
    }
    set negativeOutline(a) {
        this.c4 = a;
        this.e("NegativeOutline");
    }
}
WaterfallSeriesDescription.$t = markType(WaterfallSeriesDescription, 'WaterfallSeriesDescription', HorizontalAnchoredCategorySeriesDescription.$);
