// igniteui-angular-core/ES2015/WriteState.js
// -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
//
// System.Xml.WriteState
//
// Author:
//   Daniel Weber (daniel-weber@austin.rr.com)
//
// (C) 2001 Daniel Weber
import { markEnum } from "./type";
/**
 */
/**
 */
export var WriteState;
/**
 */
(function (WriteState) {
    /**
     * A write method has not been called.
     */
    WriteState[WriteState["Start"] = 0] = "Start";
    /**
     * The prolog is being written.
     */
    WriteState[WriteState["Prolog"] = 1] = "Prolog";
    /**
     * An element start tag is being written.
     */
    WriteState[WriteState["Element"] = 2] = "Element";
    /**
     * An attribute is being written.
     */
    WriteState[WriteState["Attribute"] = 3] = "Attribute";
    /**
     * Element content is being written.
     */
    WriteState[WriteState["Content"] = 4] = "Content";
    /**
     * The close method has been called.
     */
    WriteState[WriteState["Closed"] = 5] = "Closed";
})(WriteState || (WriteState = {}));
export let WriteState_$type = markEnum('WriteState', 'Start,0|Prolog,1|Element,2|Attribute,3|Content,4|Closed,5');
