// igniteui-angular-core/ES2015/XamMultiScaleImageView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, runOn, Delegate_$type, Type, markType, typeCast, INotifyPropertyChanged_$type, EventArgs, Point_$type } from "./type";
import { ISchedulableRender_$type } from "./ISchedulableRender";
import { List$1 } from "./List$1";
import { Image } from "./Image";
import { RenderingContext } from "./RenderingContext";
import { Dictionary$2 } from "./Dictionary$2";
import { Tile } from "./Tile";
import { Rect } from "./Rect";
import { CanvasRenderScheduler } from "./CanvasRenderScheduler";
import { StackPool$1 } from "./StackPool$1";
import { WriteableBitmap } from "./WriteableBitmap";
import { Uri } from "./Uri";
import { DownloadingMultiScaleImageEventArgs } from "./DownloadingMultiScaleImageEventArgs";
import { CanvasViewRenderer } from "./CanvasViewRenderer";
import { truncate, intDivide, logBase } from "./number";
import { DependencyObject } from "./DependencyObject";
import { Control } from "./Control";
import { DependencyProperty } from "./DependencyProperty";
import { IEasingFunction_$type } from "./IEasingFunction";
import { LRUCache$2 } from "./LRUCache$2";
import { TilePositionComparer } from "./TilePositionComparer";
import { Convert } from "./Convert";
import { MathUtil } from "./MathUtil";
import { PropertyMetadata } from "./PropertyMetadata";
import { dateNow } from "./date";
export class XamMultiScaleImageView extends Base {
    constructor(a) {
        super();
        this.b = null;
        this.h = null;
        this.g = new Dictionary$2(Base.$, List$1.$.specialize(Delegate_$type), 0);
        this.j = new List$1(Tile.$, 0);
        this.i = new List$1(Tile.$, 0);
        this.p = -1;
        this.o = -1;
        this.k = null;
        this.m = null;
        this.c = null;
        this._isDirty = false;
        this.au = Rect.empty;
        this.l = null;
        this.b = a;
        this.b.ag.l(this);
        this.h = new List$1(Image.$, 0);
        this.b.ao = ((() => {
            let $ret = new StackPool$1(Image.$);
            $ret.j = runOn(this, this.at);
            $ret.b = runOn(this, this.y);
            $ret.c = runOn(this, this.aa);
            $ret.d = runOn(this, this.z);
            return $ret;
        })());
        for (let b = 0; b < 4; ++b) {
            this.i.add(null);
        }
    }
    at() {
        return new Image();
    }
    y(a) {
        this.h.add(a);
    }
    aa(a) {
        this.h.remove(a);
    }
    z(a) {
    }
    ao(a, b, c) {
        a.n = b;
        a.o = c;
    }
    r(a) {
        for (let b = 0; b < this.i.count; ++b) {
            if (a == this.i._inner[b]) {
                if (a.f != null && a.f.ac != null) {
                    let c = a.f.ac;
                    let d = c.c;
                    if (this.g.containsKey(d)) {
                        let e = this.g.item(d);
                        for (let f = 0; f < e.count; f++) {
                            e._inner[f]();
                        }
                        e.clear();
                        this.g.removeItem(d);
                    }
                }
                this.i._inner[b] = null;
                this.q();
                return;
            }
        }
        for (let g = 0; g < this.j.count; ++g) {
            if (this.j._inner[g] == a) {
                this.j.removeAt(g);
                break;
            }
        }
    }
    u(a) {
        this.j.add(a);
        this.j.aa((b, c) => {
            let d = 0;
            let e = 0;
            if (b.e != null) {
                d = b.e.ac.b;
            }
            if (c.e != null) {
                e = c.e.ac.b;
            }
            if (d < e) {
                return -1;
            }
            else if (d > e) {
                return 1;
            }
            return 0;
        });
        this.q();
    }
    q() {
        let index_ = -1;
        if (this.j.count > 0) {
            for (let a = 0; a < this.i.count; ++a) {
                if (this.i._inner[a] == null) {
                    index_ = a;
                    break;
                }
            }
        }
        if (index_ >= 0) {
            this.i._inner[index_] = this.j._inner[0];
            let b = new WriteableBitmap(this.b.an.m, this.b.an.k);
            this.i._inner[index_].f.ac = b;
            let ele_ = this.l.createElement("img");
            let c = ele_.getNativeElement();
            b.c = c;
            let self_ = this;
            let d = new List$1(Delegate_$type, 0);
            this.g.addItem(c, d);
            d.add(ele_.listen("load", (e) => this.ab(e, index_)));
            d.add(ele_.listen("readystatechange", (e) => this.ab(e, index_)));
            d.add(ele_.listen("error", (e) => this.ac(e, index_)));
            this.j.removeAt(0);
            let e = this.b.an.r(this.i._inner[index_].d + 8, this.i._inner[index_].b, this.i._inner[index_].c);
            let f = e.value;
            let g = ((() => {
                let $ret = new DownloadingMultiScaleImageEventArgs();
                $ret.uri = f;
                $ret.image = c;
                return $ret;
            })());
            this.b.bk(g);
            f = g.uri;
            if (f != null) {
                c.src = f;
            }
        }
    }
    ab(a, b) {
        let e_ = a.originalEvent;
        let img_ = (e_.target);
        let c = (img_.complete || (img_.readyState == 'complete' && e_.type == 'readystatechange'));
        if (c) {
            this.downloadCompleted(a, b);
        }
    }
    ac(a, b) {
        this.downloadError(a, b);
    }
    downloadError(a, b) {
        let c = this.i._inner[b];
        this.i._inner[b] = null;
        if (c != null && c.f != null && c.f.ac != null) {
            let e_ = a.originalEvent;
            let d = (e_.target);
            if (this.g.containsKey(d)) {
                let e = this.g.item(d);
                for (let f = 0; f < e.count; f++) {
                    e._inner[f]();
                }
                e.clear();
                this.g.removeItem(d);
            }
        }
        this.q();
        this.ad();
    }
    downloadCompleted(a, b) {
        let c = this.i._inner[b];
        let d = true;
        if (c != null && c.f != null && c.f.ac != null) {
            d = false;
            let ele_ = c.f.ac.c;
            if (!ele_.complete) {
                d = true;
            }
            if ((ele_.width == 0 && ele_.height == 0)) {
                d = true;
            }
        }
        this.i._inner[b] = null;
        if (d) {
            this.q();
            this.ad();
            return;
        }
        let e_ = a.originalEvent;
        let e = (e_.target);
        if (this.g.containsKey(e)) {
            let f = this.g.item(e);
            for (let g = 0; g < f.count; g++) {
                f._inner[g]();
            }
            f.clear();
            this.g.removeItem(e);
        }
        this.b.be(c, c.f.ac);
        if (c.f != null) {
            this.b.bz(c);
        }
        this.q();
        this.ad();
    }
    a(a, b, c, d) {
        let e = new WriteableBitmap(a, a);
        e.c = b.c;
        if (!b.d.isEmpty) {
            c += truncate(Math.round(b.d.left));
            d += truncate(Math.round(b.d.top));
        }
        e.d = new Rect(0, c, d, a, a);
        return e;
    }
    am(a) {
        a.s = 0;
    }
    an(a) {
        a.s = 1;
    }
    f() {
        return true;
    }
    s(a) {
        if (this.b.ak != null) {
            this.b.ak.deferredRefresh();
        }
        else {
            if (this.k != null) {
                this.k.setTimeout(a, 0);
            }
            else {
                window.setTimeout(a, 0);
            }
        }
    }
    aq() {
        if (this.p == -1) {
            this.p = window.setInterval(runOn(this.b, this.b.by), 50);
        }
    }
    as() {
        if (this.p != -1) {
            window.clearInterval(this.p);
            this.p = -1;
        }
    }
    ap() {
        if (this.o == -1) {
            this.o = window.setInterval(runOn(this.b, this.b.bh), 50);
        }
    }
    ar() {
        if (this.o != -1) {
            window.clearInterval(this.o);
            this.o = -1;
        }
    }
    t() {
        if (this.p != -1) {
            this.ar();
            this.b.bn();
        }
    }
    ae(a) {
        this.k = a;
        this.k.rootWrapper.setStyleProperty("position", "relative");
        this.m = this.k.createElement("canvas");
        this.m.setStyleProperty("position", "absolute");
        this.m.setStyleProperty("top", "0px");
        this.m.setStyleProperty("left", "0px");
        this.k.append(this.m);
        this.c = new RenderingContext(new CanvasViewRenderer(), this.k.get2DCanvasContext(this.m));
        this.af(this.k.rootWrapper.width(), this.k.rootWrapper.height());
    }
    af(a, b) {
        this.m.setAttribute("width", truncate((a * this.b.a1)).toString());
        this.m.setAttribute("height", truncate((b * this.b.a1)).toString());
        this.m.setStyleProperty("width", a.toString() + "px");
        this.m.setStyleProperty("height", b.toString() + "px");
        this.b.b9 = new Rect(0, 0, 0, a, b);
        this.b.bu();
    }
    aj() {
        this.ad();
    }
    get index() {
        return 0;
    }
    postRender() {
    }
    get isDirty() {
        return this._isDirty;
    }
    set isDirty(a) {
        this._isDirty = a;
    }
    ad() {
        if (this.c == null) {
            return;
        }
        if (!this.isDirty) {
            this.isDirty = true;
            this.b.ag.n();
        }
    }
    undirty(a) {
        this.isDirty = false;
        this.ak();
    }
    ak() {
        if (this.c == null) {
            return;
        }
        if (this.c.d && this.b.a1 != 1) {
            this.c.z();
            this.c.aa(this.b.a1, this.b.a1);
        }
        if (!this.au.isEmpty) {
            this.c.k(this.au.left, this.au.top, this.au.width, this.au.height);
        }
        this.au = this.b.b9;
        for (let a = 0; a < this.h.count; a++) {
            let b = this.h._inner[a];
            if (b.s == 0) {
                this.al(b);
            }
        }
        for (let c = 0; c < this.h.count; c++) {
            let d = this.h._inner[c];
            if (d.s == 1) {
                this.al(d);
            }
        }
        if (this.c.d && this.b.a1 != 1) {
            this.c.y();
        }
        this.b.bj();
        this.b.bl();
    }
    d() {
        let a = true;
        for (let b = 0; b < this.i.count; b++) {
            if (this.i._inner[b] != null) {
                a = false;
            }
        }
        return a;
    }
    al(a) {
        if (this.c == null) {
            return;
        }
        let b = a.ac;
        let c = a._opacity;
        if (b == null || b.c == null) {
            return;
        }
        if (this.e(b.c)) {
            return;
        }
        if (!b.d.isEmpty) {
            if (b.d.width < 1 || b.d.height < 1) {
                return;
            }
            this.c.n(b.c, c, b.d.left, b.d.top, b.d.width, b.d.height, Math.round(a.n + this.b.b9.left), Math.round(a.o + this.b.b9.top), a.width, a.height);
        }
        else {
            this.c.m(b.c, c, Math.round(a.n + this.b.b9.left), Math.round(a.o + this.b.b9.top), a.width, a.height);
        }
    }
    e(a) {
        let img_ = a;
        if (!img_.complete) {
            return true;
        }
        if ((img_.width == 0 && img_.height == 0)) {
            return true;
        }
        return false;
    }
    x() {
        this.ad();
    }
    ag(a) {
        this.c = a;
        this.ad();
    }
    ah(a) {
        this.l = a;
    }
    ai(a) {
        this.b.b9 = a;
        this.b.bu();
    }
    preRender() {
    }
    isValid() {
        return true;
    }
}
XamMultiScaleImageView.$t = markType(XamMultiScaleImageView, 'XamMultiScaleImageView', Base.$, [ISchedulableRender_$type]);
export class XamMultiScaleTileSource extends DependencyObject {
    constructor(a, b, c, d, e) {
        super();
        this.o = 0;
        this.n = 0;
        this.m = 0;
        this.k = 0;
        this.l = 0;
        this.j = null;
        this.o = a;
        this.n = b;
        this.m = c;
        this.k = d;
        this.l = e;
    }
    get q() {
        return this.o;
    }
    set q(a) {
        this.o = a;
        this.t(0, 0, 0, 0);
    }
    get p() {
        return this.n;
    }
    set p(a) {
        this.n = a;
        this.t(0, 0, 0, 0);
    }
    r(a, b, c) {
        let d = new List$1(Base.$, 0);
        this.s(a, b, c, d);
        let e = null;
        if (d.count > 0) {
            e = typeCast(Uri.$, d._inner[0]);
        }
        return e;
    }
    t(a, b, c, d) {
        if (this.j != null) {
            this.j.bi(a, b, c, d);
        }
    }
}
XamMultiScaleTileSource.$t = markType(XamMultiScaleTileSource, 'XamMultiScaleTileSource', DependencyObject.$);
export class XamMultiScaleImage extends Control {
    constructor() {
        super();
        this.ao = null;
        this.am = null;
        this.aj = null;
        this.b6 = null;
        this.az = 0;
        this.propertyChanged = null;
        this.a7 = 0;
        this.a8 = 0;
        this.ax = new Date();
        this.b7 = null;
        this.a0 = 0;
        this.av = new List$1(Tile.$, 0);
        this.at = false;
        this.ap = new LRUCache$2(Tile.$, WriteableBitmap.$, 1, 0x7FFFFFFF, new TilePositionComparer());
        this.aw = new List$1(Tile.$, 0);
        this.imageTilesReady = null;
        this.ar = false;
        this.b9 = null;
        this.ag = null;
        this.imagesChanged = null;
        this.downloadingImage = null;
        this.ay = 1;
        this.b9 = Rect.empty;
        this.ag = new CanvasRenderScheduler();
        this.am = new XamMultiScaleImageView(this);
        this.ab = XamMultiScaleImage.$;
        this.b6 = this.b8;
        this.az = this.a2;
    }
    get ak() {
        return this.aj;
    }
    set ak(a) {
        if (this.aj != null) {
            this.aj.unRegister(this);
        }
        this.aj = a;
        if (this.aj != null) {
            this.aj.register(this, runOn(this, this.bv));
        }
    }
    bm(a) {
        if (this.propertyChanged != null) {
            this.propertyChanged(this, a);
        }
        switch (a.propertyName) {
            case XamMultiScaleImage.$$p[0]:
                if (this.an != null) {
                    this.an.j = this;
                }
                this.bt();
                this.bw();
                this.bu();
                break;
            case XamMultiScaleImage.$$p[3]:
                this.bx();
                break;
            case XamMultiScaleImage.$$p[4]:
                this.bx();
                break;
            case XamMultiScaleImage.$$p[2]:
                if (!this.au) {
                    this.am.t();
                }
                break;
        }
    }
    bw() {
        this.b0();
        if (this.an != null) {
            this.a7 = Convert.toInt32(logBase(this.an.m, 2));
            this.a8 = Convert.toInt32(logBase(this.an.q, 2));
        }
    }
    bi(a, b, c, d) {
        this.bt();
        this.bw();
        this.bu();
    }
    bx() {
        if (this.au) {
            this.ax = dateNow();
            this.b7 = this.b6;
            this.a0 = this.az;
            this.am.aq();
        }
        else {
            this.b6 = this.b8;
            this.az = this.a2;
            this.bu();
        }
    }
    by() {
        let a = 2;
        let b = dateNow().getTime() - this.ax.getTime();
        let c = b / 1000;
        let d = MathUtil.b((c) / a, 0, 1);
        let e = this.ai != null ? this.ai.ease(d) : d;
        let f = 1 - e;
        this.az = this.a0 * f + this.a2 * e;
        this.b6 = { $type: Point_$type, x: this.b7.x * f + this.b8.x * e, y: this.b7.y * f + this.b8.y * e };
        if (d >= 1) {
            this.am.as();
        }
        else {
        }
        this.bu();
    }
    a5(a, b, c) {
        for (let d = 0; d < this.av.count; ++d) {
            if (this.av._inner[d].b == a && this.av._inner[d].c == b && this.av._inner[d].d == c) {
                return d;
            }
        }
        return -1;
    }
    bv(a) {
        this.at = false;
        this.ar = false;
        if (this.an == null || !this.am.f() || this.b9.width == 0 || this.b9.height == 0) {
            return;
        }
        let b = truncate(Math.ceil(this.b9.width / this.an.m));
        let c = truncate(Math.max(1, Math.floor(-logBase(this.az / b, 2))));
        if (c >= this.a8 - 8) {
            c = (this.a8 - 8) - 1;
        }
        let d = truncate(Math.round(Math.pow(2, c)));
        let e = this.az;
        let f = this.b9.height * e / this.b9.width;
        let g = this.an.q / Math.pow(2, c);
        let h = this.an.p / Math.pow(2, c);
        let i = Math.max(truncate(Math.floor((this.b6.x * this.an.q) / g)), 0);
        let j = Math.min(truncate(Math.ceil(((this.b6.x + e) * this.an.q) / g)), d);
        let k = Math.max(truncate(Math.floor((this.b6.y * this.an.p) / h)), 0);
        let l = Math.min(truncate(Math.ceil(((this.b6.y + f) * this.an.q) / h)), d);
        let m = ((i * g) - (this.b6.x * this.an.q)) / g;
        let n = ((k * h) - (this.b6.y * this.an.p)) / h;
        let o = (e * this.an.q / g) * (this.an.m / this.b9.width);
        let p = new List$1(Tile.$, 0);
        for (let q = i; q < j; ++q) {
            for (let r = k; r < l; ++r) {
                let s = this.a5(q, r, c);
                if (s >= 0) {
                    p.add(this.av._inner[s]);
                    this.av.removeAt(s);
                }
                else {
                    p.add(((() => {
                        let $ret = new Tile();
                        $ret.b = q;
                        $ret.c = r;
                        $ret.d = c;
                        return $ret;
                    })()));
                }
            }
        }
        this.ao.f = true;
        this.b0();
        this.av = p;
        for (let t = 0; t < this.av.count; ++t) {
            if (this.av._inner[t].f == null) {
                this.av._inner[t].f = this.ao.a();
                this.av._inner[t].f._opacity = 1;
                this.am.am(this.av._inner[t].f);
                let u = this.al(this.av._inner[t]);
                if (u != null) {
                    this.av._inner[t].f.ac = u;
                }
                else {
                    let v = null;
                    let w = ((() => {
                        let $ret = new Tile();
                        $ret.b = this.av._inner[t].b;
                        $ret.c = this.av._inner[t].c;
                        $ret.d = this.av._inner[t].d;
                        return $ret;
                    })());
                    while (w.d >= 0 && v == null) {
                        w.b = w.b >> 1;
                        w.c = w.c >> 1;
                        w.d = w.d - 1;
                        v = this.al(w);
                    }
                    if (v != null) {
                        let x = truncate(Math.pow(2, this.av._inner[t].d - w.d));
                        let y = intDivide(256, x);
                        let z = y * (this.av._inner[t].b % x);
                        let aa = y * (this.av._inner[t].c % x);
                        this.av._inner[t].e = this.ao.a();
                        this.av._inner[t].e._opacity = 1;
                        this.am.an(this.av._inner[t].e);
                        u = this.am.a(y, v, z, aa);
                        this.av._inner[t].e.ac = u;
                    }
                    this.am.u(this.av._inner[t]);
                }
            }
            let ab = this.an.m / o;
            let ac = this.an.k / o;
            let ad = (this.av._inner[t].b - i + m) * ab;
            let ae = (this.av._inner[t].c - k + n) * ac;
            this.av._inner[t].f.width = ab + 0.5;
            this.av._inner[t].f.height = ac + 0.5;
            this.am.ao(this.av._inner[t].f, ad, ae);
            if (this.av._inner[t].e != null) {
                this.av._inner[t].e.width = ab + 0.5;
                this.av._inner[t].e.height = ac + 0.5;
                this.am.ao(this.av._inner[t].e, ad, ae);
            }
        }
        this.ao.f = false;
        this.am.aj();
    }
    bu() {
        if (this.an == null || !this.am.f() || this.b9.width == 0 || this.b9.height == 0) {
            return;
        }
        if (this.at) {
            return;
        }
        this.at = true;
        this.am.s(runOn(this, this.bv));
    }
    b0() {
        for (let a = 0; a < this.av.count; ++a) {
            this.am.r(this.av._inner[a]);
            this.bf(this.av._inner[a]);
            if (this.av._inner[a].f != null) {
                this.ao.n(this.av._inner[a].f);
                this.av._inner[a].f.ac = null;
                this.av._inner[a].f = null;
            }
        }
    }
    bt() {
        this.aq.i();
    }
    al(a) {
        return this.aq.item(a);
    }
    be(a, b) {
        this.aq.item(a, b);
    }
    get aq() {
        return this.ap;
    }
    set aq(a) {
        this.ap = a;
    }
    as() {
        return this.aw.count != 0;
    }
    bj() {
        let a = true;
        if (this.as()) {
            a = false;
        }
        if (!this.am.d()) {
            a = false;
        }
        if (a && !this.ar) {
            if (this.imageTilesReady != null) {
                this.imageTilesReady(this, new EventArgs());
            }
        }
        this.ar = a;
    }
    bz(a) {
        if (a.e != null) {
            a.a = dateNow();
            this.aw.add(a);
            this.am.ap();
        }
        else {
            this.bj();
        }
    }
    bf(a) {
        if (a.e != null) {
            this.ao.n(a.e);
            a.e.ac = null;
            a.e = null;
            for (let b = 0; b < this.aw.count; ++b) {
                if (this.aw._inner[b] == a) {
                    this.aw.removeAt(b);
                    break;
                }
            }
            if (this.aw.count == 0) {
                this.am.ar();
            }
        }
    }
    bh() {
        let a = dateNow();
        let b = 0.5;
        for (let c = 0; c < this.aw.count;) {
            let d = a.getTime() - this.aw._inner[c].a.getTime();
            let e = d / 1000;
            let f = (e) / b;
            f = MathUtil.b(f, 0, 1);
            this.aw._inner[c].e._opacity = 1 - f;
            if (f >= 1) {
                this.ao.n(this.aw._inner[c].e);
                this.aw._inner[c].e.ac = null;
                this.aw._inner[c].e = null;
                this.aw.removeAt(c);
            }
            else {
                ++c;
            }
            this.am.x();
        }
        if (this.aw.count == 0) {
            this.am.ar();
        }
    }
    bn() {
        this.az = this.a2;
        this.b6 = this.b8;
        this.bu();
    }
    bp(a) {
        let b = a;
        this.ag.j(b);
        this.am.ae(a);
    }
    bq(a) {
        this.am.ag(a);
    }
    br(a) {
        this.am.ah(a);
    }
    bs(a) {
        this.am.ai(a);
    }
    bl() {
        if (this.imagesChanged != null) {
            this.imagesChanged(this, new EventArgs());
        }
    }
    bg() {
        this.at = false;
    }
    bk(a) {
        if (this.downloadingImage != null) {
            this.downloadingImage(this, a);
        }
    }
    get a1() {
        return this.ay;
    }
    bo(a) {
        this.ay = a;
    }
    get a6() {
        return this.aq.h;
    }
    set a6(a) {
        let b = a != this.a6;
        if (b) {
            this.aq = new LRUCache$2(Tile.$, WriteableBitmap.$, 1, a, new TilePositionComparer());
        }
    }
}
XamMultiScaleImage.$t = markType(XamMultiScaleImage, 'XamMultiScaleImage', Control.$, [INotifyPropertyChanged_$type]);
Type.dep(DependencyProperty, PropertyMetadata, XamMultiScaleImage, 'bm', ['Source:an:b1', [XamMultiScaleTileSource.$, null], 'SpringsEasingFunction:ai:b2', [IEasingFunction_$type, null], 'UseSprings:au:b3', [0, false], 'ViewportOrigin:b8:b4', [Point_$type, { $type: Point_$type, x: 0, y: 0 }], 'ViewportWidth:a2:b5', [1, 1]]);
