// igniteui-angular-core/ES2015/XYChartDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { DomainChartDescription } from "./DomainChartDescription";
import { markType } from "./type";
export class XYChartDescription extends DomainChartDescription {
    constructor() {
        super();
        this.e1 = null;
        this.fh = null;
        this.c7 = 0;
        this.c9 = 0;
        this.c8 = 0;
        this.c6 = 0;
        this.dp = 0;
        this.dr = 0;
        this.dq = 0;
        this.dn = 0;
        this.e4 = null;
        this.fl = null;
        this.di = 0;
        this.d0 = 0;
        this.dh = 0;
        this.dz = 0;
        this.dk = 0;
        this.d2 = 0;
        this.dj = 0;
        this.d1 = 0;
        this.dg = 0;
        this.dy = 0;
        this.ff = null;
        this.fw = null;
        this.e5 = null;
        this.fm = null;
        this.fg = null;
        this.fx = null;
        this.e3 = null;
        this.fk = null;
        this.e8 = null;
        this.fp = null;
        this.da = 0;
        this.ds = 0;
        this.db = 0;
        this.dt = 0;
        this.fa = null;
        this.fr = null;
        this.fb = null;
        this.fs = null;
        this.dc = 0;
        this.du = 0;
        this.dd = 0;
        this.dv = 0;
        this.fc = null;
        this.ft = null;
        this.de = 0;
        this.dw = 0;
        this.fd = null;
        this.fu = null;
        this.e9 = null;
        this.fq = null;
        this.c5 = 0;
        this.dm = 0;
        this.c4 = 0;
        this.dl = 0;
        this.df = 0;
        this.dx = 0;
        this.c0 = false;
        this.c1 = false;
        this.fe = null;
        this.fv = null;
        this.e2 = null;
        this.fi = null;
        this.e6 = null;
        this.fn = null;
        this.e7 = null;
        this.fo = null;
        this.fj = null;
    }
    get_type() {
        return "XYChart";
    }
    get xAxisFormatLabelRef() {
        return this.e1;
    }
    set xAxisFormatLabelRef(a) {
        this.e1 = a;
        this.e("XAxisFormatLabelRef");
    }
    get yAxisFormatLabelRef() {
        return this.fh;
    }
    set yAxisFormatLabelRef(a) {
        this.fh = a;
        this.e("YAxisFormatLabelRef");
    }
    get xAxisLabelLeftMargin() {
        return this.c7;
    }
    set xAxisLabelLeftMargin(a) {
        this.c7 = a;
        this.e("XAxisLabelLeftMargin");
    }
    get xAxisLabelTopMargin() {
        return this.c9;
    }
    set xAxisLabelTopMargin(a) {
        this.c9 = a;
        this.e("XAxisLabelTopMargin");
    }
    get xAxisLabelRightMargin() {
        return this.c8;
    }
    set xAxisLabelRightMargin(a) {
        this.c8 = a;
        this.e("XAxisLabelRightMargin");
    }
    get xAxisLabelBottomMargin() {
        return this.c6;
    }
    set xAxisLabelBottomMargin(a) {
        this.c6 = a;
        this.e("XAxisLabelBottomMargin");
    }
    get yAxisLabelLeftMargin() {
        return this.dp;
    }
    set yAxisLabelLeftMargin(a) {
        this.dp = a;
        this.e("YAxisLabelLeftMargin");
    }
    get yAxisLabelTopMargin() {
        return this.dr;
    }
    set yAxisLabelTopMargin(a) {
        this.dr = a;
        this.e("YAxisLabelTopMargin");
    }
    get yAxisLabelRightMargin() {
        return this.dq;
    }
    set yAxisLabelRightMargin(a) {
        this.dq = a;
        this.e("YAxisLabelRightMargin");
    }
    get yAxisLabelBottomMargin() {
        return this.dn;
    }
    set yAxisLabelBottomMargin(a) {
        this.dn = a;
        this.e("YAxisLabelBottomMargin");
    }
    get xAxisLabelTextColor() {
        return this.e4;
    }
    set xAxisLabelTextColor(a) {
        this.e4 = a;
        this.e("XAxisLabelTextColor");
    }
    get yAxisLabelTextColor() {
        return this.fl;
    }
    set yAxisLabelTextColor(a) {
        this.fl = a;
        this.e("YAxisLabelTextColor");
    }
    get xAxisTitleMargin() {
        return this.di;
    }
    set xAxisTitleMargin(a) {
        this.di = a;
        this.e("XAxisTitleMargin");
    }
    get yAxisTitleMargin() {
        return this.d0;
    }
    set yAxisTitleMargin(a) {
        this.d0 = a;
        this.e("YAxisTitleMargin");
    }
    get xAxisTitleLeftMargin() {
        return this.dh;
    }
    set xAxisTitleLeftMargin(a) {
        this.dh = a;
        this.e("XAxisTitleLeftMargin");
    }
    get yAxisTitleLeftMargin() {
        return this.dz;
    }
    set yAxisTitleLeftMargin(a) {
        this.dz = a;
        this.e("YAxisTitleLeftMargin");
    }
    get xAxisTitleTopMargin() {
        return this.dk;
    }
    set xAxisTitleTopMargin(a) {
        this.dk = a;
        this.e("XAxisTitleTopMargin");
    }
    get yAxisTitleTopMargin() {
        return this.d2;
    }
    set yAxisTitleTopMargin(a) {
        this.d2 = a;
        this.e("YAxisTitleTopMargin");
    }
    get xAxisTitleRightMargin() {
        return this.dj;
    }
    set xAxisTitleRightMargin(a) {
        this.dj = a;
        this.e("XAxisTitleRightMargin");
    }
    get yAxisTitleRightMargin() {
        return this.d1;
    }
    set yAxisTitleRightMargin(a) {
        this.d1 = a;
        this.e("YAxisTitleRightMargin");
    }
    get xAxisTitleBottomMargin() {
        return this.dg;
    }
    set xAxisTitleBottomMargin(a) {
        this.dg = a;
        this.e("XAxisTitleBottomMargin");
    }
    get yAxisTitleBottomMargin() {
        return this.dy;
    }
    set yAxisTitleBottomMargin(a) {
        this.dy = a;
        this.e("YAxisTitleBottomMargin");
    }
    get xAxisTitleTextColor() {
        return this.ff;
    }
    set xAxisTitleTextColor(a) {
        this.ff = a;
        this.e("XAxisTitleTextColor");
    }
    get yAxisTitleTextColor() {
        return this.fw;
    }
    set yAxisTitleTextColor(a) {
        this.fw = a;
        this.e("YAxisTitleTextColor");
    }
    get xAxisLabelTextStyle() {
        return this.e5;
    }
    set xAxisLabelTextStyle(a) {
        this.e5 = a;
        this.e("XAxisLabelTextStyle");
    }
    get yAxisLabelTextStyle() {
        return this.fm;
    }
    set yAxisLabelTextStyle(a) {
        this.fm = a;
        this.e("YAxisLabelTextStyle");
    }
    get xAxisTitleTextStyle() {
        return this.fg;
    }
    set xAxisTitleTextStyle(a) {
        this.fg = a;
        this.e("XAxisTitleTextStyle");
    }
    get yAxisTitleTextStyle() {
        return this.fx;
    }
    set yAxisTitleTextStyle(a) {
        this.fx = a;
        this.e("YAxisTitleTextStyle");
    }
    get xAxisLabelRef() {
        return this.e3;
    }
    set xAxisLabelRef(a) {
        this.e3 = a;
        this.e("XAxisLabelRef");
    }
    get yAxisLabelRef() {
        return this.fk;
    }
    set yAxisLabelRef(a) {
        this.fk = a;
        this.e("YAxisLabelRef");
    }
    get xAxisMajorStroke() {
        return this.e8;
    }
    set xAxisMajorStroke(a) {
        this.e8 = a;
        this.e("XAxisMajorStroke");
    }
    get yAxisMajorStroke() {
        return this.fp;
    }
    set yAxisMajorStroke(a) {
        this.fp = a;
        this.e("YAxisMajorStroke");
    }
    get xAxisMajorStrokeThickness() {
        return this.da;
    }
    set xAxisMajorStrokeThickness(a) {
        this.da = a;
        this.e("XAxisMajorStrokeThickness");
    }
    get yAxisMajorStrokeThickness() {
        return this.ds;
    }
    set yAxisMajorStrokeThickness(a) {
        this.ds = a;
        this.e("YAxisMajorStrokeThickness");
    }
    get xAxisMinorStrokeThickness() {
        return this.db;
    }
    set xAxisMinorStrokeThickness(a) {
        this.db = a;
        this.e("XAxisMinorStrokeThickness");
    }
    get yAxisMinorStrokeThickness() {
        return this.dt;
    }
    set yAxisMinorStrokeThickness(a) {
        this.dt = a;
        this.e("YAxisMinorStrokeThickness");
    }
    get xAxisStrip() {
        return this.fa;
    }
    set xAxisStrip(a) {
        this.fa = a;
        this.e("XAxisStrip");
    }
    get yAxisStrip() {
        return this.fr;
    }
    set yAxisStrip(a) {
        this.fr = a;
        this.e("YAxisStrip");
    }
    get xAxisStroke() {
        return this.fb;
    }
    set xAxisStroke(a) {
        this.fb = a;
        this.e("XAxisStroke");
    }
    get yAxisStroke() {
        return this.fs;
    }
    set yAxisStroke(a) {
        this.fs = a;
        this.e("YAxisStroke");
    }
    get xAxisStrokeThickness() {
        return this.dc;
    }
    set xAxisStrokeThickness(a) {
        this.dc = a;
        this.e("XAxisStrokeThickness");
    }
    get yAxisStrokeThickness() {
        return this.du;
    }
    set yAxisStrokeThickness(a) {
        this.du = a;
        this.e("YAxisStrokeThickness");
    }
    get xAxisTickLength() {
        return this.dd;
    }
    set xAxisTickLength(a) {
        this.dd = a;
        this.e("XAxisTickLength");
    }
    get yAxisTickLength() {
        return this.dv;
    }
    set yAxisTickLength(a) {
        this.dv = a;
        this.e("YAxisTickLength");
    }
    get xAxisTickStroke() {
        return this.fc;
    }
    set xAxisTickStroke(a) {
        this.fc = a;
        this.e("XAxisTickStroke");
    }
    get yAxisTickStroke() {
        return this.ft;
    }
    set yAxisTickStroke(a) {
        this.ft = a;
        this.e("YAxisTickStroke");
    }
    get xAxisTickStrokeThickness() {
        return this.de;
    }
    set xAxisTickStrokeThickness(a) {
        this.de = a;
        this.e("XAxisTickStrokeThickness");
    }
    get yAxisTickStrokeThickness() {
        return this.dw;
    }
    set yAxisTickStrokeThickness(a) {
        this.dw = a;
        this.e("YAxisTickStrokeThickness");
    }
    get xAxisTitle() {
        return this.fd;
    }
    set xAxisTitle(a) {
        this.fd = a;
        this.e("XAxisTitle");
    }
    get yAxisTitle() {
        return this.fu;
    }
    set yAxisTitle(a) {
        this.fu = a;
        this.e("YAxisTitle");
    }
    get xAxisMinorStroke() {
        return this.e9;
    }
    set xAxisMinorStroke(a) {
        this.e9 = a;
        this.e("XAxisMinorStroke");
    }
    get yAxisMinorStroke() {
        return this.fq;
    }
    set yAxisMinorStroke(a) {
        this.fq = a;
        this.e("YAxisMinorStroke");
    }
    get xAxisLabelAngle() {
        return this.c5;
    }
    set xAxisLabelAngle(a) {
        this.c5 = a;
        this.e("XAxisLabelAngle");
    }
    get yAxisLabelAngle() {
        return this.dm;
    }
    set yAxisLabelAngle(a) {
        this.dm = a;
        this.e("YAxisLabelAngle");
    }
    get xAxisExtent() {
        return this.c4;
    }
    set xAxisExtent(a) {
        this.c4 = a;
        this.e("XAxisExtent");
    }
    get yAxisExtent() {
        return this.dl;
    }
    set yAxisExtent(a) {
        this.dl = a;
        this.e("YAxisExtent");
    }
    get xAxisTitleAngle() {
        return this.df;
    }
    set xAxisTitleAngle(a) {
        this.df = a;
        this.e("XAxisTitleAngle");
    }
    get yAxisTitleAngle() {
        return this.dx;
    }
    set yAxisTitleAngle(a) {
        this.dx = a;
        this.e("YAxisTitleAngle");
    }
    get xAxisInverted() {
        return this.c0;
    }
    set xAxisInverted(a) {
        this.c0 = a;
        this.e("XAxisInverted");
    }
    get yAxisInverted() {
        return this.c1;
    }
    set yAxisInverted(a) {
        this.c1 = a;
        this.e("YAxisInverted");
    }
    get xAxisTitleAlignment() {
        return this.fe;
    }
    set xAxisTitleAlignment(a) {
        this.fe = a;
        this.e("XAxisTitleAlignment");
    }
    get yAxisTitleAlignment() {
        return this.fv;
    }
    set yAxisTitleAlignment(a) {
        this.fv = a;
        this.e("YAxisTitleAlignment");
    }
    get xAxisLabelHorizontalAlignment() {
        return this.e2;
    }
    set xAxisLabelHorizontalAlignment(a) {
        this.e2 = a;
        this.e("XAxisLabelHorizontalAlignment");
    }
    get yAxisLabelHorizontalAlignment() {
        return this.fi;
    }
    set yAxisLabelHorizontalAlignment(a) {
        this.fi = a;
        this.e("YAxisLabelHorizontalAlignment");
    }
    get xAxisLabelVerticalAlignment() {
        return this.e6;
    }
    set xAxisLabelVerticalAlignment(a) {
        this.e6 = a;
        this.e("XAxisLabelVerticalAlignment");
    }
    get yAxisLabelVerticalAlignment() {
        return this.fn;
    }
    set yAxisLabelVerticalAlignment(a) {
        this.fn = a;
        this.e("YAxisLabelVerticalAlignment");
    }
    get xAxisLabelVisibility() {
        return this.e7;
    }
    set xAxisLabelVisibility(a) {
        this.e7 = a;
        this.e("XAxisLabelVisibility");
    }
    get yAxisLabelVisibility() {
        return this.fo;
    }
    set yAxisLabelVisibility(a) {
        this.fo = a;
        this.e("YAxisLabelVisibility");
    }
    get yAxisLabelLocation() {
        return this.fj;
    }
    set yAxisLabelLocation(a) {
        this.fj = a;
        this.e("YAxisLabelLocation");
    }
}
XYChartDescription.$t = markType(XYChartDescription, 'XYChartDescription', DomainChartDescription.$);
